---
title: "Strain sharing with inStrain"
author: "Simon Yersin"
date: "2025-05-01"
output: html_document
---

The script include the analysis of strain transmission using inStrain. 
A. include metadata, checkm (quality of genomes) and gtdb (taxonomy)
B. Analysis of inStrain results of only the MAGs
C. Analysis of inStrain results including MAGs and isolates

Steps:
1.
First, we used dRep, dRep is a python program which performs rapid pair-wise comparison of genome sets. One of it’s major purposes is for genome de-replication, but it can do a lot more.
We used dRep to dereplicate genome with the following parameters to obtain genome representing strains:
Warning during dRep, we set the parameters at completeness 75% and contamination 10%
Secondary cluster: 98% ANI 

2. 
InStrain is a tool for analysis of co-occurring genome populations from metagenomes that allows highly accurate genome comparisons, analysis of coverage, microdiversity, and linkage, and sensitive SNP detection with gene localization and synonymous non-synonymous identification.

The output of inStrain compare is the genome wide comparison using ani=0.99999 and cov=0.5
inStrain compare was run on each individual (ex: AGN001, which had saliva, and fecal samples)
- conANI – consensus ANI: ANI calculated based on consensus sequences. Each position on the genome is represented by the most common allele and minor allele are ignored
- popANI – populatin ANI: performed by inStrain, it considers both major and minor alleles. If two populations share any allele at a loci, including minor alleles, it does not count as a difference when calculating popANI

Organisms in different samples that are linked by a recent transmission event should have >=99.999% popANI and >=50% percent_genome_compared 

the result includes genomes complete at 75% and contaminated at less than 10%

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Loading library
```{r}
library(readxl)
library(dplyr)
library(stringr)
library(ggplot2)
library(tidyr)
library(data.table)
library(purrr)
library(readr)
```

# A. Load tables
## Metadata
```{r}
md <- as.data.frame(read_xlsx("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Metadata/Metadata_satellite.xlsx"))
md$id <- str_remove(md$id, "1429")
rownames(md) <- md$id

md_full <- as.data.frame(read_xlsx("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Metadata/Metadata_satellite_full.xlsx"))
rownames(md_full) <- md_full$id_modif
```

## CheckM tables
Quality of the genomes (MAGs): completeness and contamination
```{r}
checkm_dir <- "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/checkM"
checkm_tables <- list.files(checkm_dir, full.names = TRUE, pattern = "*.tsv")
checkm_mags <- checkm_tables %>%
  map_dfr(read.delim)
colnames(checkm_mags)[1] <- "Bin_ID"
colnames(checkm_mags)[2] <- "Marker_lineage"
colnames(checkm_mags)[3] <- "Num_genomes"
colnames(checkm_mags)[4] <- "Num_markers"
colnames(checkm_mags)[5] <- "Num_markers_sets"
colnames(checkm_mags)[6] <- "0"
colnames(checkm_mags)[7] <- "1"
colnames(checkm_mags)[8] <- "2"
colnames(checkm_mags)[9] <- "3"
colnames(checkm_mags)[10] <- "4"
colnames(checkm_mags)[11] <- "5"
colnames(checkm_mags)[14] <- "Strain_heterogeneity"
```

## GTDB tables
Taxonomy of the genomes (MAGs)
```{r, warning=FALSE}
gtdb_dir <- "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/GTDB"
gtdb_tables <- list.files(gtdb_dir, full.names = TRUE, pattern = "*.bac120.summary.tsv")
standardize_columns <- function(file) {
  df <- read.delim(file, stringsAsFactors = FALSE)
  # Standardize all columns
  df[, 4:7] <- df[, 4:7] %>%
    mutate(across(everything(), ~ case_when(
      . == "N/A" ~ NA_real_,  # Transform "N/A" to NA of type double
      TRUE ~ as.numeric(.)
    )))
  return(df)
}
gtdb_mags <- gtdb_tables %>%
  map_dfr(standardize_columns)
```

merging tables
```{r}
mags_df <- merge(checkm_mags, gtdb_mags, by.x="Bin_ID", by.y="user_genome")
mags_df$Completeness <- as.numeric(mags_df$Completeness)
mags_df$Contamination <- as.numeric(mags_df$Contamination)
# 10'975 genomes
# 3 genomes are not bacteria, not included with GTDB-TK
```

# B. MAGs analysis
## dRep
Filter genome >=75% completeness and <10% contamination --> 3'221 MAGs
Resulting cluster with 98% identity: 2147
```{r}
# dRep genomes with primary cluster ANI 90% and secondary cluster ANI 98%, coverage 30%
drepl_all_genomes <- read.csv2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/dRep/Only_MAGs/Widb.csv",
                               sep = ",")

# Clean genome name extensions
drepl_all_genomes$genome <- str_remove_all(drepl_all_genomes$genome, ".fasta")
drepl_all_genomes$genome <- str_remove_all(drepl_all_genomes$genome, ".fa")

# 2147 genomes after dereplication
# Merge with tax and quality
drepl_all_genomes_tax <- merge(drepl_all_genomes, mags_df, by.x = "genome", by.y = "Bin_ID", all.x = TRUE)

setdiff(drepl_all_genomes$genome, mags_df$Bin_ID)
# AGN005FE.22.fa Not bacteria

# Clean names
drepl_all_genomes_tax$genome <- str_remove_all(drepl_all_genomes_tax$genome, ".scaffolds.min1000")
drepl_all_genomes_tax$genome <- str_remove_all(drepl_all_genomes_tax$genome, "_assembly")

drepl_df <- drepl_all_genomes_tax[,c(1,2,8,26:29)]
write_csv2(drepl_df, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/Only_MAGs/dRep_WinningGenomesMAGs_df.csv")

# Filter for S. salivarius
drepl_ssal <- drepl_all_genomes_tax %>% filter(str_detect(drepl_all_genomes_tax$classification, "Streptococcus salivarius"))
write.csv(drepl_ssal, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/Only_MAGs/dRep_salivarius_WinningGenomesMAGs.csv")
# 12 dereplicated Streptococcus salivarius genomes (from MAGs)

# Clusters from dRep
drepl_cdb <- read.csv2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/dRep/Only_MAGs/Cdb.csv", sep = ",")

drepl_cdb$genome <- str_remove_all(drepl_cdb$genome, ".fa")

drepl_cdb_tax <- merge(drepl_cdb, mags_df, by.x = "genome", by.y = "Bin_ID", all.x = TRUE)


drepl_cdb_tax$genome <- str_remove_all(drepl_cdb_tax$genome, ".scaffolds.min1000")
drepl_cdb_tax$genome <- str_remove_all(drepl_cdb_tax$genome, "_assembly")
drepl_cdb_tax$genome <- str_replace(drepl_cdb_tax$genome, "SAT892", "SAT692")
# Filter for cluster of salivarius
drepl_cdb_ssal <- drepl_cdb %>% filter(secondary_cluster %in% drepl_ssal$cluster)
# 14 S. streptococcus salivarius sort into 1 clusters
write.csv(drepl_cdb_ssal, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/Only_MAGs/dRep_salivarius_clustersOnlyMAGs.csv")
```


## inStrain
```{r}
# Import genome wide comparison table (result from inStrain compare)
instrain_mags <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/Only_MAGs/instrain_compare_mags_genomeWide_compare_merged.tsv", sep = "\t")
# 449 observations 
instrain_mags$genome <- str_remove_all(instrain_mags$genome, ".fa")
instrain_mags$genome <- str_remove_all(instrain_mags$genome, ".scaffolds.min1000")

# Get checkM and gtdb data about the genomes
is_mags_df <- mags_df
is_mags_df$Bin_ID <- str_remove_all(is_mags_df$Bin_ID, ".scaffolds.min1000")
is_mags_df <- is_mags_df[,c(1,12,13,15)]

# Merge with inStrain table and cleaning
instrain_mags_tax <- merge(is_mags_df, instrain_mags, by.x="Bin_ID", by.y="genome")
instrain_mags_tax <- instrain_mags_tax %>%
    separate(classification, 
             into = c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species"), 
             sep = ";", 
             remove = FALSE, 
             fill = "right")
instrain_mags_tax <- instrain_mags_tax %>%
    dplyr::mutate(across(c(Kingdom, Phylum, Class, Order, Family, Genus, Species), 
                  ~ gsub("^[d,p,c,o,f,g,s]__", "", .)))
instrain_mags_tax$Species[which(instrain_mags_tax$Species == "")] <- instrain_mags_tax$Genus[which(instrain_mags_tax$Species== "")]
instrain_mags_tax$Completeness <- as.numeric(instrain_mags_tax$Completeness)
instrain_mags_tax$Contamination <- as.numeric(instrain_mags_tax$Contamination)
# Cleaning sample names
instrain_mags_tax$name1 <- str_replace_all(instrain_mags_tax$name1, "_sorted_allGenomesMapped.bam", "")
instrain_mags_tax$name2 <- str_replace_all(instrain_mags_tax$name2, "_sorted_allGenomesMapped.bam", "")

# Numeric for popANI and 
instrain_mags_tax$popANI <- as.numeric(instrain_mags_tax$popANI)
instrain_mags_tax$percent_compared <- as.numeric(instrain_mags_tax$percent_compared)

# filtering dataset to only include strain-sharing comparisons based on inStrain recommended cutoffs
instrain_mags_tax_filt <- instrain_mags_tax %>%
  filter(popANI >= 0.99999 & percent_compared >= 0.5)
# Remove sample AGN43DU
instrain_mags_tax_filt <- instrain_mags_tax_filt %>%
  filter(name1 != "AGN043DU") # 5 sharing events removed 

# Add column with sample name
instrain_mags_tax_filt <- instrain_mags_tax_filt %>%
  mutate(SampleID = str_replace(name1, "(SA|FE|GA|DU)$", ""))
# Add sharing compartments
instrain_mags_tax_filt <- instrain_mags_tax_filt %>%
  mutate(Compartment = paste0(str_extract(name1, "(SA|FE|GA|DU)$"), "_", str_extract(name2, "(SA|FE|GA|DU)$")))
# Add in if the representative is a MAG (AGN) or a isolate (SAT) --> only MAGs for now
instrain_mags_tax_filt <- instrain_mags_tax_filt %>%
  mutate(IsoMags = paste0(str_extract(Bin_ID, "(AGN|SAT)")))
#Add metadata
instrain_mags_md <- merge(instrain_mags_tax_filt, md, by.x = "SampleID", by.y = "id")
# Clean names
instrain_mags_md$Genus[which(instrain_mags_md$Genus=="Haemophilus_A")] <- "Haemophilus"
instrain_mags_md$Genus[which(instrain_mags_md$Genus=="Haemophilus_D")] <- "Haemophilus"

# How many sharing events: 114
# 17 are the same strains in the same individual
# Sharing event 97
# How many individals
n_distinct(instrain_mags_md$SampleID)
# 27 individuals
# How many species
n_distinct(instrain_mags_md$Species)
# 52
# How many genera?
n_distinct(instrain_mags_md$Genus)
# 19
```

```{r}


pal_instrain <- c("#377EB8","#8DD3C7","#4DAF4A","#d5a6bd","#E41A1C","#bdd5a6", "#d5a7a6","#d5d5a6",
                  "#FCCDE5","#FFED6F","#7FC97F","#BEAED4","#FDC086","#ffe599","#D95F02",
                  "#7570B3","#E7298A","#E6AB02","#6fa8dc")


pdf("Figures/inStrain_all_legend.pdf")
instrain_mags_md %>%
  ggplot(aes(x=Compartment, fill=Genus)) +
  geom_bar(color = "black") +
  scale_fill_manual(values = pal_instrain) +
  theme_minimal() +
  theme(
        legend.position = "bottom") +
  labs(fill = "Genus") + 
  guides(fill = guide_legend(nrow = 3)) +
  coord_flip()
dev.off()
```
Table for pptx
```{r}
instrain_mags_df <- instrain_mags_md %>% select(SampleID, Bin_ID, Compartment, Species)
instrain_mags_df <- pivot_wider(instrain_mags_df, names_from = "Compartment", values_from = "Species")
colnames(instrain_mags_df)[3] <- "Saliva_Feces"
colnames(instrain_mags_df)[4] <- "Saliva_Gastric"
colnames(instrain_mags_df)[5] <- "Gastric_Feces"
colnames(instrain_mags_df)[6] <- "Saliva_Duodenum"
colnames(instrain_mags_df)[7] <- "Gastric_Duodenum"
colnames(instrain_mags_df)[8] <- "Duodenum_Feces"
write.csv(instrain_mags_df, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/Only_MAGs/inStrain_sharing.csv")

instrain_mags_md %>% group_by(Genus) %>% summarise(Count = n_distinct(Bin_ID))

freq_instrain <- instrain_mags_md %>%
    ggplot(aes(x = reorder(Genus, Genus, function(x) -length(x)))) +  # Reorder by frequency
    geom_bar(fill = "grey", color = "black") +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 70, hjust = 1, face="italic")) +
  facet_grid(~Compartment, scales = "free_x") +
    labs(x = element_blank(),
         y = "Nb of sharing events")
# Save
pdf("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/Frequency_shared_species_MAGs.pdf", width = 12)
freq_instrain
dev.off()
```


Table for upset plot --> only MAGs (unecessary) see later
```{r}
instrain_mags_md_SGS <- instrain_mags_md %>%
  filter(str_detect(instrain_mags_md$Compartment, "DU", negate=TRUE)) 

#Save table for S salivarius for upset plot
instrain_mags_md_ssal <- instrain_mags_md_SGS %>% filter(Species == "Streptococcus salivarius" | Species == "Streptococcus salivarius_D")
instrain_mags_md_ssal <- instrain_mags_md_ssal[,c(1:23)]

write.csv(instrain_mags_md_ssal, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/Only_MAGs/inStrain_upsetplot_onlyMAGs.csv")
```






# C. Isolates + MAGs analysis
Add CheckM and GTDB tables:
```{r}
#CheckM
checkm_iso <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/Assemblies_checkM_stats.tsv")
colnames(checkm_iso)[1] <- "Bin_ID"
colnames(checkm_iso)[2] <- "Marker_lineage"
colnames(checkm_iso)[3] <- "Num_genomes"
colnames(checkm_iso)[4] <- "Num_markers"
colnames(checkm_iso)[5] <- "Num_markers_sets"
colnames(checkm_iso)[6] <- "0"
colnames(checkm_iso)[7] <- "1"
colnames(checkm_iso)[8] <- "2"
colnames(checkm_iso)[9] <- "3"
colnames(checkm_iso)[10] <- "4"
colnames(checkm_iso)[11] <- "5"
colnames(checkm_iso)[14] <- "Strain_heterogeneity"

# Merging with checkM mags
checkm_merged <- rbind(checkm_mags, checkm_iso)

# GTDB-Tk
gtdb_iso <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/gtdbtk.bac120.summary.tsv")
# Merging with gtdb mags
gtdb_merged <- rbind(gtdb_mags, gtdb_iso)
```

```{r}
genome_df <- merge(checkm_merged, gtdb_merged, by.x="Bin_ID", by.y="user_genome")
genome_df$Completeness <- as.numeric(genome_df$Completeness)
genome_df$Contamination <- as.numeric(genome_df$Contamination)
# 11'079 genomes
```


## dRep
dRep is a python program which performs rapid pair-wise comparison of genome sets. One of it’s major purposes is for genome de-replication, but it can do a lot more.
We used dRep to dereplicate genome with the following parameters to obtain genome representing strains:
Warning during dRep, we set the parameters at completeness 75% and contamination 10%
Secondary cluster: 98% ANI 
```{r}
# dRep genomes with primary cluster ANI 90% and secondary cluster ANI 98%, coverage 30%
drepl_all_genomes_mi <- read.csv2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/dRep/With_isolates/Widb_strains.csv",
                               sep = ",")

# Clean genome name extensions
drepl_all_genomes_mi$genome <- str_remove_all(drepl_all_genomes_mi$genome, ".fasta")
drepl_all_genomes_mi$genome <- str_remove_all(drepl_all_genomes_mi$genome, ".fa")
# 2180 genomes after dereplication
# Merge with tax and quality
drepl_all_genomes_mi_tax <- merge(drepl_all_genomes_mi, genome_df, by.x = "genome", by.y = "Bin_ID", all.x = TRUE)

setdiff(drepl_all_genomes_mi$genome, genome_df$Bin_ID)
# AGN005FE.22.fa Not bacteria

# Clean names
drepl_all_genomes_mi_tax$genome <- str_remove_all(drepl_all_genomes_mi_tax$genome, ".scaffolds.min1000")
drepl_all_genomes_mi_tax$genome <- str_remove_all(drepl_all_genomes_mi_tax$genome, "_assembly")


# Filter for S. salivarius
drepl_ssal_mi <- drepl_all_genomes_mi_tax %>% filter(str_detect(drepl_all_genomes_mi_tax$classification, "Streptococcus salivarius"))
write.csv(drepl_ssal_mi, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/With_isolates/dRep_salivarius_WinningGenomes_MAGsISO.csv")
# 40 dereplicated Streptococcus salivarius genomes (MAGs+Isolates)

# Clusters from dRep
drepl_cdb_mi <- read.csv2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/dRep/With_isolates/Cdb_strain.csv", sep = ",")

drepl_cdb_mi$genome <- str_remove_all(drepl_cdb_mi$genome, ".fa")

drepl_cdb_mi_tax <- merge(drepl_cdb_mi, genome_df, by.x = "genome", by.y = "Bin_ID", all.x = TRUE)


drepl_cdb_mi_tax$genome <- str_remove_all(drepl_cdb_mi_tax$genome, ".scaffolds.min1000")
drepl_cdb_mi_tax$genome <- str_remove_all(drepl_cdb_mi_tax$genome, "_assembly")
drepl_cdb_mi_tax$genome <- str_replace(drepl_cdb_mi_tax$genome, "SAT892", "SAT692")
# Filter for cluster of salivarius
drepl_cdb_mi_ssal <- drepl_cdb_mi %>% filter(secondary_cluster %in% drepl_ssal_mi$cluster)
# 104 S. streptococcus salivarius sort into 40 clusters
write.csv(drepl_cdb_mi_ssal, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/With_isolates/dRep_salivarius_clusters_MAGsISO.csv")
```

Isolates clusters
Not sure what this is, see assembly analysus script
```{r}
# Sharing events between two different body site
sharing_event_iso <- sharing_clusters %>%
  group_by(PatientID, Strain_cluster) %>%
  mutate(NSource = n_distinct(Source)) %>%  
  mutate(Sharing = case_when(NSource >= 2 ~ "Sharing", NSource ==1 ~ "No sharing")) %>% 
  ungroup() %>% 
  filter(Sharing == "Sharing")
n_distinct(sharing_event_iso$Strain_cluster) # 12 sharing events

sharing_event_iso <- merge(sharing_event_iso, drepl_cdb_ssal[,c(1:2)], by.x = "StrainID", by.y = "genome")
sharing_event_iso <- merge(sharing_event_iso, drepl_ssal[,c(1,8)], by.x = "secondary_cluster", by.y = "cluster", all.x = TRUE)
colnames(sharing_event_iso)[8] <- "WinningGenome"


se_iso_w <- sharing_event_iso %>%
  group_by(Strain_cluster, PatientID, Source, WinningGenome) %>%
  summarise(SharedStrains = paste(StrainID, collapse = ", "), .groups = "drop") %>%
  pivot_wider(names_from = Source, values_from = SharedStrains)
colnames(se_iso_w)[1] <- "Sharing Cluster"
se_iso_w <- se_iso_w[,c(1:4, 6,7,5)]
se_iso_w <- se_iso_w %>% arrange(PatientID)
write.csv(se_iso_w, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/Iso_salivarius_StrainSharingEvents.csv")
```



## inStrain
Load and clean tables
```{r}
# Import genome wide comparison table (result from inStrain compare)
instrain <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/With_isolates/instrain_compare_genomeWide_compare_merged.tsv", sep = "\t")
# 440 observations 
instrain$genome <- str_remove_all(instrain$genome, ".fa")
instrain$genome <- str_remove_all(instrain$genome, ".scaffolds.min1000")
instrain$genome <- str_remove_all(instrain$genome, "_assembly")

# Get checkM and gtdb data aout the genomes
is_genome_df <- genome_df
is_genome_df$Bin_ID <- str_remove_all(is_genome_df$Bin_ID, ".scaffolds.min1000")
is_genome_df$Bin_ID <- str_remove_all(is_genome_df$Bin_ID, "_assembly")
is_genome_df <- is_genome_df[,c(1,12,13,15)]

# Merge with inStrain table and cleaning
instrain_tax <- merge(is_genome_df, instrain, by.x="Bin_ID", by.y="genome")
instrain_tax <- instrain_tax %>%
    separate(classification, 
             into = c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species"), 
             sep = ";", 
             remove = FALSE, 
             fill = "right")
instrain_tax <- instrain_tax %>%
    dplyr::mutate(across(c(Kingdom, Phylum, Class, Order, Family, Genus, Species), 
                  ~ gsub("^[d,p,c,o,f,g,s]__", "", .)))
instrain_tax$Species[which(instrain_tax$Species == "")] <- instrain_tax$Genus[which(instrain_tax$Species== "")]
instrain_tax$Completeness <- as.numeric(instrain_tax$Completeness)
instrain_tax$Contamination <- as.numeric(instrain_tax$Contamination)
# Cleaning sample names
instrain_tax$name1 <- str_replace_all(instrain_tax$name1, "_sorted_allGenomesMapped.bam", "")
instrain_tax$name2 <- str_replace_all(instrain_tax$name2, "_sorted_allGenomesMapped.bam", "")

# Numeric for popANI and 
instrain_tax$popANI <- as.numeric(instrain_tax$popANI)
instrain_tax$percent_compared <- as.numeric(instrain_tax$percent_compared)

# filtering dataset to only include strain-sharing comparisons based on inStrain recommended cutoffs
instrain_tax_filt <- instrain_tax %>%
  filter(popANI >= 0.99999 & percent_compared >= 0.5)
# Remove sample AGN43DU
instrain_tax_filt <- instrain_tax_filt %>%
  filter(name1 != "AGN043DU") # 5 sharing events removed 
# 121 strain sharing events

# Add column with sample name (without SA and FE)
instrain_tax_filt <- instrain_tax_filt %>%
  mutate(SampleID = str_replace(name1, "(SA|FE|GA|DU)$", ""))
# Add sharing compartments
instrain_tax_filt <- instrain_tax_filt %>%
  mutate(Compartment = paste0(str_extract(name1, "(SA|FE|GA|DU)$"), "_", str_extract(name2, "(SA|FE|GA|DU)$")))
# Add in if the representative is a MAG (AGN) or a isolate (SAT)
instrain_tax_filt <- instrain_tax_filt %>%
  mutate(IsoMags = paste0(str_extract(Bin_ID, "(AGN|SAT)")))
#Add metadata
instrain_md <- merge(instrain_tax_filt, md, by.x = "SampleID", by.y = "id")
# Clean names
instrain_md$Genus[which(instrain_md$Genus=="Haemophilus_A")] <- "Haemophilus"
instrain_md$Genus[which(instrain_md$Genus=="Haemophilus_D")] <- "Haemophilus"

# How many sharing events: 121
# XX are the same strains in the same individual
# Sharing event XX
# How many individals
n_distinct(instrain_md$SampleID)
# 27 individuals
# How many species
n_distinct(instrain_md$Species)
# 53
# Difference with only MAGs
setdiff(instrain_md$Species, instrain_mags_md$Species)
# "Streptococcus parasanguinis_K" "Streptococcus sp901875575" 
# How many genera?
n_distinct(instrain_md$Genus)
# 19
# Same genera than MAGs only


# Which sharing events are different from only MAGs: only the ones with isolates from Streptococcus, the rest are identical
```


inStrain S. salivarius for upset plot
```{r}
instrain_ssal <- instrain_md %>% filter(Species == "Streptococcus salivarius" | Species == "Streptococcus salivarius_D")
# 8 sharing events: 7 isolates + 1 MAGs passing above threshold

instrain_ssal_mags <- instrain_mags_md %>% filter(Species == "Streptococcus salivarius" | Species == "Streptococcus salivarius_D")
# 5 sharing events for S. salivarius

# Combine both
instrain_ssal <- rbind(instrain_ssal, instrain_ssal_mags)

# Remove duplicates
instrain_ssal <- instrain_ssal %>%
  filter(Bin_ID != "AGN022GA.6" & Bin_ID != "AGN036DU.1")
# Identified both method: 1x AGN022 and 1x AGN036
# Detected only by MAGS: 2x AGN010 and 1x AGN033
# Detected with isolates: AGN019 goes above threshold, 2x AGN032, 1x AGN036, 2x AGN039 (5 new transmission from isolates)
# 11 transmission found with inStrain

write.csv(instrain_ssal[,1:23], "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/inStrain_salivarius_StrainSharing.csv")
```



### STOP HERE ###
```{r}
# Remove duodenal samples
instrain_md_SGS <- instrain_md %>%
  filter(str_detect(instrain_md$Compartment, "DU", negate=TRUE)) 
n_distinct(instrain_md_SGS$SampleID)

# Create a frequency plot for the shared species within related dyads
freq_species <- instrain_md %>%
  filter(str_detect(instrain_md$Compartment, "DU", negate=TRUE)) %>%
    ggplot(aes(x = reorder(Species, Species, function(x) -length(x)))) +  # Reorder by frequency
    geom_bar(aes(fill = SampleID), color = "black") +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, face="italic")) +
    labs(title = "Shared Species within the same individual",
         x = "Shared Species within the same individual",
         y = "Frequency",
         fill = "Isolate (SAT) or MAGs (AGN)") +
  facet_grid(stunted~Compartment, labeller = labeller(stunted = c("0" ="Non-stunted", "1"="Stunted")), scales = "free")

instrain_md %>%
  filter(str_detect(instrain_md$Compartment, "DU", negate=TRUE)) %>%
    ggplot(aes(x = reorder(Species, Species, function(x) -length(x)))) +  # Reorder by frequency
    geom_bar(aes(fill = IsoMags), color = "black") +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, face="italic")) +
    labs( x = "Shared Species within the same individual",
         y = "Frequency",
         fill = "Isolate (SAT) or MAGs (AGN)") +
  facet_grid(~Compartment, scales = "free")

pdf("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/Frequency_shared_species.pdf",
    height = 6, width = 13)
freq_species
dev.off()

# Frequency plot with duodenum, only in stunted children
freq_species_stunted <- instrain_md %>%
  filter(stunted == "1") %>%
    ggplot(aes(x = reorder(Species, Species, function(x) -length(x)))) +  # Reorder by frequency
    geom_bar(aes (fill=SampleID), color = "black") +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, face="italic")) +
    labs(title = "Shared Species within the same individual",
         x = "Shared Species within the same individual",
         y = "Frequency") +
  facet_grid(stunted~Compartment, labeller = labeller(stunted = c("0" ="Non-stunted", "1"="Stunted")))
freq_species_stunted

# Create a frequency plot for the shared genus within related dyads
instrain_md %>%
    ggplot(aes(x = reorder(Genus, Genus, function(x) -length(x)))) +  # Reorder by frequency
    geom_bar(fill = "grey", color = "black") +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, face="italic")) +
    labs(title = "Frequency of Shared genus within related dyads",
         x = "Shared genus within related dyads",
         y = "Frequency") +
  facet_grid(~statut_nut)



# shared strains within same individual in function of nutritional status
n_distinct(instrain_md$SampleID)
classification_table <- instrain_md %>%
  distinct(SampleID, .keep_all = TRUE) %>%  # Keep only one row per VG_ID
  pull(stunted) %>%  # Pull the 'classif' column as a vector
  table()  # Create a table for the unique 'classif' values

# View the result
print(classification_table)

# Calculate expected counts (the distribution is not equal across all groups)
  # Define proportions
expected_proportions_df <- md %>%
    count(stunted) %>%  # Count the number of occurrences of each classif
      mutate(proportion = n / sum(n))  # Calculate the proportion for each class

# Perform the Chi-square test
chi_test <- chisq.test(classification_table, p = expected_proportions_df$proportion)
print(chi_test)
# There is no more strain sharing pairs between a nutritional group and another one (p-value = 0.1352)

# Print the expected and observed counts to check
print(expected_proportions_df$n)
print(classification_table)
```
inStrain between saliva and stool
```{r}
instrain_sal_sto <- instrain_md %>% filter(str_detect(instrain_md$name1, "FE") & str_detect(instrain_md$name2, "SA"))
# 11 strain sharing event between the saliva and stool samples
# Which samples:
unique(instrain_sal_sto$SampleID)
# in 6 samples: "AGN001" "AGN002" "AGN031" "AGN032" "AGN039" "AGN046"
# There are 6 individuals that share a strain between saliva and stool

# Create a frequency plot for the shared species within same individual
instrain_sal_sto %>%
    ggplot(aes(x = reorder(Species, Species, function(x) -length(x)))) +  # Reorder by frequency
    geom_bar(fill = "grey", color = "black") +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, face="italic")) +
    labs(title = "Frequency of Shared Strain",
         x = "Shared Species within related dyads",
         y = "Frequency")

# Create a frequency plot for the shared genus within related dyads
instrain_sal_sto %>%
    ggplot(aes(x = reorder(Genus, Genus, function(x) -length(x)))) +  # Reorder by frequency
    geom_bar(fill = "grey", color = "black") +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, face="italic")) +
    labs(title = "Frequency of Shared genus within related dyads",
         x = "Shared genus within related dyads",
         y = "Frequency")




```

```{r}
instrain_lm <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/instrain_compare_genomeWide_compare_merged.tsv", sep = "\t") %>%
  mutate(genome = str_replace(genome, '.fasta', '')) %>%
  mutate(genome = str_replace(genome, '.fa', '')) %>%
  mutate(name1 = str_replace(name1, '.sorted.bam', '')) %>%
  mutate(name2 = str_replace(name2, '.sorted.bam', ''))
instrain_lm$popANI <- as.numeric(instrain_lm$popANI)
instrain_lm$conANI <- as.numeric(instrain_lm$conANI)
model <- lm(conANI ~ popANI, data = instrain_lm)
summary(model)
predict(model, data.frame(popANI = c(0.99999)))
coef(model)

ggplot(instrain_lm, aes(x = popANI, y = conANI)) +
  geom_point(shape = 21, fill = 'black', alpha = 0.5) +
  stat_smooth(method = 'lm', col = 'red')
```
