---
title: "Isolate analysis"
author: "Simon Yersin"
date: "2025-08-08"
output: html_document
---

Warning: SAT892 was misslabeled, the correct name for this isolate is SAT692 (short reads)

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Load library
```{r, warning=FALSE}
library(readxl)
library(dplyr)
library(stringr)
library(ggplot2)
library(tidyr)
library(ComplexHeatmap)
library(circlize)
library(igraph)
library(ggtree)
library(treeio)
library(ggnewscale)
library(purrr)
library(geiger)
library(ape)
```
# Isolates analysis
## Load tables
CheckM Isolates
```{r}
checkm_iso <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/Assemblies_checkM_stats.tsv")
colnames(checkm_iso)
colnames(checkm_iso)[1] <- "Bin_ID"
colnames(checkm_iso)[2] <- "Marker_lineage"
colnames(checkm_iso)[3] <- "Num_genomes"
colnames(checkm_iso)[4] <- "Num_markers"
colnames(checkm_iso)[5] <- "Num_markers_sets"
colnames(checkm_iso)[6] <- "0"
colnames(checkm_iso)[7] <- "1"
colnames(checkm_iso)[8] <- "2"
colnames(checkm_iso)[9] <- "3"
colnames(checkm_iso)[10] <- "4"
colnames(checkm_iso)[11] <- "5"
colnames(checkm_iso)[14] <- "Strain_heterogeneity"

# Adding extension in the name, matching the file names for dRep
checkm_iso$Bin_ID <- str_remove_all(checkm_iso$Bin_ID, "_assembly")
checkm_iso$Bin_ID <- str_remove_all(checkm_iso$Bin_ID, ".scaffolds.min1000")
```

GTDB isolates
```{r, warning=FALSE}
gtdb_iso <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/gtdbtk.bac120.summary.tsv")
gtdb_iso$user_genome <- str_remove_all(gtdb_iso$user_genome, "_assembly")
gtdb_iso$user_genome <- str_remove_all(gtdb_iso$user_genome, ".scaffolds.min1000")

iso_df <- merge(checkm_iso, gtdb_iso, by.x="Bin_ID", by.y="user_genome")
```

```{r}
# List of isolates
isolate_list <- read_xlsx("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Isolations/List/20250219_Satellite_Isolates.xlsx")

# 944 isolates from how many patient?
n_distinct(isolate_list$`Patient ID`) # 23 individuals
isolate_list <- isolate_list %>%
  mutate(PatientID_Source = paste(`Patient ID`, "_", Source))
#unique(isolate_list$PatientID_Source)
```

How many Streptococcus
```{r}
# 474 Streptococcus isolates
streptococcus_list <- isolate_list %>%
  filter(`Strain ID` %in% isolate_list$`Strain ID`[which(str_detect(isolate_list$`Strain name`, "Streptococcus"))])
# From how many patients:
unique(isolate_list$PatientID_Source)
unique(isolate_list$`Patient ID`)
unique(isolate_list$`Growth conditions`)

count_strep_iso <- streptococcus_list %>%
  group_by(`Strain name`, Source) %>%
  summarise(count = n_distinct(`Strain ID`))

pdf("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Figures/Nb_isolates.pdf")
ggplot(count_strep_iso, aes(x=`Strain name` , y= count, fill = Source)) +
  geom_bar(stat = "identity", position = "stack") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
dev.off()
```
```{r}
unique(isolate_list$`Strain name`)
```



Long reads stats
```{r}
# Flye stats
longreads_stats <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/flye_summary.tsv")
longreads_stats$filename <- str_remove_all(longreads_stats$filename, ".log")

# Name config
longreads_name_config <- read_xlsx("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/name_config.xlsx",
                                   col_names = FALSE)
colnames(longreads_name_config)[2] <- "filename"
colnames(longreads_name_config)[3] <- "IsoNames"
longreads_stats <- merge(longreads_stats, longreads_name_config[,2:3], by = "filename")

# Isolate metadata
longreads_stats <- merge(isolate_list[,1:6], longreads_stats, by.x = "Strain ID", by.y = "IsoNames", all.y = TRUE)

# CheckM
longread_checkm <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/Assemblies_checkM_stats.tsv")
longread_checkm$Bin.Id <- str_remove_all(longread_checkm$Bin.Id, "_assembly")
longreads_stats <- merge(longreads_stats, longread_checkm, by.x = "Strain ID", by.y = "Bin.Id")

# Taxonomy GTDB
longreads_gtdb <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/gtdbtk.bac120.summary.tsv")
longreads_gtdb$user_genome <- str_remove_all(longreads_gtdb$user_genome, "_assembly")
longreads_df <- merge(longreads_stats, longreads_gtdb, by.x = "Strain ID", by.y = "user_genome")

# Adding metadata for type strain
longreads_df$Project[which(longreads_df$`Strain ID` == "SsalTypeStrain")] <- "TypeStrain"
longreads_df$Source[which(longreads_df$`Strain ID` == "SsalTypeStrain")] <- "Blood"
longreads_df$`Strain name`[which(longreads_df$`Strain ID` == "SsalTypeStrain")] <- "Streptococcus salivarius"

longreads_df$Contamination <- as.numeric(longreads_df$Contamination)
longreads_df$Completeness <- as.numeric(longreads_df$Completeness)

longreads_dt <- longreads_df

# Filtering out samples: AFB651 (E. coli) and SAT 350 (contamination)
longreads_df <- longreads_df %>% filter(str_detect(longreads_df$classification, "g__Streptococcus"))
longreads_df <- longreads_df %>% filter(Contamination < 5 & Completeness > 95)
# 27 genomes

# Keep only Satellite and type strain
longreads_df <- longreads_df %>% filter(Project %in% c("Satellite", "TypeStrain"))
# 25 genomes: 24 Satellite and 1 Type strain

# Keep only Satellite strains
longreads_sat <- longreads_df %>% filter(Project == "Satellite")

# Stats
mean(longreads_sat$Total_length) # 2'257'823
mean(longreads_sat$Mean_coverage) # 149.4583
mean(longreads_sat$Fragments) # 1.625 fragments

# Tyep strain
longreads_ts <- longreads_df %>% filter(Project == "TypeStrain")
```

Short reads stats
```{r}
shortreads_stats <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/merged_stats.tsv")
shortreads_stats$filename <- str_remove_all(shortreads_stats$filename, ".min1000.assembly.stats")

# Renaming SAT892 for SAT692
shortreads_stats$filename <- str_replace(shortreads_stats$filename, "SAT892", "SAT692")
# Isolate metadata
shortreads_stats <- merge(isolate_list[,1:6], shortreads_stats, by.x = "Strain ID", by.y = "filename")

# CheckM
shortreads_checkm <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/Assemblies_checkM_stats.tsv")
shortreads_checkm$Bin.Id <- str_remove_all(shortreads_checkm$Bin.Id, ".scaffolds.min1000")
# Renaming SAT892 for SAT692
shortreads_checkm$Bin.Id <- str_replace(shortreads_checkm$Bin.Id, "SAT892", "SAT692")
shortreads_stats <- merge(shortreads_stats, shortreads_checkm, by.x = "Strain ID", by.y = "Bin.Id")

# Taxonomy GTDB
shortreads_gtdb <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/gtdbtk.bac120.summary.tsv")
shortreads_gtdb$user_genome <- str_remove_all(shortreads_gtdb$user_genome, ".scaffolds.min1000")
# Renaming SAT892 for SAT692
shortreads_gtdb$user_genome <- str_replace(shortreads_gtdb$user_genome, "SAT892", "SAT692")
shortreads_df <- merge(shortreads_stats, shortreads_gtdb, by.x = "Strain ID", by.y = "user_genome")

# Filter out Completness > 95% and contamination < 5% (SAT622 removed)
shortreads_df$Contamination <- as.numeric(shortreads_df$Contamination)
shortreads_df$Completeness <- as.numeric(shortreads_df$Completeness)

# Add coverage
iso_cov <- read.csv2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/average_coverage.csv", sep = ",")
iso_cov$file <- str_remove_all(iso_cov$file, ".depth")
iso_cov$file <- str_sub(iso_cov$file, end = -8)
iso_cov$file <- str_remove_all(iso_cov$file, "./")
shortreads_df <- merge(shortreads_df, iso_cov, by.x = "Strain ID", by.y = "file", all.x = TRUE)

shortreads_dt <- shortreads_df
# 75 genomes

shortreads_df <- shortreads_df %>% filter(Contamination < 5 & Completeness > 95)
# 74 genomes (removed SAT622)

# Keep only salivarius and parasanguinis
shortreads_sal <- shortreads_df %>% filter(str_detect(shortreads_df$classification, "s__Streptococcus salivarius"))
shortreads_par <- shortreads_df %>% filter(str_detect(shortreads_df$classification, "s__Streptococcus parasanguinis"))

# Stats
mean(shortreads_sal$total_length) # 2'223'160
mean(shortreads_sal$number) # 25.07937

mean(shortreads_par$total_length) # 2'177'953
mean(shortreads_par$number) # 29
```

Combine metadata short and long
Counting
```{r}
salivarius_iso <- rbind(longreads_sat[,c(1:6, 27)], shortreads_sal[,c(1:6, 34)])
salivarius_iso_count <- salivarius_iso %>%
  group_by(`Patient ID`, Source) %>%
  summarise(count = n_distinct(`Strain ID`))
sum(salivarius_iso_count$count) # 87 salivarius isolates of High Quality
salivarius_iso_count <- pivot_wider(salivarius_iso_count, names_from = "Source", values_from = "count", values_fill = 0)
```

Isolate data table
```{r}
longreads_dt <- longreads_dt %>% mutate(Gaps = 0)
longreads_dt <- longreads_dt %>% mutate(N_count = 0)
colnames(shortreads_dt)[7] <- "filename"
colnames(shortreads_dt)[8] <- "Total_length"
colnames(shortreads_dt)[9] <- "Fragments"
colnames(shortreads_dt)[53] <- "Mean_coverage"
colnames(shortreads_dt)[11] <- "Largest_frg"
colnames(shortreads_dt)[15] <- "Fragments_N50"
iso_df <- rbind(longreads_dt[,c(1:7,27, 24:26,8,13,9:11,46,47)], shortreads_dt[,c(1:7, 34, 31:33,8,53,9,11,15,13,14)])

# Completing table
iso_df
iso_df$Project[which(str_detect(iso_df$`Strain ID`, "AFB"))] <- "Afribiota"

iso_df$`Patient ID`[which(iso_df$`Strain ID`=="AFB477")] <- "CPB512"
iso_df$Source[which(iso_df$`Strain ID`=="AFB477")] <- "Stool"
iso_df$`Growth conditions`[which(iso_df$`Strain ID`=="AFB477")] <- "Aerobic, BHIS"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="AFB477")] <- "Streptococcus salivarius"

iso_df$`Patient ID`[which(iso_df$`Strain ID`=="AFB570")] <- "CPB460"
iso_df$Source[which(iso_df$`Strain ID`=="AFB570")] <- "Stool"
iso_df$`Growth conditions`[which(iso_df$`Strain ID`=="AFB570")] <- "Aerobic, BHI+I"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="AFB570")] <- "Streptococcus salivarius"

iso_df$`Patient ID`[which(iso_df$`Strain ID`=="AFB651")] <- "CPB460"
iso_df$Source[which(iso_df$`Strain ID`=="AFB651")] <- "Stool"
iso_df$`Growth conditions`[which(iso_df$`Strain ID`=="AFB651")] <- "Aerobic, BHIS"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="AFB651")] <- "Escherichia coli"

iso_df$`Strain name`[which(iso_df$`Strain ID`=="SAT371")] <- "Streptococcus sp001813295"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="SAT526")] <- "Streptococcus"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="SAT529")] <- "Streptococcus"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="SAT622")] <- "Streptococcus"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="SAT941")] <- "Streptococcus salivarius"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="SAT692")] <- "Streptococcus salivarius_D"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="SAT749")] <- "Streptococcus salivarius_D"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="SAT361")] <- "Streptococcus salivarius_D"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="SAT417")] <- "Streptococcus salivarius"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="SAT734")] <- "Streptococcus salivarius"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="SAT813")] <- "Streptococcus salivarius"
iso_df$`Strain name`[which(iso_df$`Strain ID`=="SAT850")] <- "Streptococcus salivarius"

iso_df$Strain.heterogeneity <- as.numeric(iso_df$Strain.heterogeneity)
iso_df$Mean_coverage <- as.numeric(iso_df$Mean_coverage)
iso_df$Gaps <- as.numeric(iso_df$Gaps)
iso_df$N_count <- as.numeric(iso_df$N_count)

library(writexl)
write_xlsx(iso_df, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Isolates_summary.xlsx")
```




# ANI analysis on isolates
```{r}
# List of S. salivarius, from Satelllite and type strain
isolates_genome_list <- c(longreads_sat$`Strain ID`, shortreads_sal$`Strain ID`, "SsalTypeStrain")
# Removed Afribiota and parasanguinis strains: 88 salivarius

fastANI <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/fastANI/SAT_fastANI.tsv", sep = "\t",
                       header = FALSE)
colnames(fastANI)[1] <- "Genome1"
colnames(fastANI)[2] <- "Genome2"
colnames(fastANI)[3] <- "ANI"
colnames(fastANI)[4] <- "Aligned_fragments"
colnames(fastANI)[5] <- "Number_query_fragment"

fastANI$Genome1 <- str_remove_all(fastANI$Genome1, "_assembly.fa")
fastANI$Genome1 <- str_remove_all(fastANI$Genome1, ".scaffolds.min1000.fa")
fastANI$Genome2 <- str_remove_all(fastANI$Genome2, "_assembly.fa")
fastANI$Genome2 <- str_remove_all(fastANI$Genome2, ".scaffolds.min1000.fa")
# Fixing misslabeled strain (read file labeled 892 while it should be 692)
fastANI$Genome1 <- str_replace(fastANI$Genome1, "SAT892", "SAT692")
fastANI$Genome2 <- str_replace(fastANI$Genome2, "SAT892", "SAT692")
# Filtering genomes
fastANI <- fastANI %>% filter(Genome1 %in% isolates_genome_list)
fastANI <- fastANI %>% filter(Genome2 %in% isolates_genome_list)

fastANI <- fastANI[,c(1:3)]
fastANIw <- pivot_wider(fastANI, names_from = "Genome2", values_from = "ANI")

# Symmetrical matrix
fastANIw <- as.data.frame(fastANIw)
rownames(fastANIw) <- fastANIw$Genome1
fastANIw$Genome1 <- NULL
fastANIw <- fastANIw[order(rownames(fastANIw)), order(colnames(fastANIw))]
fastANIw <- as.matrix(fastANIw)
fastANIw <- apply(fastANIw, 2, as.numeric)
rownames(fastANIw) <- colnames(fastANIw)

symmetrize_matrix <- function(mat) {
  for (i in 1:nrow(mat)) {
    for (j in 1:ncol(mat)) {
      if (!is.na(mat[i, j]) && !is.na(mat[j, i])) {
        mat[i, j] <- (mat[i, j] + mat[j, i]) / 2
        mat[j, i] <- mat[i, j]
      }
    }
  }
  return(mat)
}

fastANI_sym <- symmetrize_matrix(as.matrix(fastANIw))
fastANI_sym <- as.data.frame(fastANI_sym)
fastANI_sym <- fastANI_sym %>% mutate(Genome1 = rownames(fastANI_sym))

# I need to add from which sample the isolate and MAGs are from 
# Add the taxonomy as well
iso_strain_list <- read_xlsx("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Isolations/List/20241212_Satellite_Isolates.xlsx")
iso_strain_list <- iso_strain_list %>% filter(`Strain ID` %in% fastANI_sym$Genome1)
iso_strain_list <- iso_strain_list[,c(2:4)]

fastANI_sym <- merge(iso_strain_list, fastANI_sym, by.y = "Genome1", by.x = "Strain ID", all.y = TRUE)
colnames(fastANI_sym)[1] <- "StrainID"
colnames(fastANI_sym)[2] <- "PatientID"

# Type strain info
fastANI_sym$PatientID[which(str_detect(fastANI_sym$StrainID, "Ssal"))] <- "Type_strain"
fastANI_sym$Source[which(str_detect(fastANI_sym$StrainID, "Ssal"))] <- "Blood"

# Adding taxonomy information
tax_ani <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/gtdbtk.bac120.summary.tsv")
tax_ani$user_genome <- str_remove_all(tax_ani$user_genome, "_assembly")
tax_ani$user_genome <- str_remove_all(tax_ani$user_genome, ".scaffolds.min1000")
tax_ani$user_genome <- str_replace(tax_ani$user_genome, "SAT892", "SAT692")
tax_ani <- tax_ani[, c(1,2)]
fastANI_sym <- merge(tax_ani, fastANI_sym, by.x="user_genome", by.y="StrainID")
colnames(fastANI_sym)[1] <- "StrainID"

hm_fastani <- fastANI_sym[,c(5:92)]
rownames(hm_fastani) <- fastANI_sym$StrainID
hm_fastani <- as.matrix(hm_fastani)

fastANI_sym$classification <- str_remove(fastANI_sym$classification,
                                     "d__Bacteria;p__Bacillota;c__Bacilli;o__Lactobacillales;f__Streptococcaceae;g__Streptococcus;")
fastANI_sym$classification[which(fastANI_sym$classification=="s__")] <- "Streptococcus"
fastANI_sym$classification <- str_remove(fastANI_sym$classification,
                                     "s__")
unique(fastANI_sym$PatientID)
row_annot <- rowAnnotation(source = fastANI_sym$Source, patientID = fastANI_sym$PatientID, strain = fastANI_sym$classification,
                                  col = list(source=c("Blood" = "#cc0000",
                                                  "Duodenum" = "#8fce00",
                                                  "Gastric aspirate" = "#6aa84f",
                                                  "Saliva" = "#3d85c6",
                                                  "Stool" = "#ce7e00"
                                                  ),
                                             patientID=c("AGN002"="#e06666", "AGN004"="#f6b26b", "AGN008"="#ffd966","AGN010"="#b6d7a8",
                                                         "AGN022"="#9fc5e8", "AGN034"="#c27ba0","AGN036"="#990000","AGN039"="#bf9000","AGN045"="#134f5c",
                                                         "AGN018"="#a2c4c9","AGN037"="#b45f06",
                                                         "AGN041"="#38761d","AGN046"="#351c75","AGN032"="#8e7cc3","Type_strain"="#cc0000"),
                                             strain=c("Streptococcus salivarius"="#f11316",
                                                      "Streptococcus salivarius_D"="#cc0009")),
                                  annotation_label = c("Source", "PatientID", "Strains"), 
                            annotation_name_side = "top")
col_fun = colorRamp2(c(90,95,98,99,99.9999,100), c("white", "white", "#85D54AFF", "#1E9B8AFF", "#33638DFF", "#440154FF"))

heatmap_ani_iso <- Heatmap(hm_fastani, name = "ANI distance",col = col_fun, show_row_dend = FALSE, show_column_dend = FALSE,
        right_annotation = row_annot, show_row_names = TRUE, row_names_side = "left",
        row_names_gp = gpar(fontsize = 8),column_names_gp = gpar(fontsize = 8), border = TRUE)
pdf("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Figures/Heatmap_ANI_Isolates.pdf", height = 10, width = 12)
heatmap_ani_iso
dev.off()
```

# strain richness
```{r}
# Perpare dataframe
fastANI_diversity <- merge(fastANI, iso_strain_list, by.x = "Genome1", by.y = "Strain ID", all.x = TRUE)
colnames(fastANI_diversity)[4] <- "PatientID1"
colnames(fastANI_diversity)[5] <- "Source1"
fastANI_diversity <- merge(fastANI_diversity, iso_strain_list, by.x = "Genome2", by.y = "Strain ID", all.x = TRUE)
colnames(fastANI_diversity)[6] <- "PatientID2"
colnames(fastANI_diversity)[7] <- "Source2"

# Keep only ANI comparison for same individual and same body site
# Keep same isolate comparison to add 1 strain richness if only 1 strain present
# Remove duplicated comparison
fastANI_richness_source <- fastANI_diversity %>%
  filter(PatientID1 == PatientID2) %>% 
  filter(Source1 == Source2) %>%
  mutate(Pair = pmin(Genome1, Genome2), Pair2 = pmax(Genome1, Genome2)) %>%
  distinct(Pair, Pair2, .keep_all = TRUE) %>%
  select(-Pair, -Pair2)
fastANI_richness_source$ANI <- as.numeric(fastANI_richness_source$ANI)
# Create a function for graphical approach to resolve the identical and non-identical strains
# Compute the strain richness
# The graph connect strain that have ANI >99.5 and keep single strain as well
# It then count the number of clusters and return it
compute_strain_richness <- function(sub_df) {
  # Get all unique genomes
  all_genomes <- unique(c(sub_df$Genome1, sub_df$Genome2))
  # Create a graph where ANI > 98 means strains are identical (linked)
  g <- graph_from_data_frame(sub_df[sub_df$ANI > 99.5, c("Genome1", "Genome2")], 
                             directed = FALSE, vertices = all_genomes)
  # Find connected components (clusters of identical strains)
  clusters <- components(g)$membership
  # Count the number of unique strain clusters
  strainrichness <- length(unique(clusters))
  #Plot graph
  plot(g, vertex.color = clusters, vertex.size = 20, vertex.label.color = "black",
       main = paste("Strain Clustering for", sub_df$PatientID2[1], "-", sub_df$Source2[1]),
       layout = layout_with_fr)

  return(strainrichness)
}

# Apply the function per PatientID & Source
strain_richness <- fastANI_richness_source %>%
  group_by(PatientID1, Source1) %>%
  summarise(StrainRichness = compute_strain_richness(cur_data()), .groups = "drop")
strain_richness <- pivot_wider(strain_richness, names_from = "Source1", values_from = "StrainRichness", values_fill = 0)
#Rename columns and merge with strain count
colnames(strain_richness)[1] <- "Patient ID"
colnames(strain_richness)[2:5] <- paste(colnames(strain_richness)[2:5], "Strain Richness")
#Merge with count of number of strain per sample source
salivarius_countRichness <- merge(salivarius_iso_count, strain_richness)
# Reorder columns
salivarius_countRichness <- salivarius_countRichness[,c(1,4,8,2,6,5,9,3,7)]
write.csv(salivarius_countRichness, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/Iso_salivarius_StrainvRichness.csv")

# Plot richness
salivarius_plotRichness <- salivarius_countRichness
salivarius_plotRichness <- salivarius_plotRichness[,c(1,3,5,7,9)]
colnames(salivarius_plotRichness) <- str_remove_all(colnames(salivarius_plotRichness), "Strain Richness")
salivarius_plotRichness <- pivot_longer(salivarius_plotRichness, cols = 2:5, names_to = "Body_site", values_to = "Richness")
# Remove 0 as no isolate were obtained
salivarius_plotRichness <- salivarius_plotRichness %>% filter(Richness > 0)

salivarius_plotRichness_df <- salivarius_plotRichness %>%
  group_by(Body_site) %>%
  summarise(
    mean_richness = mean(Richness, na.rm = TRUE),
    se = sd(Richness, na.rm = TRUE) / sqrt(n()),
    .groups = 'drop'
  )

salivarius_plotRichness_df$Body_site <- str_replace_all(salivarius_plotRichness_df$Body_site, " aspirate", "")
salivarius_plotRichness_df$Body_site <- str_replace_all(salivarius_plotRichness_df$Body_site, " ", "")
salivarius_plotRichness_df$Body_site <- str_replace_all(salivarius_plotRichness_df$Body_site, "Stool", "Feces")
salivarius_plotRichness_df$Body_site = factor(salivarius_plotRichness_df$Body_site,
                                               levels = c("Saliva", "Gastric", "Duodenum", "Feces"))

# plot
pdf("Figures/StrainRichness.pdf")
ggplot(salivarius_plotRichness_df, aes(x = Body_site, y = mean_richness, color = Body_site)) +
  geom_point(size = 5) +
  geom_errorbar(aes(ymin = mean_richness - se, ymax = mean_richness + se), width = 0.2) +
  scale_color_manual(values = c(
    Feces = "#DCC04B",
    Saliva = "#9AB9C2",
    Gastric = "#BBDEEC",
    Duodenum = "#EDE692"
  )) +
  labs(y = "Strain richness") +
  theme_minimal(base_size = 14) +
  theme(axis.title.x = element_blank(),
        legend.position = "bottom") +
    guides(color=guide_legend(title = "Body site"))
dev.off()
```

# Strain sharing event
```{r}
# Prepare dataframe
fastANI_diversity_patient <- fastANI_diversity
fastANI_diversity_patient$PatientID1[which(fastANI_diversity_patient$Genome1=="SsalTypeStrain")] <- "TypeStrain"
fastANI_diversity_patient$PatientID2[which(fastANI_diversity_patient$Genome2=="SsalTypeStrain")] <- "TypeStrain"
fastANI_diversity_patient$Source1[which(fastANI_diversity_patient$Genome1=="SsalTypeStrain")] <- "TypeStrain"
fastANI_diversity_patient$Source2[which(fastANI_diversity_patient$Genome2=="SsalTypeStrain")] <- "TypeStrain"
# Remove ANI comparison for same genomes
# Remove duplicated comparison
# Keep only comparison for same individual
fastANI_diversity_patient <- fastANI_diversity_patient %>%
  filter(Genome1 != Genome2) %>%
  filter(PatientID1 == PatientID2) %>%
  mutate(Pair = pmin(Genome1, Genome2), Pair2 = pmax(Genome1, Genome2)) %>%
  distinct(Pair, Pair2, .keep_all = TRUE) %>%
  select(-Pair, -Pair2)
# Combine strain name with PatientID and Source of the isolates
fastANI_diversity_patient$ANI <- as.numeric(fastANI_diversity_patient$ANI)
fastANI_diversity_patient$Genome1 <- str_c(fastANI_diversity_patient$Genome1, "_",fastANI_diversity_patient$PatientID1 , "_", fastANI_diversity_patient$Source1)
fastANI_diversity_patient$Genome2 <- str_c(fastANI_diversity_patient$Genome2, "_",fastANI_diversity_patient$PatientID2,  "_", fastANI_diversity_patient$Source2)
# List all strains
all_genomes_iso <- unique(c(fastANI_diversity_patient$Genome1, fastANI_diversity_patient$Genome2))
# Resolve graphically the cluster
igraph_iso <- graph_from_data_frame(fastANI_diversity_patient[fastANI_diversity_patient$ANI > 99.9, c("Genome1", "Genome2")], 
                             directed = FALSE, vertices = all_genomes_iso)
# Save cluster
sharing_clusters <- as.data.frame(components(igraph_iso)$membership)
colnames(sharing_clusters)[1] <- "Strain_cluster"
sharing_clusters <- cbind(sharing_clusters, str_split_fixed(rownames(sharing_clusters), "_", n=3))
colnames(sharing_clusters)[2] <- "StrainID"
colnames(sharing_clusters)[3] <- "PatientID"
colnames(sharing_clusters)[4] <- "Source"
#Sharing event saliva to stool
sharing_sf <- sharing_clusters %>% filter(Source %in% c("Saliva", "Stool"))
sharing_sf <- sharing_sf %>%
  group_by(PatientID, Strain_cluster) %>%
  summarise(NSource = n_distinct(Source)) %>%  
  mutate(Sharing = case_when(NSource ==2 ~ "Sharing",
                                                         NSource ==1 ~ "No sharing"))
n_distinct(sharing_sf$PatientID[which(sharing_sf$Sharing == "Sharing")]) # Sharing detected in 9 individuals
n_distinct(sharing_sf$Strain_cluster[which(sharing_sf$Sharing == "Sharing")]) # 10 sharing event detected
# In saliva and gastric
sharing_sg <- sharing_clusters %>% filter(Source %in% c("Saliva", "Gastric aspirate"))
sharing_sg <- sharing_sg %>%
  group_by(PatientID, Strain_cluster) %>%
  summarise(NSource = n_distinct(Source)) %>%  
  mutate(Sharing = case_when(NSource ==2 ~ "Sharing",
                                                         NSource ==1 ~ "No sharing"))
n_distinct(sharing_sg$PatientID[which(sharing_sg$Sharing == "Sharing")]) # Sharing detected in 9 individuals
n_distinct(sharing_sg$Strain_cluster[which(sharing_sg$Sharing == "Sharing")]) # 10 sharing event detected

write.csv(sharing_clusters, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Output_tables/Iso_salivarius_StrainSharing.csv")

# Save plot
pdf("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Figures/Iso_SharingEvent_igraph.pdf",
    height = 16, width = 16)
plot(igraph_iso, vertex.color = components(igraph_iso)$membership, vertex.size = 5, vertex.label.color = "black",
     layout = layout_with_fr)
dev.off()
```

```{r}
fastANI_test <- fastANI_diversity %>%
  filter(PatientID1 == PatientID2) %>% 
  mutate(Pair = pmin(Genome1, Genome2), Pair2 = pmax(Genome1, Genome2)) %>%
  distinct(Pair, Pair2, .keep_all = TRUE) %>%
  select(-Pair, -Pair2)
```


# Phylogeny
```{r}
tree <- read.iqtree("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Phylogeny/May06/SAT_isolates.treefile")

tree_data <- as.data.frame(as_tibble(tree))
tree_phylo <- ape::as.phylo(tree)

# Find node number for rooting using outgroup of S. parasanguinis
ggtree(tree, branch.length = "none") +
  geom_tiplab() +
  geom_text2(aes(label = node), hjust = -0.3, color = "red")

# Rooting tree at node 110
tree_rooted <- ape::root(tree_phylo, node = 161)


tree_delta <- tree_rooted

# Adding metadata
#iso_df$Bin_ID[which(iso_df$Bin_ID=="SAT892")] <- "SAT692"
phylo_md <- iso_df %>% filter(`Strain ID` %in% tree_data$label)
phylo_md <- merge(phylo_md, isolate_list, by.x = "Strain ID", by.y = "Strain ID", all.x = TRUE)
# Add info for type strain
phylo_md$`Patient ID.x`[which(str_detect(phylo_md$`Strain ID`, "Ssal"))] <- "Type_strain"
phylo_md$Source.x[which(str_detect(phylo_md$`Strain ID`, "Ssal"))] <- "Blood"

# Add info for type strain
phylo_md$`Patient ID.x`[which(str_detect(phylo_md$`Strain ID`, "AFB"))] <- "Afribiota"
phylo_md$Source.x[which(str_detect(phylo_md$`Strain ID`, "AFB"))] <- "Stool"

phylo_md$classification <- str_remove_all(phylo_md$classification,
                                          "d__Bacteria;p__Bacillota;c__Bacilli;o__Lactobacillales;f__Streptococcaceae;g__Streptococcus;s__")
# Create heatmap data
heatmap_data <- phylo_md[, c("classification", "Source.x", "Patient ID.x")]
colnames(heatmap_data)[1] <- "Taxonomy"
colnames(heatmap_data)[3] <- "PatientID"
rownames(heatmap_data) <- phylo_md$`Strain ID`
# Define colors
source_colors <- c(
  "Blood" = "#cc0000",
  "Duodenum" = "#8fce00",
  "Gastric aspirate" = "#6aa84f",
  "Saliva" = "#3d85c6",
  "Stool" = "#ce7e00"
)

patientID_colors <- c(
  "AFB477"="#fafabb", "AFB570"="#999900", "AGN005"="#9090bb",
  "AGN002"="#e06666", "AGN004"="#f6b26b", "AGN008"="#ffd966", "AGN010"="#b6d7a8",
  "AGN014"="#741b47", "AGN015"="#8fce20", "AGN019"="#ffab99",
  "AGN022"="#9fc5e8", "AGN025"="#faa120", "AGN026"="#99aacc", "AGN028" = "#cc5010",
  "AGN030" = "#aaffff", "AGN031" = "#109090", "AGN035" = "#cc99ac", "AGN043" = "#505050",
  "AGN034"="#c27ba0", "AGN036"="#990000", "AGN039"="#bf9000", "AGN045"="#134f5c",
  "AGN018"="#a2c4c9", "AGN037"="#b45f06", "AGN044"="#caca90", "AGN047" ="#0099cc",
  "AGN041"="#38761d", "AGN046"="#351c75", "AGN032"="#8e7cc3", "Type_strain"="#cc0000"
)

strain_colors <- c(
  "Streptococcus salivarius" = "#f11316",
  "Streptococcus salivarius_D" = "#cc0099",
  "Streptococcus parasanguinis" = "#999900",
  "Streptococcus parasanguinis_K" = "#999900"
)

combined_colors <- c(strain_colors, source_colors, patientID_colors)
all_breaks <- names(combined_colors)

# Plot tree

p_tree <- ggtree(tree_rooted, branch.length = "none") +
  geom_tiplab(size=2)

p_tree <- gheatmap(p_tree, heatmap_data, offset = 1.5, width = 0.1, font.size = 2,
         colnames = TRUE, colnames_position = "top", colnames_angle = 90, colnames_offset_y = 2,
         legend_title = "") +
  scale_fill_manual(values = combined_colors, breaks = all_breaks) +
    scale_x_ggtree()

#Combine this with ANI heatmap - showing ANI distance > 99.9% = sharing events
pdf("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Figures/Phylogeny_Isolates.pdf", height = 10, width = 12)
p_tree
dev.off()

# Remove parasanguinis
tree_rooted_pruned <- drop.tip(tree_rooted, c("SAT340","SAT513","SAT551", "SAT894"))

p_tree_subs <- ggtree(tree_rooted_pruned) +
  geom_tiplab(size = 2) +
  #geom_treescale(x = 0, y = 0, width = 0.01, fontsize = 3) +
  geom_text2(
    aes(subset = !isTip, label = label),
    size = 2, vjust = -0.5, hjust = 2) +
  theme_tree2() +
  scale_x_continuous(name = "Substitutions per site")

p_tree_subs
pdf("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/Figures/Phylogeny_Isolates_SubRate.pdf", height = 10, width = 12)
p_tree_subs
dev.off()


```

## Transition rate analysis
```{r}
tree_trans <- read.iqtree("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Phylogeny/May06/SAT_isolates.treefile")
tree_trans <- ape::as.phylo(tree_trans)

tree_trans <- ape::root(tree_trans, node = 161)
#Drop tip of unwanted isolates
tree_trans <- drop.tip(tree_trans, c("SAT340","SAT513","SAT551", "SAT894", "AFB477", "AFB570", "SsalTypeStrain"))
ordered_tips <- tree_trans$tip.label

hist(tree_trans$edge.length)
sum(tree_trans$edge.length)

# Load Traits
phylo_md <- phylo_md <- read.csv2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Phylogeny/May06/SAT_isolates_md.csv")
colnames(phylo_md)[2] <- "StrainID"
colnames(phylo_md)[4] <- "PatientID"
colnames(phylo_md)[5] <- "Source"
phylo_md$Source[which(phylo_md$Source=="Stool")] <- "Feces"
phylo_md$Source[which(phylo_md$Source=="Gastric aspirate")] <- "Gastric"


trait_source <- phylo_md[,c("StrainID", "Source")]
rownames(trait_source ) <- trait_source$StrainID
trait_source  <- trait_source[ordered_tips, "Source"]
names(trait_source ) <- tree_trans$tip.label 

trait_indiv  <- phylo_md[,c("StrainID", "PatientID")]
rownames(trait_indiv) <- trait_indiv$StrainID
trait_indiv <- trait_indiv[ordered_tips, "PatientID"]
names(trait_indiv) <- tree_trans$tip.label 
```

```{r}
p <- ggtree(tree_trans) +
  geom_tiplab(align = TRUE, size = 2.5)

# Define colors
pal_source_colors <- c(
  #"Blood" = "#cc0000",
  "Duodenum" = "#EDE692",
  "Gastric" = "#BBDEEC",
  "Saliva" = "#9AB9C2",
  "Feces" = "#DCC04B"
)

# First heatmap: Source — very close to the tree
p <- gheatmap(p,
              data.frame(Source = trait_source),
              offset = 0.003,  
              width = 0.06, 
              colnames = TRUE) +
  scale_fill_manual(values = pal_source_colors, name = "Source")

# New fill scale for second trait
p <- p + ggnewscale::new_scale_fill()

pal_patientID_colors <- c(
  "AGN005"="#9090bb",
  "AGN002"="#e06666", "AGN004"="#f6b26b", "AGN008"="#ffd966", "AGN010"="#b6d7a8",
  "AGN014"="#741b47", "AGN015"="#8fce20", "AGN019"="#ffab99",
  "AGN022"="#9fc5e8", "AGN025"="#faa120", "AGN026"="#99aacc", "AGN028" = "#cc5010",
  "AGN030" = "#aaffff", "AGN031" = "#109090", "AGN035" = "#cc99ac", "AGN043" = "#505050",
  "AGN034"="#c27ba0", "AGN036"="#990000", "AGN039"="#bf9000", "AGN045"="#134f5c",
  "AGN018"="#a2c4c9", "AGN037"="#b45f06", "AGN044"="#caca90", "AGN047" ="#0099cc",
  "AGN041"="#38761d", "AGN046"="#351c75", "AGN032"="#8e7cc3"
)

# Second heatmap: Individual — immediately next to Source
p <- gheatmap(p,
              data.frame(PatientID = trait_indiv),
              offset = 0,  
              width = 0.06, 
              colnames = TRUE) +
  scale_fill_manual(values = pal_patientID_colors, name = "Patient ID")

pdf("Figures/Phylo_Traits.pdf")
print(p)
dev.off()
```

```{r}
?fitDiscrete
source_ER_model_fit <-  fitDiscrete(tree_trans,trait_source , model="ER")
Indi_ER_model_fit <-  fitDiscrete(tree_trans,trait_indiv, model="ER")

source_ER_model_fit
Indi_ER_model_fit 

# Q matrix transitions per unit branch length

# by hand we expect 30 transition for inidvidual, and XXX for source 

Transition_source <- source_ER_model_fit$res[1,"q12"]
Transition_indiv <- Indi_ER_model_fit$res[1,"q0102"]

Transition_source
Transition_indiv
Transition_source/Transition_indiv # 60

# expected number of transitions 
Transition_source*sum(tree_trans$edge.length)
Transition_indiv*sum(tree_trans$edge.length)

# the compartment model has very low loglik (10^-200) and is thus not valid - 
# this is likely due to very short branches where compartment transition occurs but the model is enable to capture properly

#######################
## ultrametric trees
#######################

tree_delta_U <- chronos(tree_trans,model = "relaxed")

# using a autocorrelated model 
# tree_delta_U <- chronos(tree_trans, model = "correlated") 
# leads to ratio = 4500 - so we are using here a highly conservative estimate 

# --- Plot ---
pU <- ggtree(tree_delta_U) +
  geom_tiplab(align = TRUE, size = 2.5)

# First heatmap: Source — very close to the tree
pU <- gheatmap(pU,
              data.frame(Source = trait_source),
              offset = 0.04,  # closer to tree
              width = 0.06,   # narrower column
              colnames = TRUE) +
  scale_fill_manual(values = pal_source_colors, name = "Source")

# New fill scale for second trait
pU <- pU + ggnewscale::new_scale_fill()

# Second heatmap: Individual — immediately next to Source
pU <- gheatmap(pU,
              data.frame(PatientID = trait_indiv),
              offset = 0.1,  # right after Source
              width = 0.06,   # match width to Source
              colnames = TRUE) +
  scale_fill_manual(values = pal_patientID_colors, name = "Patient ID")

ggsave(filename = "Figures/Phylo_ultrametric_relaxed_Traits.pdf")
print(pU)
dev.off()

source_U_ER_model_fit <-  fitDiscrete(tree_delta_U,trait_source , model="ER")
Indi_ER_U_model_fit <-  fitDiscrete(tree_delta_U,trait_indiv, model="ER")

source_U_ER_model_fit
Indi_ER_U_model_fit
# models hows OK loglik 

obs_Transition_source_U <- source_U_ER_model_fit$res[1,"q12"]
obs_Transition_indiv_U <- Indi_ER_U_model_fit$res[1,"q0102"]

obs_Transition_source_U
obs_Transition_indiv_U
obs_Transition_source_U/obs_Transition_indiv_U

# expected number of transitions 
obs_Transition_source_U*sum(tree_delta_U$edge.length)
obs_Transition_indiv_U*sum(tree_delta_U$edge.length)

# Estimation of random transition ratio (shuffling isolates labels on trait data)

random_fits <- function(x){
  print(x)
  trait_sourceR <- trait_source
  names(trait_sourceR) <- sample(names((trait_sourceR)))
  
  trait_indivR <- trait_indiv
  names(trait_indivR ) <- sample(names((trait_indivR)))
  
  
  source_U_ER_model_fit <-  fitDiscrete(tree_delta_U,trait_sourceR  , model="ER")
  Indi_ER_U_model_fit <-  fitDiscrete(tree_delta_U,trait_indivR, model="ER")
  Transition_source_U <- source_U_ER_model_fit$res[1,"q12"]
  Transition_indiv_U <- Indi_ER_U_model_fit$res[1,"q0102"]
  
  return(c(Transition_source_U,
  Transition_indiv_U,
  Transition_source_U/Transition_indiv_U))
  }

random_res <- t(sapply(FUN = random_fits,X = 1:499))
random_res <- as_tibble(random_res)
colnames(random_res) <- c("Transition_rate_Source", "Transition_rate_indiv","Ratio_Transition_Source/Indiv")
random_res
saveRDS(random_res,"500Random_ratio_relaexdclokU_ERmodel.rds")

hist_random <- hist(random_res$`Ratio_Transition_Source/Indiv`, n=50,
     main=NULL,
     
     xlab= "Ratio Transition rate Source/ Transition rate Indiv",
     xlim = c(0,max(random_res$`Ratio_Transition_Source/Indiv`,obs_Transition_source_U/obs_Transition_indiv_U))) 

median(random_res$`Ratio_Transition_Source/Indiv`)
mean(random_res$`Ratio_Transition_Source/Indiv`)
pval = sum(obs_Transition_source_U/obs_Transition_indiv_U<random_res$`Ratio_Transition_Source/Indiv`)/1000
pval # 0.019
```



# Roary pangenome
```{r}
pangenome_genes_df <- read.csv2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/pangenome/gene_presence_absence.csv", sep = ",")
pangenome_genes <- pangenome_genes_df[,-102]
pangenome_genes[pangenome_genes == ""] <- NA
pangenome_genes <- pangenome_genes %>%
  select(-(2:14)) %>% 
  mutate(across(-Gene, ~ ifelse(is.na(.), 0, 1))) # Change to presence (1) and absence (0)
pangenome_genes <- pangenome_genes %>%
  rowwise() %>%
  mutate(presence = sum(c_across(-Gene))) %>%
  ungroup()
pangenome_genes_sorted <- pangenome_genes %>%
  arrange(desc(presence))
ggplot(pangenome_genes, aes(x = presence)) +
  geom_histogram(binwidth = 1, fill = "#4F86C6", alpha = 0.7) +
  labs(x = "Number of genomes",
       y = "Number of genes",
       title = "Pangenome Frequency Plot") +
  theme_minimal()
```


pie chart with core, soft, shell, cloud gene
```{r}
total_strains <- 87

pangenome_genes_piechart <- pangenome_genes %>% select(-c("Gene", "presence"))
core <- sum(rowSums(pangenome_genes_piechart) == total_strains)
softcore <- sum(rowSums(pangenome_genes_piechart) >= total_strains * 0.95 & rowSums(pangenome_genes_piechart) < total_strains)
shell <- sum(rowSums(pangenome_genes_piechart) >= total_strains * 0.15 & rowSums(pangenome_genes_piechart) < total_strains * 0.95)
cloud <- sum(rowSums(pangenome_genes_piechart) < total_strains * 0.15)

pangenome_df <- data.frame(
  Category = c("Core", "Soft-core", "Shell", "Cloud"),
  Count = c(core, softcore, shell, cloud)
)

pangenome_df <- pangenome_df %>%
  mutate(
    Percent = Count / sum(Count) * 100,
    Label = paste0(Category, "\n", round(Percent, 1), "% (", Count, " genes)")
  )

pangenome_df <- pangenome_df %>%
  arrange(desc(Category)) %>%
  mutate(
    ypos = cumsum(Count) - 0.5 * Count
  )
library(ggrepel)
ggplot(pangenome_df, aes(x = "", y = Count, fill = Category)) +
  geom_bar(stat = "identity", width = 1, color = "white") +
  coord_polar("y", start = 0) +
  theme_void() +
  scale_fill_brewer(palette = "Paired") +
  geom_text_repel(
    aes(y = ypos, label = Label),
    size = 4,
    nudge_x = 1,  # pushes labels outside
    show.legend = FALSE
  )

```

Heatmap with tree

```{r}
tree_rooted_roary <- drop.tip(tree_rooted, c("SAT340","SAT513","SAT551", "SAT894", "SsalTypeStrain", "AFB477", "AFB570"))
# Basic ggtree
p_roary_tree <- ggtree(tree_rooted_roary, branch.length = "none") +
  geom_tiplab(size = 2)

# Heatmap with tree
tip_order <- p_roary_tree$data %>%
  filter(isTip) %>%
  arrange(y) %>%
  pull(label)

pangenome_genes_hm <- pangenome_genes %>%
  filter(presence < 87) %>%
  filter(presence > 0.15*total_strains)
pangenome_genes_hm_label <- pangenome_genes_hm$Gene
pangenome_genes_hm <- pangenome_genes_hm[,-c(1,89)]
setdiff(colnames(pangenome_genes_hm), tip_order)
# Reorder columns
pangenome_genes_hm <- pangenome_genes_hm %>%
  select(all_of(tip_order))
heatmap_data <- as.data.frame(t(pangenome_genes_hm))
colnames(heatmap_data) <- paste0("Gene_", seq_len(ncol(heatmap_data))) # optional: generic gene names
heatmap_data <- tibble::rownames_to_column(heatmap_data, var = "strain")
heatmap_data <- as.data.frame(pangenome_genes_hm)
rownames(heatmap_data) <- pangenome_genes_hm_label
heatmap_data <- t(heatmap_data)




# Add heatmap
p_heatmap <- gheatmap(p_roary_tree,
                      heatmap_data,
                      offset = 12,     # space between tree and heatmap
                      width = 15,      # width of heatmap
                      colnames = FALSE)


p_heatmap
```

```{r}
tree_rooted_roary <- drop.tip(tree_rooted, c("SAT340","SAT513","SAT551", "SAT894", "SsalTypeStrain", "AFB477", "AFB570"))

# Basic ggtree
p_roary_tree <- ggtree(tree_rooted_roary, branch.length = "none") +
  geom_tiplab(size = 2)

# Heatmap with tree
tip_order <- p_roary_tree$data %>%
  filter(isTip) %>%
  arrange(y) %>%
  pull(label)

pangenome_genes_hm <- pangenome_genes %>%
  filter(presence < total_strains * 0.15) %>%
  filter(presence > total_strains * 0.02)
pangenome_genes_hm_label <- pangenome_genes_hm$Gene
pangenome_genes_hm <- pangenome_genes_hm[,-c(1,89)]
setdiff(colnames(pangenome_genes_hm), tip_order)
# Reorder columns
pangenome_genes_hm <- pangenome_genes_hm %>%
  select(all_of(tip_order))
heatmap_data <- as.data.frame(pangenome_genes_hm)
rownames(heatmap_data) <- pangenome_genes_hm_label
heatmap_data <- t(heatmap_data)


# Basic ggtree
p_roary_tree <- ggtree(tree_rooted_roary, branch.length = "none") +
  geom_tiplab(size = 2)

# Add heatmap
p_heatmap_cloud <- gheatmap(p_roary_tree,
                      heatmap_data,
                      offset = 12,     # space between tree and heatmap
                      width = 15,      # width of heatmap
                      colnames = FALSE)

p_heatmap_cloud

```


```{r}
pangenome_diff <- pangenome_genes
pangenome_diff <- t(pangenome_diff)
colnames(pangenome_diff) <- pangenome_diff[1,]
pangenome_diff <- pangenome_diff[-c(1,89),]
pangenome_diff <- as.data.frame(pangenome_diff)
pangenome_diff <- pangenome_diff %>% mutate(StrainID = rownames(pangenome_diff))

sharing_clusters_df <- sharing_clusters 

sharing_clusters_df <- sharing_clusters_df %>% mutate(SharedCluster = rownames(sharing_clusters_df))
sharing_clusters_df <- sharing_clusters_df %>% mutate(Indiv_Source = str_c(PatientID, "_", Source))


sharing_clusters_df <- sharing_clusters_df %>%
  group_by(Strain_cluster, PatientID) %>%
  mutate(
    unique_sources = n_distinct(Source),
    SharingStatus = ifelse(unique_sources > 1, "shared", "notshared")
  ) %>%
  ungroup() %>%
  select(-unique_sources)


pangenome_diff <- merge(sharing_clusters_df, pangenome_diff, by = "StrainID")
```

Overall shared vs not shared
```{r}
pangenome_diff_long <- pivot_longer(pangenome_diff, cols = c(5:7429), names_to = "Gene", values_to = "Presence_Absence")
pangenome_diff_long <- pangenome_diff_long %>% mutate(Indiv_Source = str_c(PatientID, "_", Source))

pangenome_diff_all <- pangenome_diff_long %>%
  group_by(Strain_cluster, PatientID) %>%
  mutate(
    unique_sources = n_distinct(Source),
    SharingStatus = ifelse(unique_sources > 1, "shared", "notshared")
  ) %>%
  ungroup() %>%
  select(-unique_sources)

# oral to stool
pangenome_diff_safe <- pangenome_diff_long %>%
  filter(Source %in% c("Saliva", "Stool")) %>%
  group_by(Strain_cluster, PatientID) %>%
  mutate(
    unique_sources = n_distinct(Source),
    SharingStatus = ifelse(unique_sources > 1, "shared", "notshared")
  ) %>%
  ungroup() %>%
  select(-unique_sources)
# Modify SAT 463 that was shared based on inStrain
pangenome_diff_safe$SharingStatus[which(pangenome_diff_safe$StrainID=="SAT463")] <- "shared"

pangenome_safe_test <- pangenome_diff_safe %>%
  group_by(SharingStatus, Gene)

```

Table for scoary
```{r}
traits_df_all <- as.data.frame(sharing_clusters)
traits_df_all <- traits_df_all %>%
  group_by(Strain_cluster, PatientID) %>%
  mutate(
    unique_sources = n_distinct(Source),
    SharingStatus = ifelse(unique_sources > 1, "shared", "notshared")
  ) %>%
  ungroup() %>%
  select(-unique_sources)
traits_df_all$SharingStatus[which(traits_df_all$StrainID=="SAT863")] <- "shared"
traits_df_all$SharingStatus[which(traits_df_all$StrainID=="SAT463")] <- "shared"
traits_df_all$SharingStatus[which(traits_df_all$StrainID=="SAT749")] <- "shared"
traits_df_all <- as.data.frame(traits_df_all)
rownames(traits_df_all) <- traits_df_all$StrainID
traits_df_all <- traits_df_all %>% select(SharingStatus)
traits_df_all$SharingStatus[which(traits_df_all$SharingStatus=="shared")] <- 1
traits_df_all$SharingStatus[which(traits_df_all$SharingStatus=="notshared")] <- 0
write.csv(traits_df_all, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/pangenome/traits.csv", sep = ",")


traits_df <- sharing_clusters 
traits_df <- traits_df %>%
  filter(Source %in% c("Saliva", "Stool")) %>%
  group_by(Strain_cluster, PatientID) %>%
  mutate(
    unique_sources = n_distinct(Source),
    SharingStatus = ifelse(unique_sources > 1, "shared", "notshared")
  ) %>%
  ungroup() %>%
  select(-unique_sources)
# Found with inStrain saliva to stool
traits_df$SharingStatus[which(traits_df$StrainID=="SAT463")] <- "shared"
traits_df <- as.data.frame(traits_df)
rownames(traits_df) <- traits_df$StrainID

traits_df_sa <- traits_df %>%
  filter(Source=="Saliva") %>%
  select(SharingStatus)
traits_df_sa$SharingStatus[which(traits_df_sa$SharingStatus=="shared")] <- 1
traits_df_sa$SharingStatus[which(traits_df_sa$SharingStatus=="notshared")] <- 0
write.csv(traits_df_sa, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/pangenome/traits_sa.csv", sep = ",")
strains_to_keep <- rownames(traits_df_sa)
metadata_cols <- colnames(pangenome_genes_df)[1:14]
strain_cols <- intersect(colnames(pangenome_genes_df), strains_to_keep)
cols_to_keep <- c(metadata_cols, strain_cols)
f_pangenome_genes_df_sa <- pangenome_genes_df %>% select(all_of(cols_to_keep))
write.csv(f_pangenome_genes_df_sa, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/pangenome/gene_presence_absence_filtered.csv", row.names = FALSE)

traits_df_fe <- traits_df %>%
  filter(Source=="Stool") %>%
  select(SharingStatus)
traits_df_fe$SharingStatus[which(traits_df_fe$SharingStatus=="shared")] <- 1
traits_df_fe$SharingStatus[which(traits_df_fe$SharingStatus=="notshared")] <- 0
write.csv(traits_df_fe, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/pangenome/traits_fe.csv", sep = ",")
strains_to_keep <- rownames(traits_df_fe)
metadata_cols <- colnames(pangenome_genes_df)[1:14]
strain_cols <- intersect(colnames(pangenome_genes_df), strains_to_keep)
cols_to_keep <- c(metadata_cols, strain_cols)
f_pangenome_genes_df_fe <- pangenome_genes_df %>% select(all_of(cols_to_keep))
write.csv(f_pangenome_genes_df_fe, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/pangenome/gene_presence_absence_filtered_fe.csv", row.names = FALSE)

traits_df <- traits_df %>%
  select(SharingStatus)
traits_df$SharingStatus[which(traits_df$SharingStatus=="shared")] <- 1
traits_df$SharingStatus[which(traits_df$SharingStatus=="notshared")] <- 0
write.csv(traits_df, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/pangenome/traits_safe.csv", sep = ",")
strains_to_keep <- rownames(traits_df)
metadata_cols <- colnames(pangenome_genes_df)[1:14]
strain_cols <- intersect(colnames(pangenome_genes_df), strains_to_keep)
cols_to_keep <- c(metadata_cols, strain_cols)
f_pangenome_genes_df <- pangenome_genes_df %>% select(all_of(cols_to_keep))
write.csv(f_pangenome_genes_df, "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/pangenome/gene_presence_absence_filtered.csv", row.names = FALSE)

```

```{r}
header_genes <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/pangenome/headers_two_columns.tsv", header = FALSE)
header_genes <- str_split_fixed(header_genes$V1, " ", 2)
header_genes <- as.data.frame(header_genes)
colnames(header_genes)[1] <- "EggnogID"
colnames(header_genes)[2] <- "Gene"

eggnog_annot <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/pangenome/EggNOG_pan_genome_annotations.emapper.annotations", header = TRUE, skip = 4)

eggnog_annot <- merge(header_genes, eggnog_annot, by.x = "EggnogID", by.y = "X.query")

# results all
scoary_all <- read.csv2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/pangenome/SharingStatus_04_07_2025_1407.results.csv", sep = ",")

scoary_all <- merge(eggnog_annot, scoary_all, by = "Gene")
scoary_all <- scoary_all %>%
  filter(Benjamini_H_p < 0.05)

# Heatmap with tree
tip_order <- p_roary_tree$data %>%
  filter(isTip) %>%
  arrange(y) %>%
  pull(label)

pangenome_genes_hm <- pangenome_genes %>%
  filter(Gene %in% scoary_all$Gene)
pangenome_genes_hm_label <- pangenome_genes_hm$Gene
pangenome_genes_hm <- pangenome_genes_hm[,-c(1,89)]
setdiff(colnames(pangenome_genes_hm), tip_order)
pangenome_genes_hm <- pangenome_genes_hm %>%
  select(all_of(tip_order))
heatmap_data <- as.data.frame(pangenome_genes_hm)
rownames(heatmap_data) <- pangenome_genes_hm_label
heatmap_data <- t(heatmap_data)
p_roary_tree <- ggtree(tree_rooted_roary, branch.length = "none") +
  geom_tiplab(size = 2)

p_heatmap <- gheatmap(
  p_roary_tree,
  heatmap_data,
  offset = 5,
  width = 6,
  colnames = TRUE,
  colnames_angle = 90, font.size = 2,
  hjust = -2)
p_heatmap
```

```{r}
# results all
scoary_safe <- read.csv2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/Isolates/pangenome/SharingStatus_04_07_2025_1342.results_fe.csv", sep = ",")
scoary_safe$Description
scoary_safe <- merge(eggnog_annot, scoary_safe, by = "Gene")
scoary_safe <- scoary_safe %>%
  filter(Benjamini_H_p < 0.05)

# Heatmap with tree
tip_order <- p_roary_tree$data %>%
  filter(isTip) %>%
  arrange(y) %>%
  pull(label)

pangenome_genes_hm <- pangenome_genes %>%
  filter(Gene %in% scoary_safe$Gene)
pangenome_genes_hm_label <- pangenome_genes_hm$Gene
pangenome_genes_hm <- pangenome_genes_hm[,-c(1,89)]
setdiff(colnames(pangenome_genes_hm), tip_order)
pangenome_genes_hm <- pangenome_genes_hm %>%
  select(all_of(tip_order))
heatmap_data <- as.data.frame(pangenome_genes_hm)
rownames(heatmap_data) <- pangenome_genes_hm_label
heatmap_data <- t(heatmap_data)
p_roary_tree <- ggtree(tree_rooted_roary, branch.length = "none") +
  geom_tiplab(size = 2)

p_heatmap <- gheatmap(
  p_roary_tree,
  heatmap_data,
  offset = 5,
  width = 6,
  colnames = TRUE,
  colnames_angle = 90, font.size = 2,
  hjust = -2)


p_heatmap
```

Within individual adaptation
```{r}
sharing_agn045 <- sharing_clusters %>%
  mutate(ClusterAGN045 = "")
cluster_agn045 <- c("SAT944", "SAT758", "SAT757", "SAT941", "SAT755", "SAT940")
sharing_agn045$ClusterAGN045[which(sharing_agn045$StrainID %in% cluster_agn045)] <- "Cluster_045"
`%nin%` = Negate(`%in%`)
sharing_agn045$ClusterAGN045[which(sharing_agn045$StrainID %nin% cluster_agn045)] <- "NotCluster_045"
cluster_NOTagn045 <- sharing_agn045$StrainID[which(sharing_agn045$ClusterAGN045=="NotCluster_045")]

pangenome_45 <- pangenome_genes[,-89]
pangenome_45$Cluster45_mean <- rowMeans(pangenome_45[, cluster_agn045])
pangenome_45$ClusterNot45_mean <- rowMeans(pangenome_45[, cluster_NOTagn045])

# Filter for genes highly present in Cluster A, absent in B
unique_to_45 <- pangenome_45 %>%
  filter(Cluster45_mean == 1, ClusterNot45_mean == 0)
unique_to_45 <- as.data.frame(unique_to_45)
rownames(unique_to_45) <- unique_to_45$Gene
unique_to_45 <- unique_to_45[,-c(1,89,90)]
unique_to_45 <- unique_to_45 %>%
  select(all_of(tip_order))
unique_to_45 <- t(unique_to_45)
# Add heatmap
p_heatmap_45 <- gheatmap(p_roary_tree,
                      unique_to_45,
                      offset = 12,     # space between tree and heatmap
                      width = 15,      # width of heatmap
                      colnames = FALSE)


eggnog_annot_45 <- eggnog_annot %>%
  filter(Gene %in% colnames(unique_to_45))

eggnog_annot_45$Description
```

```{r}
unique(eggnog_annot$COG_category)
```

```{r}
sharing_agn022 <- sharing_clusters %>%
  filter(PatientID == "AGN022")
  mutate(ClusterAGN022 = "")
cluster_agn022 <- c("SAT776", "SAT810")
cluster_NOTagn022 <- c("SAT774", "SAT808", "SAT727")

pangenome_22 <- pangenome_genes[,-89]
pangenome_22$Cluster22_mean <- rowMeans(pangenome_45[, cluster_agn022])
pangenome_22$ClusterNot22_mean <- rowMeans(pangenome_45[, cluster_NOTagn022])

# Filter for genes highly present in Cluster A, absent in B
unique_to_22 <- pangenome_22 %>%
  filter(Cluster22_mean == 1, ClusterNot22_mean == 0)
unique_to_22 <- as.data.frame(unique_to_22)
eggnog_annot_22 <- eggnog_annot %>%
  filter(Gene %in% unique_to_22$Gene)
eggnog_annot_22$Description
```



