---
title: "Satellite - MetaPhlAn profile analysis"
author: "Simon Yersin"
date: "2024-06-28"
output: html_document
---
Project: Satellite - Afribiota
The Satellite project samples include oral, gastric aspirates, duodenal aspirates and stools from stunted and non-stunted children age 2-5 years, from Bangui, Central African Republic.
DNA extraction was done with either promega or qiagen kit. Extracted DNA was send for shotgun metagenomic sequencing 12Gb at Novogene (NVUK2024042951).
Reads were cleaned locally using:
    fastp for quality trimming and adapter removal (v.0.23.4)
Then reads were transferred on the Urblauna cluster for human host reads removal using:
    bowtie2 aligner using the human reference genome (GRCh38, downloaded on the 21st of Nov. 2022) for host reads removal. (v.2.5.1)
Samples: Total = 129
    - Oral: 39
    - Gastric: 31
    - Duodenum: 15
    - Stool: 44
    
Taxonomic annotations and abundance profiling of the clean reads is done with MetaPhlAn4 (v. 4.1.1) and ChocoPhlAn database SGB_202307
    
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Load library
```{r, warning=FALSE, message=FALSE}
library(readxl)
library(phyloseq)
library(stringr)
library(dplyr)
library(RColorBrewer)
library(tidyr)
library(MicEco)

library(readr)

library(ggplot2)
library(usedist)
library(vegan)
library(rstatix)

library(circlize)
library(ComplexHeatmap)
library(microbiomeMarker)
library(SIAMCAT)
library(purrr)
library(DESeq2)
library(glue)
library(broom)
library(ANCOMBC)
```

# Load metadata
The metadata table was prepared in the R script: Satellite_metadata_cleaning.sh
```{r}
md_mpa <- as.data.frame(read_xlsx("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Metadata/Metadata_satellite_full.xlsx"))
rownames(md_mpa) <- md_mpa$id_modif

md_ps <- sample_data(md_mpa)
```

IMPORTANT METADATA
id = general sample ID (1429AGN###)
sexe = sex of the individual
nagem = age in months 
nheight = height of the individual
nweight = weight of the individual
statut_nut = nutritional status at recruitment (different than HAZ)
voie_accou = delivery (vaginal or C-section)
atb_suivi = antibiotic (yes or no)
ph_estomac
ph_intestin
crp = C-reactive protein
SeqBatch = sequencing batch (either VH or SY)

# Load MetaPhlAn profiles
Relative abundance
```{r}
tax_table <- read.delim("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Read_based_output/METAPHLAN/MERGED/Satellite_metaphlan_relab.txt", skip = 1)

## Create a taxonomy table
# Obtain a table with only the tax names
tax_names <- str_split_fixed(tax_table$clade_name, "\\|[p,c,o,f,g,s,t]__", 8)
tax_names <- as.data.frame(tax_names)
tax_names[,1] <- str_remove(tax_names[,1], "k__")
# Keeping only the rows with the complete taxonomy
tax_names <- tax_names[-(which(tax_names[,8] == "")),] 
# Changing colnames
colnames(tax_names) <- c("Kingdom", "Phylum", "Class","Order","Family","Genus","Species", "SGB")
# Adding the rownames
tax_names <- tax_names %>%
  mutate(ASV = paste("SGB", seq(1:nrow(tax_names)), sep=""))
rownames(tax_names) <- tax_names$ASV

asv_num <- tax_names$ASV
tax_names <- tax_names[,-9]
tax_names <- as.matrix(tax_names)
# We have now a taxonomy table, similar than the one obtain with DADA2
# Set as tax tablt phyloseq object
taxonomy_table <- tax_table(tax_names)

## Abundance table
abun_table <- cbind(str_split_fixed(tax_table$clade_name, "\\|[p,c,o,f,g,s,t]__", 8),tax_table)
abun_table <- as.data.frame(abun_table)
abun_table[,1] <- str_remove(abun_table[,1], "k__")
abun_table <- abun_table[-(which(abun_table[,8] == "")),] 
rownames(abun_table) <- asv_num
abun_table <- abun_table[,-c(1:9)]
# We have now a table with the column being each sample
# The row are the ASV assigned to a bacteria in the taxonomy table

# Clear the column name
colnames(abun_table) <- str_remove(colnames(abun_table), "_metaphlan")
# Set as otu_table phyloseq object
abun_table <- as.matrix(abun_table)
class(abun_table) <- "numeric"

abundance_table = otu_table(abun_table, taxa_are_rows = TRUE)

# We have now a tax table and an otu table, we can create our phyloseq
physeq <- phyloseq(taxonomy_table, abundance_table)
```

Absolute abundance
```{r}
tax_table_abs <- read.delim("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Read_based_output/METAPHLAN/MERGED/Satellite_metaphlan_abs.txt")

## Create a taxonomy table
# Obtain a table with only the tax names
tax_names_abs <- str_split_fixed(tax_table_abs$clade_name, "\\|[p,c,o,f,g,s,t]__", 8)
tax_names_abs <- as.data.frame(tax_names_abs)
tax_names_abs[,1] <- str_remove(tax_names_abs[,1], "k__")
# Keeping only the rows with the complete taxonomy
tax_names_abs <- tax_names_abs[-(which(tax_names_abs[,8] == "")),] 
# Changing colnames
colnames(tax_names_abs) <- c("Kingdom", "Phylum", "Class","Order","Family","Genus","Species", "SGB")
# Adding the rownames
tax_names_abs <- tax_names_abs %>%
  mutate(ASV = paste("SGB", seq(1:nrow(tax_names_abs)), sep=""))
rownames(tax_names_abs) <- tax_names_abs$ASV

asv_num_abs <- tax_names_abs$ASV
tax_names_abs <- tax_names_abs[,-9]
tax_names_abs <- as.matrix(tax_names_abs)
# We have now a taxonomy table, similar than the one obtain with DADA2
# Set as tax table phyloseq object
taxonomy_table_abs <- tax_table(tax_names_abs)

## Abundance table
abs_abun_table <- cbind(str_split_fixed(tax_table_abs$clade_name, "\\|[p,c,o,f,g,s,t]__", 8),tax_table_abs)
abs_abun_table <- as.data.frame(abs_abun_table)
abs_abun_table[,1] <- str_remove(abs_abun_table[,1], "k__")
abs_abun_table <- abs_abun_table[-(which(abs_abun_table[,8] == "")),] 
rownames(abs_abun_table) <- asv_num_abs
abs_abun_table <- abs_abun_table[,-c(1:10)]
# We have now a table with the column being each sample
# The row are the ASV assigned to a bacteria in the taxonomy table

# Clear the column name
colnames(abs_abun_table) <- str_remove(colnames(abs_abun_table), "_metaphlan_profile")
# Set as otu_table phyloseq object
abs_abun_table <- as.matrix(abs_abun_table)
class(abs_abun_table) <- "numeric"

abs_abundance_table = otu_table(abs_abun_table, taxa_are_rows = TRUE)

# We have now a tax table and an otu table, we can create our phyloseq
physeq_abs <- phyloseq(taxonomy_table_abs, abs_abundance_table)
```

Relative abundance in GTDB taxonomy
```{r}
tax_table_gtdb <- read.delim("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Read_based_output/METAPHLAN/MERGED/Satellite_metaphlan_gtdb.txt", skip = 1)

## Create a taxonomy table
# Obtain a table with only the tax names
tax_names_gtdb <- str_split_fixed(tax_table_gtdb$clade_name, "\\;[p,c,o,f,g,s,t]__", 7)
tax_names_gtdb <- as.data.frame(tax_names_gtdb)
tax_names_gtdb[,1] <- str_remove(tax_names_gtdb[,1], "d__")
# Keeping only the rows with the complete taxonomy
tax_names_gtdb <- tax_names_gtdb[-(which(tax_names_gtdb[,7] == "")),] 
# Changing colnames
colnames(tax_names_gtdb) <- c("Kingdom", "Phylum", "Class","Order","Family","Genus","Species")
# Adding the rownames
tax_names_gtdb <- tax_names_gtdb %>%
  mutate(GTDB = paste("TAX", seq(1:nrow(tax_names_gtdb)), sep=""))
rownames(tax_names_gtdb) <- tax_names_gtdb$GTDB

asv_num_gtdb <- tax_names_gtdb$GTDB
tax_names_gtdb <- tax_names_gtdb[,-8]
tax_names_gtdb <- as.matrix(tax_names_gtdb)
# We have now a taxonomy table, similar than the one obtain with DADA2
# Set as tax table phyloseq object
taxonomy_table_gtdb <- tax_table(tax_names_gtdb)

## Abundance table
abun_table_gtdb <- cbind(str_split_fixed(tax_table_gtdb$clade_name, "\\;[p,c,o,f,g,s,t]__", 7),tax_table_gtdb)
abun_table_gtdb <- as.data.frame(abun_table_gtdb)
abun_table_gtdb[,1] <- str_remove(abun_table_gtdb[,1], "d__")
abun_table_gtdb <- abun_table_gtdb[-(which(abun_table_gtdb[,7] == "")),] 
rownames(abun_table_gtdb) <- asv_num_gtdb
abun_table_gtdb <- abun_table_gtdb[,-c(1:8)]
# We have now a table with the column being each sample
# The row are the TAX assigned to a bacteria in the taxonomy table

# Clear the column name
colnames(abun_table_gtdb) <- str_remove(colnames(abun_table_gtdb), "_gtdb")
# Set as otu_table phyloseq object
abun_table_gtdb <- as.matrix(abun_table_gtdb)
class(abun_table_gtdb) <- "numeric"

abundance_table_gtdb = otu_table(abun_table_gtdb, taxa_are_rows = TRUE)

# We have now a tax table and an otu table, we can create our phyloseq
physeq_gtdb <- phyloseq(taxonomy_table_gtdb, abundance_table_gtdb)
```

# Phyloseq
Merging tables
```{r}
physeq <- merge_phyloseq(physeq, md_ps)
# Filter out AGN043DU (Duodenum but non-stunted based on haz score)
physeq <- subset_samples(physeq, id_modif != "AGN043DU")
physeq
# 128 samples
# 1998 taxa

physeq_abs <- merge_phyloseq(physeq_abs, md_ps)
physeq_abs <- subset_samples(physeq_abs, id_modif != "AGN043DU")
physeq_abs
# 128 samples
# 1998 taxa

physeq_gtdb <- merge_phyloseq(physeq_gtdb, md_ps)
physeq_gtdb <- subset_samples(physeq_gtdb, id_modif != "AGN043DU")
physeq_gtdb
# 128 samples
# 1640 taxa
```
Filtering out Eukaryota, and other unwanted taxa
```{r}
physeq_filtered <- physeq %>%
  subset_taxa(Family != "Mitochondria") %>%
  subset_taxa(Order != "Chloroplast") %>%
  subset_taxa(Kingdom != "Eukaryota")
# Removing taxa with abundance 0
physeq_filtered <- prune_taxa(taxa_sums(physeq_filtered) > 0, physeq_filtered)
physeq_ra <- transform_sample_counts(physeq_filtered, function(x) x / sum(x))
physeq_ra
# Removed 8 taxa by filtering

#Absolute abundance
physeq_abs_filtered <- physeq_abs %>%
  subset_taxa(Family != "Mitochondria") %>%
  subset_taxa(Order != "Chloroplast") %>%
  subset_taxa(Kingdom != "Eukaryota")
# Removing taxa with abundance 0
physeq_abs_filtered <- prune_taxa(taxa_sums(physeq_abs_filtered) > 0, physeq_abs_filtered)
# Removed 8 taxa by filtering


# GTDB taxonomy
# No taxa filtering for the GTDB taxonomy
physeq_gtdb <- transform_sample_counts(physeq_gtdb, function(x) x / sum(x))
```

# Palette
Color palette
```{r}
palbrew_info <- brewer.pal.info[brewer.pal.info$category == "qual", ] 
palbrew_all <- unlist(mapply(brewer.pal,palbrew_info$maxcolors,rownames(palbrew_info)))
# Set color palette
pal_2  <- palbrew_all[c(69,66,67,68)]

# Bar plots color palette
pal_saliva <- c("#377EB8","#8DD3C7","#4DAF4A","#FFFFB3","#FB8072","#80B1D3","#B3DE69",
                "#A65628","#F781BF","#E5C494","#FCCDE5","#7FC97F","#FDC086","#666666",
                "#c27ba0","#D95F02","#7570B3","#E7298A","#66A61E","#E6AB02","#6fa8dc")

pal_gastric <- c("#377EB8","#4DAF4A","#E41A1C","#80B1D3","#B3DE69","#A65628","#F781BF",
                "#9FCAF9","#18af11","#E5C494","#FCCDE5","#FFED6F","#7FC97F","#BEAED4",
                "#FDC086","#666666","#D95F02","#7570B3","#E7298A","#E6AB02","#6fa8dc")

pal_duodenum <- c("#377EB8","#4DAF4A","#FF7F00","#BC80BD","#FB8072","#80B1D3","#B3DE69",
                "#A65628","#F781BF","#FFFF33","#E5C494","#FCCDE5","#BEAED4","#FDC086",
                "#666666","#c27ba0","#D95F02","#7570B3","#E7298A","#E6AB02","#6fa8dc")

pal_feces <- c("#997EB8","#9CCCC9","#D41A1C","#A69972","#FFFFB3","#FA9909","#96C60E",
               "#F77199","#E9F199","#AAEEFF","#DF099A","#CACAD9","#B3DE99","#9fc5e8",
               "#666666","#ea9919","#e06666","#f1c232","#c27ba0","#93aa7d","#45818e")


# Palettes unused for now (stopped at diff abundance)
pal_1 <- palbrew_all[c(63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14)]
pal_3 <- palbrew_all[16:26]
pal_4 <- palbrew_all[c(63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14,
                       63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14)]
pal_5 <- palbrew_all[c(63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14,
                       63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14,
                       63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14,
                       63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14)]
pal_6 <- palbrew_all[c(63,46,69,47,48,16,72,66,67,50,71,53,61,70,74,1)]

pal_6 <- palbrew_all[c(61,63,65,66,67,68,69,60,72,73,71,1,29,70,30,46)]

pie(rep(1, length(pal_6)), col = pal_6 , main="") 
```
# Analysis
Samples
```{r}
md_ps_final <- as.matrix(sample_data(physeq_filtered))
md_ps_final <- as.data.frame(md_ps_final)

# Oral samples
n_distinct(md_ps_final$id[which(md_ps_final$body_site =="Saliva")]) # 39 samples
n_distinct(md_ps_final$id[which(md_ps_final$body_site =="Gastric")]) # 31 samples
n_distinct(md_ps_final$id[which(md_ps_final$body_site =="Duodenum")]) # 14 samples
n_distinct(md_ps_final$id[which(md_ps_final$body_site =="Feces")]) # 44 samples

sample_summary_table <- md_ps_final %>%
  group_by(body_site, Stunting_status) %>%
  summarise(n = n_distinct(id), .groups = "drop") %>%
  pivot_wider(names_from = Stunting_status, values_from = n, values_fill = 0) %>%
  mutate(Total = rowSums(across(where(is.numeric))))

# Pairs
n_distinct(intersect(md_ps_final$id[which(md_ps_final$body_site =="Gastric")],
          md_ps_final$id[which(md_ps_final$body_site =="Saliva")])) # 27 samples with both saliva and gastric
n_distinct(intersect(md_ps_final$id[which(md_ps_final$body_site =="Duodenum")],
          md_ps_final$id[which(md_ps_final$body_site =="Saliva")])) # 13 samples with both saliva and duodenal
n_distinct(intersect(md_ps_final$id[which(md_ps_final$body_site =="Saliva")],
          md_ps_final$id[which(md_ps_final$body_site =="Feces")])) # 39 samples with both saliva and fecal samples
n_distinct(intersect(md_ps_final$id[which(md_ps_final$body_site =="Gastric")],
          md_ps_final$id[which(md_ps_final$body_site =="Duodenum")])) # 11 samples with both gastric and duodenal samples
n_distinct(intersect(md_ps_final$id[which(md_ps_final$body_site =="Gastric")],
          md_ps_final$id[which(md_ps_final$body_site =="Feces")])) # 31 samples with both gastric and fecal samples
n_distinct(intersect(md_ps_final$id[which(md_ps_final$body_site =="Duodenum")],
          md_ps_final$id[which(md_ps_final$body_site =="Feces")])) # 15 with both duodenal and feces

n_distinct(intersect(intersect(md_ps_final$id[which(md_ps_final$body_site =="Gastric")],
          md_ps_final$id[which(md_ps_final$body_site =="Duodenum")]),
          md_ps_final$id[which(md_ps_final$body_site =="Saliva")])) # 12 samples with saliva, gastric and duodenal samples
n_distinct(intersect(intersect(md_ps_final$id[which(md_ps_final$body_site =="Gastric")],
          md_ps_final$id[which(md_ps_final$body_site =="Duodenum")]),
          md_ps_final$id[which(md_ps_final$body_site =="Feces")])) # 12 samples with fecal, gastric and duodenal samples
n_distinct(intersect(intersect(md_ps_final$id[which(md_ps_final$body_site =="Saliva")],
          md_ps_final$id[which(md_ps_final$body_site =="Duodenum")]),
          md_ps_final$id[which(md_ps_final$body_site =="Feces")])) # 14 samples with fecal, saliva and duodenal samples
n_distinct(intersect(intersect(md_ps_final$id[which(md_ps_final$body_site =="Gastric")],
          md_ps_final$id[which(md_ps_final$body_site =="Saliva")]),
          md_ps_final$id[which(md_ps_final$body_site =="Feces")])) # 27 samples with fecal, gastric and saliva samples


n_distinct(intersect(intersect(md_ps_final$id[which(md_ps_final$body_site =="Gastric")],
          md_ps_final$id[which(md_ps_final$body_site =="Saliva")]),
          intersect(md_ps_final$id[which(md_ps_final$body_site =="Duodenum")],
          md_ps_final$id[which(md_ps_final$body_site =="Feces")]))) # 11 samples with all body sites
```


Agglomeration at Species level
```{r}
physeq_sp <- tax_glom(physeq_ra, taxrank = "Species")
physeq_sp <- prune_taxa(taxa_sums(physeq_sp)> 0, physeq_sp)
physeq_sp
# 1906 species in the dataset

# Absolute abundance table
physeq_sp_abs <- tax_glom(physeq_abs_filtered, taxrank = "Species")
physeq_sp_abs <- prune_taxa(taxa_sums(physeq_sp_abs)> 0, physeq_sp_abs)
physeq_sp_abs
# 1906 species in the dataset

# GTDB relative abundance table
ps_gtdb_sp <- tax_glom(physeq_gtdb, taxrank = "Species")
ps_gtdb_sp <- prune_taxa(taxa_sums(ps_gtdb_sp)> 0, ps_gtdb_sp)
ps_gtdb_sp
# 1640 species in the dataset
```

```{r}
physeq_sp

prevdf_sp_all = apply(X= otu_table(physeq_sp),
               MARGIN = ifelse(taxa_are_rows(physeq_sp),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_all = data.frame(Prevalence = prevdf_sp_all,
                       Percentage_Prev = 100*prevdf_sp_all/128,
                       Mean_Abundance = 100*taxa_sums(physeq_sp)/128,
                       Total_Abundance = taxa_sums(physeq_sp),
                        tax_table(physeq_sp))
```

# Diversity and Richness
## Alpha-diversity
Observed index
counts the number of distinct species (or other taxonomic units) observed in the sample

Shannon index
Considers both species richness (the number of different species) and species evenness (the relative abundance of each species)
Higher Values: A higher Shannon index value indicates greater diversity within the community. This means there is a more even distribution of species abundances.
Lower Values: A lower Shannon index value suggests lower diversity, indicating that one or a few species dominate the community.

Simpson's index
accounts for both richness and evenness, providing a measure of diversity that considers the abundance distribution of species
probability that two individuals randomly selected from a sample will belong to the same species. It is a measure of dominance, with values ranging from 0 to 1
Lower Values: Indicate higher diversity (many species with similar abundances).
Higher Values: Indicate lower diversity (few species dominate)

Inverse Simpson's index
Provides an intuitive measure of diversity; higher values indicate higher diversity
Higher Values: Indicate higher diversity (many species with similar abundances)
Lower Values: Indicate lower diversity (few species dominate)

Index computation
In metagenomic sequencing we do not consider Chao1 or ACE index as they are taking into account rare species (singletons and doubletons) to estimate the "unobserved" species. We do not have singletons and doubletons using metaG.
```{r}
results_adiv <- estimate_richness(physeq_sp_abs)
results_adiv <- results_adiv %>%
  mutate(SampleID = rownames(results_adiv))
results_adiv <- merge(results_adiv, md_mpa, by.x = "SampleID", by.y="id_modif")

results_fe <- results_adiv %>%
  filter(body_site == "Feces")

results_ga <- results_adiv %>%
  filter(body_site == "Gastric")

results_du <- results_adiv %>%
  filter(body_site == "Duodenum")

results_sa <- results_adiv %>%
  filter(body_site == "Saliva")

# Wilcoxon tests
results_adiv_long <- results_adiv %>%
  pivot_longer(cols = c("Observed", "Shannon", "Simpson", "InvSimpson"),
               names_to = "Metric", values_to = "value")
```

Wilcoxon tests
```{r}
wilcox_adiv_results <- results_adiv_long %>%
  group_by(Metric) %>%
  pairwise_wilcox_test(
    value ~ body_site,
    p.adjust.method = "BH"
  )
write_csv2(wilcox_adiv_results, "Tables/AlphaDiv_WilcoxBH_results.csv")

```

Rarefaction curves
```{r}
physeq_sp_abs_ga <- subset_samples(physeq_sp_abs, body_site == "Gastric")
rare_curve_df_ga <- otu_table(physeq_sp_abs_ga)
class(rare_curve_df_ga) <- "matrix"
rare_curve_df_ga <- t(rare_curve_df_ga)
pdf("Figures/RareCurve_Gastric.pdf", width = 10)
rarecurve(rare_curve_df_ga, step=50, cex=0.1, lwd=1, ylab = "Species")
dev.off()

physeq_sp_abs_du <- subset_samples(physeq_sp_abs, body_site == "Duodenum")
rare_curve_df_du <- otu_table(physeq_sp_abs_du)
class(rare_curve_df_du) <- "matrix"
rare_curve_df_du <- t(rare_curve_df_du)
pdf("Figures/RareCurve_Duodenum.pdf", width = 10)
rarecurve(rare_curve_df_du, step=50, cex=0.1, lwd=1, ylab = "Species")
dev.off()

physeq_sp_abs_sa <- subset_samples(physeq_sp_abs, body_site == "Saliva")
rare_curve_df_sa <- otu_table(physeq_sp_abs_sa)
class(rare_curve_df_sa) <- "matrix"
rare_curve_df_sa <- t(rare_curve_df_sa)
pdf("Figures/RareCurve_Saliva.pdf", width = 10)
rarecurve(rare_curve_df_sa, step=50, cex=0.1, lwd=1, ylab = "Species")
dev.off()

physeq_sp_abs_fe <- subset_samples(physeq_sp_abs, body_site == "Feces")
rare_curve_df_fe <- otu_table(physeq_sp_abs_fe)
class(rare_curve_df_fe) <- "matrix"
rare_curve_df_fe <- t(rare_curve_df_fe)
pdf("Figures/RareCurve_Feces.pdf", width = 10)
rarecurve(rare_curve_df_fe, step=50, cex=0.1, lwd=1, ylab = "Species")
dev.off()
```


Tests
Histograms
```{r}
ggplot(results_adiv, aes(x = Observed)) +
  geom_histogram(binwidth = 100, falpha = 0.7) +
  labs(title = "Histogram of Observed Index",
       x = "Observed Index",
       y = "Frequency")

ggplot(results_adiv, aes(x = Shannon)) +
  geom_histogram(binwidth = 0.5, falpha = 0.7) +
  labs(title = "Histogram of Shannon Index",
       x = "Shannon Index",
       y = "Frequency")

ggplot(results_adiv, aes(x = Simpson)) +
  geom_histogram(binwidth = 0.02, falpha = 0.7) +
  labs(title = "Histogram of Simpson Index",
       x = "Simpson Index",
       y = "Frequency")

ggplot(results_adiv, aes(x = InvSimpson)) +
  geom_histogram(binwidth = 5, falpha = 0.7) +
  labs(title = "Histogram of InvSimpson Index",
       x = "InvSimpson Index",
       y = "Frequency")
```


ANOVA (analysis of variance) 
for categorical metadata with normal distribution
If significant with multiple categories in the explanatory variables: Tukey test
```{r}
# Body site
summary(aov(Shannon ~ body_site, data = results_adiv))
# 0.047
TukeyHSD(aov(Shannon ~ body_site, data = results_adiv))
# Not significant
summary(aov(Observed ~ body_site, data = results_adiv))
# 5.79e-16
TukeyHSD(aov(Observed ~ body_site, data = results_adiv))
# Significant for: Feces - Gastric/Saliva/Duodenum, and Saliva - Gastric
summary(aov(Simpson ~ body_site, data = results_adiv))
# 0.00129
TukeyHSD(aov(Simpson ~ body_site, data = results_adiv))
# Significant for saliva - feces and saliva - gastric
summary(aov(InvSimpson ~ body_site, data = results_adiv))
# 0.0441
TukeyHSD(aov(InvSimpson ~ body_site, data = results_adiv))
# Not significant

# Body site -> affect alpha diversity

# For all body sites:
# Stunting status (Stunted and Non-stunted)
summary(aov(Observed ~ Stunting_status, data = results_adiv))
summary(aov(Shannon ~ Stunting_status, data = results_adiv))
summary(aov(Simpson ~ Stunting_status, data = results_adiv))
summary(aov(InvSimpson ~ Stunting_status, data = results_adiv))
# Not significant, stunting status does not affect Observed diversity

# Stunting level
summary(aov(Observed ~ Stunting_level, data = results_adiv))
summary(aov(Shannon ~ Stunting_level, data = results_adiv))
summary(aov(Simpson ~ Stunting_level, data = results_adiv))
summary(aov(InvSimpson ~ Stunting_level, data = results_adiv))
# Nothing significant

#Presence of parasites
summary(aov(Shannon ~ giardiase + entamoeba + parasite_aut, data = results_adiv))
summary(aov(Shannon ~ entamoeba + parasite_aut, data = results_adiv))
summary(aov(Shannon ~ entamoeba, data = results_adiv))
summary(aov(Shannon ~ parasite_aut, data = results_adiv))
# Shannon (but not observed) diversity influenced by the presence of Entamoeba parasite and other parasite
summary(aov(Simpson ~ giardiase + entamoeba + parasite_aut, data = results_adiv))
# Presence of other parasite affect Simpson index
summary(aov(InvSimpson ~ giardiase + entamoeba + parasite_aut, data = results_adiv))
# InvSimpson diversity influenced by the presence of Entamoeba parasite and other parasite
summary(aov(Observed ~ giardiase + entamoeba + parasite_aut, data = results_fe))
# Presence of Entamoeba affect number of species observed in the feces

# Antibiotics
summary(aov(Observed ~ atb_suivi, data = results_adiv))
summary(aov(Shannon ~ atb_suivi, data = results_adiv))
summary(aov(Simpson ~ atb_suivi, data = results_adiv))
summary(aov(InvSimpson ~ atb_suivi, data = results_adiv))
# Nothing significant


# Inflammation
summary(aov(Observed ~ Gut_inflammation_AAT + Gut_inflammation_Calpro + Systemic_inflammation, data = results_adiv))
summary(aov(Shannon ~ Gut_inflammation_AAT + Gut_inflammation_Calpro + Systemic_inflammation, data = results_adiv))
summary(aov(Simpson ~ Gut_inflammation_AAT + Gut_inflammation_Calpro + Systemic_inflammation, data = results_adiv))
summary(aov(InvSimpson ~ Gut_inflammation_AAT + Gut_inflammation_Calpro + Systemic_inflammation, data = results_adiv))
# Nothing significant

# Leaky gut
summary(aov(Observed ~ Leaky_gut, data = results_adiv))
summary(aov(Shannon ~ Leaky_gut, data = results_adiv))
summary(aov(Simpson ~ Leaky_gut, data = results_adiv))
summary(aov(InvSimpson ~ Leaky_gut, data = results_adiv))
# Nothing significant
```

Correlation test and plot (for continuous variables)
Indicates the strength and direction of the linear relationship (values range from -1 to 1)
```{r}
cor.test(results_adiv$Observed, results_adiv$haz_cont, method = "pearson")
ggplot(results_adiv, aes(x = Observed, y = haz_cont)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE, color = "black") +
  labs(title = paste("Correlation:", round(cor.test(results_adiv$Observed, results_adiv$haz_cont, method = "pearson")$estimate, 2)),
       x = "Observed",
       y = "haz")

cor.test(results_adiv$Shannon, results_adiv$haz_cont, method = "pearson")
ggplot(results_adiv, aes(x = Shannon, y = haz_cont)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE, color = "black") +
  labs(title = paste("Correlation:", round(cor.test(results_adiv$Shannon, results_adiv$haz_cont, method = "pearson")$estimate, 2)),
       x = "Shannon",
       y = "haz")
# No correlations between haz and alpha diversity index

# Tested correlation between Shannon or Observed (both all data and only feces) with crp, calprotectin, ATT levels and Mannitol-lactilol ratio. 
# No correlations!

```

Linear model
Assess relationship between two continuous data 
Understand the effect of one variable on another
(For a categorical variable and a continuous index, aov then TUkey test instead of linear model)
```{r}
# No need as nothing was correlated
```

Plots
```{r}
md_alpha <- as.matrix(sample_data(physeq_sp_abs))
body_site_levels <- c("Saliva", "Gastric", "Duodenum", "Feces")
body_site_data <- data.frame(
  body_site = body_site_levels,
  bs_num = 1:length(body_site_levels))
md_alpha <- as.data.frame(md_alpha)
md_alpha <- dplyr::left_join(md_alpha, body_site_data, by= "body_site")
sample_data(physeq_sp_abs) <- cbind(sample_data(physeq_sp_abs), md_alpha$bs_num)
colnames(sample_data(physeq_sp_abs))[554] <- "bs_num"

plot_rich <- plot_richness(physeq_sp_abs, x= "bs_num",
                               measures=c("Observed", "Shannon", "Simpson", "InvSimpson"),
                               title = "Alpha diversity indexes by body sites",
                               color = "body_site") +
  scale_x_continuous(breaks=1:length(body_site_levels), labels=body_site_levels) +
  scale_color_manual(name= "Body sites", values = pal_2) +
  geom_boxplot(aes(x = bs_num, y = value), alpha = 0.1) +
  xlab("Body site") +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    plot.title = element_text(size = 14),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    strip.text = element_text(size = 14),
    legend.position = "bottom"
  )

pdf("Figures/AlphaDiv_METAG_satellite.pdf", width = 8, height = 6)
plot_rich
dev.off()
```

Test for seq batch in the feces
```{r}
plot_seqbatch <- physeq_sp_abs %>% subset_samples(body_site == "Feces") %>%
  plot_richness(x= "SeqBatch",
                               measures=c("Observed"),
                               title = "Number of taxa observed in the stool samples: split by sequencing batch",
                               color = "SeqBatch") +
  scale_color_manual(values = pal_2) +
  geom_boxplot() +
  xlab("Body site") +
  theme_bw() +
  theme(axis.text.x=element_text(angle = 45, hjust = .9))

#SeqBatch for feces
wilcox.test(Observed ~ SeqBatch, data = results_fe)
wilcox.test(Shannon ~ SeqBatch, data = results_fe)
wilcox.test(Simpson ~ SeqBatch, data = results_fe)
wilcox.test(InvSimpson ~ SeqBatch, data = results_fe)

ggsave(filename = "Figures/ObservedSeqBatch_METAG_satellite.svg", plot = plot_seqbatch, width = 6, height = 8)
```

Summary alpha-diversity
There are no difference in alpha diversity indexes except between body sites.
Notably, fecal samples have more observed species than samples from other body sites.
Stunting, inflammation, or leaky gut does not affect the alpha diversity.
However the presence of some intestinal parasites affect the alpha diversity (not plotted).


## Beta-diversity
All dataset
we first measure "distance" between samples using Bray-Curtis
Bray-Curtis is a dissimilarity measure:
  0 indicates complete similarity (same composition)
  1 indicates complete dissimilarity (no common species)
We first compute the distance (obtaining a distance matrix), then do a PERMANOVA to test for significance differences in community composition between groups
We can then plot the distance (or "dissimilarity") 
And finally, we can use a PCoA to reduce the complexity of multivariate data by representing it in a few (two) dimensions for vizualisation
```{r}
physeq_bdiv <- physeq_ra

# Bray-Curtis
dist.bc <- phyloseq::distance(physeq_bdiv, method = "bray")

# Permanova
meta_pcoa = as.matrix(sample_data(physeq_bdiv))
meta_pcoa <- as.data.frame(meta_pcoa)
permanova_all <- vegan::adonis2(dist.bc ~ body_site, data = meta_pcoa, permutations = 9999) # Significant

# Does stunting level explain some of the variation?
vegan::adonis2(dist.bc ~ body_site + Stunting_level,
                                data = meta_pcoa,permutations = 9999, by="terms") # Significant
# Gastric and Duodenum cluster together
# Saliva group away from gastric-duodenal
# structure of the anova table
permanova_all$`Pr(>F)` # 1e-04

pairwise_p <- numeric()
# Pairwise comparison
fe_sa_ado <- meta_pcoa %>% filter(body_site == "Feces" | body_site == "Saliva")
fe_sa_ado_dist <- dist_subset(dist.bc, fe_sa_ado$id_modif)
fe_sa_tado <- adonis2(fe_sa_ado_dist ~ body_site, data = fe_sa_ado,permutations = 1000)
pairwise_p["FE_SA"] <- fe_sa_tado$`Pr(>F)`[1]

fe_ga_ado <- meta_pcoa %>% filter(body_site == "Feces" | body_site == "Gastric")
fe_ga_ado_dist <- dist_subset(dist.bc, fe_ga_ado$id_modif)
fe_ga_tado <- adonis2(fe_ga_ado_dist ~ body_site, data = fe_ga_ado,permutations = 1000)
pairwise_p["FE_GA"] <- fe_ga_tado$`Pr(>F)`[1]

fe_du_ado <- meta_pcoa %>% filter(body_site == "Feces" | body_site == "Duodenum")
fe_du_ado_dist <- dist_subset(dist.bc, fe_du_ado$id_modif)
fe_du_tado <- adonis2(fe_du_ado_dist ~ body_site, data = fe_du_ado,permutations = 1000)
pairwise_p["FE_DU"] <- fe_du_tado$`Pr(>F)`[1]

ga_sa_ado <- meta_pcoa %>% filter(body_site == "Gastric" | body_site == "Saliva")
ga_sa_ado_dist <- dist_subset(dist.bc, ga_sa_ado$id_modif)
ga_sa_tado <- adonis2(ga_sa_ado_dist ~ body_site, data = ga_sa_ado,permutations = 1000)
pairwise_p["GA_SA"] <- ga_sa_tado$`Pr(>F)`[1]

du_ga_ado <- meta_pcoa %>% filter(body_site == "Duodenum" | body_site == "Saliva")
du_ga_ado_dist <- dist_subset(dist.bc, du_ga_ado$id_modif)
du_ga_tado <- adonis2(du_ga_ado_dist ~ body_site, data = du_ga_ado,permutations = 1000)
pairwise_p["DU_SA"] <- du_ga_tado$`Pr(>F)`[1]

ga_du_ado <- meta_pcoa %>% filter(body_site == "Gastric" | body_site == "Duodenum")
ga_du_ado_dist <- dist_subset(dist.bc, ga_du_ado$id_modif)
ga_du_tado <- adonis2(ga_du_ado_dist ~ body_site, data = ga_du_ado,permutations = 1000)
pairwise_p["GA_DU"] <- ga_du_tado$`Pr(>F)`[1]

pairwise_padj <- p.adjust(pairwise_p, method = "BH")
pairwise_padj
# Feces significant from everything
# Saliva significant from everything
# Gastric and duodenum not different from each other
write.csv(pairwise_padj, "Tables/BetaDiv_Pairwise_adonis2.csv")

# Plot distance intra and inter groups
dist.bc_df <- as.data.frame(as.matrix(dist.bc))
dist.bc_df <- dist.bc_df %>% 
  mutate(Row = rownames(dist.bc_df))
dist.bc_long <- dist.bc_df %>%
  pivot_longer(cols = -Row, names_to = "Column", values_to = "distances")
dist.bc_unique <- dist.bc_long %>%
  filter(Row != Column) %>%
  filter(Row < Column)
dist.bc_unique <- dist.bc_unique %>%
  mutate(Body_site_comp = "")
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "FE") & str_detect(dist.bc_unique$Column, "FE"))] <- "FE_FE"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "FE") & str_detect(dist.bc_unique$Column, "SA"))] <- "FE_SA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "FE") & str_detect(dist.bc_unique$Column, "GA"))] <- "FE_GA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "FE") & str_detect(dist.bc_unique$Column, "DU"))] <- "FE_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "SA") & str_detect(dist.bc_unique$Column, "SA"))] <- "SA_SA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "SA") & str_detect(dist.bc_unique$Column, "GA"))] <- "SA_GA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "SA") & str_detect(dist.bc_unique$Column, "DU"))] <- "SA_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "GA") & str_detect(dist.bc_unique$Column, "GA"))] <- "GA_GA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "GA") & str_detect(dist.bc_unique$Column, "DU"))] <- "GA_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "DU") & str_detect(dist.bc_unique$Column, "DU"))] <- "DU_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "FE") & str_detect(dist.bc_unique$Row, "FE"))] <- "FE_FE"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "FE") & str_detect(dist.bc_unique$Row, "SA"))] <- "FE_SA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "FE") & str_detect(dist.bc_unique$Row, "GA"))] <- "FE_GA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "FE") & str_detect(dist.bc_unique$Row, "DU"))] <- "FE_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "SA") & str_detect(dist.bc_unique$Row, "SA"))] <- "SA_SA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "SA") & str_detect(dist.bc_unique$Row, "GA"))] <- "SA_GA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "SA") & str_detect(dist.bc_unique$Row, "DU"))] <- "SA_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "GA") & str_detect(dist.bc_unique$Row, "GA"))] <- "GA_GA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "GA") & str_detect(dist.bc_unique$Row, "DU"))] <- "GA_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "DU") & str_detect(dist.bc_unique$Row, "DU"))] <- "DU_DU"

pdf("Figures/BetaDiv_distanceIntraInter.pdf")
dist.bc_unique %>%
  ggplot(aes(x=Body_site_comp, y=distances)) +
  geom_point() +
  geom_boxplot() +
  theme_bw()
dev.off()

#PCoA
test_PCoAbray <- ordinate(physeq_bdiv, "PCoA", distance = dist.bc)

physeq_bdiv@sam_data$Stunting_level <- factor(physeq_bdiv@sam_data$Stunting_level,
                                            levels = c("Normal_growth", "Mild_stunting", "Moderate_stunting", "Severe_stunting"),
                                            labels = c("Normal growth", "Mild stunting", "Moderate stunting", "Severe stunting"))

plot_ordi <- plot_ordination(physeq_bdiv, test_PCoAbray, type = "samples", color = "body_site", shape = "Stunting_level") +
  geom_point(size = 3) +
  theme_bw() +
  scale_shape_manual(values=c(15:18)) +
  scale_color_manual(values = pal_2) +
  labs(shape = "Nutritional status", color = "Body sites") +
  theme(
    axis.text.x = element_text(size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    plot.title = element_text(size = 14),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    strip.text = element_text(size = 14),
    )
# Clear separation between feces and other body sites

pdf("Figures/BetaDiv_METAG_satellite_axis12.pdf", width = 8, height = 6)
plot_ordi
dev.off()
#ggsave(filename = "Figures/PCoA_METAG_satellite.svg", plot = plot_ordi, width = 10, height = 8)

plot_ordi_13 <- plot_ordination(physeq_bdiv, test_PCoAbray, type = "samples", color = "body_site", shape = "Stunting_level", axes = c(1,3)) +
  geom_point(size = 3) +
  theme_bw() +
  scale_shape_manual(values=c(15:18)) +
  scale_color_manual(values = pal_2) +
  labs(shape = "Nutritional status", color = "Body sites") +
  theme(
    axis.text.x = element_text(size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    plot.title = element_text(size = 14),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    strip.text = element_text(size = 14),
    )

pdf("Figures/BetaDiv_METAG_satellite_axis13.pdf", width = 8, height = 6)
plot_ordi_13
dev.off()

plot_ordi_32 <- plot_ordination(physeq_bdiv, test_PCoAbray, type = "samples", color = "body_site", shape = "Stunting_level", axes = c(3,2)) +
  geom_point(size = 3) +
  theme_bw() +
  scale_shape_manual(values=c(15:18)) +
  scale_color_manual(values = pal_2) +
  labs(shape = "Nutritional status", color = "Body sites") +
  theme(
    axis.text.x = element_text(size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    plot.title = element_text(size = 14),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    strip.text = element_text(size = 14),
    )

pdf("Figures/BetaDiv_METAG_satellite_axis32.pdf", width = 8, height = 6)
plot_ordi_32
dev.off()
```

Covariates analysis using EnvFit
sexe, voie_accou, lait_enf, age_allait, eau_traitee, nb_repas, alim_malnut, haz, Stunting_level, Stunting_status, parasite, Gut_inflammation_AAT, Gut_inflammation_Calpro, Systemic_inflammation, ferritine_seuil, body_site

```{r}
# Test on 3 axis
vectorPCOA <- as.data.frame(test_PCoAbray$vectors[,1:3])
# Get metadata
META_envfit = as(sample_data(physeq_ra), "matrix")
meta_envfit <- as.data.frame(META_envfit)
# Covariates:
envfit_sd <- meta_envfit[, c("sexe", "voie_accou", "lait_enf", "age_allaite",
                "eau_traitee", "nb_repas", "alim_malnut", "haz", "Stunting_level",
                "Stunting_status", "parasite", "Gut_inflammation_AAT", "Gut_inflammation_Calpro",
                "Systemic_inflammation", "ferritine_seuil", "body_site")]

env.fit_results <- envfit(vectorPCOA, envfit_sd, perm = 999, na.rm=TRUE)
env.fit_results$factors$pvals
env.fit_results$factors$r
# Variance mostly explained by body site (make sense)
# Only one significant 
```

By body site using filtered absolute abundance
Feces only
```{r}
# Bray-Curtis
physeq_ra_fe <- subset_samples(physeq_ra, body_site == "Feces")
dist.bc_fe <- phyloseq::distance(physeq_ra_fe, method = "bray")
test_PCoAbray_fe <- ordinate(physeq_ra_fe, "PCoA", distance = dist.bc_fe)

plot_ordination(physeq_ra_fe, test_PCoAbray_fe, type = "samples", color = "Stunting_level", shape = "Stunting_status") +
  theme_bw() +
  scale_shape_manual(values=c(0:9)) +
  scale_color_brewer(palette = "Set1") +
  ggtitle("PCoA (Bray-Curtis distance) on fecal samples")

# No clear separation between stunted and non-stunted children in the feces (expected)
meta_pcoa_fe = as.matrix(sample_data(physeq_ra_fe))
meta_pcoa_fe <- as.data.frame(meta_pcoa_fe)
vegan::adonis2(dist.bc_fe ~ SeqBatch, data = meta_pcoa_fe, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_fe ~ Stunting_level, data = meta_pcoa_fe, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_fe ~ Stunting_status, data = meta_pcoa_fe, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_fe ~ Systemic_inflammation, data = meta_pcoa_fe, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_fe ~ sexe, data = meta_pcoa_fe, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_fe ~ Gut_inflammation_AAT, data = meta_pcoa_fe, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_fe ~ Gut_inflammation_Calpro, data = meta_pcoa_fe, permutations = 9999) # Not significant
#Parasites
meta_pcoa_fe_para <- meta_pcoa_fe
meta_pcoa_fe_para <- meta_pcoa_fe_para[-which(is.na(meta_pcoa_fe_para$Parasites)),]
dist.bc_fe_para <- dist_subset(dist.bc_fe, meta_pcoa_fe_para$id_modif)
vegan::adonis2(dist.bc_fe_para ~ Parasites, data = meta_pcoa_fe_para, permutations = 9999) # Not significant
#Leaky gut
meta_pcoa_fe_leaky <- meta_pcoa_fe
meta_pcoa_fe_leaky <- meta_pcoa_fe_leaky[-which(is.na(meta_pcoa_fe_leaky$Leaky_gut)),]
dist.bc_fe_leaky <- dist_subset(dist.bc_fe, meta_pcoa_fe_leaky$id_modif)
vegan::adonis2(dist.bc_fe_leaky ~ Leaky_gut, data = meta_pcoa_fe_leaky, permutations = 9999) # Not significant

# Using envfit
vectorPCOA_fe <- as.data.frame(test_PCoAbray_fe$vectors[,1:2])
# Get metadata
META_envfit_fe = as(sample_data(physeq_ra_fe), "matrix")
meta_envfit_fe <- as.data.frame(META_envfit_fe)
# Covariates:
envfit_sd_fe <- meta_envfit_fe[, c("sexe", "voie_accou", "lait_enf",
                "eau_traitee", "nb_repas", "alim_malnut", "haz", "Stunting_level",
                "Stunting_status", "parasite", "Gut_inflammation_AAT", "Gut_inflammation_Calpro",
                "Systemic_inflammation", "ferritine_seuil", "SeqBatch")]

env.fit_results_fe <- envfit(vectorPCOA_fe, envfit_sd_fe, perm = 999, na.rm=TRUE)
env.fit_results_fe$factors$pvals
env.fit_results_fe$factors$r
# Nothing
```

Saliva, gastric and duodenal samples
```{r}
# Bray-Curtis
physeq_ra_upper <- subset_samples(physeq_ra, body_site != "Feces")
dist.bc_upper <- phyloseq::distance(physeq_ra_upper, method = "bray")
test_PCoAbray_upper <- ordinate(physeq_ra_upper, "PCoA", distance = dist.bc_upper)

plot_ordination(physeq_ra_upper, test_PCoAbray_upper, type = "samples", color = "body_site", shape = "body_site") +
  theme_bw() +
  scale_shape_manual(values=c(0:9)) +
  scale_color_brewer(palette = "Set1") +
  ggtitle("PCoA (Bray-Curtis distance) on saliva, gastric and duodenal samples")

# Clear separation between saliva and other samples
meta_pcoa_upper = as.matrix(sample_data(physeq_ra_upper))
meta_pcoa_upper <- as.data.frame(meta_pcoa_upper)
#Significant
vegan::adonis2(dist.bc_upper ~ body_site, data = meta_pcoa_upper, permutations = 9999) # 1e-04 ***

vectorPCOA_upper <- test_PCoAbray_upper$vectors[,1:2]
# Get metadata
META_envfit_upper = as(sample_data(physeq_ra_upper), "matrix")
meta_envfit_upper <- as.data.frame(META_envfit_upper)
# Covariates:
envfit_sd_upper <- meta_envfit_upper[, c("sexe", "voie_accou", "lait_enf",
                "eau_traitee", "nb_repas", "alim_malnut", "haz", "Stunting_level",
                "Stunting_status", "parasite", "Gut_inflammation_AAT", "Gut_inflammation_Calpro",
                "Systemic_inflammation", "ferritine_seuil", "body_site")]

env.fit_results_upper <- envfit(vectorPCOA_upper, envfit_sd_upper, perm = 999, na.rm=TRUE)
env.fit_results_upper$factors$pvals
env.fit_results_upper$factors$r
# Variance mostly explained by body site (~40%)
# Also stunting status --> does it hold when separating the body sites?
```

Saliva
```{r}
# Bray-Curtis
physeq_ra_sa <- subset_samples(physeq_ra, body_site == "Saliva")
dist.bc_sa <- phyloseq::distance(physeq_ra_sa, method = "bray")
test_PCoAbray_sa <- ordinate(physeq_ra_sa, "PCoA", distance = dist.bc_sa)

plot_ordination(physeq_ra_sa, test_PCoAbray_sa, type = "samples", color = "Stunting_level", shape = "Stunting_level") +
  theme_bw() +
  scale_shape_manual(values=c(0:9)) +
  scale_color_brewer(palette = "Set1") +
  ggtitle("PCoA (Bray-Curtis distance) on saliva samples")
# No clear separation between stunted and non-stunted children in the saliva samples

meta_pcoa_sa = as.matrix(sample_data(physeq_ra_sa))
meta_pcoa_sa <- as.data.frame(meta_pcoa_sa)
#Significant
vegan::adonis2(dist.bc_sa ~ Systemic_inflammation, data = meta_pcoa_sa, permutations = 9999) # 0.0385 *
#Systemic inflammation
plot_ordination(physeq_ra_sa, test_PCoAbray_sa, type = "samples", color = "Systemic_inflammation", shape = "Systemic_inflammation") +
    stat_ellipse(mapping= aes(fill = Systemic_inflammation),geom = "polygon", alpha = 0.2, show.legend = FALSE) +
  theme_bw() +
  scale_shape_manual(values=c(0:9)) +
  scale_color_brewer(palette = "Set1") +
  ggtitle("PCoA (Bray-Curtis distance) on saliva samples")

# Not significant:
vegan::adonis2(dist.bc_sa ~ Stunting_level, data = meta_pcoa_sa, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_sa ~ Stunting_status, data = meta_pcoa_sa, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_sa ~ sexe, data = meta_pcoa_sa, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_sa ~ Gut_inflammation_AAT, data = meta_pcoa_sa, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_sa ~ Gut_inflammation_Calpro, data = meta_pcoa_sa, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_sa ~ Parasites, data = meta_pcoa_sa, permutations = 9999) # Not significant
# Leaky gut
meta_pcoa_sa_leaky <- meta_pcoa_sa
meta_pcoa_sa_leaky <- meta_pcoa_sa_leaky[-which(is.na(meta_pcoa_sa_leaky$Leaky_gut)),]
dist.bc_sa_leaky <- dist_subset(dist.bc_sa, meta_pcoa_sa_leaky$id_modif)
vegan::adonis2(dist.bc_sa_leaky ~ Leaky_gut, data = meta_pcoa_sa_leaky, permutations = 9999) # Not significant

# Using EnvFit
vectorPCOA_sa <- test_PCoAbray_sa$vectors[,1:2]
# Get metadata
META_envfit_sa = as(sample_data(physeq_ra_sa), "matrix")
meta_envfit_sa <- as.data.frame(META_envfit_sa)
# Covariates:
envfit_sd_sa <- meta_envfit_sa[, c("sexe", "voie_accou", "lait_enf",
                "eau_traitee", "nb_repas", "alim_malnut", "haz", "Stunting_level",
                "Stunting_status", "parasite", "Gut_inflammation_AAT", "Gut_inflammation_Calpro",
                "Systemic_inflammation", "ferritine_seuil")]

env.fit_results_sa <- envfit(vectorPCOA_sa, envfit_sd_sa, perm = 999, na.rm=TRUE)
env.fit_results_sa$factors$pvals
env.fit_results_sa$factors$r
# Nothing
plot_ordination(physeq_ra_sa, test_PCoAbray_sa, type = "samples", color = "haz", shape = "haz") +
  theme_bw() +
  scale_shape_manual(values=c(0:9)) +
  scale_color_brewer(palette = "Set1") +
  ggtitle("PCoA (Bray-Curtis distance) on saliva samples")
```

Gastric and duodenal samples
```{r}
# Bray-Curtis
physeq_ra_gd <- subset_samples(physeq_bdiv, body_site == "Duodenum" | body_site == "Gastric")
dist.bc_gd <- phyloseq::distance(physeq_ra_gd, method = "bray")
test_PCoAbray_gd <- ordinate(physeq_ra_gd, "PCoA", distance = dist.bc_gd)

plot_ordi_gd <- plot_ordination(physeq_ra_gd, test_PCoAbray_gd, type = "samples", color = "body_site", shape = "Stunting_level") +
  geom_point(size = 3) +
  theme_bw() +
  scale_shape_manual(values=c(15:18)) +
  scale_color_manual(values = pal_2[c(1,3)]) +
  labs(shape = "Nutritional status", color = "Body sites") +
  theme(
    axis.text.x = element_text(size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    plot.title = element_text(size = 14),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    strip.text = element_text(size = 14),
    ) +
  ggtitle("PCoA (Bray-Curtis distance) on gastric and duodenal samples")

pdf("Figures/BetaDiv_METAG_satellite_GastricDuodenum.pdf", width = 8, height = 6)
plot_ordi_gd
dev.off()

meta_pcoa_gd = as.matrix(sample_data(physeq_ra_gd))
meta_pcoa_gd <- as.data.frame(meta_pcoa_gd)

# Using PERMANOVA
vegan::adonis2(dist.bc_gd ~ body_site, data = meta_pcoa_gd, permutations = 9999) # Not significant
# Significant
vegan::adonis2(dist.bc_gd ~ Stunting_level, data = meta_pcoa_gd, permutations = 9999) # 0.0119 **
# Control for body site first:
vegan::adonis2(dist.bc_gd ~ body_site + Stunting_level, data = meta_pcoa_gd,
              by="terms",  permutations = 9999) # Stunting still significant
#Parasites
meta_pcoa_gd_para <- meta_pcoa_gd
meta_pcoa_gd_para <- meta_pcoa_gd_para[-which(is.na(meta_pcoa_gd_para$Parasites)),]
dist.bc_gd_para <- dist_subset(dist.bc_gd, meta_pcoa_gd_para$id_modif)
vegan::adonis2(dist.bc_gd_para ~ Parasites, data = meta_pcoa_gd_para, permutations = 9999) # 0.0105 *

# Age allaite
vegan::adonis2(dist.bc_gd ~ age_allaite, data = meta_pcoa_gd, permutations = 9999, na.action = na.omit) #5e-04
# Not significant
vegan::adonis2(dist.bc_gd ~ body_site, data = meta_pcoa_gd, permutations = 9999) # Not significant

vegan::adonis2(dist.bc_gd ~ Stunting_status, data = meta_pcoa_gd, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_gd ~ sexe, data = meta_pcoa_gd, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_gd ~ Systemic_inflammation, data = meta_pcoa_gd, permutations = 9999)
vegan::adonis2(dist.bc_gd ~ Gut_inflammation_AAT, data = meta_pcoa_gd, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_gd ~ Gut_inflammation_Calpro, data = meta_pcoa_gd, permutations = 9999) # Not significant
#Leaky gut
meta_pcoa_gd_leaky <- meta_pcoa_gd
meta_pcoa_gd_leaky <- meta_pcoa_gd_leaky[-which(is.na(meta_pcoa_gd_leaky$Leaky_gut)),]
dist.bc_gd_leaky <- dist_subset(dist.bc_gd, meta_pcoa_gd_leaky$id_modif)
vegan::adonis2(dist.bc_gd_leaky ~ Leaky_gut, data = meta_pcoa_gd_leaky, permutations = 9999) # Not significant

# USing EnvFit
vectorPCOA_gd <- test_PCoAbray_gd$vectors[,1:2]
# Get metadata
META_envfit_gd = as(sample_data(physeq_ra_gd), "matrix")
meta_envfit_gd <- as.data.frame(META_envfit_gd)
# Covariates:
envfit_sd_gd <- meta_envfit_gd[, c("sexe", "voie_accou", "lait_enf",
                "eau_traitee", "nb_repas", "alim_malnut", "haz", "Stunting_level",
                "Stunting_status", "parasite", "Gut_inflammation_AAT", "Gut_inflammation_Calpro",
                "Systemic_inflammation", "ferritine_seuil")]

env.fit_results_gd <- envfit(vectorPCOA_gd, envfit_sd_gd, perm = 999, na.rm=TRUE)
env.fit_results_gd$factors$pvals
env.fit_results_gd$factors$r
# age_allait?
```

Gastric
```{r}
# Bray-Curtis
physeq_ra_g <- subset_samples(physeq_ra, body_site == "Gastric")
dist.bc_g <- phyloseq::distance(physeq_ra_g, method = "bray")
test_PCoAbray_g <- ordinate(physeq_ra_g, "PCoA", distance = dist.bc_g)

plot_ordination(physeq_ra_g, test_PCoAbray_g, type = "samples", color = "Stunting_level", shape = "Stunting_level") +
  theme_bw() +
  scale_shape_manual(values=c(0:9)) +
  scale_color_brewer(palette = "Set1") +
  ggtitle("PCoA (Bray-Curtis distance) on saliva, gastric and duodenal samples")

meta_pcoa_g = as.matrix(sample_data(physeq_ra_g))
meta_pcoa_g <- as.data.frame(meta_pcoa_g)

# Using PERMANOVA
# Significant
vegan::adonis2(dist.bc_g ~ Stunting_level, data = meta_pcoa_g, permutations = 9999) # not significant
#Parasites
meta_pcoa_g_para <- meta_pcoa_g
meta_pcoa_g_para <- meta_pcoa_g_para[-which(is.na(meta_pcoa_g_para$Parasites)),]
dist.bc_g_para <- dist_subset(dist.bc_g, meta_pcoa_g_para$id_modif)

# Age allaite
vegan::adonis2(dist.bc_g ~ age_allaite, data = meta_pcoa_g, permutations = 9999, na.action = na.omit)
# Not significant
vegan::adonis2(dist.bc_g ~ Stunting_status, data = meta_pcoa_g, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_g ~ sexe, data = meta_pcoa_g, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_g ~ Systemic_inflammation, data = meta_pcoa_g, permutations = 9999)
vegan::adonis2(dist.bc_g ~ Gut_inflammation_AAT, data = meta_pcoa_g, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_g ~ Gut_inflammation_Calpro, data = meta_pcoa_g, permutations = 9999) # Not significant
#Leaky gut
meta_pcoa_g_leaky <- meta_pcoa_g
meta_pcoa_g_leaky <- meta_pcoa_g_leaky[-which(is.na(meta_pcoa_g_leaky$Leaky_gut)),]
dist.bc_g_leaky <- dist_subset(dist.bc_g, meta_pcoa_g_leaky$id_modif)
vegan::adonis2(dist.bc_g_leaky ~ Leaky_gut, data = meta_pcoa_g_leaky, permutations = 9999) # Not significant

# USing EnvFit
vectorPCOA_g <- test_PCoAbray_g$vectors[,1:2]
# Get metadata
META_envfit_g = as(sample_data(physeq_ra_g), "matrix")
meta_envfit_g <- as.data.frame(META_envfit_g)
# Covariates:
envfit_sd_g <- meta_envfit_g[, c("sexe", "voie_accou", "lait_enf",
                "eau_traitee", "nb_repas", "alim_malnut", "haz", "Stunting_level",
                "Stunting_status", "parasite", "Gut_inflammation_AAT", "Gut_inflammation_Calpro",
                "Systemic_inflammation", "ferritine_seuil")]

env.fit_results_g <- envfit(vectorPCOA_g, envfit_sd_g, perm = 999, na.rm=TRUE)
env.fit_results_g$factors$pvals
env.fit_results_g$factors$r
# age_allait?
```

Duodenal
```{r}
# Bray-Curtis
physeq_ra_d <- subset_samples(physeq_ra, body_site == "Duodenum")
dist.bc_d <- phyloseq::distance(physeq_ra_d, method = "bray")
test_PCoAbray_d <- ordinate(physeq_ra_d, "PCoA", distance = dist.bc_d)

plot_ordination(physeq_ra_d, test_PCoAbray_d, type = "samples", color = "Stunting_level", shape = "Stunting_level") +
  theme_bw() +
  scale_shape_manual(values=c(0:9)) +
  scale_color_brewer(palette = "Set1") +
  ggtitle("PCoA (Bray-Curtis distance) on saliva, gastric and duodenal samples")

meta_pcoa_d = as.matrix(sample_data(physeq_ra_d))
meta_pcoa_d <- as.data.frame(meta_pcoa_d)

# Using PERMANOVA
# Significant
vegan::adonis2(dist.bc_d ~ Stunting_level, data = meta_pcoa_d, permutations = 9999) # not significant
#Parasites
meta_pcoa_d_para <- meta_pcoa_d
meta_pcoa_d_para <- meta_pcoa_d_para[-which(is.na(meta_pcoa_d_para$Parasites)),]
dist.bc_d_para <- dist_subset(dist.bc_d, meta_pcoa_d_para$id_modif)

# Age allaite
vegan::adonis2(dist.bc_d ~ age_allaite, data = meta_pcoa_d, permutations = 9999, na.action = na.omit)
# Not significant
vegan::adonis2(dist.bc_d ~ Stunting_status, data = meta_pcoa_d, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_d ~ sexe, data = meta_pcoa_d, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_d ~ Systemic_inflammation, data = meta_pcoa_d, permutations = 9999)
vegan::adonis2(dist.bc_d ~ Gut_inflammation_AAT, data = meta_pcoa_d, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_d ~ Gut_inflammation_Calpro, data = meta_pcoa_d, permutations = 9999) # Not significant
#Leaky gut
meta_pcoa_d_leaky <- meta_pcoa_d
meta_pcoa_d_leaky <- meta_pcoa_d_leaky[-which(is.na(meta_pcoa_d_leaky$Leaky_gut)),]
dist.bc_d_leaky <- dist_subset(dist.bc_d, meta_pcoa_d_leaky$id_modif)
vegan::adonis2(dist.bc_d_leaky ~ Leaky_gut, data = meta_pcoa_d_leaky, permutations = 9999) # Not significant

# USing EnvFit
vectorPCOA_d <- test_PCoAbray_d$vectors[,1:2]
# Get metadata
META_envfit_d = as(sample_data(physeq_ra_d), "matrix")
meta_envfit_d <- as.data.frame(META_envfit_d)
# Covariates:
envfit_sd_d <- meta_envfit_d[, c("sexe", "voie_accou", "lait_enf",
                "eau_traitee", "nb_repas", "alim_malnut", "haz", "Stunting_level",
                "Stunting_status", "parasite", "Gut_inflammation_AAT", "Gut_inflammation_Calpro",
                "Systemic_inflammation", "ferritine_seuil")]

env.fit_results_d <- envfit(vectorPCOA_d, envfit_sd_d, perm = 999, na.rm=TRUE)
env.fit_results_d$factors$pvals
env.fit_results_d$factors$r
# age_allait?
```



# Composition
## Taxa exploration
Phylum
```{r}
get_taxa_unique(physeq_ra, "Kingdom") # Bacteria, Archaea
get_taxa_unique(physeq_ra, "Phylum") # 21 phyla

physeq_phy <- tax_glom(physeq_ra, "Phylum")

# Frequency plot
plot_bar(physeq_phy, fill = "Phylum") +
  scale_fill_manual(values = pal_1) +
  theme_bw() + 
  ggtitle("Relative abundance: Phylum level") +
  theme(
      axis.ticks.x=element_blank(),
      strip.text = element_text(size=15),
        legend.position = "bottom") +
  facet_grid(~body_site, scales = "free") +
  ylab("Relative abundance")

# Feces: dominated by Bacteroidota and Firmicutes
# Saliva: Proteobacteria, Firmicutes, Bacteroidota, Actinobacteria
# Gastric and Duodenal: Proteobacteria, Firmicutes, Bacteroidota and a bit less Actinobacteria

# Box plot
physeq_phy_m <- psmelt(physeq_phy)
ggplot(physeq_phy_m, aes(x = Phylum, y = Abundance)) +
  geom_boxplot(aes(fill = Phylum), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Phylum), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9))+
  facet_grid(body_site~Nutritional_status, scales = "free")

ps_venn(physeq_phy,
        group = "body_site",
        plot = FALSE)
# Unique to feces: Verrucomicrobia, Candidatus_Melainabacteria, Bacillota, Lentisphaerae, Bacteria_unclassified
# Unique Gastric: Chloroflexi
```
Phylum GTDB
```{r}
get_taxa_unique(physeq_gtdb, "Kingdom") # Bacteria, Archaea
get_taxa_unique(physeq_gtdb, "Phylum") # 21 phyla

physeq_gtdb_phy <- tax_glom(physeq_gtdb, "Phylum")

# Frequency plot
plot_bar(physeq_gtdb_phy, fill = "Phylum") +
  scale_fill_manual(values = pal_1) +
  theme_bw() + 
  ggtitle("Relative abundance: Phylum level") +
  theme(
      axis.ticks.x=element_blank(),
      strip.text = element_text(size=15),
        legend.position = "bottom") +
  facet_grid(~body_site, scales = "free") +
  ylab("Relative abundance")

# Feces: dominated by Bacteroidota and Firmicutes
# Saliva: Proteobacteria, Firmicutes, Bacteroidota, Actinobacteria
# Gastric and Duodenal: Proteobacteria, Firmicutes, Bacteroidota and a bit less Actinobacteria

# Box plot
physeq_phy_m <- psmelt(physeq_gtdb_phy)
ggplot(physeq_phy_m, aes(x = Phylum, y = Abundance)) +
  geom_boxplot(aes(fill = Phylum), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Phylum), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9))+
  facet_grid(body_site~Nutritional_status, scales = "free")

ps_venn(physeq_gtdb_phy,
        group = "body_site",
        plot = FALSE)
# Unique to feces: Verrucomicrobia, Candidatus_Melainabacteria, Bacillota, Lentisphaerae, Bacteria_unclassified
# Unique Gastric: Chloroflexi
```
Family
```{r}
physeq_fam <- tax_glom(physeq_ra, taxrank = "Family")
# 370 families in the dataset
physeq_fam_m <- psmelt(physeq_fam)

# GTDB
ps_gtdb_fam <- tax_glom(physeq_gtdb, taxrank = "Family")
# 153 families in the dataset
ps_gtdb_fam_m <- psmelt(ps_gtdb_fam)
```

Saliva
```{r}
physeq_fam_sa <- subset_samples(physeq_fam, body_site =="Saliva")
physeq_fam_sa <- prune_taxa(taxa_sums(physeq_fam_sa)> 0, physeq_fam_sa)
# 95 families in Saliva samples

TopN_famSA <- names(sort(taxa_sums(physeq_fam_sa), TRUE)[1:12])
familySA <- prune_taxa(TopN_famSA, physeq_fam_sa)
familySA_melt <- psmelt(familySA)

ggplot(familySA_melt, aes(x = Family, y = Abundance)) +
  geom_boxplot(aes(fill = Family), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Family), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9))+
  facet_grid(~Nutritional_status, scales = "free") +
  ylim(0,0.5)


# GTDB
ps_gtdb_fam_sa <- subset_samples(ps_gtdb_fam, body_site =="Saliva")
ps_gtdb_fam_sa <- prune_taxa(taxa_sums(ps_gtdb_fam_sa)> 0, ps_gtdb_fam_sa)
# 95 families in Saliva samples

TopN_gtdb_famSA <- names(sort(taxa_sums(ps_gtdb_fam_sa), TRUE)[1:12])
familySA_gtdb <- prune_taxa(TopN_gtdb_famSA, ps_gtdb_fam_sa)
familySA_gtdb_melt <- psmelt(familySA_gtdb)

ggplot(familySA_gtdb_melt, aes(x = Family, y = Abundance)) +
  geom_boxplot(aes(fill = Family), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Family), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9))+
  facet_grid(~Nutritional_status, scales = "free")+
  ylim(0,0.5)
```

Gastric
```{r}
physeq_fam_ga <- subset_samples(physeq_fam, body_site =="Gastric")
physeq_fam_ga <- prune_taxa(taxa_sums(physeq_fam_ga)> 0, physeq_fam_ga)
# 89 families in Gastric samples

TopN_famga <- names(sort(taxa_sums(physeq_fam_ga), TRUE)[1:12])
familyga <- prune_taxa(TopN_famga, physeq_fam_ga)
familyga_melt <- psmelt(familyga)

ggplot(familyga_melt, aes(x = Family, y = Abundance)) +
  geom_boxplot(aes(fill = Family), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Family), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9))+
  facet_grid(~Nutritional_status, scales = "free")
```

Feces
```{r}
physeq_fam_fe <- subset_samples(physeq_fam, body_site =="Feces")
physeq_fam_fe <- prune_taxa(taxa_sums(physeq_fam_fe)> 0, physeq_fam_fe)
# 320 families in Feces samples

TopN_famfe <- names(sort(taxa_sums(physeq_fam_fe), TRUE)[1:20])
familyfe <- prune_taxa(TopN_famfe, physeq_fam_fe)
familyfe_melt <- psmelt(familyfe)

ggplot(familyfe_melt, aes(x = Family, y = Abundance)) +
  geom_boxplot(aes(fill = Family), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Family), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9))+
  facet_grid(~Nutritional_status, scales = "free")


# GTDB
ps_gtdb_fam_fe <- subset_samples(ps_gtdb_fam, body_site =="Feces")
ps_gtdb_fam_fe <- prune_taxa(taxa_sums(ps_gtdb_fam_fe)> 0, ps_gtdb_fam_fe)
# 320 families in Feces samples

TopN_gtdb_famfe <- names(sort(taxa_sums(ps_gtdb_fam_fe), TRUE)[1:20])
familyfe_gtdb <- prune_taxa(TopN_gtdb_famfe, ps_gtdb_fam_fe)
familyfe_gtdb_melt <- psmelt(familyfe_gtdb)

ggplot(familyfe_gtdb_melt, aes(x = Family, y = Abundance)) +
  geom_boxplot(aes(fill = Family), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Family), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9))+
  facet_grid(~Nutritional_status, scales = "free")
```
Shared families
```{r}
venn_list <- ps_venn(physeq_fam,
        group = "body_site",
        plot = FALSE)
# Only shared Families between all body sites
physeq_fam_shared <- subset_taxa(physeq_fam, rownames(tax_table(physeq_fam)) %in% venn_list$Duodenum__Feces__Gastric__Saliva)

physeq_fam_shared_melt <- psmelt(physeq_fam_shared)
physeq_fam_shared_melt$body_site = factor(physeq_fam_shared_melt$body_site, levels = c("Saliva", "Gastric", "Duodenum", "Feces"))

ggplot(physeq_fam_shared_melt, aes(x = Family, y = Abundance)) +
  geom_boxplot(aes(fill = Family), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Family), size =1) +
  scale_color_manual(values = pal_4) +
  scale_fill_manual(values = pal_4) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9)) +
  facet_grid(body_site~., scales = "free")
```


Genus
```{r}
physeq_gen <- tax_glom(physeq_ra, taxrank = "Genus")
# 1040 genera in the dataset

ps_gtdb_gen <- tax_glom(physeq_gtdb, taxrank = "Genus")
# 629 genera in the dataset
```
Saliva
```{r}
physeq_gen_sa <- subset_samples(physeq_gen, body_site =="Saliva")
physeq_gen_sa <- prune_taxa(taxa_sums(physeq_gen_sa)> 0, physeq_gen_sa)
# 221 genera in Saliva samples

TopN_genSA <- names(sort(taxa_sums(physeq_gen_sa), TRUE)[1:20])
genusSA <- prune_taxa(TopN_genSA, physeq_gen_sa)
genusSA_melt <- psmelt(genusSA)

ggplot(genusSA_melt, aes(x = Genus, y = Abundance)) +
  geom_boxplot(aes(fill = Genus), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Genus), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9))+
  facet_grid(~Nutritional_status, scales = "free")

# GTDB
ps_gtdb_gen_sa <- subset_samples(ps_gtdb_gen, body_site =="Saliva")
ps_gtdb_gen_sa <- prune_taxa(taxa_sums(ps_gtdb_gen_sa)> 0, ps_gtdb_gen_sa)
# 137 genera in Saliva samples

TopN_gtdb_genSA <- names(sort(taxa_sums(ps_gtdb_gen_sa), TRUE)[1:20])
genusSA_gtdb <- prune_taxa(TopN_gtdb_genSA, ps_gtdb_gen_sa)
genusSA_gtdb_melt <- psmelt(genusSA_gtdb)

ggplot(genusSA_gtdb_melt, aes(x = Genus, y = Abundance)) +
  geom_boxplot(aes(fill = Genus), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Genus), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9))+
  facet_grid(~Nutritional_status, scales = "free")
```

Gastric
```{r}
physeq_gen_ga <- subset_samples(physeq_gen, body_site =="Gastric")
physeq_gen_ga <- prune_taxa(taxa_sums(physeq_gen_ga)> 0, physeq_gen_ga)
# 191 genera in Gastric gamples

TopN_genga <- names(sort(taxa_sums(physeq_gen_ga), TRUE)[1:20])
genusga <- prune_taxa(TopN_genga, physeq_gen_ga)
genusga_melt <- psmelt(genusga)

ggplot(genusga_melt, aes(x = Genus, y = Abundance)) +
  geom_boxplot(aes(fill = Genus), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Genus), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9))+
  facet_grid(~Nutritional_status, scales = "free")
```
Feces
```{r}
physeq_gen_fe <- subset_samples(physeq_gen, body_site =="Feces")
physeq_gen_fe <- prune_taxa(taxa_sums(physeq_gen_fe)> 0, physeq_gen_fe)
# 888 genera in fecal femples

TopN_genfe <- names(sort(taxa_sums(physeq_gen_fe), TRUE)[1:20])
genusfe <- prune_taxa(TopN_genfe, physeq_gen_fe)
genusfe_melt <- psmelt(genusfe)

ggplot(genusfe_melt, aes(x = Genus, y = Abundance)) +
  geom_boxplot(aes(fill = Genus), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Genus), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9))+
  facet_grid(~Nutritional_status, scales = "free")


# GTDB
ps_gtdb_gen_fe <- subset_samples(ps_gtdb_gen, body_site =="Feces")
ps_gtdb_gen_fe <- prune_taxa(taxa_sums(ps_gtdb_gen_fe)> 0, ps_gtdb_gen_fe)
# 888 genera in fecal femples

TopN_gtdb_genfe <- names(sort(taxa_sums(ps_gtdb_gen_fe), TRUE)[1:20])
genusfe_gtdb <- prune_taxa(TopN_gtdb_genfe, ps_gtdb_gen_fe)
genusfe_gtdb_melt <- psmelt(genusfe_gtdb)

ggplot(genusfe_gtdb_melt, aes(x = Genus, y = Abundance)) +
  geom_boxplot(aes(fill = Genus), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Genus), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9))+
  facet_grid(~Nutritional_status, scales = "free")
```

Shared genera
```{r}
venn_list_gen <- ps_venn(physeq_gen,
        group = "body_site",
        plot = FALSE)
# Only shared Families between all body sites
physeq_gen_shared <- subset_taxa(physeq_gen, rownames(tax_table(physeq_gen)) %in% venn_list_gen$Duodenum__Feces__Gastric__Saliva)

physeq_gen_shared_melt <- psmelt(physeq_gen_shared)
physeq_gen_shared_melt$body_site = factor(physeq_gen_shared_melt$body_site, levels = c("Saliva", "Gastric", "Duodenum", "Feces"))

ggplot(physeq_gen_shared_melt, aes(x = Genus, y = Abundance)) +
  geom_boxplot(aes(fill = Genus), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Genus), size =1) +
  scale_color_manual(values = pal_5) +
  scale_fill_manual(values = pal_5) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9)) +
  facet_grid(body_site~., scales = "free")

# GTDB
venn_list_gen <- ps_venn(ps_gtdb_gen,
        group = "body_site",
        plot = FALSE)
# Only shared Families between all body sites
ps_gtdb_gen_shared <- subset_taxa(ps_gtdb_gen, rownames(tax_table(ps_gtdb_gen)) %in% venn_list_gen$Duodenum__Feces__Gastric__Saliva)

ps_gtdb_gen_shared_melt <- psmelt(ps_gtdb_gen_shared)
ps_gtdb_gen_shared_melt$body_site = factor(ps_gtdb_gen_shared_melt$body_site, levels = c("Saliva", "Gastric", "Duodenum", "Feces"))

ggplot(ps_gtdb_gen_shared_melt, aes(x = Genus, y = Abundance)) +
  geom_boxplot(aes(fill = Genus), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Genus), size =1) +
  scale_color_manual(values = pal_5) +
  scale_fill_manual(values = pal_5) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9)) +
  facet_grid(body_site~., scales = "free")
```

Species
Streptococcus salivarius box plot
```{r}
physeq_sp_abs_m <- psmelt(physeq_sp_abs)
physeq_sp_abs_m_Ssal <- physeq_sp_abs_m %>% filter(Species == "Streptococcus_salivarius")



physeq_sp_m <- psmelt(physeq_sp)
physeq_sp_m_Ssal <- physeq_sp_m %>% filter(Species == "Streptococcus_salivarius")
#physeq_sp_m_Ssal <- physeq_sp_m_Ssal %>% mutate(log10Abundance = log10(Abundance+1e-6))
physeq_sp_m_Ssal$body_site <- factor(physeq_sp_m_Ssal$body_site, levels = c("Saliva", "Gastric", "Duodenum", "Feces"))
library(ggpubr)
comparisons <- list(
  c("Saliva", "Gastric"), c("Saliva", "Duodenum"), c("Gastric", "Feces"),
  c("Gastric", "Duodenum"),
  c("Duodenum", "Feces"),
  c("Saliva", "Feces")
)


Ssal_boxplot <- ggplot(physeq_sp_m_Ssal, aes(x = body_site, y = Abundance)) +
  geom_boxplot(fill = "steelblue", alpha = 0.7) +
  scale_y_continuous(trans = "log10") + 
  stat_compare_means(
    comparisons = comparisons,
    method = "wilcox.test",
    p.adjust.method = "fdr",
    label = "p.signif"
  ) +
  labs(x = "Body compartments", y = "Abundance (log10)", title = "Streptococcus salivarius abundance by body compartment (AGN045GA 0%") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggsave(filename = "Figures/Boxplot_Streptococcus_salivarius_BodySite_Satellite.svg", plot = Ssal_boxplot)
```


## Oral microbiota
```{r}
ps_ra_saliva <- subset_samples(physeq_ra, body_site =="Saliva")
ps_ra_saliva <- prune_taxa(taxa_sums(ps_ra_saliva)> 0, ps_ra_saliva)
ps_ra_saliva_gen <- tax_glom(ps_ra_saliva, "Genus") # 221 genera

# Get percentage
saliva_relab_gen <- as.data.frame(tax_table(ps_ra_saliva_gen))[,c(2,6)]
saliva_relab_gen <- saliva_relab_gen %>%
  mutate(Rel_abun = round(100*apply(otu_table(ps_ra_saliva_gen), 1, mean), digits = 3),
         Standard_dev = round(100*apply(otu_table(ps_ra_saliva_gen), 1, sd), digits = 3))
ps_ra_saliva_sp <- tax_glom(ps_ra_saliva, "Species") 
saliva_relab_sp <- as.data.frame(tax_table(ps_ra_saliva_sp))[,c(6,7)]
saliva_relab_sp <- saliva_relab_sp %>%
  mutate(Rel_abun = round(100*apply(otu_table(ps_ra_saliva_sp), 1, mean), digits = 3),
         Standard_dev = round(100*apply(otu_table(ps_ra_saliva_sp), 1, sd), digits = 3))

TopGen_sal <- names(sort(taxa_sums(ps_ra_saliva_gen), TRUE)[1:20])
SalGen20 <- prune_taxa(TopGen_sal, ps_ra_saliva_gen)
topotuG_sal <- otu_table(SalGen20)
toptaxaG_sal <- tax_table(SalGen20)
SalGen20 = phyloseq(topotuG_sal, toptaxaG_sal)
SalGen20 = merge_phyloseq(sample_data(ps_ra_saliva_gen), SalGen20)
# Grouping other Genus
OtherG_sal <- names(sort(taxa_sums(ps_ra_saliva_gen), TRUE)[c(21:221)])
SalG_other <- prune_taxa(OtherG_sal, ps_ra_saliva_gen)
others_otu_sal <- t(as.matrix(apply(otu_table(SalG_other), 2, sum)))
rownames(others_otu_sal) <- "Others Genera"
others_otu_sal=otu_table(others_otu_sal, taxa_are_rows = TRUE)
other_taxG_sal <- c("Others", "Others","Others","Others","Others","Others","Others", "Others")
other_taxG_sal <- as.matrix(other_taxG_sal)
rownames(other_taxG_sal) <- colnames(tax_table(SalG_other))
other_taxG_sal <- t(other_taxG_sal)
rownames(other_taxG_sal) <- "Others Genera"
other_taxG_sal =  tax_table(other_taxG_sal)
others_salG= phyloseq(others_otu_sal,other_taxG_sal)
others_salG = merge_phyloseq(sample_data(ps_ra_saliva_gen), others_salG)
others_salG
# Regrouping phyloseq
salG <- merge_phyloseq(SalGen20, others_salG)
salG_melt <- psmelt(salG)

# Fix unassigned SGB names to higher taxonomic ranks
salG_melt <- salG_melt %>%
  mutate(Name = case_when(
    str_detect(Genus, "GGB", negate = TRUE) ~ Genus,
    str_detect(Genus, "GGB", negate = FALSE) ~ paste(Genus, Family)
  ))

salG_melt$Stunting_status[which(salG_melt$Stunting_status=="Non_stunted")] <- "Non-stunted"

boxplot_oral <- ggplot(salG_melt, aes(x = Name, y = Abundance)) +
  geom_boxplot(aes(fill = Name), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Name), size =1) +
  scale_color_manual(values = pal_saliva) +
  scale_fill_manual(values = pal_saliva) +
  facet_grid(Stunting_status~., scales = "free_x") +
  ylab("Relative abundance")+
  theme_bw() +
  theme(
    legend.position = "none", 
    axis.text.x =  element_text(angle = 45, hjust = .9, face = 3),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    axis.title.x = element_blank(),
    plot.title = element_text(size = 14),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    strip.text = element_text(size = 14)) +
  ggtitle("Oral microbiota composition")

pdf("Figures/Composition_BoxPlot_saliva.pdf", width = 8, height = 6)
boxplot_oral
dev.off()

barplot_oral <- salG_melt %>%
  ggplot(aes_string(x = "id_modif", y = "Abundance", fill = "Name")) +
  geom_bar(stat = "identity", position = "stack") + 
  facet_grid(~Stunting_status, scales = "free_x") +
  scale_fill_manual(values = pal_saliva) +
  theme_bw() + 
  theme(legend.position = "right", 
        axis.title.x=element_blank(),
        axis.text.x=element_text(angle = 60, hjust = 1),
        axis.ticks.x=element_blank()) +
  guides(fill=guide_legend(ncol=2))+
  ylab("Relative abundance")+
  labs(fill = "Genus") +
  theme(
    axis.text.x =  element_blank(),
    axis.text.y = element_text(size = 11),
    axis.title = element_text(size = 11),
    plot.title = element_text(size = 11),
    legend.title = element_text(size = 11),
    legend.text = element_text(size = 11, face = 3),
    strip.text = element_text(size = 11)) +
  ggtitle("Oral microbiota composition")

pdf("Figures/Composition_BarPlot_saliva.pdf", width = 16, height = 4)
barplot_oral
dev.off()
```
## Gastric microbiota
```{r}
ps_ra_ga <- subset_samples(physeq_ra, body_site =="Gastric")
ps_ra_ga <- prune_taxa(taxa_sums(ps_ra_ga)> 0, ps_ra_ga)
ps_ra_ga_gen <- tax_glom(ps_ra_ga, "Genus") # 191 genera

# Relative abundance 
gastric_relab_gen <- as.data.frame(tax_table(ps_ra_ga_gen))[,c(2,6)]
gastric_relab_gen <- gastric_relab_gen %>%
  mutate(Rel_abun = round(100*apply(otu_table(ps_ra_ga_gen), 1, mean), digits = 3),
         Standard_dev = round(100*apply(otu_table(ps_ra_ga_gen), 1, sd), digits = 3))
ps_ra_ga_sp <- tax_glom(ps_ra_ga, "Species") 
gastric_relab_sp <- as.data.frame(tax_table(ps_ra_ga_sp))[,c(6,7)]
gastric_relab_sp <- gastric_relab_sp %>%
  mutate(Rel_abun = round(100*apply(otu_table(ps_ra_ga_sp), 1, mean), digits = 3),
         Standard_dev = round(100*apply(otu_table(ps_ra_ga_sp), 1, sd), digits = 3))
gastric_prevdf_sp = apply(X= otu_table(ps_ra_ga_sp),
               MARGIN = ifelse(taxa_are_rows(ps_ra_ga_sp),
              yes =1, no =2), FUN = function(x){sum(x>0)})
gastric_prevdf_sp = data.frame(Prevalence = gastric_prevdf_sp,
                        Percentage_Prev = 100*gastric_prevdf_sp/31,
                        tax_table(ps_ra_ga_sp))
# Top 20 Genera
TopGen_ga <- names(sort(taxa_sums(ps_ra_ga_gen), TRUE)[1:20])
gaGen20 <- prune_taxa(TopGen_ga, ps_ra_ga_gen)
topotuG_ga <- otu_table(gaGen20)
toptaxaG_ga <- tax_table(gaGen20)
gaGen20 = phyloseq(topotuG_ga, toptaxaG_ga)
gaGen20 = merge_phyloseq(sample_data(ps_ra_ga_gen), gaGen20)
# Grouping other Genilies
OtherG_ga <- names(sort(taxa_sums(ps_ra_ga_gen), TRUE)[c(21:191)])
gaG_other <- prune_taxa(OtherG_ga, ps_ra_ga_gen)
others_otu_ga <- t(as.matrix(apply(otu_table(gaG_other), 2, sum)))
rownames(others_otu_ga) <- "Others Genera"
others_otu_ga=otu_table(others_otu_ga, taxa_are_rows = TRUE)
other_taxG_ga <- c("Others", "Others","Others","Others","Others","Others","Others", "Others")
other_taxG_ga <- as.matrix(other_taxG_ga)
rownames(other_taxG_ga) <- colnames(tax_table(gaG_other))
other_taxG_ga <- t(other_taxG_ga)
rownames(other_taxG_ga) <- "Others Genera"
other_taxG_ga =  tax_table(other_taxG_ga)
others_gaG= phyloseq(others_otu_ga,other_taxG_ga)
others_gaG = merge_phyloseq(sample_data(ps_ra_ga_gen), others_gaG)
others_gaG
# Regrouping phyloseq
gaG <- merge_phyloseq(gaGen20, others_gaG)
gaG_melt <- psmelt(gaG)

gaG_melt <- gaG_melt %>%
  mutate(Name = case_when(
    str_detect(Genus, "GGB", negate = TRUE) ~ Genus,
    str_detect(Genus, "GGB", negate = FALSE) ~ paste(Genus, Family)
  ))

gaG_melt$Stunting_status[which(gaG_melt$Stunting_status=="Non_stunted")] <- "Non-stunted"

boxplot_ga <- ggplot(gaG_melt, aes(x = Name, y = Abundance)) +
  geom_boxplot(aes(fill = Name), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Name), size =1) +
  scale_color_manual(values = pal_gastric) +
  scale_fill_manual(values = pal_gastric) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9, face = 3)) +
  ggtitle("Gastric microbiota composition - Metagnomic data - Top 20 Genera")

#ggsave(filename = "Figures/Boxplot_ga_METAG_satellite.svg", plot = boxplot_ga, width = 10, height = 8)

barplot_ga <- gaG_melt %>%
  ggplot(aes_string(x = "id_modif", y = "Abundance", fill = "Name")) +
  geom_bar(stat = "identity", position = "stack") + 
  facet_grid(~Stunting_status, scales = "free_x") +
  scale_fill_manual(values = pal_gastric) +
  theme_bw() + 
  theme(legend.position = "right", 
        axis.title.x=element_blank(),
        axis.text.x=element_text(angle = 60, hjust = 1),
        axis.ticks.x=element_blank()) +
  guides(fill=guide_legend(ncol=2))+
  ylab("Relative abundance")+
  ggtitle("Gastric microbiota composition")+
  labs(fill = "Genus") +
  theme(
    axis.text.x =  element_blank(),
    axis.text.y = element_text(size = 11),
    axis.title = element_text(size = 11),
    plot.title = element_text(size = 11),
    legend.title = element_text(size = 11),
    legend.text = element_text(size = 11, face = 3),
    strip.text = element_text(size = 11))

pdf("Figures/Composition_BarPlot_gastric.pdf", width = 12, height =4)
barplot_ga
dev.off()

#ggsave(filename = "Figures/Barplot_ga_METAG_satellite.svg", plot = barplot_ga, width = 10, height = 8)
```
## Duodenal microbiota
```{r}
ps_ra_du <- subset_samples(physeq_ra, body_site =="Duodenum" )
ps_ra_du <- prune_taxa(taxa_sums(ps_ra_du)> 0, ps_ra_du)
ps_ra_du_gen <- tax_glom(ps_ra_du, "Genus") # 177 genera

# Duodenum
duodenum_relab_gen <- as.data.frame(tax_table(ps_ra_du_gen))[,c(2,6)]
duodenum_relab_gen <- duodenum_relab_gen %>%
  mutate(Rel_abun = round(100*apply(otu_table(ps_ra_du_gen), 1, mean), digits = 3),
         Standard_dev = round(100*apply(otu_table(ps_ra_du_gen), 1, sd), digits = 3))
ps_ra_du_sp <- tax_glom(ps_ra_du, "Species") 
duodenum_relab_sp <- as.data.frame(tax_table(ps_ra_du_sp))[,c(6,7)]
duodenum_relab_sp <- duodenum_relab_sp %>%
  mutate(Rel_abun = round(100*apply(otu_table(ps_ra_du_sp), 1, mean), digits = 3),
         Standard_dev = round(100*apply(otu_table(ps_ra_du_sp), 1, sd), digits = 3))

# Top 20 Genera
TopGen_du <- names(sort(taxa_sums(ps_ra_du_gen), TRUE)[1:20])
duGen20 <- prune_taxa(TopGen_du, ps_ra_du_gen)
topotuG_du <- otu_table(duGen20)
toptaxaG_du <- tax_table(duGen20)
duGen20 = phyloseq(topotuG_du, toptaxaG_du)
duGen20 = merge_phyloseq(sample_data(ps_ra_du_gen), duGen20)
# Grouping other Genilies
OtherG_du <- names(sort(taxa_sums(ps_ra_du_gen), TRUE)[c(21:177)])
duG_other <- prune_taxa(OtherG_du, ps_ra_du_gen)
others_otu_du <- t(as.matrix(apply(otu_table(duG_other), 2, sum)))
rownames(others_otu_du) <- "Others Genera"
others_otu_du=otu_table(others_otu_du, taxa_are_rows = TRUE)
other_taxG_du <- c("Others", "Others","Others","Others","Others","Others","Others", "Others")
other_taxG_du <- as.matrix(other_taxG_du)
rownames(other_taxG_du) <- colnames(tax_table(duG_other))
other_taxG_du <- t(other_taxG_du)
rownames(other_taxG_du) <- "Others Genera"
other_taxG_du =  tax_table(other_taxG_du)
others_duG= phyloseq(others_otu_du,other_taxG_du)
others_duG = merge_phyloseq(sample_data(ps_ra_du_gen), others_duG)
others_duG
# Regrouping phyloseq
duG <- merge_phyloseq(duGen20, others_duG)
duG_melt <- psmelt(duG)

duG_melt <- duG_melt %>%
  mutate(Name = case_when(
    str_detect(Genus, "GGB", negate = TRUE) ~ Genus,
    str_detect(Genus, "GGB", negate = FALSE) ~ paste(Genus, Family)
  ))


boxplot_du <- ggplot(duG_melt, aes(x = Name, y = Abundance)) +
  geom_boxplot(aes(fill = Name), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Name), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9, face = 3)) +
  ggtitle("Duodenal microbiota composition - Metagnomic data - Top 20 Genera")

#ggsave(filename = "Figures/Boxplot_du_METAG_satellite.svg", plot = boxplot_du, width = 10, height = 8)

sort(unique(duG_melt$Name))

barplot_du <- duG_melt %>%
  ggplot(aes_string(x = "id_modif", y = "Abundance", fill = "Name")) +
  geom_bar(stat = "identity", position = "stack") + 
  facet_grid(~Stunting_status, scales = "free_x") +
  scale_fill_manual(values = pal_duodenum) +
  theme_bw() + 
  theme(legend.position = "right", 
        axis.title.x=element_blank(),
        axis.text.x=element_text(angle = 60, hjust = 1),
        axis.ticks.x=element_blank()) +
  guides(fill=guide_legend(ncol=2))+
  ylab("Relative abundance")+
  ggtitle("Duodenal microbiota composition")+
  labs(fill = "Genus") +
  theme(
    axis.text.x =  element_blank(),
    axis.text.y = element_text(size = 11),
    axis.title = element_text(size = 11),
    plot.title = element_text(size = 11),
    legend.title = element_text(size = 11),
    legend.text = element_text(size = 11, face = 3),
    strip.text = element_text(size = 11))

pdf("Figures/Composition_BarPlot_duodenum.pdf", width = 10, height = 4)
barplot_du
dev.off()
```
Correlation species - haz
```{r}
ps_ra_du_sp <- tax_glom(ps_ra_du, "Species")
# 444 species
prevdf_sp_du = apply(X= otu_table(ps_ra_du_sp),
               MARGIN = ifelse(taxa_are_rows(ps_ra_du_sp),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_du = data.frame(Prevalence = prevdf_sp_du,
                        Percentage_Prev_feces = 100*prevdf_sp_du/14,
                        tax_table(ps_ra_du_sp))
prevdf_sp_du <- prevdf_sp_du %>% filter(Percentage_Prev_feces == 100)
# 9 species "core taxa"
unique(prevdf_sp_du$Species)

ps_ra_du_core <- subset_taxa(ps_ra_du_sp, Species %in% prevdf_sp_du$Species)
ps_ra_du_core <- psmelt(ps_ra_du_core)

cor_results <- ps_ra_du_core %>%
  group_by(Species) %>%  # replace 'OTU' with your taxa column name if different (e.g. Species)
  summarise(
    cor_spearman = cor(Abundance, haz_cont, method = "spearman", use = "complete.obs"),
    p_spearman = cor.test(Abundance, haz_cont, method = "spearman")$p.value,
    
    cor_pearson = cor(Abundance, haz_cont, method = "pearson", use = "complete.obs"),
    p_pearson = cor.test(Abundance, haz_cont, method = "pearson")$p.value,
    
    n = n()
  ) %>%
  ungroup() %>%
  mutate(p_adj_spearman = p.adjust(p_spearman, method = "fdr"),
         p_adj_pearson = p.adjust(p_pearson, method = "fdr"))
```
## Fecal microbiota
```{r}
ps_ra_feces <- subset_samples(physeq_ra, body_site =="Feces")
ps_ra_feces <- prune_taxa(taxa_sums(ps_ra_feces)> 0, ps_ra_feces)
ps_ra_feces_gen <- tax_glom(ps_ra_feces, "Genus") # 888 genera

# Get percentage
feces_relab_gen <- as.data.frame(tax_table(ps_ra_feces_gen))[,c(2,6)]
feces_relab_gen <- feces_relab_gen %>%
  mutate(Rel_abun = round(100*apply(otu_table(ps_ra_feces_gen), 1, mean), digits = 3),
         Standard_dev = round(100*apply(otu_table(ps_ra_feces_gen), 1, sd), digits = 3))
ps_ra_feces_sp <- tax_glom(ps_ra_feces, "Species") 
feces_relab_sp <- as.data.frame(tax_table(ps_ra_feces_sp))[,c(6,7)]
feces_relab_sp <- feces_relab_sp %>%
  mutate(Rel_abun = round(100*apply(otu_table(ps_ra_feces_sp), 1, mean), digits = 3),
         Standard_dev = round(100*apply(otu_table(ps_ra_feces_sp), 1, sd), digits = 3))

TopGen_fec <- names(sort(taxa_sums(ps_ra_feces_gen), TRUE)[1:20])
fecGen20 <- prune_taxa(TopGen_fec, ps_ra_feces_gen)
topotuG_fec <- otu_table(fecGen20)
toptaxaG_fec <- tax_table(fecGen20)
fecGen20 = phyloseq(topotuG_fec, toptaxaG_fec)
fecGen20 = merge_phyloseq(sample_data(ps_ra_feces_gen), fecGen20)
# Grouping other Genilies
OtherG_fec <- names(sort(taxa_sums(ps_ra_feces_gen), TRUE)[c(21:888)])
fecG_other <- prune_taxa(OtherG_fec, ps_ra_feces_gen)
others_otu_fec <- t(as.matrix(apply(otu_table(fecG_other), 2, sum)))
rownames(others_otu_fec) <- "Others Genera"
others_otu_fec=otu_table(others_otu_fec, taxa_are_rows = TRUE)
other_taxG_fec <- c("Others", "Others","Others","Others","Others","Others","Others", "Others")
other_taxG_fec <- as.matrix(other_taxG_fec)
rownames(other_taxG_fec) <- colnames(tax_table(fecG_other))
other_taxG_fec <- t(other_taxG_fec)
rownames(other_taxG_fec) <- "Others Genera"
other_taxG_fec =  tax_table(other_taxG_fec)
others_fecG= phyloseq(others_otu_fec,other_taxG_fec)
others_fecG = merge_phyloseq(sample_data(ps_ra_feces_gen), others_fecG)
others_fecG
# Regrouping phyloseq
fecG <- merge_phyloseq(fecGen20, others_fecG)
fecG_melt <- psmelt(fecG)

fecG_melt <- fecG_melt %>%
  mutate(Name = case_when(
    str_detect(Genus, "GGB", negate = TRUE) ~ Genus,
    str_detect(Genus, "GGB", negate = FALSE) ~ paste(Genus, Family)
  ))

sort(unique(fecG_melt$Name))

fecG_melt$Stunting_status[which(fecG_melt$Stunting_status=="Non_stunted")] <- "Non-stunted"

boxplot_feces <- ggplot(fecG_melt, aes(x = Name, y = Abundance)) +
  geom_boxplot(aes(fill = Name), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Name), size =1) +
  scale_color_manual(values = pal_feces) +
  scale_fill_manual(values = pal_feces) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9, face = 3)) +
  ggtitle("Fecal microbiota composition - Metagenomic data - Top 20 Genera")

#ggsave(filename = "Figures/Boxplot_feces_METAG_satellite.svg", plot = boxplot_feces, width = 10, height = 8)

barplot_feces <- fecG_melt %>%
  ggplot(aes_string(x = "id_modif", y = "Abundance", fill = "Name")) +
  geom_bar(stat = "identity", position = "stack") + 
  facet_grid(~Stunting_status, scales = "free_x") +
  scale_fill_manual(values = pal_feces) +
  theme_bw() + 
  theme(legend.position = "right", 
        axis.title.x=element_blank(),
        axis.text.x=element_text(angle = 60, hjust = 1),
        axis.ticks.x=element_blank()) +
  guides(fill=guide_legend(ncol=2))+
  ylab("Relative abundance")+
  ggtitle("Fecal microbiota composition")+
  labs(fill = "Genus") +
  theme(
    axis.text.x =  element_blank(),
    axis.text.y = element_text(size = 11),
    axis.title = element_text(size = 11),
    plot.title = element_text(size = 11),
    legend.title = element_text(size = 11),
    legend.text = element_text(size = 11, face = 3),
    strip.text = element_text(size = 11))

pdf("Figures/Composition_BarPlot_feces.pdf", width = 16, height = 4)
barplot_feces
dev.off()
```

### Top 10
Saliva
```{r}
ps_ra_saliva <- subset_samples(physeq_ra, body_site =="Saliva")
ps_ra_saliva <- prune_taxa(taxa_sums(ps_ra_saliva)> 0, ps_ra_saliva)
ps_ra_saliva_gen <- tax_glom(ps_ra_saliva, "Genus") # 221 genera

TopGen_sal <- names(sort(taxa_sums(ps_ra_saliva_gen), TRUE)[1:10])
SalGen10 <- prune_taxa(TopGen_sal, ps_ra_saliva_gen)
topotuG_sal <- otu_table(SalGen10)
toptaxaG_sal <- tax_table(SalGen10)
SalGen10 = phyloseq(topotuG_sal, toptaxaG_sal)
SalGen10 = merge_phyloseq(sample_data(ps_ra_saliva_gen), SalGen10)
# Grouping other Genus
OtherG_sal <- names(sort(taxa_sums(ps_ra_saliva_gen), TRUE)[c(11:221)])
SalG_other <- prune_taxa(OtherG_sal, ps_ra_saliva_gen)
others_otu_sal <- t(as.matrix(apply(otu_table(SalG_other), 2, sum)))
rownames(others_otu_sal) <- "Others Genera"
others_otu_sal=otu_table(others_otu_sal, taxa_are_rows = TRUE)
other_taxG_sal <- c("Others", "Others","Others","Others","Others","Others","Others", "Others")
other_taxG_sal <- as.matrix(other_taxG_sal)
rownames(other_taxG_sal) <- colnames(tax_table(SalG_other))
other_taxG_sal <- t(other_taxG_sal)
rownames(other_taxG_sal) <- "Others Genera"
other_taxG_sal =  tax_table(other_taxG_sal)
others_salG= phyloseq(others_otu_sal,other_taxG_sal)
others_salG = merge_phyloseq(sample_data(ps_ra_saliva_gen), others_salG)
others_salG
# Regrouping phyloseq
salG10 <- merge_phyloseq(SalGen10, others_salG)
salG10_melt <- psmelt(salG10)

unique(salG10_melt$Genus)

# Fix unassigned SGB names to higher taxonomic ranks
salG10_melt <- salG10_melt %>%
  mutate(Name = case_when(
    str_detect(Genus, "GGB", negate = TRUE) ~ Genus,
    str_detect(Genus, "GGB", negate = FALSE) ~ paste("GGB ", Family)
  ))

boxplot_oral10 <- ggplot(salG10_melt, aes(x = Name, y = Abundance)) +
  geom_boxplot(aes(fill = Name), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Name), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9, face = 3)) +
  ggtitle("SALIVA")


ggsave(filename = "Figures/Boxplot10_oral_METAG_satellite.svg", plot = boxplot_oral10, width = 6 , height = 8)
```
Gastric microbiota
```{r}
ps_ra_ga <- subset_samples(physeq_ra, body_site =="Gastric")
ps_ra_ga <- prune_taxa(taxa_sums(ps_ra_ga)> 0, ps_ra_ga)
ps_ra_ga_gen <- tax_glom(ps_ra_ga, "Genus") # 191 genera

# Top 10 Genera
TopGen_ga <- names(sort(taxa_sums(ps_ra_ga_gen), TRUE)[1:10])
gaGen10 <- prune_taxa(TopGen_ga, ps_ra_ga_gen)
topotuG_ga <- otu_table(gaGen10)
toptaxaG_ga <- tax_table(gaGen10)
gaGen10 = phyloseq(topotuG_ga, toptaxaG_ga)
gaGen10 = merge_phyloseq(sample_data(ps_ra_ga_gen), gaGen10)
# Grouping other Genilies
OtherG_ga <- names(sort(taxa_sums(ps_ra_ga_gen), TRUE)[c(11:203)])
gaG_other <- prune_taxa(OtherG_ga, ps_ra_ga_gen)
others_otu_ga <- t(as.matrix(apply(otu_table(gaG_other), 2, sum)))
rownames(others_otu_ga) <- "Others Genera"
others_otu_ga=otu_table(others_otu_ga, taxa_are_rows = TRUE)
other_taxG_ga <- c("Others", "Others","Others","Others","Others","Others","Others", "Others")
other_taxG_ga <- as.matrix(other_taxG_ga)
rownames(other_taxG_ga) <- colnames(tax_table(gaG_other))
other_taxG_ga <- t(other_taxG_ga)
rownames(other_taxG_ga) <- "Others Genera"
other_taxG_ga =  tax_table(other_taxG_ga)
others_gaG= phyloseq(others_otu_ga,other_taxG_ga)
others_gaG = merge_phyloseq(sample_data(ps_ra_ga_gen), others_gaG)
others_gaG
# Regrouping phyloseq
gaGen10 <- merge_phyloseq(gaGen10, others_gaG)
gaG10_melt <- psmelt(gaGen10)

gaG10_melt <- gaG10_melt %>%
  mutate(Name = case_when(
    str_detect(Genus, "GGB", negate = TRUE) ~ Genus,
    str_detect(Genus, "GGB", negate = FALSE) ~ paste("GGB ", Family)
  ))


boxplot_ga <- ggplot(gaG10_melt, aes(x = Name, y = Abundance)) +
  geom_boxplot(aes(fill = Name), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Name), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9, face = 3)) +
  ggtitle("GASTRIC ASPIRATES")

ggsave(filename = "Figures/Boxplot10_ga_METAG_satellite.svg", plot = boxplot_ga, width = 6, height = 8)
```
Duodenal microbiota
```{r}
ps_ra_du <- subset_samples(physeq_ra, body_site =="Duodenum" )
ps_ra_du <- prune_taxa(taxa_sums(ps_ra_du)> 0, ps_ra_du)
ps_ra_du_gen <- tax_glom(ps_ra_du, "Genus") # 203 genera

# Top 20 Genera
TopGen_du <- names(sort(taxa_sums(ps_ra_du_gen), TRUE)[1:10])
duGen10 <- prune_taxa(TopGen_du, ps_ra_du_gen)
topotuG_du <- otu_table(duGen10)
toptaxaG_du <- tax_table(duGen10)
duGen10 = phyloseq(topotuG_du, toptaxaG_du)
duGen10 = merge_phyloseq(sample_data(ps_ra_du_gen), duGen10)
# Grouping other Genilies
OtherG_du <- names(sort(taxa_sums(ps_ra_du_gen), TRUE)[c(11:203)])
duG_other <- prune_taxa(OtherG_du, ps_ra_du_gen)
others_otu_du <- t(as.matrix(apply(otu_table(duG_other), 2, sum)))
rownames(others_otu_du) <- "Others Genera"
others_otu_du=otu_table(others_otu_du, taxa_are_rows = TRUE)
other_taxG_du <- c("Others", "Others","Others","Others","Others","Others","Others", "Others")
other_taxG_du <- as.matrix(other_taxG_du)
rownames(other_taxG_du) <- colnames(tax_table(duG_other))
other_taxG_du <- t(other_taxG_du)
rownames(other_taxG_du) <- "Others Genera"
other_taxG_du =  tax_table(other_taxG_du)
others_duG= phyloseq(others_otu_du,other_taxG_du)
others_duG = merge_phyloseq(sample_data(ps_ra_du_gen), others_duG)
others_duG
# Regrouping phyloseq
duG10 <- merge_phyloseq(duGen10, others_duG)
duG10_melt <- psmelt(duG10)

duG10_melt <- duG10_melt %>%
  mutate(Name = case_when(
    str_detect(Genus, "GGB", negate = TRUE) ~ Genus,
    str_detect(Genus, "GGB", negate = FALSE) ~ paste("GGB ", Family)
  ))


boxplot_du10 <- ggplot(duG10_melt, aes(x = Name, y = Abundance)) +
  geom_boxplot(aes(fill = Name), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Name), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9, face = 3)) +
  ggtitle("DUODENAL ASPIRATES")

ggsave(filename = "Figures/Boxplot10_du_METAG_satellite.svg", plot = boxplot_du10, width = 6, height = 8)
```
Fecal microbiota
```{r}
ps_ra_feces <- subset_samples(physeq_ra, body_site =="Feces")
ps_ra_feces <- prune_taxa(taxa_sums(ps_ra_feces)> 0, ps_ra_feces)
ps_ra_feces_gen <- tax_glom(ps_ra_feces, "Genus") # 888 genera

TopGen_fec <- names(sort(taxa_sums(ps_ra_feces_gen), TRUE)[1:10])
fecGen10 <- prune_taxa(TopGen_fec, ps_ra_feces_gen)
topotuG_fec <- otu_table(fecGen10)
toptaxaG_fec <- tax_table(fecGen10)
fecGen10 = phyloseq(topotuG_fec, toptaxaG_fec)
fecGen10 = merge_phyloseq(sample_data(ps_ra_feces_gen), fecGen10)
# Grouping other Genilies
OtherG_fec <- names(sort(taxa_sums(ps_ra_feces_gen), TRUE)[c(11:888)])
fecG_other <- prune_taxa(OtherG_fec, ps_ra_feces_gen)
others_otu_fec <- t(as.matrix(apply(otu_table(fecG_other), 2, sum)))
rownames(others_otu_fec) <- "Others Genera"
others_otu_fec=otu_table(others_otu_fec, taxa_are_rows = TRUE)
other_taxG_fec <- c("Others", "Others","Others","Others","Others","Others","Others", "Others")
other_taxG_fec <- as.matrix(other_taxG_fec)
rownames(other_taxG_fec) <- colnames(tax_table(fecG_other))
other_taxG_fec <- t(other_taxG_fec)
rownames(other_taxG_fec) <- "Others Genera"
other_taxG_fec =  tax_table(other_taxG_fec)
others_fecG= phyloseq(others_otu_fec,other_taxG_fec)
others_fecG = merge_phyloseq(sample_data(ps_ra_feces_gen), others_fecG)
others_fecG
# Regrouping phyloseq
fecG10 <- merge_phyloseq(fecGen10, others_fecG)
fecG10_melt <- psmelt(fecG10)

fecG10_melt <- fecG10_melt %>%
  mutate(Name = case_when(
    str_detect(Genus, "GGB", negate = TRUE) ~ Genus,
    str_detect(Genus, "GGB", negate = FALSE) ~ paste("GGB ", Family)
  ))

boxplot_feces <- ggplot(fecG10_melt, aes(x = Name, y = Abundance)) +
  geom_boxplot(aes(fill = Name), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Name), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9, face = 3)) +
  ggtitle("STOOL")

ggsave(filename = "Figures/Boxplot10_feces_METAG_satellite.svg", plot = boxplot_feces, width = 6, height = 8)
```

# Differential abundance analysis
Simple statistical test:
Wilcoxon test: Non-parametric test comparing two groups. As it is sensitive the sample size, if the data has many zeros or small effect size, wilcoxon may not detect differences

Linear models (LM): fits a regression model to each taxon (lm(Abundance ~ Group)). Assumes normal distribution of residuals which is often violated in microbiome data

Compositional Data Methods: 
ALDEx2: Bayesian method using Monte Carlo sampling to account for compositional effects. Focus on effect size rather than raw p-values. If the effect size are small, ALDEx2 might find no significant taxa.

ANCOM: uses log-ration transformation to identify taxa that differ significantly across group. Very conservative, if many taxa change together, ANCOM might fail to detect them. Avoid using this one!

ANCOM-BC: adjust for compositional bias while controlling for false discoveries. More powerful than ANCOM but still strict, so fewer taxa might pass significance thresholds
Prevalence cutoff (prv_cut): exclude taxa with prevalence lower than this threshold
Structural zeros (struc_zero): detect taxa totally absent from one group if set to true, and treat them differently

Count-Based Methods:
DESeq2: Normalize counts using geometric mean, models count data with a negative binomial distribution, and uses Wald tests for differential abundance. More sensitive for low-abundance taxa because it models variance across samples. Can be inflated if library size vary greatly.
Hard time trusting it for metagenomic data

LEfSe: uses Kruskal-Wallis + LDA Effect Size to find biomarkers. Can be biased by low-abundance taxa and high variance taxa. It may not be as reproducible as DESeq2.

SIAMCAT: Machine learning-based approach for microbiome classification. Detect complex patterns beyond simple statistical differences. May be overfitting if not properly tuned.

Maaslin3: GLM-based method that models relative abundance while accounting for covariates


Potential issues: 
Small effect size (i.e. strenght of relationship): examples: wilcoxon, LM and aldex2 don't handle low-abundance taxa wll and require strong difference to detect significance.
Compositional bias: aldex2 assumes that relative abundances must be corrected. If library size vary, simple tests may not be valid.
Multiple Testing Correction: Bonferroni or FDR correction may be too strict in high dimensional data.


## Saliva
```{r}
diffabun_sa <- subset_samples(physeq_ra, body_site == "Saliva")
diffabun_sa <- tax_glom(diffabun_sa, "Species")
diffabun_sa <- subset_taxa(diffabun_sa, taxa_sums(diffabun_sa) > 0)
tax_table(diffabun_sa) <- tax_table(diffabun_sa)[, -8]

tax_df_sa <- as.data.frame(tax_table(diffabun_sa))
tax_df_sa <- tax_df_sa %>%
  mutate(Species = case_when(
    !str_detect(Species, "GGB") ~ Species,
    !str_detect(Genus, "GGB") ~ Species,
    !str_detect(Family, "FGB") ~ paste(Family, Species),
    !str_detect(Order, "OFGB") ~ paste(Order, Species),
    !str_detect(Class, "CFGB") ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species),
    TRUE ~ Species
  ))
tax_table(diffabun_sa) <- as.matrix(tax_df_sa)

diffabun_sa_m <- psmelt(diffabun_sa)
```
Wilcoxon test
```{r}
diffabun_sa_wil <- diffabun_sa_m %>%
  group_by(Species) %>%
  summarise(pval = wilcox.test(Abundance ~ Stunting_status, exact = FALSE)$p.value,.groups = "drop")
  
diffabun_sa_wil <- diffabun_sa_wil %>%
  mutate(pval_adj = p.adjust(pval, method = "BH"))

significant_taxa_sa_wil <- diffabun_sa_wil %>%
  filter(pval_adj < 0.05)  # Keep only significant results
significant_taxa_sa_wil
```
LM
```{r}
lm_model_sa <- diffabun_sa_m %>%
  group_by(Species) %>%
  summarise(model = list(lm(Abundance ~ Stunting_status, data = cur_data())),
    .groups = "drop")

lm_model_sa <- lm_model_sa %>%
  mutate(summary = map(model, summary))

lm_model_sa <- lm_model_sa %>%
  mutate(
    coeff = map(summary, ~ coef(.x)),  # Extracts coefficient table
    estimate = map_dbl(coeff, ~ .["Stunting_statusStunted", "Estimate"]),  # Effect size
    std_error = map_dbl(coeff, ~ .["Stunting_statusStunted", "Std. Error"]),  # Std error
    p_value = map_dbl(coeff, ~ .["Stunting_statusStunted", "Pr(>|t|)"])  # p-value
  ) %>%
  select(Species, estimate, std_error, p_value)  # Keep relevant columns

lm_model_sa <- lm_model_sa %>%
  mutate(p_value_adj = p.adjust(p_value, method = "BH"))

significant_taxa_sa <- lm_model_sa %>%
  filter(p_value_adj < 0.05)  # Keep only significant results
significant_taxa_sa
```
Lefse
```{r}
lef_out_sa <- run_lefse(diffabun_sa, wilcoxon_cutoff = 0.01, group = "Stunting_status", taxa_rank = "Species",
                        kw_cutoff = 0.05, norm = "CPM", multigrp_strat = TRUE, lda_cutoff = 2)

marker_table(lef_out_sa)$pdaj_bh <- p.adjust(marker_table(lef_out_sa)$pvalue, method = "BH")
pdf("Figures/Lefse_SalivaStunting_metag.pdf", height = 6, width = 6)
plot_ef_bar(lef_out_sa)
dev.off()
```
ANCOM-BC
```{r}
ancombc_out_sa <- run_ancombc(diffabun_sa, group = "Stunting_status", taxa_rank = "Species",
                              p_adjust = "BH")
# Warning: No marker was identified
# Trying contiuous variable: haz
ancombc_out_sa_cont <- ancombc(
  diffabun_sa,
  formula = "haz_cont",
  p_adj_method = "BH",
  prv_cut = 0.1,
  tax_level = "Species"
)
View(ancombc_out_sa_cont$res$q_val)
# SGB298
diffabun_sa_melt <- psmelt(diffabun_sa)
diffabun_sa_melt_sgb <- diffabun_sa_melt %>% 
  filter(OTU == "SGB298")

pdf("Figures/ANCOMBC_Salivahaz_Sgordonii.pdf", height = 6, width = 6)
ggplot(diffabun_sa_melt_sgb, aes(x = haz_cont, y = Abundance + 1e-6)) +
  geom_point(color = "#3366CC", alpha = 0.7) +
  geom_smooth(method = "lm", color = "darkred", se = TRUE) +
  labs(
    title = "Streptococcus gordonii",
    x = "height-for-age z-score",
    y = "Abundance"
  ) +
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    plot.title = element_text(size = 14, face = 3),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    strip.text = element_text(size = 14),
  )
dev.off()
```
Aldex
```{r}
aldex_out_sa <- run_aldex(diffabun_sa, group = "Stunting_status", taxa_rank = "Species")
plot_ef_bar(aldex_out_sa)
```
DESeq2
```{r}
diffabun_abs_sa <- subset_samples(physeq_abs, body_site == "Saliva")
diffabun_abs_sa <- tax_glom(diffabun_abs_sa, "Species")
diffabun_abs_sa <- subset_taxa(diffabun_abs_sa, taxa_sums(diffabun_abs_sa) > 0)
tax_table(diffabun_abs_sa) <- tax_table(diffabun_abs_sa)[, -8]

tax_dfa_sa <- as.data.frame(tax_table(diffabun_abs_sa))
tax_dfa_sa <- tax_dfa_sa %>%
  mutate(Species = case_when(
    !str_detect(Species, "GGB") ~ Species,
    !str_detect(Genus, "GGB") ~ Species,
    !str_detect(Family, "FGB") ~ paste(Family, Species),
    !str_detect(Order, "OFGB") ~ paste(Order, Species),
    !str_detect(Class, "CFGB") ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species),
    TRUE ~ Species
  ))
tax_table(diffabun_abs_sa) <- as.matrix(tax_dfa_sa)

dds_sa <- phyloseq_to_deseq2(diffabun_abs_sa, ~ Stunting_status)
dds_sa <- DESeq(dds_sa)
res_sa <- results(dds_sa, contrast = c("Stunting_status", "Stunted", "Non_stunted"))
res_sa_df <- as.data.frame(res_sa)
res_sa_df <- cbind(res_sa_df, tax_table(diffabun_abs_sa))

sig_res_sa_df <- res_sa_df %>% filter(padj < 0.05)

sig_res_sa_df <- sig_res_sa_df %>% arrange(log2FoldChange)

# Convert Species to factor (for ordered plotting)
sig_res_sa_df$Species <- factor(sig_res_sa_df$Species, levels = sig_res_sa_df$Species)

pdf("Figures/Deseq2_SalivaStunting_metag.pdf", height = 8, width = 10)
ggplot(sig_res_sa_df, aes(x = Species, y = log2FoldChange, size = -log10(padj), color = log2FoldChange)) +
  geom_point(alpha = 0.8) + 
  coord_flip() +  # Horizontal plot
  theme_minimal() +
  scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0) +
  labs(title = "Differentially Abundant Taxa (Saliva - DESeq2)",
       x = "Species", 
       y = "Log2 Fold Change", 
       size = "-log10(p-value)")
dev.off()
```

## Gastric
```{r}
diffabun_ga <- subset_samples(physeq_ra, body_site == "Gastric")
diffabun_ga <- tax_glom(diffabun_ga, "Species")
diffabun_ga <- subset_taxa(diffabun_ga, taxa_sums(diffabun_ga) > 0)
tax_table(diffabun_ga) <- tax_table(diffabun_ga)[, -8]

tax_df_ga <- as.data.frame(tax_table(diffabun_ga))
tax_df_ga <- tax_df_ga %>%
  mutate(Species = case_when(
    !str_detect(Species, "GGB") ~ Species,
    !str_detect(Genus, "GGB") ~ Species,
    !str_detect(Family, "FGB") ~ paste(Family, Species),
    !str_detect(Order, "OFGB") ~ paste(Order, Species),
    !str_detect(Class, "CFGB") ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species),
    TRUE ~ Species
  ))
tax_table(diffabun_ga) <- as.matrix(tax_df_ga)


diffabun_ga_m <- psmelt(diffabun_ga)
```
Wilcoxon test
```{r}
diffabun_ga_wil <- diffabun_ga_m %>%
  group_by(Species) %>%
  summarise(pval = wilcox.test(Abundance ~ Stunting_status, exact = FALSE)$p.value,.groups = "drop")
  
diffabun_ga_wil <- diffabun_ga_wil %>%
  mutate(pval_adj = p.adjust(pval, method = "BH"))

significant_taxa_ga_wil <- diffabun_ga_wil %>%
  filter(pval_adj < 0.05)  # Keep only significant results
significant_taxa_ga_wil
```
LM
```{r}
lm_model_ga <- diffabun_ga_m %>%
  group_by(Species) %>%
  summarise(model = list(lm(Abundance ~ Stunting_status, data = cur_data())),
    .groups = "drop")

lm_model_ga <- lm_model_ga %>%
  mutate(summary = map(model, summary))

lm_model_ga <- lm_model_ga %>%
  mutate(
    coeff = map(summary, ~ coef(.x)),  # Extracts coefficient table
    estimate = map_dbl(coeff, ~ .["Stunting_statusStunted", "Estimate"]),  # Effect size
    std_error = map_dbl(coeff, ~ .["Stunting_statusStunted", "Std. Error"]),  # Std error
    p_value = map_dbl(coeff, ~ .["Stunting_statusStunted", "Pr(>|t|)"])  # p-value
  ) %>%
  select(Species, estimate, std_error, p_value)  # Keep relevant columns

lm_model_ga <- lm_model_ga %>%
  mutate(p_value_adj = p.adjust(p_value, method = "BH"))

significant_taxa_ga <- lm_model_ga %>%
  filter(p_value_adj < 0.05)  # Keep only significant results
significant_taxa_ga
```
Lefse
```{r}
lef_out_ga <- run_lefse(diffabun_ga, wilcoxon_cutoff = 0.01, group = "Stunting_status", taxa_rank = "Species",
                        kw_cutoff = 0.05, norm = "CPM", multigrp_strat = TRUE, lda_cutoff = 2)
marker_table(lef_out_ga)$pdaj_bh <- p.adjust(marker_table(lef_out_ga)$pvalue, method = "BH")
marker_table(lef_out_ga)$feature[which(marker_table(lef_out_ga)$pdaj_bh<0.05)]
pdf("Figures/Lefse_GastricStunting_metag.pdf", height = 6, width = 6)
plot_ef_bar(lef_out_ga)
dev.off()
```
ANCOM-BC
```{r}
ancombc_out_ga <- run_ancombc(diffabun_ga, group = "Stunting_status", taxa_rank = "Species",
                              p_adjust = "BH")
# Warning: No marker was identified
# Trying contiuous variable: haz
ancombc_out_ga_cont <- ancombc(
  diffabun_ga,
  formula = "haz_cont",
  p_adj_method = "fdr",
  prv_cut = 0.1,
  tax_level = "Species"
)
View(ancombc_out_ga_cont$res$q_val)
# No significant taxa
```
Aldex
```{r}
aldex_out_ga <- run_aldex(diffabun_ga, group = "Stunting_status", taxa_rank = "Species")
plot_ef_bar(aldex_out_ga)
```
DESeq2
```{r}
diffabun_abs_ga <- subset_samples(physeq_abs, body_site == "Gastric")
diffabun_abs_ga <- tax_glom(diffabun_abs_ga, "Species")
diffabun_abs_ga <- subset_taxa(diffabun_abs_ga, taxa_sums(diffabun_abs_ga) > 0)
tax_table(diffabun_abs_ga) <- tax_table(diffabun_abs_ga)[, -8]

tax_dfa_ga <- as.data.frame(tax_table(diffabun_abs_ga))
tax_dfa_ga <- tax_dfa_ga %>%
  mutate(Species = case_when(
    !str_detect(Species, "GGB") ~ Species,
    !str_detect(Genus, "GGB") ~ Species,
    !str_detect(Family, "FGB") ~ paste(Family, Species),
    !str_detect(Order, "OFGB") ~ paste(Order, Species),
    !str_detect(Class, "CFGB") ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species),
    TRUE ~ Species
  ))
tax_table(diffabun_abs_ga) <- as.matrix(tax_dfa_ga)

dds_ga <- phyloseq_to_deseq2(diffabun_abs_ga, ~ Stunting_status)
dds_ga <- DESeq(dds_ga)
res_ga <- results(dds_ga, contrast = c("Stunting_status", "Stunted", "Non_stunted"))
res_ga_df <- as.data.frame(res_ga)
res_ga_df <- cbind(res_ga_df, tax_table(diffabun_abs_ga))

sig_res_ga_df <- res_ga_df %>% filter(padj < 0.05)

sig_res_ga_df <- sig_res_ga_df %>% arrange(log2FoldChange)

# Convert Species to factor (for ordered plotting)
sig_res_ga_df$Species <- factor(sig_res_ga_df$Species, levels = sig_res_ga_df$Species)

pdf("Figures/Deseq2_GastricStunting_metag.pdf", height = 8, width = 10)
ggplot(sig_res_ga_df, aes(x = Species, y = log2FoldChange, size = -log10(padj), color = log2FoldChange)) +
  geom_point(alpha = 0.8) + 
  coord_flip() +  # Horizontal plot
  theme_minimal() +
  scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0) +
  labs(title = "Differentially Abundant Taxa (Gastric - DESeq2)",
       x = "Species", 
       y = "Log2 Fold Change", 
       size = "-log10(p-value)")
dev.off()
```

## Duodenal
```{r}
diffabun_du <- subset_samples(physeq_ra, body_site == "Duodenum")
diffabun_du <- tax_glom(diffabun_du, "Species")
diffabun_du <- subset_taxa(diffabun_du, taxa_sums(diffabun_du) > 0)
tax_table(diffabun_du) <- tax_table(diffabun_du)[, -8]

tax_df_du <- as.data.frame(tax_table(diffabun_du))
tax_df_du <- tax_df_du %>%
  mutate(Species = case_when(
    !str_detect(Species, "GGB") ~ Species,
    !str_detect(Genus, "GGB") ~ Species,
    !str_detect(Family, "FGB") ~ paste(Family, Species),
    !str_detect(Order, "OFGB") ~ paste(Order, Species),
    !str_detect(Class, "CFGB") ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species),
    TRUE ~ Species
  ))
tax_table(diffabun_du) <- as.matrix(tax_df_du)

# Trying contiuous variable: haz
ancombc_out_du_cont <- ancombc(
  diffabun_du,
  formula = "haz_cont",
  p_adj_method = "BH",
  prv_cut = 0.1,
  tax_level = "Species"
)
View(ancombc_out_du_cont$res$q_val)

ancombc_out_du_cont$res$q_val$taxon[which(ancombc_out_du_cont$res$q_val$haz_cont<0.05)]
diffabun_du_m <- psmelt(diffabun_du)
unique(diffabun_du_m$Species[which(diffabun_du_m$OTU=="SGB326")])
unique(diffabun_du_m$Species[which(diffabun_du_m$OTU=="SGB451")])
unique(diffabun_du_m$Species[which(diffabun_du_m$OTU=="SGB543")])
unique(diffabun_du_m$Species[which(diffabun_du_m$OTU=="SGB584")])
unique(diffabun_du_m$Species[which(diffabun_du_m$OTU=="SGB1182")])

pdf("Figures/ANCOMBC_Duohaz_Solobacterium.pdf", height = 6, width = 6)
diffabun_du_m_SGB326 <- diffabun_du_m %>% 
  filter(OTU == "SGB326")
ggplot(diffabun_du_m_SGB326, aes(x = haz_cont, y = Abundance + 1e-6)) +
  geom_point(color = "#3366CC", alpha = 0.7) +
  geom_smooth(method = "lm", color = "darkred", se = TRUE) +
  labs(
    title = "Solobacterium unclassified",
    x = "height-for-age z-score",
    y = "Abundance"
  ) +
  theme_minimal()+
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    plot.title = element_text(size = 14, face = 3),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    strip.text = element_text(size = 14),
  )
dev.off()

pdf("Figures/ANCOMBC_Duohaz_Prevotellaceae.pdf", height = 6, width = 6)
diffabun_du_m_SGB543 <- diffabun_du_m %>% 
  filter(OTU == "SGB543")
ggplot(diffabun_du_m_SGB543, aes(x = haz_cont, y = Abundance + 1e-6)) +
  geom_point(color = "#3366CC", alpha = 0.7) +
  geom_smooth(method = "lm", color = "darkred", se = TRUE) +
  labs(
    title = "Prevotellaceae unclassified",
    x = "height-for-age z-score",
    y = "Abundance"
  ) +
  theme_minimal()+
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    plot.title = element_text(size = 14, face = 3),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    strip.text = element_text(size = 14),
  )
dev.off()

diffabun_du_m_SGB584 <- diffabun_du_m %>% 
  filter(OTU == "SGB584")
pdf("Figures/ANCOMBC_Duohaz_Tserpentiformis.pdf", height = 6, width = 6)
ggplot(diffabun_du_m_SGB584, aes(x = haz_cont, y = Abundance + 1e-6)) +
  geom_point(color = "#3366CC", alpha = 0.7) +
  geom_smooth(method = "lm", color = "darkred", se = TRUE) +
  labs(
    title = "Tannerella serpentiformis",
    x = "height-for-age z-score",
    y = "Abundance"
  ) +
  theme_minimal()+
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    plot.title = element_text(size = 14, face = 3),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    strip.text = element_text(size = 14),
  )
dev.off()

pdf("Figures/ANCOMBC_Duohaz_Mnonliquefaciens.pdf", height = 6, width = 6)
diffabun_du_m_SGB1182 <- diffabun_du_m %>% 
  filter(OTU == "SGB1182")
ggplot(diffabun_du_m_SGB1182, aes(x = haz_cont, y = Abundance + 1e-6)) +
  geom_point(color = "#3366CC", alpha = 0.7) +
  geom_smooth(method = "lm", color = "darkred", se = TRUE) +
  labs(
    title = "Moraxella nonliquefaciens",
    x = "height-for-age z-score",
    y = "Abundance"
  ) +
  theme_minimal()+
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    plot.title = element_text(size = 14, face = 3),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    strip.text = element_text(size = 14),
  )
dev.off()



pdf("Figures/ANCOMBC_Duohaz_Porphyromonadaceae.pdf", height = 6, width = 6)
diffabun_du_m_SGB451 <- diffabun_du_m %>% 
  filter(OTU == "SGB451")
ggplot(diffabun_du_m_SGB584, aes(x = haz_cont, y = Abundance + 1e-6)) +
  geom_point(color = "#3366CC", alpha = 0.7) +
  geom_smooth(method = "lm", color = "darkred", se = TRUE) +
  labs(
    title = "Porphyromonadaceae unclassified",
    x = "height-for-age z-score",
    y = "Abundance"
  ) +
  theme_minimal()+
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    plot.title = element_text(size = 14, face = 3),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    strip.text = element_text(size = 14),
  )
dev.off()
```

## Stool
```{r}
diffabun_fe <- subset_samples(physeq_ra, body_site == "Feces")
diffabun_fe <- tax_glom(diffabun_fe, "Species")
diffabun_fe <- subset_taxa(diffabun_fe, taxa_sums(diffabun_fe) > 0)
tax_table(diffabun_fe) <- tax_table(diffabun_fe)[, -8]

tax_df_fe <- as.data.frame(tax_table(diffabun_fe))
tax_df_fe <- tax_df_fe %>%
  mutate(Species = case_when(
    !str_detect(Species, "GGB") ~ Species,
    !str_detect(Genus, "GGB") ~ Species,
    !str_detect(Family, "FGB") ~ paste(Family, Species),
    !str_detect(Order, "OFGB") ~ paste(Order, Species),
    !str_detect(Class, "CFGB") ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species),
    TRUE ~ Species
  ))
tax_table(diffabun_fe) <- as.matrix(tax_df_fe)


diffabun_fe_m <- psmelt(diffabun_fe)
```
Wilcoxon
```{r}
diffabun_fe_wil <- diffabun_fe_m %>%
  group_by(Species) %>%
  summarise(pval = wilcox.test(Abundance ~ Stunting_status, exact = FALSE)$p.value,.groups = "drop")
  
diffabun_fe_wil <- diffabun_fe_wil %>%
  mutate(pval_adj = p.adjust(pval, method = "BH"))

significant_taxa_fe_wil <- diffabun_fe_wil %>%
  filter(pval_adj < 0.05)  # Keep only significant results
significant_taxa_fe_wil
```
LM
```{r}
lm_model_fe <- diffabun_fe_m %>%
  group_by(Species) %>%
  summarise(model = list(lm(Abundance ~ Stunting_status, data = cur_data())),
    .groups = "drop")

lm_model_fe <- lm_model_fe %>%
  mutate(summary = map(model, summary))

lm_model_fe <- lm_model_fe %>%
  mutate(
    coeff = map(summary, ~ coef(.x)),  # Extracts coefficient table
    estimate = map_dbl(coeff, ~ .["Stunting_statusStunted", "Estimate"]),  # Effect size
    std_error = map_dbl(coeff, ~ .["Stunting_statusStunted", "Std. Error"]),  # Std error
    p_value = map_dbl(coeff, ~ .["Stunting_statusStunted", "Pr(>|t|)"])  # p-value
  ) %>%
  select(Species, estimate, std_error, p_value)  # Keep relevant columns

lm_model_fe <- lm_model_fe %>%
  mutate(p_value_adj = p.adjust(p_value, method = "BH"))

significant_taxa_fe <- lm_model_fe %>%
  filter(p_value_adj < 0.05)  # Keep only significant results
significant_taxa_fe
```
Lefse
```{r}
lef_out_fe <- run_lefse(diffabun_fe, wilcoxon_cutoff = 0.01, group = "Stunting_status", taxa_rank = "Species",
                        kw_cutoff = 0.05, norm = "CPM", multigrp_strat = TRUE, lda_cutoff = 2)
marker_table(lef_out_fe)$pdaj_bh <- p.adjust(marker_table(lef_out_fe)$pvalue, method = "BH")
marker_table(lef_out_fe)$feature[which(marker_table(lef_out_fe)$pdaj_bh<0.05)]
pdf("Figures/Lefse_StoolStunting_metag.pdf", height = 6, width = 6)
plot_ef_bar(lef_out_fe)
dev.off()
```
ANCOM-BC
```{r}
ancombc_out_fe <- run_ancombc(diffabun_fe, group = "Stunting_status", taxa_rank = "Species",
                              p_adjust = "BH")
# Warning: No marker was identified
# Trying contiuous variable: haz
ancombc_out_fe_cont <- ancombc(
  diffabun_fe,
  formula = "haz_cont",
  p_adj_method = "BH",
  prv_cut = 0.1,
  tax_level = "Species"
)
View(ancombc_out_fe_cont$res$q_val)
# N.B. SGB156 still significative after correction with age, calprotectin inflammation, AAT inflammation
diffabun_fe_melt <- psmelt(diffabun_fe)
diffabun_fe_melt_sgb <- diffabun_fe_melt %>% 
  filter(OTU == "SGB156")
Bwads_haz <- ggplot(diffabun_fe_melt_sgb, aes(x = haz_cont, y = Abundance + 1e-6)) +
  geom_point(color = "#3366CC", alpha = 0.7) +
  geom_smooth(method = "lm", color = "darkred", se = TRUE) +
  labs(
    title = "Bilophila wadsworthia",
    x = "height-for-age z-score",
    y = "Abundance"
  ) +
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 14),
    axis.text.y = element_text(size = 14),
    axis.title = element_text(size = 14),
    plot.title = element_text(size = 14, face = 3),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 14),
    strip.text = element_text(size = 14),
  )

pdf("Figures/ANCOMBC_StoolStunting_metaghaz.pdf", height = 6, width = 6)
Bwads_haz
dev.off()


# Test calpro
ancombc_out_fe_calpro <- ancombc(
  diffabun_fe,
  formula = "CALPROTECTINE..g.g.",
  p_adj_method = "BH",
  prv_cut = 0.1,
  tax_level = "Species"
)
View(ancombc_out_fe_calpro$res$q_val)
# E. coli 

# Test AAT
ancombc_out_fe_aat <- ancombc(
  diffabun_fe,
  formula = "AAT...mg.g.",
  p_adj_method = "BH",
  prv_cut = 0.1,
  tax_level = "Species"
)
# Nothing
```
Aldex
```{r}
aldex_out_fe <- run_aldex(diffabun_fe, group = "Stunting_status", taxa_rank = "Species")
plot_ef_bar(aldex_out_fe)
```
DESeq2
```{r}
diffabun_abs_fe <- subset_samples(physeq_abs, body_site == "Feces")
diffabun_abs_fe <- tax_glom(diffabun_abs_fe, "Species")
diffabun_abs_fe <- subset_taxa(diffabun_abs_fe, taxa_sums(diffabun_abs_fe) > 0)
tax_table(diffabun_abs_fe) <- tax_table(diffabun_abs_fe)[, -8]

tax_dfa_fe <- as.data.frame(tax_table(diffabun_abs_fe))
tax_dfa_fe <- tax_dfa_fe %>%
  mutate(Species = case_when(
    !str_detect(Species, "GGB") ~ Species,
    !str_detect(Genus, "GGB") ~ Species,
    !str_detect(Family, "FGB") ~ paste(Family, Species),
    !str_detect(Order, "OFGB") ~ paste(Order, Species),
    !str_detect(Class, "CFGB") ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species),
    TRUE ~ Species
  ))
tax_table(diffabun_abs_fe) <- as.matrix(tax_dfa_fe)

dds_fe <- phyloseq_to_deseq2(diffabun_abs_fe, ~ Stunting_status)
dds_fe <- DESeq(dds_fe)
res_fe <- results(dds_fe, contrast = c("Stunting_status", "Stunted", "Non_stunted"))
res_fe_df <- as.data.frame(res_fe)
res_fe_df <- cbind(res_fe_df, tax_table(diffabun_abs_fe))

sig_res_fe_df <- res_fe_df %>% filter(padj < 0.05)

sig_res_fe_df <- sig_res_fe_df %>% arrange(log2FoldChange)

# Convert Species to factor (for ordered plotting)
sig_res_fe_df$Species <- factor(sig_res_fe_df$Species, levels = sig_res_fe_df$Species)

pdf("Figures/Deseq2_StoolStunting_metag.pdf", height = 14, width = 10)
ggplot(sig_res_fe_df, aes(x = Species, y = log2FoldChange, size = -log10(padj), color = log2FoldChange)) +
  geom_point(alpha = 0.8) + 
  coord_flip() +  # Horizontal plot
  theme_minimal() +
  scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0) +
  labs(title = "Differentially Abundant Taxa (Stool - DESeq2)",
       x = "Species", 
       y = "Log2 Fold Change", 
       size = "-log10(p-value)")
dev.off()
```



# Shared species

To do for heatmap: 
Split the heatmap: keep only the mostly and highly shared for the main figure
Create a table with the rest 

Do venn diagram but change color based on figure 1

WATCH OUT FOR DUODNEUM IT"S 14 SAMPLES NOW!!!!!

Compartment specific species and shared species 
```{r}
physeq_cat <- tax_glom(physeq_ra, taxrank = "Species")
physeq_cat <- prune_taxa(taxa_sums(physeq_cat)> 0, physeq_cat)
# 1906 species in the dataset, 128 samples
```

Classifying species 
1) predominantly in oral samples (= oral origin)
2) predominantly fecal (=fecal origin)
3) prevalent in both fecal and oral samples (=shared)

For children with paired saliva and fecal samples
First remove children for which we do not have paired saliva and fecal sample

Should add a coverage filter (Schmidt et al.)
average vertical coverage (depth) >= 0.25
Average vertical coverage (sequencing depth) and horizontal coverage (breadth) per microbial genome in each sample were quantified using the qaCompute utility in metaSNV

OR a filter to select only the species for which we have a MAG
(might have to change the metaphlan table to GTDB taxonomy and merge them)
## Classification
Saliva - Feces
```{r}
# Remove samples without paired saliva - stool samples
physeq_sp_sf <- subset_samples(physeq_cat, id != "1429AGN007" & id != "1429AGN012" & id != "1429AGN014" & id != "1429AGN029" & id != "1429AGN040")
#Keep only saliva and stool samples --> 78 samples = 39 individuals
physeq_sp_sf <- subset_samples(physeq_sp_sf, body_site == "Saliva" | body_site == "Feces")
#physeq_sp_sf <- subset_taxa(physeq_sp_sf, taxa_sums(physeq_sp_sf)/78 > 10^-6)
# Per taxa total abundance filtering at 10^-4
physeq_sp_sf <- subset_taxa(physeq_sp_sf, taxa_sums(physeq_sp_sf)/78 > 10^-4)
# Prevalence filtering
prevdf_sp_sf = apply(X= otu_table(physeq_sp_sf),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_sf),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_sf = data.frame(Prevalence = prevdf_sp_sf,
                        Percentage_Prev_feces = 100*prevdf_sp_sf/78,
                        tax_table(physeq_sp_sf))
# Filter for taxa in at least 10% of the samples (saliva + stool samples)
physeq_sp_sf = subset_taxa(physeq_sp_sf,  rownames(tax_table(physeq_sp_sf)) %in% rownames(prevdf_sp_sf)[which(prevdf_sp_sf$Percentage_Prev >= 10)])
# 547 taxa have prevalence > 10% in fecal and saliva sample AND average relative abundance > 10^-4
# For now, only prevalence and abundance filter and only saliva and feces
# Compute prevalence and abundance per compartment
# Prevalence and abundance in feces for all species
physeq_sp_fe <- subset_samples(physeq_sp_sf, body_site == "Feces")
prevdf_sp_fe = apply(X= otu_table(physeq_sp_fe),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_fe),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_fe = data.frame(Prevalence_feces = prevdf_sp_fe,
                       Percentage_Prev_feces = 100*prevdf_sp_fe/39,
                       Mean_Abundance_feces = 100*taxa_sums(physeq_sp_fe)/39,
                       Total_Abundance_feces = taxa_sums(physeq_sp_fe),
                        tax_table(physeq_sp_fe))
prevdf_sp_fe <- prevdf_sp_fe[, -12]
# Prevalence and abundance in saliva for all species
physeq_sp_sa <- subset_samples(physeq_sp_sf, body_site == "Saliva")
prevdf_sp_sa = apply(X= otu_table(physeq_sp_sa),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_sa),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_sa = data.frame(Prevalence_saliva = prevdf_sp_sa,
                       Percentage_Prev_saliva = 100*prevdf_sp_sa/39,
                       Mean_Abundance_saliva = 100*taxa_sums(physeq_sp_sa)/39,
                       Total_Abundance_saliva = taxa_sums(physeq_sp_sa),
                        tax_table(physeq_sp_sa))
prevdf_sp_sa <- prevdf_sp_sa[, -12]
# Remerging dataframe saliva + stool
prevdf_sp <- cbind(prevdf_sp_sa[,c(1:4)], prevdf_sp_fe)
# Adding categories:
prevdf_sp <- prevdf_sp %>%
  mutate(Intersect_category = "")
# Unique to a compartment: 0 prevalence in the other
prevdf_sp$Intersect_category[which(prevdf_sp$Percentage_Prev_saliva > 0 
                                    & prevdf_sp$Percentage_Prev_feces == 0)] <- "Uniquely oral"
prevdf_sp$Intersect_category[which(prevdf_sp$Percentage_Prev_saliva == 0 
                                    & prevdf_sp$Percentage_Prev_feces > 0)] <- "Uniquely fecal"
# Low prevalence taxa: <10 in both compartment 
prevdf_sp$Intersect_category[which(prevdf_sp$Percentage_Prev_saliva < 10 & prevdf_sp$Percentage_Prev_feces < 10)] <- "Low prevalence"
# Completely shared taxa: 100% prevalence in both compartment
prevdf_sp$Intersect_category[which(prevdf_sp$Percentage_Prev_saliva == 100 & prevdf_sp$Percentage_Prev_feces == 100)] <- "Completely shared"
# Compartmental origin: >10 in one and <10 in the other
prevdf_sp$Intersect_category[which(prevdf_sp$Percentage_Prev_saliva > 10 
                                    & prevdf_sp$Percentage_Prev_feces < 10 & prevdf_sp$Percentage_Prev_feces > 0)] <- "Oral origin"
prevdf_sp$Intersect_category[which(prevdf_sp$Percentage_Prev_saliva < 10 & prevdf_sp$Percentage_Prev_saliva > 0 
                                    & prevdf_sp$Percentage_Prev_feces > 10)] <- "Fecal origin"
# Often shared: found in both compartment with >10 and <60
prevdf_sp$Intersect_category[which(prevdf_sp$Percentage_Prev_saliva >= 10 & prevdf_sp$Percentage_Prev_saliva < 100
                                    & prevdf_sp$Percentage_Prev_feces >= 10 )] <- "Mostly shared"
prevdf_sp$Intersect_category[which(prevdf_sp$Percentage_Prev_saliva >= 10
                                    & prevdf_sp$Percentage_Prev_feces >= 10 & prevdf_sp$Percentage_Prev_feces < 100)] <- "Mostly shared"
# Highly shared: >60 and < 100
prevdf_sp$Intersect_category[which(prevdf_sp$Percentage_Prev_saliva >= 60 & prevdf_sp$Percentage_Prev_saliva < 100
                                    & prevdf_sp$Percentage_Prev_feces >= 60)] <- "Highly shared"
prevdf_sp$Intersect_category[which(prevdf_sp$Percentage_Prev_saliva >= 60 
                                    & prevdf_sp$Percentage_Prev_feces >= 60 & prevdf_sp$Percentage_Prev_feces < 100)] <- "Highly shared"
# Change category as factor
prevdf_sp$Intersect_category <- as.factor(prevdf_sp$Intersect_category)
# Create levels and order them
prevdf_sp$Intersect_category <- factor(prevdf_sp$Intersect_category, levels=c("Uniquely oral", "Oral origin", "Low prevalence",
                                                                                "Mostly shared", "Highly shared", "Completely shared",
                                                                                "Fecal origin", "Uniquely fecal"), ordered = TRUE)
prevdf_sp_ordered <- prevdf_sp[order(prevdf_sp$Intersect_category, prevdf_sp$Percentage_Prev_saliva, prevdf_sp$Percentage_Prev_feces),]

# Rename unasigned species (SGB) with the higher taxnomic rank
prevdf_sp_ordered <- prevdf_sp_ordered %>%
  mutate(Name = case_when(
    str_detect(Species, "GGB", negate = TRUE) ~ Species,
    str_detect(Genus, "GGB", negate = TRUE) ~ Species,
    str_detect(Family, "FGB", negate = TRUE) ~ paste(Family,Species),
    str_detect(Order, "OFGB", negate = TRUE) ~ paste(Order, Species),
    str_detect(Class, "CFGB", negate = TRUE) ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species)
  ))

#write_csv(prevdf_sp_ordered, "Tables/MetaG_PrevAbun_SalivaStool.csv")

# Count table per intersect category
df_shared_sf <- prevdf_sp_ordered %>%
  group_by(Intersect_category) %>%
  summarise(n_species = n_distinct(Name),
            sum_meanAbundance_sa = sum(Mean_Abundance_saliva),
            sum_meanAbundance_fe = sum(Mean_Abundance_feces))
# Save table
#write_csv(df_shared_sf, "Tables/MetaG_intersect_cat_count.csv")

# Venn diagram 
# Order body sites
sample_data(physeq_sp_sf)$body_site <- factor(
  sample_data(physeq_sp_sf)$body_site,
  levels = c("Saliva", "Feces")  # Replace with the actual levels present
)
```

Gastric - Duodenum analysis 
```{r}
physeq_sp_gd <- subset_samples(physeq_cat, body_site == "Duodenum" | body_site == "Gastric")
# Remove samples with unpaired saliva and feces
physeq_sp_gd <- subset_samples(physeq_sp_gd, id %in% intersect(md_ps_final$id[which(md_ps_final$body_site =="Gastric")],
          md_ps_final$id[which(md_ps_final$body_site =="Duodenum")]))

physeq_sp_gd <- subset_taxa(physeq_sp_gd, taxa_sums(physeq_sp_gd)/22 > 10^-4)

prevdf_sp_gd = apply(X= otu_table(physeq_sp_gd),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_gd),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_gd = data.frame(Prevalence = prevdf_sp_gd,
                        Percentage_Prev = 100*prevdf_sp_gd/22,
                        tax_table(physeq_sp_gd))

physeq_sp_gd = subset_taxa(physeq_sp_gd,  rownames(tax_table(physeq_sp_gd)) %in% rownames(prevdf_sp_gd)[which(prevdf_sp_gd$Percentage_Prev >= 10)])
# 253 taxa

# Prevalence and abundance in gastric for all species
physeq_sp_gd_ga <- subset_samples(physeq_sp_gd, body_site == "Gastric")
prevdf_sp_gd_ga = apply(X= otu_table(physeq_sp_gd_ga),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_gd_ga),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_gd_ga = data.frame(Prevalence_ga = prevdf_sp_gd_ga,
                       Percentage_Prev_ga = 100*prevdf_sp_gd_ga/11,
                       Mean_Abundance_ga = 100*taxa_sums(physeq_sp_gd_ga)/11,
                       Total_Abundance_ga = taxa_sums(physeq_sp_gd_ga),
                        tax_table(physeq_sp_gd_ga))
prevdf_sp_gd_ga <- prevdf_sp_gd_ga[, -12]


# Prevalence and abundance in duodenal for all species
physeq_sp_gd_da <- subset_samples(physeq_sp_gd, body_site == "Duodenum")
prevdf_sp_gd_da = apply(X= otu_table(physeq_sp_gd_da),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_gd_da),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_gd_da = data.frame(Prevalence_da = prevdf_sp_gd_da,
                       Percentage_Prev_da = 100*prevdf_sp_gd_da/11,
                       Mean_Abundance_da = 100*taxa_sums(physeq_sp_gd_da)/11,
                       Total_Abundance_da = taxa_sums(physeq_sp_gd_da),
                        tax_table(physeq_sp_gd_da))
prevdf_sp_gd_da <- prevdf_sp_gd_da[, -12]


# Merging dataset
prevdf_sp_gd <- cbind(prevdf_sp_gd_ga[,c(1:4)], prevdf_sp_gd_da)

# Adding categories:
prevdf_sp_gd <- prevdf_sp_gd %>%
  mutate(Intersect_category = "")
prevdf_sp_gd$Intersect_category[which(prevdf_sp_gd$Percentage_Prev_ga > 10 
                                    & prevdf_sp_gd$Percentage_Prev_da < 10)] <- "Gastric origin"

prevdf_sp_gd$Intersect_category[which(prevdf_sp_gd$Percentage_Prev_ga < 10
                                    & prevdf_sp_gd$Percentage_Prev_da > 10)] <- "Duodenal origin"
#Unique to a body site
prevdf_sp_gd$Intersect_category[which(prevdf_sp_gd$Percentage_Prev_ga > 0 
                                    & prevdf_sp_gd$Percentage_Prev_da == 0)] <- "Uniquely gastric"

prevdf_sp_gd$Intersect_category[which(prevdf_sp_gd$Percentage_Prev_ga == 0 
                                    & prevdf_sp_gd$Percentage_Prev_da > 0)] <- "Uniquely duodenal"

prevdf_sp_gd$Intersect_category[which(prevdf_sp_gd$Percentage_Prev_ga < 10 
                                   & prevdf_sp_gd$Percentage_Prev_da < 10)] <- "Low prevalence"

# Shared
prevdf_sp_gd$Intersect_category[which(prevdf_sp_gd$Percentage_Prev_ga >= 10
                                    & prevdf_sp_gd$Percentage_Prev_da >= 10)] <- "Mostly shared"

prevdf_sp_gd$Intersect_category[which(prevdf_sp_gd$Percentage_Prev_ga >= 60
                                    & prevdf_sp_gd$Percentage_Prev_da >= 60)] <- "Highly shared"

prevdf_sp_gd$Intersect_category[which(prevdf_sp_gd$Percentage_Prev_ga == 100 & 
                                         prevdf_sp_gd$Percentage_Prev_da == 100)] <- "Completely shared"

prevdf_sp_gd$Intersect_category <- as.factor(prevdf_sp_gd$Intersect_category)
prevdf_sp_gd$Intersect_category <- factor(prevdf_sp_gd$Intersect_category, levels=c("Uniquely gastric", "Gastric origin",
                                                                                "Mostly shared", "Highly shared", "Completely shared",
                                                                                "Duodenal origin", "Uniquely duodenal"), ordered = TRUE)


prevdf_sp_gd_ordered <- prevdf_sp_gd[order(prevdf_sp_gd$Intersect_category, -prevdf_sp_gd$Percentage_Prev_ga,
                                             prevdf_sp_gd$Percentage_Prev_ga),]

# Rename unasigned species (SGB)
prevdf_sp_gd_ordered <- prevdf_sp_gd_ordered %>%
  mutate(Name = case_when(
    str_detect(Species, "GGB", negate = TRUE) ~ Species,
    str_detect(Genus, "GGB", negate = TRUE) ~ Species,
    str_detect(Family, "FGB", negate = TRUE) ~ paste(Family,Species),
    str_detect(Order, "OFGB", negate = TRUE) ~ paste(Order, Species),
    str_detect(Class, "CFGB", negate = TRUE) ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species)
  ))

#write_csv(prevdf_sp_gd_ordered, "Tables/MetaG_PrevAbun_GastricDuodenum.csv")

# Count table per intersect category
df_shared_gd <- prevdf_sp_gd_ordered %>%
  group_by(Intersect_category) %>%
  summarise(n_species = n_distinct(Name),
            sum_meanAbundance_ga = sum(Mean_Abundance_ga),
            sum_meanAbundance_du = sum(Mean_Abundance_da))

#write_csv(df_shared_gd, "Tables/MetaG_intersect_cat_count_gd.csv")

# Order body sites
sample_data(physeq_sp_gd)$body_site <- factor(
  sample_data(physeq_sp_gd)$body_site,
  levels = c("Gastric", "Duodenum")  # Replace with the actual levels present
)
```

Samples with saliva, upper GIT and fecal samples
```{r}
# Keep children with saliva - gastric - stool samples: 27 children
# It automatically keeps the 11 duodenal samples - all children with a duodenal samples have a gastric sample
physeq_sp_sgf <- subset_samples(physeq_cat, id %in% intersect(intersect(md_ps_final$id[which(md_ps_final$body_site =="Gastric")],
          md_ps_final$id[which(md_ps_final$body_site =="Saliva")]),
          md_ps_final$id[which(md_ps_final$body_site =="Feces")]))

# Don't remove duodenal samples now: gastric + duodenal = Upper GIT
# physeq_sp_sgf <- subset_samples(physeq_sp_sgf, body_site == "Saliva" | body_site == "Feces" | body_site == "Gastric")
# 27 children have at least saliva, gastric and fecal samples, and 11 of them have a duodenal sample
# 92 samples

# Abundance filtering
physeq_sp_sgf <- subset_taxa(physeq_sp_sgf, taxa_sums(physeq_sp_sgf)/92 > 10^-4)
# 589 taxa (6 less than if you remove duodenal)
# Prevalence filtering
prevdf_sp_sgf = apply(X= otu_table(physeq_sp_sgf),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_sgf),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_sgf = data.frame(Prevalence = prevdf_sp_sgf,
                        Percentage_Prev = 100*prevdf_sp_sgf/92,
                        tax_table(physeq_sp_sgf))
physeq_sp_sgf = subset_taxa(physeq_sp_sgf,
                            rownames(tax_table(physeq_sp_sgf)) %in% rownames(prevdf_sp_sgf)[which(prevdf_sp_sgf$Percentage_Prev >= 10)])
# 455 taxa have prevalence > 10% AND average relative abundance > 10^-4

# Rename Gastric and Duodenum as Upper GIT
sample_data(physeq_sp_sgf)$body_site[which(sample_data(physeq_sp_sgf)$body_site=="Gastric" |
                                             sample_data(physeq_sp_sgf)$body_site=="Duodenum")] <- "Upper GIT"
# 38 samples in Upper GIT

# Prevalence and abundance in feces for all species
physeq_sp_sgf_fe <- subset_samples(physeq_sp_sgf, body_site == "Feces")
prevdf_sp_sgf_fe = apply(X= otu_table(physeq_sp_sgf_fe),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_sgf_fe),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_sgf_fe = data.frame(Prevalence_feces = prevdf_sp_sgf_fe,
                       Percentage_Prev_feces = 100*prevdf_sp_sgf_fe/27,
                       Mean_Abundance_feces = 100*taxa_sums(physeq_sp_sgf_fe)/27,
                       Total_Abundance_feces = taxa_sums(physeq_sp_sgf_fe),
                        tax_table(physeq_sp_sgf_fe))
prevdf_sp_sgf_fe <- prevdf_sp_sgf_fe[, -12]

# Prevalence and abundance in saliva for all species
physeq_sp_sgf_sa <- subset_samples(physeq_sp_sgf, body_site == "Saliva")
prevdf_sp_sgf_sa = apply(X= otu_table(physeq_sp_sgf_sa),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_sgf_sa),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_sgf_sa = data.frame(Prevalence_saliva = prevdf_sp_sgf_sa,
                       Percentage_Prev_saliva = 100*prevdf_sp_sgf_sa/27,
                       Mean_Abundance_saliva = 100*taxa_sums(physeq_sp_sgf_sa)/27,
                       Total_Abundance_saliva = taxa_sums(physeq_sp_sgf_sa),
                        tax_table(physeq_sp_sgf_sa))
prevdf_sp_sgf_sa <- prevdf_sp_sgf_sa[, -12]

# Prevalence and abundance in upper GIT for all species
physeq_sp_sgf_ga <- subset_samples(physeq_sp_sgf, body_site == "Upper GIT")
prevdf_sp_sgf_ga = apply(X= otu_table(physeq_sp_sgf_ga),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_sgf_ga),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_sgf_ga = data.frame(Prevalence_ga = prevdf_sp_sgf_ga,
                       Percentage_Prev_ga = 100*prevdf_sp_sgf_ga/38,
                       Mean_Abundance_ga = 100*taxa_sums(physeq_sp_sgf_ga)/38,
                       Total_Abundance_ga = taxa_sums(physeq_sp_sgf_ga),
                        tax_table(physeq_sp_sgf_ga))
prevdf_sp_sgf_ga <- prevdf_sp_sgf_ga[, -12]

# Merging dataset
prevdf_sp_sgf <- cbind(prevdf_sp_sgf_sa[,c(1:4)], prevdf_sp_sgf_fe[,c(1:4)], prevdf_sp_sgf_ga)

# Adding categories:
prevdf_sp_sgf <- prevdf_sp_sgf %>%
  mutate(Intersect_category = "")
prevdf_sp_sgf$Intersect_category[which(prevdf_sp_sgf$Percentage_Prev_saliva > 10 
                                    & prevdf_sp_sgf$Percentage_Prev_feces < 10
                                    & prevdf_sp_sgf$Percentage_Prev_ga < 10)] <- "Oral origin"

prevdf_sp_sgf$Intersect_category[which(prevdf_sp_sgf$Percentage_Prev_saliva < 10
                                    & prevdf_sp_sgf$Percentage_Prev_feces > 10 
                                    & prevdf_sp_sgf$Percentage_Prev_ga < 10)] <- "Fecal origin"

prevdf_sp_sgf$Intersect_category[which(prevdf_sp_sgf$Percentage_Prev_saliva < 10
                                    & prevdf_sp_sgf$Percentage_Prev_ga > 10 
                                    & prevdf_sp_sgf$Percentage_Prev_feces < 10 )] <- "Upper GIT origin"
#Unique to a body site
prevdf_sp_sgf$Intersect_category[which(prevdf_sp_sgf$Percentage_Prev_saliva > 0 
                                    & prevdf_sp_sgf$Percentage_Prev_feces == 0
                                     & prevdf_sp_sgf$Percentage_Prev_ga == 0)] <- "Uniquely oral"
prevdf_sp_sgf$Intersect_category[which(prevdf_sp_sgf$Percentage_Prev_saliva == 0 
                                    & prevdf_sp_sgf$Percentage_Prev_feces > 0
                                     & prevdf_sp_sgf$Percentage_Prev_ga == 0)] <- "Uniquely fecal"
prevdf_sp_sgf$Intersect_category[which(prevdf_sp_sgf$Percentage_Prev_ga > 0 
                                    & prevdf_sp_sgf$Percentage_Prev_feces == 0
                                     & prevdf_sp_sgf$Percentage_Prev_saliva == 0)] <- "Uniquely upper GIT"


prevdf_sp_sgf$Intersect_category[which(prevdf_sp_sgf$Percentage_Prev_saliva < 10 
                                   & prevdf_sp_sgf$Percentage_Prev_feces < 10 &
                                     prevdf_sp_sgf$Percentage_Prev_ga < 10)] <- "Low prevalence"

# Shared by pairs
prevdf_sp_sgf$Intersect_category[which(prevdf_sp_sgf$Percentage_Prev_saliva > 10 
                                    & prevdf_sp_sgf$Percentage_Prev_feces < 10
                                    & prevdf_sp_sgf$Percentage_Prev_ga > 10)] <- "Shared in upper digestive tract"
prevdf_sp_sgf$Intersect_category[which(prevdf_sp_sgf$Percentage_Prev_saliva < 10
                                    & prevdf_sp_sgf$Percentage_Prev_feces > 10 
                                    & prevdf_sp_sgf$Percentage_Prev_ga > 10)] <- "Shared in lower digestive tract"
prevdf_sp_sgf$Intersect_category[which(prevdf_sp_sgf$Percentage_Prev_saliva > 10
                                    & prevdf_sp_sgf$Percentage_Prev_feces > 10 
                                    & prevdf_sp_sgf$Percentage_Prev_ga < 10)] <- "Saliva-feces"

prevdf_sp_sgf$Intersect_category[which(prevdf_sp_sgf$Percentage_Prev_saliva >= 10
                                    & prevdf_sp_sgf$Percentage_Prev_feces >= 10 
                                    & prevdf_sp_sgf$Percentage_Prev_ga >= 10 )] <- "Mostly shared"


prevdf_sp_sgf$Intersect_category[which(prevdf_sp_sgf$Percentage_Prev_saliva >= 60
                                    & prevdf_sp_sgf$Percentage_Prev_feces >= 60
                                    & prevdf_sp_sgf$Percentage_Prev_ga >= 60)] <- "Highly shared"

# No taxa are present in all samples! 1 gastric sample does not have Streptococcus salivarius
prevdf_sp_sgf$Intersect_category[which(prevdf_sp_sgf$Percentage_Prev_saliva == 100 & 
                                         prevdf_sp_sgf$Percentage_Prev_feces == 100 &
                                         prevdf_sp_sgf$Percentage_Prev_ga == 100)] <- "Completely shared"

prevdf_sp_sgf$Intersect_category <- as.factor(prevdf_sp_sgf$Intersect_category)
prevdf_sp_sgf$Intersect_category <- factor(prevdf_sp_sgf$Intersect_category, levels=c("Uniquely oral", "Oral origin",
                                                                                      "Shared in upper digestive tract", 
                                                                              "Upper GIT origin", "Uniquely upper GIT",
                                                                                "Mostly shared", "Highly shared", "Completely shared",
                                                                              "Shared in lower digestive tract",
                                                                                "Fecal origin", "Uniquely fecal", "Saliva-feces"), ordered = TRUE)

prevdf_sp_sgf_ordered <- prevdf_sp_sgf[order(prevdf_sp_sgf$Intersect_category, -prevdf_sp_sgf$Percentage_Prev_saliva,
                                             prevdf_sp_sgf$Percentage_Prev_ga, prevdf_sp_sgf$Percentage_Prev_feces),]

# Rename unasigned species (SGB)
prevdf_sp_sgf_ordered <- prevdf_sp_sgf_ordered %>%
  mutate(Name = case_when(
    str_detect(Species, "GGB", negate = TRUE) ~ Species,
    str_detect(Genus, "GGB", negate = TRUE) ~ Species,
    str_detect(Family, "FGB", negate = TRUE) ~ paste(Family,Species),
    str_detect(Order, "OFGB", negate = TRUE) ~ paste(Order, Species),
    str_detect(Class, "CFGB", negate = TRUE) ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species)
  ))

write_csv(prevdf_sp_sgf_ordered, "Tables/MetaG_PrevAbun_SalivaUpperGITFeces.csv")

# Count table per intersect category
df_shared_sgf <- prevdf_sp_sgf_ordered %>%
  group_by(Intersect_category) %>%
  summarise(n_species = n_distinct(Name),
            sum_meanAbundance_sa = sum(Mean_Abundance_saliva),
            sum_meanAbundance_fe = sum(Mean_Abundance_feces),
            sum_meanAbundance_ga = sum(Mean_Abundance_ga))

write_csv(df_shared_sgf, "Tables/MetaG_intersect_cat_count_sgf.csv")

# Order body sites
sample_data(physeq_sp_sgf)$body_site <- factor(
  sample_data(physeq_sp_sgf)$body_site,
  levels = c("Saliva", "Upper GIT", "Feces")
)
```

## Venn diagrams 
Saliva - feces
```{r}
pdf("Figures/Venndiagramm_FecesOral_metag.pdf", height = 6, width = 6)
MicEco::ps_venn(physeq_sp_sf,
        group = "body_site",
        plot = TRUE, 
        fill = c("#9AB9C2","#DCC04B", "white"),
        label=list(cex=2),
        quantities = list(cex=4)
        )
dev.off()
```

Gastric - duodenum
```{r}
pdf("Figures/Venndiagramm_GastricDuodenum_metag.pdf", height = 6, width = 6)
MicEco::ps_venn(physeq_sp_gd,
        group = "body_site",
        plot = TRUE, 
        fill = c("#BBDEEC", "#EDE692","white"),
        label=list(cex=2),
        quantities = list(cex=4)
        )
dev.off()
```

Saliva - Upper GIT - Feces
```{r}
# Venn diagram 
pdf("Figures/Venndiagramm_FecesOralUpperGIT_metag.pdf", height = 8, width = 8)
MicEco::ps_venn(physeq_sp_sgf,
        group = "body_site",
        plot = TRUE, 
        fill = c("#9AB9C2", "#B8CA7A", "#DCC04B"),
        label=list(cex=2),
        quantities = list(cex=4)
        )
dev.off()
```


## Heatmaps
saliva - feces prevalence
saliva - feces
```{r}
prevdf_sp_hm <- prevdf_sp_ordered
rownames(prevdf_sp_hm) <- prevdf_sp_hm$Name

# Set color range
col_fun = colorRamp2(c(0,100), c("white","#8f3b3b"))

# Matrix for heatmap
hm_tab_prev <- t(prevdf_sp_hm[,c(2, 6)])
rownames(hm_tab_prev)[1] <- "Saliva"
rownames(hm_tab_prev)[2] <- "Feces"

# Set row annotations
row_annot_prev <- rowAnnotation(ls = prevdf_sp_hm$Intersect_category,
                                  col = list(ls=c("Uniquely oral" = "#9AB9C2",
                                                  "Oral origin" = "#c2d5da",
                                                  "Mostly shared" = "#e6aaaa",
                                                  "Highly shared" = "#cd5555",
                                                  "Completely shared" = "#8f3b3b",
                                                  "Fecal origin" = "#ead993",
                                                  "Uniquely fecal" = "#DCC04B")),
                                  annotation_label = " ", annotation_name_side = "top", annotation_name_rot = 90)

# Heatmap
heatmap_sharing_prev <- Heatmap(t(hm_tab_prev), name = "Prevalence",
                             row_order =  rownames(prevdf_sp_hm), column_order = c("Saliva", "Feces"),
        col = col_fun, left_annotation = row_annot_prev, show_column_dend = FALSE, show_row_names =  FALSE, column_names_side = "top",
        column_names_rot = 90, border = FALSE, row_title = NULL)    

pdf("Figures/Heatmap_Prevalence_SaliveStool.pdf", height = 6, width = 3)
heatmap_sharing_prev
dev.off()
```
saliva - upper GIT - feces
```{r}
prevdf_sp_all <- prevdf_sp_sgf_ordered
prevdf_sp_all$Name <- str_replace_all(prevdf_sp_all$Name, "_", " ")
rownames(prevdf_sp_all) <- prevdf_sp_all$Name

# Split heatmap into 3 heatmaps: oral, fecal, shared
# Keep only shared taxa (highly)
prevdf_sp_hm_sharedsf <- prevdf_sp_all %>% filter(Intersect_category %in% c("Highly shared"))

# Set color range
col_fun = colorRamp2(c(0,0,0.1,1,5,10), c("white","white", "#B8DE29FF", "#29AF7FFF","#33638DFF", "#440154FF"))

# Sharing Saliva - Feces HM
hm_tab_sharedsf <- t(prevdf_sp_hm_sharedsf[,c(3, 11, 7)])
rownames(hm_tab_sharedsf)[1] <- "Saliva"
rownames(hm_tab_sharedsf)[2] <- "Upper GIT"
rownames(hm_tab_sharedsf)[3] <- "Feces"

# Split
split_hm_shared = rep("Highly shared", 14)
split_hm_shared <- as.factor(split_hm_shared)

heatmap_sharing_shared <- Heatmap(t(hm_tab_sharedsf), name = "Mean relative abundance", row_split = split_hm_shared,
                             row_order =  rownames(prevdf_sp_hm_sharedsf),
        col = col_fun, show_column_dend = FALSE, row_names_side = "right", column_names_side = "top",
        column_names_rot = 90, border = TRUE, row_title = NULL, row_names_gp = gpar(fontsize = 11, fontface = 3),
        heatmap_legend_param = list(direction = "horizontal", legend_width = unit(4,"cm")))   
# Save
pdf("Figures/Heatmap_sharing_SalivaUpperGITStool_METAG.pdf", height = 6, width = 4)
draw(heatmap_sharing_shared, merge_legend = TRUE, heatmap_legend_side = "bottom")
dev.off()
```


# Old heatmaps
saliva - feces
```{r}
prevdf_sp_hm <- prevdf_sp_ordered
rownames(prevdf_sp_hm) <- prevdf_sp_hm$Name

# Split heatmap into 3 heatmaps: oral, fecal, shared
prevdf_sp_hm_oral <- prevdf_sp_hm %>% filter(Intersect_category %in% c("Uniquely oral", "Oral origin"))
prevdf_sp_hm_fecal <- prevdf_sp_hm %>% filter(Intersect_category %in% c("Uniquely fecal", "Fecal origin"))
prevdf_sp_hm_sharedsf <- prevdf_sp_hm %>% filter(Intersect_category %in% c("Mostly shared", "Highly shared", "Completely shared"))

# Set color range
max(prevdf_sp_hm$Mean_Abundance_feces) # 9.8
max(prevdf_sp$Mean_Abundance_saliva) # 6.6
col_fun = colorRamp2(c(0,0,0.1,1,5,10), c("white","white", "#B8DE29FF", "#29AF7FFF","#33638DFF", "#440154FF"))

# Oral HM
# Matrix for heatmap
hm_tab_oral <- t(prevdf_sp_hm_oral[,c(3, 7)])
rownames(hm_tab_oral)[1] <- "Saliva"
rownames(hm_tab_oral)[2] <- "Feces"
# Set row annotations
row_annot_oral <- rowAnnotation(ls = prevdf_sp_hm_oral$Intersect_category,
                                  col = list(ls=c("Uniquely oral" = "khaki2",
                                                  "Oral origin" = "orange2")),
                                  annotation_label = "Sharing", annotation_name_side = "top", annotation_name_rot = 45)
# Set split
split_hm_oral = rep("Uniquely oral", 168)
split_hm_oral[110:168] = "Oral origin"
split_hm_oral <- as.factor(split_hm_oral)
split_hm_oral <- factor(split_hm_oral, levels=c("Uniquely oral", "Oral origin"), ordered = TRUE)
# Heatmap
heatmap_sharing_oral <- Heatmap(t(hm_tab_oral), name = "Mean relative abundance", row_split = split_hm_oral,
                             row_order =  rownames(prevdf_sp_hm_oral),
        col = col_fun, right_annotation = row_annot_oral, show_column_dend = FALSE, row_names_side = "left", column_names_side = "top",
        column_names_rot = 45, border = TRUE, row_title = NULL, row_names_gp = gpar(fontsize = 4, fontface = 3))    
# Save
pdf("Figures/Heatmap_sharing_Oral_METAG.pdf", height = 12, width = 6)
heatmap_sharing_oral
dev.off()

# Feces HM
hm_tab_fecal <- t(prevdf_sp_hm_fecal[,c(3, 7)])
rownames(hm_tab_fecal)[1] <- "Saliva"
rownames(hm_tab_fecal)[2] <- "Feces"
# Set row annotations
row_annot_fecal <- rowAnnotation(ls = prevdf_sp_hm_fecal$Intersect_category,
                                  col = list(ls=c("Fecal origin" = "darkcyan",
                                                  "Uniquely fecal" = "lightblue2" )),
                                  annotation_label = "Sharing", annotation_name_side = "top", annotation_name_rot = 45)
# Split 
split_hm_fecal = rep("Fecal origin", 308)
split_hm_fecal[8:308] = "Uniquely fecal"
split_hm_fecal <- as.factor(split_hm_fecal)
split_hm_fecal <- factor(split_hm_fecal, levels=c("Fecal origin", "Uniquely fecal"), ordered = TRUE)

heatmap_sharing_fecal <- Heatmap(t(hm_tab_fecal), name = "Mean relative abundance", row_split = split_hm_fecal,
                             row_order =  rownames(prevdf_sp_hm_fecal),
        col = col_fun, right_annotation = row_annot_fecal, show_column_dend = FALSE, row_names_side = "left", column_names_side = "top",
        column_names_rot = 45, border = TRUE, row_title = NULL, row_names_gp = gpar(fontsize = 4, fontface = 3))    
# Save
pdf("Figures/Heatmap_sharing_Feces_METAG.pdf", height = 18, width = 6)
heatmap_sharing_fecal
dev.off()


# Sharing Saliva - Feces HM
hm_tab_sharedsf <- t(prevdf_sp_hm_sharedsf[,c(3, 7)])
rownames(hm_tab_sharedsf)[1] <- "Saliva"
rownames(hm_tab_sharedsf)[2] <- "Feces"
# Set row annotations
row_annot_shared <- rowAnnotation(ls = prevdf_sp_hm_sharedsf$Intersect_category,
                                  col = list(ls=c("Mostly shared" = "pink2",
                                                  "Highly shared" = "indianred3",
                                                  "Completely shared" = "#440154FF")),
                                  annotation_label = "Sharing", annotation_name_side = "top", annotation_name_rot = 45)
# Split
split_hm_shared = rep("Mostly shared", 71)
split_hm_shared[59:70] = "Highly shared"
split_hm_shared[71] = "Completely shared"
split_hm_shared <- as.factor(split_hm_shared)
split_hm_shared <- factor(split_hm_shared, levels=c("Mostly shared", "Highly shared", "Completely shared"), ordered = TRUE)

heatmap_sharing_shared <- Heatmap(t(hm_tab_sharedsf), name = "Mean relative abundance", row_split = split_hm_shared,
                             row_order =  rownames(prevdf_sp_hm_sharedsf),
        col = col_fun, right_annotation = row_annot_shared, show_column_dend = FALSE, row_names_side = "left", column_names_side = "top",
        column_names_rot = 45, border = TRUE, row_title = NULL, row_names_gp = gpar(fontsize = 6, fontface = 3))   
# Save
pdf("Figures/Heatmap_sharing_SF_METAG.pdf", height = 8, width = 6)
heatmap_sharing_shared
dev.off()

```

gastric - duodenum
```{r}
prevdf_sp_gd_hm <- prevdf_sp_gd_ordered
rownames(prevdf_sp_gd_hm) <- prevdf_sp_gd_hm$Name
hm_tab_gd <- t(prevdf_sp_gd_hm[,c(3,7)])

# Split heatmap into 3 heatmaps: gastric + oral, shared (ignore fecal for now)
prevdf_sp_hm_gd <- prevdf_sp_gd_hm %>% filter(Intersect_category %in% c("Uniquely gastric", "Gastric origin",
                                                                              "Duodenal origin", "Uniquely duodenal"))
prevdf_sp_hm_sharedgd <- prevdf_sp_gd_hm %>% filter(Intersect_category %in% c("Mostly shared", "Highly shared",
                                                                           "Completely shared"))
# Set color 
max(prevdf_sp_gd_hm$Mean_Abundance_ga) # 5
max(prevdf_sp_gd_hm$Mean_Abundance_da) # 5
col_fun = colorRamp2(c(0,0,0.1,1,5), c("white","white", "#B8DE29FF", "#33638DFF", "#440154FF"))

# Gastric-Duodenum HM
# Matrix for heatmap
hm_tab_gd <- t(prevdf_sp_hm_gd[,c(3, 7)])
rownames(hm_tab_gd)[1] <- "Gastric"
rownames(hm_tab_gd)[2] <- "Duodenum"
# Set row annotations
row_annot_gd <- rowAnnotation(ls = prevdf_sp_hm_gd$Intersect_category,
                                  col = list(ls=c("Uniquely gastric" = "yellow4",
                                                  "Gastric origin" = "olivedrab4",
                                                  "Duodenal origin" = "#6575aa",
                                                  "Uniquely duodenal" = "#5099aa"
                                                  )),
                                  annotation_label = "Sharing", annotation_name_side = "top", annotation_name_rot = 45)
# Set split
split_hm_gd = rep("Uniquely gastric", 22)
split_hm_gd[2:8] = "Gastric origin"
split_hm_gd[9:20] = "Duodenal origin"
split_hm_gd[21:22] = "Uniquely duodenal"
split_hm_gd <- as.factor(split_hm_gd)
split_hm_gd <- factor(split_hm_gd, levels=c("Uniquely gastric", "Gastric origin","Duodenal origin", "Uniquely duodenal"), ordered = TRUE)
# heatmap
heatmap_sharing_gd <- Heatmap(t(hm_tab_gd), name = "Mean relative abundance", row_split = split_hm_gd,
                             row_order =  rownames(prevdf_sp_hm_gd),
        col = col_fun, right_annotation = row_annot_gd, show_column_dend = FALSE, row_names_side = "left", column_names_side = "top",
        column_names_rot = 45, border = TRUE, row_title = NULL, row_names_gp = gpar(fontsize = 8, fontface = 3))   

pdf("Figures/Heatmap_GastricDuodenum_METAG.pdf", height = 6, width = 6)
heatmap_sharing_gd
dev.off()



# Gastric-Duodenum shared HM
# Matrix for heatmap
hm_tab_sharedgd <- t(prevdf_sp_hm_sharedgd[,c(3, 7)])
rownames(hm_tab_sharedgd)[1] <- "Gastric"
rownames(hm_tab_sharedgd)[2] <- "Duodenum"
# Set row annotations
row_annot_sharedgd <- rowAnnotation(ls = prevdf_sp_hm_sharedgd$Intersect_category,
                                  col = list(ls=c("Mostly shared" = "pink2",
                                                  "Highly shared" = "indianred3",
                                                  "Completely shared" = "#440154FF"
                                                  )),
                                  annotation_label = "Sharing", annotation_name_side = "top", annotation_name_rot = 45)
# Set split
split_hm_sharedgd = rep("Mostly shared", 233)
split_hm_sharedgd[148:227] = "Highly shared"
split_hm_sharedgd[228:233] = "Completely shared"
split_hm_sharedgd <- as.factor(split_hm_sharedgd)
split_hm_sharedgd <- factor(split_hm_sharedgd, levels=c("Mostly shared", "Highly shared","Completely shared"), ordered = TRUE)
# heatmap
heatmap_sharing_sharedgd <- Heatmap(t(hm_tab_sharedgd), name = "Mean relative abundance", row_split = split_hm_sharedgd,
                             row_order =  rownames(prevdf_sp_hm_sharedgd),
        col = col_fun, right_annotation = row_annot_sharedgd, show_column_dend = FALSE, row_names_side = "left", column_names_side = "top",
        column_names_rot = 45, border = TRUE, row_title = NULL, row_names_gp = gpar(fontsize = 8, fontface = 3))   

pdf("Figures/Heatmap_GastricDuodenumShared_METAG.pdf", height = 18, width = 6)
heatmap_sharing_sharedgd
dev.off()
```


## Other grouping
```{r}
prevdf_sp_sgf_hm <- prevdf_sp_sgf_ordered
rownames(prevdf_sp_sgf_hm) <- prevdf_sp_sgf_hm$Name

# Split heatmap into 3 heatmaps: gastric + oral, shared (ignore fecal for now)
prevdf_sp_hm_gastric <- prevdf_sp_sgf_hm %>% filter(Intersect_category %in% c("Oral origin","Uniquely gastric", "Gastric origin", "Saliva-feces"))
prevdf_sp_hm_sharedsgf <- prevdf_sp_sgf_hm %>% filter(Intersect_category %in% c("Shared in upper tract", "Mostly shared", "Highly shared",
                                                                           "Completely shared", "Shared in lower tract"))
# Set color range
max(prevdf_sp_sgf_hm$Mean_Abundance_feces) # 9.8
max(prevdf_sp_sgf_hm$Mean_Abundance_saliva) # 6.6
max(prevdf_sp_sgf_hm$Mean_Abundance_ga) # 7.2
col_fun = colorRamp2(c(0,0,0.1,1,5,10), c("white","white", "#B8DE29FF", "#29AF7FFF","#33638DFF", "#440154FF"))

# Gastric HM
# Matrix for heatmap
hm_tab_ga <- t(prevdf_sp_hm_gastric[,c(3,11, 7)])
rownames(hm_tab_ga)[1] <- "Saliva"
rownames(hm_tab_ga)[3] <- "Feces"
rownames(hm_tab_ga)[2] <- "Gastric"
# Set row annotations
row_annot_ga <- rowAnnotation(ls = prevdf_sp_hm_gastric$Intersect_category,
                                  col = list(ls=c("Oral origin" = "orange2",
                                                  "Gastric origin" = "olivedrab4" ,
                                                  "Uniquely gastric" = "yellow4",
                                                  "Saliva-feces" = "honeydew4")),
                                  annotation_label = "Sharing", annotation_name_side = "top", annotation_name_rot = 45)
# Set split
split_hm_ga = rep("Oral origin", 18)
split_hm_ga[15] = "Gastric origin"
split_hm_ga[16:17] = "Uniquely gastric"
split_hm_ga[18] = "Saliva-feces"
split_hm_ga <- as.factor(split_hm_ga)
split_hm_ga <- factor(split_hm_ga, levels=c("Oral origin", "Gastric origin", "Uniquely gastric", "Saliva-feces"), ordered = TRUE)
# heatmap
heatmap_sharing_gastric <- Heatmap(t(hm_tab_ga), name = "Mean relative abundance", row_split = split_hm_ga,
                             row_order =  rownames(prevdf_sp_hm_gastric),
        col = col_fun, right_annotation = row_annot_ga, show_column_dend = FALSE, row_names_side = "left", column_names_side = "top",
        column_names_rot = 45, border = TRUE, row_title = NULL, row_names_gp = gpar(fontsize = 8, fontface = 3))   

pdf("Figures/Heatmap_sharing_Gastric_METAG.pdf", height = 6, width = 4)
draw(heatmap_sharing_gastric, heatmap_legend_side = "bottom", annotation_legend_side = "bottom")
dev.off()

# Gastric HM
# Matrix for heatmap
hm_tab_sharedsgf <- t(prevdf_sp_hm_sharedsgf[,c(3,11, 7)])
rownames(hm_tab_sharedsgf)[1] <- "Saliva"
rownames(hm_tab_sharedsgf)[3] <- "Feces"
rownames(hm_tab_sharedsgf)[2] <- "Gastric"
# Set row annotations
row_annot_sharedsgf <- rowAnnotation(ls = prevdf_sp_hm_sharedsgf$Intersect_category,
                                  col = list(ls=c("Shared in upper tract" = "darkgoldenrod",
                                                  "Mostly shared" = "pink2",
                                                  "Highly shared" = "indianred3",
                                                  "Completely shared" = "#440154FF",
                                                  "Shared in lower tract" = "darkorchid")),
                                  annotation_label = "Sharing", annotation_name_side = "top", annotation_name_rot = 45)
# Set split
split_hm_sgf = rep("Shared in upper tract", 230)
split_hm_sgf[164:215] = "Mostly shared"
split_hm_sgf[216:229] = "Highly shared"
split_hm_sgf[230] = "Shared in lower tract"
split_hm_sgf <- as.factor(split_hm_sgf)
split_hm_sgf <- factor(split_hm_sgf, levels=c("Shared in upper tract", "Mostly shared",
                                            "Highly shared", "Shared in lower tract" ), ordered = TRUE)
# heatmap
heatmap_sharing_sgf <- Heatmap(t(hm_tab_sharedsgf), name = "Mean relative abundance", row_split = split_hm_sgf,
                             row_order =  rownames(prevdf_sp_hm_sharedsgf),
        col = col_fun, right_annotation = row_annot_sharedsgf, show_column_dend = FALSE, row_names_side = "left", column_names_side = "top",
        column_names_rot = 45, border = TRUE, row_title = NULL, row_names_gp = gpar(fontsize = 6, fontface = 3))   

pdf("Figures/Heatmap_sharing_SGF_METAG.pdf", height = 16, width = 6)
heatmap_sharing_sgf
dev.off()
```

Upper GI
```{r}
physeq_sp_upper <- subset_samples(physeq_sp, body_site == "Duodenum" | body_site == "Gastric" | body_site == "Saliva")
# Remove samples with unpaired saliva, gastric and duodenal
#physeq_sp_upper <- subset_samples(physeq_sp_upper, id %in% intersect(md_ps_final$id[which(md_ps_final$body_site =="Gastric")],
#          md_ps_final$id[which(md_ps_final$body_site =="Duodenum")]))

physeq_sp_upper <- subset_taxa(physeq_sp_upper, taxa_sums(physeq_sp_upper)/85 > 10^-4)

prevdf_sp_upper = apply(X= otu_table(physeq_sp_upper),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_upper),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_upper = data.frame(Prevalence = prevdf_sp_upper,
                        Percentage_Prev = 100*prevdf_sp_upper/85,
                        tax_table(physeq_sp_upper))

physeq_sp_upper = subset_taxa(physeq_sp_upper,  rownames(tax_table(physeq_sp_upper)) %in% rownames(prevdf_sp_upper)[which(prevdf_sp_upper$Percentage_Prev >= 10)])
# 290 taxa

# Prevalence and abundance in gastric for all species
physeq_sp_upper_ga <- subset_samples(physeq_sp_upper, body_site == "Gastric")
prevdf_sp_upper_ga = apply(X= otu_table(physeq_sp_upper_ga),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_upper_ga),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_upper_ga = data.frame(Prevalence_ga = prevdf_sp_upper_ga,
                       Percentage_Prev_ga = 100*prevdf_sp_upper_ga/31,
                       Mean_Abundance_ga = 100*taxa_sums(physeq_sp_upper_ga)/31,
                       Total_Abundance_ga = taxa_sums(physeq_sp_upper_ga),
                        tax_table(physeq_sp_upper_ga))
prevdf_sp_upper_ga <- prevdf_sp_upper_ga[, -12]


# Prevalence and abundance in duodenal for all species
physeq_sp_upper_da <- subset_samples(physeq_sp_upper, body_site == "Duodenum")
prevdf_sp_upper_da = apply(X= otu_table(physeq_sp_upper_da),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_upper_da),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_upper_da = data.frame(Prevalence_da = prevdf_sp_upper_da,
                       Percentage_Prev_da = 100*prevdf_sp_upper_da/15,
                       Mean_Abundance_da = 100*taxa_sums(physeq_sp_upper_da)/15,
                       Total_Abundance_da = taxa_sums(physeq_sp_upper_da),
                        tax_table(physeq_sp_upper_da))
prevdf_sp_upper_da <- prevdf_sp_upper_da[, -12]

# Prevalence and abundance in saliva for all species
physeq_sp_upper_sa <- subset_samples(physeq_sp_upper, body_site == "Saliva")
prevdf_sp_upper_sa = apply(X= otu_table(physeq_sp_upper_sa),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_upper_sa),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_upper_sa = data.frame(Prevalence_sa = prevdf_sp_upper_sa,
                       Percentage_Prev_sa = 100*prevdf_sp_upper_sa/39,
                       Mean_Abundance_sa = 100*taxa_sums(physeq_sp_upper_sa)/39,
                       Total_Abundance_sa = taxa_sums(physeq_sp_upper_sa),
                        tax_table(physeq_sp_upper_sa))
prevdf_sp_upper_sa <- prevdf_sp_upper_sa[, -12]


# Merging dataset
prevdf_sp_upper <- cbind(prevdf_sp_upper_sa[,c(1:4)], prevdf_sp_upper_ga[,c(1:4)], prevdf_sp_upper_da)

# Adding categories:
prevdf_sp_upper <- prevdf_sp_upper %>%
  mutate(Intersect_category = "")
prevdf_sp_upper$Intersect_category[which(prevdf_sp_upper$Percentage_Prev_sa > 10 
                                    & prevdf_sp_upper$Percentage_Prev_da < 10
                                    & prevdf_sp_upper$Percentage_Prev_ga < 10)] <- "Oral origin"

prevdf_sp_upper$Intersect_category[which(prevdf_sp_upper$Percentage_Prev_sa < 10
                                    & prevdf_sp_upper$Percentage_Prev_da > 10 
                                    & prevdf_sp_upper$Percentage_Prev_ga < 10)] <- "Duodenal origin"

prevdf_sp_upper$Intersect_category[which(prevdf_sp_upper$Percentage_Prev_sa < 10
                                    & prevdf_sp_upper$Percentage_Prev_ga > 10 
                                    & prevdf_sp_upper$Percentage_Prev_da < 10 )] <- "Gastric origin"
#Unique to a body site
prevdf_sp_upper$Intersect_category[which(prevdf_sp_upper$Percentage_Prev_sa > 0 
                                    & prevdf_sp_upper$Percentage_Prev_da == 0
                                     & prevdf_sp_upper$Percentage_Prev_ga == 0)] <- "Uniquely oral"
prevdf_sp_upper$Intersect_category[which(prevdf_sp_upper$Percentage_Prev_sa == 0 
                                    & prevdf_sp_upper$Percentage_Prev_da > 0
                                     & prevdf_sp_upper$Percentage_Prev_ga == 0)] <- "Uniquely duodenal"
prevdf_sp_upper$Intersect_category[which(prevdf_sp_upper$Percentage_Prev_ga > 0 
                                    & prevdf_sp_upper$Percentage_Prev_da == 0
                                     & prevdf_sp_upper$Percentage_Prev_sa == 0)] <- "Uniquely gastric"


prevdf_sp_upper$Intersect_category[which(prevdf_sp_upper$Percentage_Prev_sa < 10 
                                   & prevdf_sp_upper$Percentage_Prev_da < 10 &
                                     prevdf_sp_upper$Percentage_Prev_ga < 10)] <- "Low prevalence"

# Shared by pairs
prevdf_sp_upper$Intersect_category[which(prevdf_sp_upper$Percentage_Prev_sa > 10 
                                    & prevdf_sp_upper$Percentage_Prev_da < 10
                                    & prevdf_sp_upper$Percentage_Prev_ga > 10)] <- "Gastric-sa"
prevdf_sp_upper$Intersect_category[which(prevdf_sp_upper$Percentage_Prev_sa < 10
                                    & prevdf_sp_upper$Percentage_Prev_da > 10 
                                    & prevdf_sp_upper$Percentage_Prev_ga > 10)] <- "Gastric-Duodenum"
prevdf_sp_upper$Intersect_category[which(prevdf_sp_upper$Percentage_Prev_sa > 10
                                    & prevdf_sp_upper$Percentage_Prev_da > 10 
                                    & prevdf_sp_upper$Percentage_Prev_ga < 10)] <- "Saliva-Duodenum"


prevdf_sp_upper$Intersect_category[which(prevdf_sp_upper$Percentage_Prev_sa >= 10
                                    & prevdf_sp_upper$Percentage_Prev_da >= 10 
                                    & prevdf_sp_upper$Percentage_Prev_ga >= 10 )] <- "Mostly shared"


prevdf_sp_upper$Intersect_category[which(prevdf_sp_upper$Percentage_Prev_sa >= 60
                                    & prevdf_sp_upper$Percentage_Prev_da >= 60
                                    & prevdf_sp_upper$Percentage_Prev_ga >= 60)] <- "Highly shared"

# No taxa are present in all samples! 1 gastric sample does not have Streptococcus sarius
prevdf_sp_upper$Intersect_category[which(prevdf_sp_upper$Percentage_Prev_sa == 100 & 
                                         prevdf_sp_upper$Percentage_Prev_da == 100 &
                                         prevdf_sp_upper$Percentage_Prev_ga == 100)] <- "Completely shared"

prevdf_sp_upper$Intersect_category <- as.factor(prevdf_sp_upper$Intersect_category)
prevdf_sp_upper$Intersect_category <- factor(prevdf_sp_upper$Intersect_category, levels=c("Uniquely oral", "Oral origin", "Gastric-Saliva", 
                                                                              "Gastric origin", "Uniquely gastric",
                                                                                "Mostly shared", "Highly shared", "Completely shared",
                                                                              "Gastric-Duodenum",
                                                                                "Duodenal origin", 
                                                                              "Uniquely duodenal", "Saliva-Duodenum"), ordered = TRUE)

prevdf_sp_upper_ordered <- prevdf_sp_upper[order(prevdf_sp_upper$Intersect_category, -prevdf_sp_upper$Percentage_Prev_sa,
                                             prevdf_sp_upper$Percentage_Prev_ga, prevdf_sp_upper$Percentage_Prev_da),]

# Rename unasigned species (SGB)
prevdf_sp_upper_ordered <- prevdf_sp_upper_ordered %>%
  mutate(Name = case_when(
    str_detect(Species, "GGB", negate = TRUE) ~ Species,
    str_detect(Genus, "GGB", negate = TRUE) ~ Species,
    str_detect(Family, "FGB", negate = TRUE) ~ paste(Family,Species),
    str_detect(Order, "OFGB", negate = TRUE) ~ paste(Order, Species),
    str_detect(Class, "CFGB", negate = TRUE) ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species)
  ))

# Count table per intersect category
df_shared_upper <- prevdf_sp_upper_ordered %>%
  group_by(Intersect_category) %>%
  summarise(n_species = n_distinct(Name))
```

Lower GI
```{r}
physeq_sp_lower <- subset_samples(physeq_sp, body_site == "Duodenum" | body_site == "Gastric" | body_site == "Feces")
# Remove samples with unpaired saliva, gastric and duodenal
physeq_sp_lower <- subset_samples(physeq_sp_lower, id %in% intersect(md_ps_final$id[which(md_ps_final$body_site =="Gastric")],
          md_ps_final$id[which(md_ps_final$body_site =="Duodenum")]))

physeq_sp_lower <- subset_taxa(physeq_sp_lower, taxa_sums(physeq_sp_lower)/36 > 10^-4)

prevdf_sp_lower = apply(X= otu_table(physeq_sp_lower),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_lower),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_lower = data.frame(Prevalence = prevdf_sp_lower,
                        Percentage_Prev = 100*prevdf_sp_lower/36,
                        tax_table(physeq_sp_lower))

physeq_sp_lower = subset_taxa(physeq_sp_lower,  rownames(tax_table(physeq_sp_lower)) %in% rownames(prevdf_sp_lower)[which(prevdf_sp_lower$Percentage_Prev >= 10)])
# 434 taxa

# Prevalence and abundance in gastric for all species
physeq_sp_lower_ga <- subset_samples(physeq_sp_lower, body_site == "Gastric")
prevdf_sp_lower_ga = apply(X= otu_table(physeq_sp_lower_ga),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_lower_ga),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_lower_ga = data.frame(Prevalence_ga = prevdf_sp_lower_ga,
                       Percentage_Prev_ga = 100*prevdf_sp_lower_ga/12,
                       Mean_Abundance_ga = 100*taxa_sums(physeq_sp_lower_ga)/12,
                       Total_Abundance_ga = taxa_sums(physeq_sp_lower_ga),
                        tax_table(physeq_sp_lower_ga))
prevdf_sp_lower_ga <- prevdf_sp_lower_ga[, -12]


# Prevalence and abundance in duodenal for all species
physeq_sp_lower_da <- subset_samples(physeq_sp_lower, body_site == "Duodenum")
prevdf_sp_lower_da = apply(X= otu_table(physeq_sp_lower_da),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_lower_da),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_lower_da = data.frame(Prevalence_da = prevdf_sp_lower_da,
                       Percentage_Prev_da = 100*prevdf_sp_lower_da/12,
                       Mean_Abundance_da = 100*taxa_sums(physeq_sp_lower_da)/12,
                       Total_Abundance_da = taxa_sums(physeq_sp_lower_da),
                        tax_table(physeq_sp_lower_da))
prevdf_sp_lower_da <- prevdf_sp_lower_da[, -12]

# Prevalence and abundance in saliva for all species
physeq_sp_lower_fe <- subset_samples(physeq_sp_lower, body_site == "Feces")
prevdf_sp_lower_fe = apply(X= otu_table(physeq_sp_lower_fe),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_lower_fe),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_lower_fe = data.frame(Prevalence_fe = prevdf_sp_lower_fe,
                       Percentage_Prev_fe = 100*prevdf_sp_lower_fe/12,
                       Mean_Abundance_fe = 100*taxa_sums(physeq_sp_lower_fe)/12,
                       Total_Abundance_fe = taxa_sums(physeq_sp_lower_fe),
                        tax_table(physeq_sp_lower_fe))
prevdf_sp_lower_fe <- prevdf_sp_lower_fe[, -12]


# Merging dataset
prevdf_sp_lower <- cbind(prevdf_sp_lower_fe[,c(1:4)], prevdf_sp_lower_ga[,c(1:4)], prevdf_sp_lower_da)

# Adding categories:
prevdf_sp_lower <- prevdf_sp_lower %>%
  mutate(Intersect_category = "")
prevdf_sp_lower$Intersect_category[which(prevdf_sp_lower$Percentage_Prev_fe > 10 
                                    & prevdf_sp_lower$Percentage_Prev_da < 10
                                    & prevdf_sp_lower$Percentage_Prev_ga < 10)] <- "Fecal origin"

prevdf_sp_lower$Intersect_category[which(prevdf_sp_lower$Percentage_Prev_fe < 10
                                    & prevdf_sp_lower$Percentage_Prev_da > 10 
                                    & prevdf_sp_lower$Percentage_Prev_ga < 10)] <- "Duodenal origin"

prevdf_sp_lower$Intersect_category[which(prevdf_sp_lower$Percentage_Prev_fe < 10
                                    & prevdf_sp_lower$Percentage_Prev_ga > 10 
                                    & prevdf_sp_lower$Percentage_Prev_da < 10 )] <- "Gastric origin"
#Unique to a body site
prevdf_sp_lower$Intersect_category[which(prevdf_sp_lower$Percentage_Prev_fe > 0 
                                    & prevdf_sp_lower$Percentage_Prev_da == 0
                                     & prevdf_sp_lower$Percentage_Prev_ga == 0)] <- "Uniquely fecal"
prevdf_sp_lower$Intersect_category[which(prevdf_sp_lower$Percentage_Prev_fe == 0 
                                    & prevdf_sp_lower$Percentage_Prev_da > 0
                                     & prevdf_sp_lower$Percentage_Prev_ga == 0)] <- "Uniquely duodenal"
prevdf_sp_lower$Intersect_category[which(prevdf_sp_lower$Percentage_Prev_ga > 0 
                                    & prevdf_sp_lower$Percentage_Prev_da == 0
                                     & prevdf_sp_lower$Percentage_Prev_fe == 0)] <- "Uniquely gastric"


prevdf_sp_lower$Intersect_category[which(prevdf_sp_lower$Percentage_Prev_fe < 10 
                                   & prevdf_sp_lower$Percentage_Prev_da < 10 &
                                     prevdf_sp_lower$Percentage_Prev_ga < 10)] <- "Low prevalence"

# Shared by pairs
prevdf_sp_lower$Intersect_category[which(prevdf_sp_lower$Percentage_Prev_fe > 10 
                                    & prevdf_sp_lower$Percentage_Prev_da < 10
                                    & prevdf_sp_lower$Percentage_Prev_ga > 10)] <- "Gastric-Feces"
prevdf_sp_lower$Intersect_category[which(prevdf_sp_lower$Percentage_Prev_fe < 10
                                    & prevdf_sp_lower$Percentage_Prev_da > 10 
                                    & prevdf_sp_lower$Percentage_Prev_ga > 10)] <- "Gastric-Duodenum"
prevdf_sp_lower$Intersect_category[which(prevdf_sp_lower$Percentage_Prev_fe > 10
                                    & prevdf_sp_lower$Percentage_Prev_da > 10 
                                    & prevdf_sp_lower$Percentage_Prev_ga < 10)] <- "Feces-Duodenum"


prevdf_sp_lower$Intersect_category[which(prevdf_sp_lower$Percentage_Prev_fe >= 10
                                    & prevdf_sp_lower$Percentage_Prev_da >= 10 
                                    & prevdf_sp_lower$Percentage_Prev_ga >= 10 )] <- "Mostly shared"


prevdf_sp_lower$Intersect_category[which(prevdf_sp_lower$Percentage_Prev_fe >= 60
                                    & prevdf_sp_lower$Percentage_Prev_da >= 60
                                    & prevdf_sp_lower$Percentage_Prev_ga >= 60)] <- "Highly shared"

# No taxa are present in all femples! 1 gastric femple does not have Streptococcus ferius
prevdf_sp_lower$Intersect_category[which(prevdf_sp_lower$Percentage_Prev_fe == 100 & 
                                         prevdf_sp_lower$Percentage_Prev_da == 100 &
                                         prevdf_sp_lower$Percentage_Prev_ga == 100)] <- "Completely shared"

prevdf_sp_lower$Intersect_category <- as.factor(prevdf_sp_lower$Intersect_category)
prevdf_sp_lower$Intersect_category <- factor(prevdf_sp_lower$Intersect_category, levels=c("Uniquely fecal", "Fecal origin", "Gastric-Feces", 
                                                                              "Gastric origin", "Uniquely gastric",
                                                                                "Mostly shared", "Highly shared", "Completely shared",
                                                                              "Gastric-Duodenum",
                                                                                "Duodenal origin", 
                                                                              "Uniquely duodenal", "Feces-Duodenum"), ordered = TRUE)

prevdf_sp_lower_ordered <- prevdf_sp_lower[order(prevdf_sp_lower$Intersect_category, -prevdf_sp_lower$Percentage_Prev_fe,
                                             prevdf_sp_lower$Percentage_Prev_ga, prevdf_sp_lower$Percentage_Prev_da),]

# Rename unasigned species (SGB)
prevdf_sp_lower_ordered <- prevdf_sp_lower_ordered %>%
  mutate(Name = case_when(
    str_detect(Species, "GGB", negate = TRUE) ~ Species,
    str_detect(Genus, "GGB", negate = TRUE) ~ Species,
    str_detect(Family, "FGB", negate = TRUE) ~ paste(Family,Species),
    str_detect(Order, "OFGB", negate = TRUE) ~ paste(Order, Species),
    str_detect(Class, "CFGB", negate = TRUE) ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species)
  ))

# Count table per intersect category
df_shared_lower <- prevdf_sp_lower_ordered %>%
  group_by(Intersect_category) %>%
  summarise(n_species = n_distinct(Name))
```

Non-stunted
```{r}
physeq_sp_sgft_NonStu <- subset_samples(physeq_sp, Stunting_status == "Non_stunted")
physeq_sp_sgft_NonStu <- subset_samples(physeq_sp_sgft_NonStu, 
                                        id %in% intersect(intersect(md_ps_final$id[which(md_ps_final$body_site =="Gastric")],
                                                                    md_ps_final$id[which(md_ps_final$body_site =="Saliva")]),
                                                          md_ps_final$id[which(md_ps_final$body_site =="Feces")]))
# Remove 043 as haz at 1.99 
physeq_sp_sgft_NonStu <- subset_samples(physeq_sp_sgft_NonStu, id != "1429AGN043")
physeq_sp_sgft_NonStu <- subset_samples(physeq_sp_sgft_NonStu, body_site == "Saliva" | body_site == "Feces" | body_site == "Gastric")
# 10 non-stunted with saliva, gastric and fecal samples

physeq_sp_sgft_NonStu <- subset_taxa(physeq_sp_sgft_NonStu, taxa_sums(physeq_sp_sgft_NonStu)/30 > 10^-6)
# 1906 - 1288 = 618 low abundant taxa removed

prevdf_filt_NonStu = apply(X= otu_table(physeq_sp_sgft_NonStu),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_sgft_NonStu),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_filt_NonStu = data.frame(Prevalence = prevdf_filt_NonStu,
                        Percentage_Prev = 100*prevdf_filt_NonStu/30,
                        tax_table(physeq_sp_sgft_NonStu))

# Filtering for species in the above dataset
physeq_sp_sgft_NonStu = subset_taxa(physeq_sp_sgft_NonStu,
                                    Species %in% prevdf_filt_NonStu$Species[which(prevdf_filt_NonStu$Percentage_Prev >= 10)])
# 803 taxa with abundance above 10^-6 and prevalence above 10% in stunted children with 3 compartments samples


# Prevalence and abundance in feces for all species
physeq_sp_NonStu_fe <- subset_samples(physeq_sp_sgft_NonStu, body_site == "Feces")
prevdf_sp_NonStu_fe = apply(X= otu_table(physeq_sp_NonStu_fe),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_NonStu_fe),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_NonStu_fe = data.frame(Prevalence_feces = prevdf_sp_NonStu_fe,
                       Percentage_Prev_feces = 100*prevdf_sp_NonStu_fe/10,
                       Mean_Abundance_feces = 100*taxa_sums(physeq_sp_NonStu_fe)/10,
                       Total_Abundance_feces = taxa_sums(physeq_sp_NonStu_fe),
                        tax_table(physeq_sp_NonStu_fe))
prevdf_sp_NonStu_fe <- prevdf_sp_NonStu_fe[, -12]

# Prevalence and abundance in saliva for all species
physeq_sp_NonStu_sa <- subset_samples(physeq_sp_sgft_NonStu, body_site == "Saliva")
prevdf_sp_NonStu_sa = apply(X= otu_table(physeq_sp_NonStu_sa),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_NonStu_sa),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_NonStu_sa = data.frame(Prevalence_saliva = prevdf_sp_NonStu_sa,
                       Percentage_Prev_saliva = 100*prevdf_sp_NonStu_sa/10,
                       Mean_Abundance_saliva = 100*taxa_sums(physeq_sp_NonStu_sa)/10,
                       Total_Abundance_saliva = taxa_sums(physeq_sp_NonStu_sa),
                        tax_table(physeq_sp_NonStu_sa))
prevdf_sp_NonStu_sa <- prevdf_sp_NonStu_sa[, -12]

# Prevalence and abundance in gastric for all species
physeq_sp_NonStu_ga <- subset_samples(physeq_sp_sgft_NonStu, body_site == "Gastric")
prevdf_sp_NonStu_ga = apply(X= otu_table(physeq_sp_NonStu_ga),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_NonStu_ga),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_NonStu_ga = data.frame(Prevalence_ga = prevdf_sp_NonStu_ga,
                       Percentage_Prev_ga = 100*prevdf_sp_NonStu_ga/10,
                       Mean_Abundance_ga = 100*taxa_sums(physeq_sp_NonStu_ga)/10,
                       Total_Abundance_ga = taxa_sums(physeq_sp_NonStu_ga),
                        tax_table(physeq_sp_NonStu_ga))
prevdf_sp_NonStu_ga <- prevdf_sp_NonStu_ga[, -12]


# Merging dataset
prevdf_sp_NonStu <- cbind(prevdf_sp_NonStu_sa[,c(1:4)], prevdf_sp_NonStu_fe[,c(1:4)], prevdf_sp_NonStu_ga)

# Adding categories:
prevdf_sp_NonStu <- prevdf_sp_NonStu %>%
  mutate(Intersect_category = "")
prevdf_sp_NonStu$Intersect_category[which(prevdf_sp_NonStu$Percentage_Prev_saliva > 10 
                                    & prevdf_sp_NonStu$Percentage_Prev_feces <= 10
                                    & prevdf_sp_NonStu$Percentage_Prev_ga <= 10)] <- "Oral origin"

prevdf_sp_NonStu$Intersect_category[which(prevdf_sp_NonStu$Percentage_Prev_saliva <= 10
                                    & prevdf_sp_NonStu$Percentage_Prev_feces > 10 
                                    & prevdf_sp_NonStu$Percentage_Prev_ga <= 10)] <- "Fecal origin"

prevdf_sp_NonStu$Intersect_category[which(prevdf_sp_NonStu$Percentage_Prev_saliva <= 10
                                    & prevdf_sp_NonStu$Percentage_Prev_ga > 10 
                                    & prevdf_sp_NonStu$Percentage_Prev_feces <= 10 )] <- "Gastric origin"
#Unique to a body site
prevdf_sp_NonStu$Intersect_category[which(prevdf_sp_NonStu$Percentage_Prev_saliva > 0 
                                    & prevdf_sp_NonStu$Percentage_Prev_feces == 0
                                     & prevdf_sp_NonStu$Percentage_Prev_ga == 0)] <- "Uniquely oral"
prevdf_sp_NonStu$Intersect_category[which(prevdf_sp_NonStu$Percentage_Prev_saliva == 0 
                                    & prevdf_sp_NonStu$Percentage_Prev_feces > 0
                                     & prevdf_sp_NonStu$Percentage_Prev_ga == 0)] <- "Uniquely fecal"
prevdf_sp_NonStu$Intersect_category[which(prevdf_sp_NonStu$Percentage_Prev_ga > 0 
                                    & prevdf_sp_NonStu$Percentage_Prev_feces == 0
                                     & prevdf_sp_NonStu$Percentage_Prev_saliva == 0)] <- "Uniquely gastric"


prevdf_sp_NonStu$Intersect_category[which(prevdf_sp_NonStu$Percentage_Prev_saliva <= 10 
                                   & prevdf_sp_NonStu$Percentage_Prev_feces <= 10 &
                                     prevdf_sp_NonStu$Percentage_Prev_ga <= 10)] <- "Low prevalence"

# Shared by pairs
prevdf_sp_NonStu$Intersect_category[which(prevdf_sp_NonStu$Percentage_Prev_saliva > 10 
                                    & prevdf_sp_NonStu$Percentage_Prev_feces < 10
                                    & prevdf_sp_NonStu$Percentage_Prev_ga > 10)] <- "Shared in upper tract"
prevdf_sp_NonStu$Intersect_category[which(prevdf_sp_NonStu$Percentage_Prev_saliva < 10
                                    & prevdf_sp_NonStu$Percentage_Prev_feces > 10 
                                    & prevdf_sp_NonStu$Percentage_Prev_ga > 10)] <- "Shared in lower tract"
prevdf_sp_NonStu$Intersect_category[which(prevdf_sp_NonStu$Percentage_Prev_saliva > 10
                                    & prevdf_sp_NonStu$Percentage_Prev_feces > 10 
                                    & prevdf_sp_NonStu$Percentage_Prev_ga < 10)] <- "Saliva-feces"

prevdf_sp_NonStu$Intersect_category[which(prevdf_sp_NonStu$Percentage_Prev_saliva >= 10
                                    & prevdf_sp_NonStu$Percentage_Prev_feces >= 10 
                                    & prevdf_sp_NonStu$Percentage_Prev_ga >= 10 )] <- "Mostly shared"


prevdf_sp_NonStu$Intersect_category[which(prevdf_sp_NonStu$Percentage_Prev_saliva >= 60
                                    & prevdf_sp_NonStu$Percentage_Prev_feces >= 60
                                    & prevdf_sp_NonStu$Percentage_Prev_ga >= 60)] <- "Highly shared"

# No taxa are present in all samples! 1 gastric sample does not have Streptococcus salivarius
prevdf_sp_NonStu$Intersect_category[which(prevdf_sp_NonStu$Percentage_Prev_saliva == 100 & 
                                         prevdf_sp_NonStu$Percentage_Prev_feces == 100 &
                                         prevdf_sp_NonStu$Percentage_Prev_ga == 100)] <- "Completely shared"

prevdf_sp_NonStu$Intersect_category <- as.factor(prevdf_sp_NonStu$Intersect_category)
prevdf_sp_NonStu$Intersect_category <- factor(prevdf_sp_NonStu$Intersect_category, 
                                              levels=c("Uniquely oral", "Oral origin", "Shared in upper tract", 
                                                                              "Gastric origin", "Uniquely gastric",
                                                                                "Mostly shared", "Highly shared", "Completely shared",
                                                                              "Shared in lower tract",
                                                                                "Fecal origin", "Uniquely fecal", "Saliva-feces"), 
                                              ordered = TRUE)

prevdf_sp_NonStu_ordered <- prevdf_sp_NonStu[order(prevdf_sp_NonStu$Intersect_category, -prevdf_sp_NonStu$Percentage_Prev_saliva,
                                             prevdf_sp_NonStu$Percentage_Prev_ga, prevdf_sp_NonStu$Percentage_Prev_feces),]

# Rename unasigned species (SGB)
prevdf_sp_NonStu_ordered <- prevdf_sp_NonStu_ordered %>%
  mutate(Name = case_when(
    str_detect(Species, "GGB", negate = TRUE) ~ Species,
    str_detect(Genus, "GGB", negate = TRUE) ~ Species,
    str_detect(Family, "FGB", negate = TRUE) ~ paste(Family,Species),
    str_detect(Order, "OFGB", negate = TRUE) ~ paste(Order, Species),
    str_detect(Class, "CFGB", negate = TRUE) ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species)
  ))

# Count table per intersect category
df_shared_NonStu <- prevdf_sp_NonStu_ordered %>%
  group_by(Intersect_category) %>%
  summarise(n_species = n_distinct(Name))
```
Saliva, gastric, duodenum, feces
Stunted children
```{r}

# Remove 043 as haz at 1.99 
physeq_sp_sgft_NonStu <- subset_samples(physeq_sp_sgft_NonStu, id != "1429AGN043")
physeq_sp_sgft_NonStu <- subset_samples(physeq_sp_sgft_NonStu, body_site == "Saliva" | body_site == "Feces" | body_site == "Gastric")
# 10 non-stunted with saliva, gastric and fecal samples


physeq_sp_filt_stu <- subset_samples(physeq_sp, Stunting_status == "Stunted") 
physeq_sp_filt_stu <- subset_samples(physeq_sp_filt_stu, 
                                        id %in% intersect(intersect(intersect(md_ps_final$id[which(md_ps_final$body_site =="Gastric")],
                                                                    md_ps_final$id[which(md_ps_final$body_site =="Saliva")]),
                                                          md_ps_final$id[which(md_ps_final$body_site =="Feces")])
                                                          ,md_ps_final$id[which(md_ps_final$body_site =="Duodenum")]))

View(sample_data(physeq_sp_filt_stu))
physeq_sp_filt_stu <- subset_samples(physeq_sp_filt_stu, id == "1429AGN032" | id == "1429AGN033" | id == "1429AGN035" |
                                          id == "1429AGN036" | id == "1429AGN037" |
                                          id == "1429AGN038" | id == "1429AGN041" | id == "1429AGN042" | id == "1429AGN043" | id == "1429AGN044" |
                                          id == "1429AGN047")
# 11 stunted with all 4 samples

physeq_sp_filt_stu <- subset_taxa(physeq_sp_filt_stu, taxa_sums(physeq_sp_filt_stu)/44 > 10^-6)
# 1906 - 1387 = 519 low abundant taxa removed

prevdf_filt_stu = apply(X= otu_table(physeq_sp_filt_stu),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_filt_stu),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_filt_stu = data.frame(Prevalence = prevdf_filt_stu,
                        Percentage_Prev = 100*prevdf_filt_stu/4,
                        tax_table(physeq_sp_filt_stu))

# Filtering for species in the above dataset and adding the duodenal samples
physeq_sp_filt_stu = subset_taxa(physeq_sp_filt_stu,  Species %in% prevdf_filt_stu$Species[which(prevdf_filt_stu$Percentage_Prev >= 10)])
# 673 taxa with abundance above 10^-6 and prevalence above 10% in stunted children with 3 compartments samples

# Prevalence and abundance in feces for all species
physeq_sp_stu_fe <- subset_samples(physeq_sp_filt_stu, body_site == "Feces")
prevdf_sp_stu_fe = apply(X= otu_table(physeq_sp_stu_fe),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_stu_fe),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_stu_fe = data.frame(Prevalence_feces = prevdf_sp_stu_fe,
                       Percentage_Prev_feces = 100*prevdf_sp_stu_fe/11,
                       Mean_Abundance_feces = 100*taxa_sums(physeq_sp_stu_fe)/11,
                       Total_Abundance_feces = taxa_sums(physeq_sp_stu_fe),
                        tax_table(physeq_sp_stu_fe))
prevdf_sp_stu_fe <- prevdf_sp_stu_fe[, -12]

# Prevalence and abundance in saliva for all species
physeq_sp_stu_sa <- subset_samples(physeq_sp_filt_stu, body_site == "Saliva")
prevdf_sp_stu_sa = apply(X= otu_table(physeq_sp_stu_sa),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_stu_sa),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_stu_sa = data.frame(Prevalence_saliva = prevdf_sp_stu_sa,
                       Percentage_Prev_saliva = 100*prevdf_sp_stu_sa/11,
                       Mean_Abundance_saliva = 100*taxa_sums(physeq_sp_stu_sa)/11,
                       Total_Abundance_saliva = taxa_sums(physeq_sp_stu_sa),
                        tax_table(physeq_sp_stu_sa))
prevdf_sp_stu_sa <- prevdf_sp_stu_sa[, -12]

# Prevalence and abundance in gastric for all species
physeq_sp_stu_ga <- subset_samples(physeq_sp_filt_stu, body_site == "Gastric")
prevdf_sp_stu_ga = apply(X= otu_table(physeq_sp_stu_ga),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_stu_ga),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_stu_ga = data.frame(Prevalence_ga = prevdf_sp_stu_ga,
                       Percentage_Prev_ga = 100*prevdf_sp_stu_ga/11,
                       Mean_Abundance_ga = 100*taxa_sums(physeq_sp_stu_ga)/11,
                       Total_Abundance_ga = taxa_sums(physeq_sp_stu_ga),
                        tax_table(physeq_sp_stu_ga))
prevdf_sp_stu_ga <- prevdf_sp_stu_ga[, -12]

# Prevalence and abundance in duodenal for all species
physeq_sp_stu_du <- subset_samples(physeq_sp_filt_stu, body_site == "Duodenum")
prevdf_sp_stu_du = apply(X= otu_table(physeq_sp_stu_du),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_stu_du),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_stu_du = data.frame(Prevalence_du = prevdf_sp_stu_du,
                       Percentage_Prev_du = 100*prevdf_sp_stu_du/11,
                       Mean_Abundance_du = 100*taxa_sums(physeq_sp_stu_du)/11,
                       Total_Abundance_du = taxa_sums(physeq_sp_stu_du),
                        tax_table(physeq_sp_stu_du))
prevdf_sp_stu_du <- prevdf_sp_stu_du[, -12]

# Merging dataset
prevdf_sp_stu <- cbind(prevdf_sp_stu_sa[,c(1:4)], prevdf_sp_stu_fe[,c(1:4)], prevdf_sp_stu_ga[,c(1:4)], prevdf_sp_stu_du)

# Adding categories:
prevdf_sp_stu <- prevdf_sp_stu %>%
  mutate(Intersect_category = "")
prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva > 10 
                                    & prevdf_sp_stu$Percentage_Prev_feces <= 10
                                    & prevdf_sp_stu$Percentage_Prev_ga <= 10
                                    & prevdf_sp_stu$Percentage_Prev_du <= 10)] <- "Oral origin"

prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva <= 10
                                    & prevdf_sp_stu$Percentage_Prev_feces > 10 
                                    & prevdf_sp_stu$Percentage_Prev_ga <= 10
                                    & prevdf_sp_stu$Percentage_Prev_du <= 10)] <- "Fecal origin"

prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva <= 10
                                    & prevdf_sp_stu$Percentage_Prev_ga > 10 
                                    & prevdf_sp_stu$Percentage_Prev_feces <= 10
                                    & prevdf_sp_stu$Percentage_Prev_du <= 10)] <- "Gastric origin"

prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva <= 10
                                    & prevdf_sp_stu$Percentage_Prev_ga <= 10 
                                    & prevdf_sp_stu$Percentage_Prev_feces <= 10
                                    & prevdf_sp_stu$Percentage_Prev_du > 10)] <- "Duodenal origin"
#Unique to a body site
prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva > 0 
                                    & prevdf_sp_stu$Percentage_Prev_feces == 0
                                     & prevdf_sp_stu$Percentage_Prev_ga == 0
                                    & prevdf_sp_stu$Percentage_Prev_du == 0)] <- "Uniquely oral"
prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva == 0 
                                    & prevdf_sp_stu$Percentage_Prev_feces > 0
                                     & prevdf_sp_stu$Percentage_Prev_ga == 0
                                    & prevdf_sp_stu$Percentage_Prev_du == 0)] <- "Uniquely fecal"
prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_ga > 0 
                                    & prevdf_sp_stu$Percentage_Prev_feces == 0
                                     & prevdf_sp_stu$Percentage_Prev_saliva == 0
                                    & prevdf_sp_stu$Percentage_Prev_du == 0)] <- "Uniquely gastric"
prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_ga == 0 
                                    & prevdf_sp_stu$Percentage_Prev_feces == 0
                                     & prevdf_sp_stu$Percentage_Prev_saliva == 0
                                    & prevdf_sp_stu$Percentage_Prev_du > 0)] <- "Uniquely duodenal"

prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva <= 10 
                                   & prevdf_sp_stu$Percentage_Prev_feces <= 10 
                                   & prevdf_sp_stu$Percentage_Prev_ga <= 10
                                   & prevdf_sp_stu$Percentage_Prev_du <= 10)] <- "Low prevalence"

# Shared by pairs
prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva > 10 
                                    & prevdf_sp_stu$Percentage_Prev_feces < 10
                                    & prevdf_sp_stu$Percentage_Prev_ga > 10
                                    & prevdf_sp_stu$Percentage_Prev_du > 10)] <- "Shared in upper tract"
prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva > 10 
                                    & prevdf_sp_stu$Percentage_Prev_feces < 10
                                    & prevdf_sp_stu$Percentage_Prev_ga > 10
                                    & prevdf_sp_stu$Percentage_Prev_du <= 10)] <- "Shared in upper tract"
prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva > 10 
                                    & prevdf_sp_stu$Percentage_Prev_feces < 10
                                    & prevdf_sp_stu$Percentage_Prev_ga <= 10
                                    & prevdf_sp_stu$Percentage_Prev_du > 10)] <- "Shared in upper tract"

prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva < 10
                                    & prevdf_sp_stu$Percentage_Prev_feces > 10 
                                    & prevdf_sp_stu$Percentage_Prev_ga > 10
                                    & prevdf_sp_stu$Percentage_Prev_du > 10)] <- "Shared in lower tract"
prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva < 10
                                    & prevdf_sp_stu$Percentage_Prev_feces > 10 
                                    & prevdf_sp_stu$Percentage_Prev_ga <= 10
                                    & prevdf_sp_stu$Percentage_Prev_du > 10)] <- "Shared in lower tract"
prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva < 10
                                    & prevdf_sp_stu$Percentage_Prev_feces > 10 
                                    & prevdf_sp_stu$Percentage_Prev_ga > 10
                                    & prevdf_sp_stu$Percentage_Prev_du <= 10)] <- "Shared in lower tract"

prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva > 10
                                    & prevdf_sp_stu$Percentage_Prev_feces > 10 
                                    & prevdf_sp_stu$Percentage_Prev_ga < 10
                                    & prevdf_sp_stu$Percentage_Prev_du < 10)] <- "Saliva-feces"

prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva <= 10
                                    & prevdf_sp_stu$Percentage_Prev_feces <= 10 
                                    & prevdf_sp_stu$Percentage_Prev_ga > 10
                                    & prevdf_sp_stu$Percentage_Prev_du > 10)] <- "Gastric-Duodenal"

prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva >= 10
                                    & prevdf_sp_stu$Percentage_Prev_feces >= 10 
                                    & prevdf_sp_stu$Percentage_Prev_ga >= 10 
                                    & prevdf_sp_stu$Percentage_Prev_du >= 10)] <- "Mostly shared"


prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva >= 60
                                    & prevdf_sp_stu$Percentage_Prev_feces >= 60
                                    & prevdf_sp_stu$Percentage_Prev_ga >= 60
                                    & prevdf_sp_stu$Percentage_Prev_du >= 60)] <- "Highly shared"

# No taxa are present in all samples! 1 gastric sample does not have Streptococcus salivarius
prevdf_sp_stu$Intersect_category[which(prevdf_sp_stu$Percentage_Prev_saliva == 100 & 
                                         prevdf_sp_stu$Percentage_Prev_feces == 100 &
                                         prevdf_sp_stu$Percentage_Prev_ga == 100
                                       & prevdf_sp_stu$Percentage_Prev_du == 100)] <- "Completely shared"

prevdf_sp_stu$Intersect_category <- as.factor(prevdf_sp_stu$Intersect_category)
prevdf_sp_stu$Intersect_category <- factor(prevdf_sp_stu$Intersect_category, 
                                              levels=c("Uniquely oral", "Oral origin", "Shared in upper tract", 
                                                                              "Gastric origin", "Uniquely gastric",
                                                        "Gastric-Duodenal", "Uniquely duodenal", "Duodenal origin",
                                                                                "Mostly shared", "Highly shared", "Completely shared",
                                                                              "Shared in lower tract",
                                                                                "Fecal origin", "Uniquely fecal", "Saliva-feces",
                                                       "Low prevalence"), 
                                              ordered = TRUE)

prevdf_sp_stu_ordered <- prevdf_sp_stu[order(prevdf_sp_stu$Intersect_category, -prevdf_sp_stu$Percentage_Prev_saliva,
                                             prevdf_sp_stu$Percentage_Prev_ga, prevdf_sp_stu$Percentage_Prev_feces),]

# Rename unasigned species (SGB)
prevdf_sp_stu_ordered <- prevdf_sp_stu_ordered %>%
  mutate(Name = case_when(
    str_detect(Species, "GGB", negate = TRUE) ~ Species,
    str_detect(Genus, "GGB", negate = TRUE) ~ Species,
    str_detect(Family, "FGB", negate = TRUE) ~ paste(Family,Species),
    str_detect(Order, "OFGB", negate = TRUE) ~ paste(Order, Species),
    str_detect(Class, "CFGB", negate = TRUE) ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species)
  ))

# Count table per intersect category
df_shared_stu <- prevdf_sp_stu_ordered %>%
  group_by(Intersect_category) %>%
  summarise(n_species = n_distinct(Name))

shared_orig_sp_stu <- prevdf_sp_stu_ordered %>% filter(Intersect_category %in% c("Mostly shared", "Highly shared", "Completely shared"))
```

```{r}
shared_orig_sp_stu <- prevdf_sp_stu_ordered %>% filter(Intersect_category %in% c("Highly shared", "Completely shared"))

shared_orig_sp_NonsSu <- prevdf_sp_NonStu_ordered %>% filter(Intersect_category %in% c("Highly shared", "Completely shared"))

setdiff(shared_orig_sp_stu$Name, shared_orig_sp_NonsSu$Name)
# 18 species shared in one group but not the other 
```

```{r}
# Prevalence and abundance in feces for all species
ps_sp_fe <- subset_samples(physeq_sp, body_site == "Feces")
prevabunsp_fe = apply(X= otu_table(ps_sp_fe),
               MARGIN = ifelse(taxa_are_rows(ps_sp_fe),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevabunsp_fe = data.frame(Prevalence_feces = prevabunsp_fe,
                       Percentage_Prev_feces = 100*prevabunsp_fe/44,
                       Mean_Abundance_feces = 100*taxa_sums(ps_sp_fe)/44,
                       Total_Abundance_feces = taxa_sums(ps_sp_fe),
                        tax_table(ps_sp_fe))
prevabunsp_fe <- prevabunsp_fe[, -12]

# Prevalence and abundance in saliva for all species
ps_sp_sa <- subset_samples(physeq_sp, body_site == "Saliva")
prevabunsp_sa = apply(X= otu_table(ps_sp_sa),
               MARGIN = ifelse(taxa_are_rows(ps_sp_sa),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevabunsp_sa = data.frame(Prevalence_saliva = prevabunsp_sa,
                       Percentage_Prev_saliva = 100*prevabunsp_sa/39,
                       Mean_Abundance_saliva = 100*taxa_sums(ps_sp_sa)/39,
                       Total_Abundance_saliva = taxa_sums(ps_sp_sa),
                        tax_table(ps_sp_sa))
prevabunsp_sa <- prevabunsp_sa[, -12]

ps_sp_ga <- subset_samples(physeq_sp, body_site == "Gastric")
prevabunsp_ga = apply(X= otu_table(ps_sp_ga),
               MARGIN = ifelse(taxa_are_rows(ps_sp_ga),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevabunsp_ga = data.frame(Prevalence_galiva = prevabunsp_ga,
                       Percentage_Prev_galiva = 100*prevabunsp_ga/31,
                       Mean_Abundance_galiva = 100*taxa_sums(ps_sp_ga)/31,
                       Total_Abundance_galiva = taxa_sums(ps_sp_ga),
                        tax_table(ps_sp_ga))
prevabunsp_ga <- prevabunsp_ga[, -12]

ps_sp_du <- subset_samples(physeq_sp, body_site == "Duodenum")
prevabunsp_du = apply(X= otu_table(ps_sp_du),
               MARGIN = ifelse(taxa_are_rows(ps_sp_du),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevabunsp_du = data.frame(Prevalence_duliva = prevabunsp_du,
                       Percentage_Prev_duliva = 100*prevabunsp_du/31,
                       Mean_Abundance_duliva = 100*taxa_sums(ps_sp_du)/31,
                       Total_Abundance_duliva = taxa_sums(ps_sp_du),
                        tax_table(ps_sp_du))
prevabunsp_du <- prevabunsp_du[, -12]
```


# Afribiota - stool
Load metaphlan merged table
```{r}
tax_table_AFB <- read.delim("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Read_based_output/METAPHLAN/MERGED/AFB_SAT_metaphlan_relab.txt", skip = 1)

## Create a taxonomy table
# Obtain a table with only the tax names
tax_names_AFB <- str_split_fixed(tax_table_AFB$clade_name, "\\|[p,c,o,f,g,s,t]__", 8)
tax_names <- as.data.frame(tax_names_AFB)
tax_names_AFB[,1] <- str_remove(tax_names_AFB[,1], "k__")
# Keeping only the rows with the complete taxonomy
tax_names_AFB <- tax_names_AFB[-(which(tax_names_AFB[,8] == "")),] 
# Changing colnames
colnames(tax_names_AFB) <- c("Kingdom", "Phylum", "Class","Order","Family","Genus","Species", "SGB")
tax_names_AFB <- as.data.frame(tax_names_AFB)
# Adding the rownames
rownames(tax_names_AFB) <- tax_names_AFB$SGB

sgb_num <- tax_names_AFB$SGB
tax_names_AFB <- as.matrix(tax_names_AFB)
# We have now a taxonomy table, similar than the one obtain with DADA2
# Set as tax tablt phyloseq object
taxonomy_table_AFB <- tax_table(tax_names_AFB)

## Abundance table
abun_table_AFB <- cbind(str_split_fixed(tax_table_AFB$clade_name, "\\|[p,c,o,f,g,s,t]__", 8),tax_table_AFB)
abun_table_AFB <- as.data.frame(abun_table_AFB)
abun_table_AFB[,1] <- str_remove(abun_table_AFB[,1], "k__")
abun_table_AFB <- abun_table_AFB[-(which(abun_table_AFB[,8] == "")),] 
rownames(abun_table_AFB) <- sgb_num
abun_table_AFB <- abun_table_AFB[,-c(1:9)]
# We have now a table with the column being each sample

# Clear the column name
colnames(abun_table_AFB) <- str_remove(colnames(abun_table_AFB), "_metaphlan4")
colnames(abun_table_AFB) <- str_remove(colnames(abun_table_AFB), "_metaphlan")
colnames(abun_table_AFB) <- str_remove_all(colnames(abun_table_AFB), "SE_")
colnames(abun_table_AFB) <- str_remove_all(colnames(abun_table_AFB), "_")
# Set as otu_table phyloseq object
abun_table_AFB <- as.matrix(abun_table_AFB)
class(abun_table_AFB) <- "numeric"

abundance_table_AFB = otu_table(abun_table_AFB, taxa_are_rows = TRUE)

# We have now a tax table and an otu table, we can create our phyloseq
physeqAFB <- phyloseq(taxonomy_table_AFB, abundance_table_AFB)
```

Merge metadata
```{r}
# Satellite
md_sat <- md_mpa[,c(1,471,439,436,424,431,438,4,8)]
md_sat <- md_sat %>% mutate(Project = "Satellite")
colnames(md_sat)[1] <- "id"
colnames(md_sat)[4] <- "Country"
md_sat <- md_sat[,c(1,10,4,2,3,5:7, 8:9)]
colnames(md_sat)[10] <- "agem"
# Inflammation status: col 444, 468, 469, 470


# Afribiota
library(readstata13)
md_afb = readstata13::read.dta13(
  "/Users/admin/Documents/VLab_projects/02_Side_projects/SIC_BA_analysis/Tables/BileacidsandmetadataOct20v2.dta",
  nonint.factors = TRUE, encoding = "latin1" )
md_afb <- as.data.frame(md_afb)
md_afb <- md_afb[,c(1,652,655,643,650,36,656, 19, 714)]
md_afb$id <- str_remove_all(md_afb$id, "1429A")
md_afb$id <- str_remove_all(md_afb$id, "1429")
md_afb <- md_afb %>% filter(id %in% colnames(abun_table_AFB))
# remove duplicate
md_afb <- md_afb[!duplicated(md_afb), ]
rownames(md_afb) <- md_afb$id
md_afb <- md_afb %>% mutate(body_site = "Feces")
md_afb$stunted[which(md_afb$stunted==1)] <- "Stunted"
md_afb$stunted[which(md_afb$stunted==0)] <- "Non_stunted"
colnames(md_afb)[2] <- "Stunting_status"
colnames(md_afb)[3] <- "Country"
colnames(md_afb)[7] <- "Project"
md_afb <- md_afb[,c(1,7,3,10,2,4,5,8:9)]
unique(md_afb$haz)
levels(md_afb$haz)[levels(md_afb$haz)=="normonutris"] <- "Normonutris"
levels(md_afb$haz)[levels(md_afb$haz)=="malnutris chronique severe"] <- "Moderately chronic malnutris"
md_afb <- md_afb %>% mutate(Stunting_level = case_when(
    haz_cont < -3 ~ "Severe_stunting",
    haz_cont >= -3 & haz_cont < -2 ~ "Moderate_stunting",
    haz_cont >= -2 & haz_cont < -1 ~ "Mild_stunting",
    haz_cont >= -1 ~ "Normal_growth"
  ))
colnames(md_afb)[9] <- "agem"
md_afb <- md_afb[,c(1:7,10,8:9)]
levels(md_afb$sexe)[levels(md_afb$sexe)=="Masculin"] <- "M"
levels(md_afb$sexe)[levels(md_afb$sexe)=="Fminin"] <- "F"

md_AFB <- rbind(md_sat, md_afb)
md_AFB_ps <- sample_data(md_AFB)
#Merge with phyloseq
physeqAFB <- merge_phyloseq(physeqAFB, md_AFB_ps)

```

```{r}
# Keep only stool sample
physeqAFB <- subset_samples(physeqAFB, body_site == "Feces")
# Keep only RCA
physeqAFB <- subset_samples(physeqAFB, Country == "RCA")
# 73 samples from central african republic

# Removing taxa with abundance 0
physeqAFB <- prune_taxa(taxa_sums(physeqAFB) > 0, physeqAFB)
physeqAFB
# 1728 taxa

physeqAFB_sp <- tax_glom(physeqAFB, "Species")
```
PCoA
```{r}
dist.bc_afb <- phyloseq::distance(physeqAFB_sp, method = "bray")

# Permanova
meta_pcoa_afb = as.matrix(sample_data(physeqAFB_sp))
meta_pcoa_afb <- as.data.frame(meta_pcoa_afb)
permanova_proj <- vegan::adonis2(dist.bc_afb ~ Project, data = meta_pcoa_afb, permutations = 9999) # Not significant

permanova_afb <- vegan::adonis2(dist.bc_afb ~ Stunting_level, data = meta_pcoa_afb, permutations = 9999) # significant!!

pairwise_p_afb <- numeric()
# Pairwise comparison
mild_mod_ado <- meta_pcoa_afb %>% filter(Stunting_level == "Mild_stunting" | Stunting_level == "Moderate_stunting")
mild_mod_ado_dist <- dist_subset(dist.bc_afb, mild_mod_ado$id)
mild_mod_tado <- adonis2(mild_mod_ado_dist ~ Stunting_level, data = mild_mod_ado, permutations = 1000)
pairwise_p_afb["mild_mod"] <- mild_mod_tado$`Pr(>F)`[1]

mild_norm_ado <- meta_pcoa_afb %>% filter(Stunting_level == "Mild_stunting" | Stunting_level == "Normal_growth")
mild_norm_ado_dist <- dist_subset(dist.bc_afb, mild_norm_ado$id)
mild_norm_tado <- adonis2(mild_norm_ado_dist ~ Stunting_level, data = mild_norm_ado,permutations = 1000)
pairwise_p_afb["mild_norm"] <- mild_norm_tado$`Pr(>F)`[1]

mild_sev_ado <- meta_pcoa_afb %>% filter(Stunting_level == "Mild_stunting" | Stunting_level == "Severe_stunting")
mild_sev_ado_dist <- dist_subset(dist.bc_afb, mild_sev_ado$id)
mild_sev_tado <- adonis2(mild_sev_ado_dist ~ Stunting_level, data = mild_sev_ado,permutations = 1000)
pairwise_p_afb["mild_sev"] <- mild_sev_tado$`Pr(>F)`[1]

norm_mod_ado <- meta_pcoa_afb %>% filter(Stunting_level == "Normal_growth" | Stunting_level == "Moderate_stunting")
norm_mod_ado_dist <- dist_subset(dist.bc_afb, norm_mod_ado$id)
norm_mod_tado <- adonis2(norm_mod_ado_dist ~ Stunting_level, data = norm_mod_ado,permutations = 1000)
pairwise_p_afb["norm_mod"] <- norm_mod_tado$`Pr(>F)`[1]

sev_norm_ado <- meta_pcoa_afb %>% filter(Stunting_level == "Severe_stunting" | Stunting_level == "Moderate_stunting")
sev_norm_ado_dist <- dist_subset(dist.bc_afb, sev_norm_ado$id)
sev_norm_tado <- adonis2(sev_norm_ado_dist ~ Stunting_level, data = sev_norm_ado,permutations = 1000)
pairwise_p_afb["sev_mod"] <- sev_norm_tado$`Pr(>F)`[1]

norm_sev_ado <- meta_pcoa_afb %>% filter(Stunting_level == "Normal_growth" | Stunting_level == "Severe_stunting")
norm_sev_ado_dist <- dist_subset(dist.bc_afb, norm_sev_ado$id)
norm_sev_tado <- adonis2(norm_sev_ado_dist ~ Stunting_level, data = norm_sev_ado,permutations = 1000)
pairwise_p_afb["norm_sev"] <- norm_sev_tado$`Pr(>F)`[1]

pairwise_p_afbadj <- p.adjust(pairwise_p_afb, method = "BH")
pairwise_p_afbadj

test_PCoAbray_afb <- ordinate(physeqAFB, "PCoA", distance = dist.bc_afb)
plot_ordination(physeqAFB, test_PCoAbray_afb, type = "samples", color = "Stunting_level", shape = "Stunting_level") +
  stat_ellipse(mapping= aes(fill = Stunting_level),geom = "polygon", alpha = 0.2, show.legend = TRUE) +
  theme_bw() +
  scale_shape_manual(values=c(15:18)) +
  scale_color_brewer(palette = "Set1")
```

Differential abundance analysis
```{r}
diffabun_afb <- physeqAFB_sp
diffabun_afb <- subset_taxa(diffabun_afb, taxa_sums(diffabun_afb) > 0)
tax_table(diffabun_afb) <- tax_table(diffabun_afb)[, -8]

tax_df_afb <- as.data.frame(tax_table(diffabun_afb))
tax_df_afb <- tax_df_afb %>%
  mutate(Species = case_when(
    !str_detect(Species, "GGB") ~ Species,
    !str_detect(Genus, "GGB") ~ Species,
    !str_detect(Family, "FGB") ~ paste(Family, Species),
    !str_detect(Order, "OFGB") ~ paste(Order, Species),
    !str_detect(Class, "CFGB") ~ paste(Class, Species),
    str_detect(Class, "CFGB") ~ paste(Phylum, Species),
    TRUE ~ Species
  ))
tax_table(diffabun_afb) <- as.matrix(tax_df_afb)

# ANCOM-BC
# Test by stunting levels
ancombc_out_afb <- run_ancombc(diffabun_afb, confounders = c("agem", "sexe", "Project"), group = "Stunting_level", taxa_rank = "Species",
                              p_adjust = "BH")
# Warning: No marker was identified
# Trying contiuous variable: haz
ancombc_out_afb_cont <- ancombc(
  diffabun_afb,
  formula = "haz_cont + Project",
  p_adj_method = "BH",
  prv_cut = 0.1,
  tax_level = "Species"
)
View(ancombc_out_afb_cont$res$q_val)
# Nothing significant
```
Lefse
```{r}
lef_out_afb <- run_lefse(diffabun_afb, wilcoxon_cutoff = 0.01, group = "Stunting_status", taxa_rank = "Species",
                        kw_cutoff = 0.05, norm = "CPM", multigrp_strat = TRUE, lda_cutoff = 2)
marker_table(lef_out_afb)$pdaj_bh <- p.adjust(marker_table(lef_out_afb)$pvalue, method = "BH")
marker_table(lef_out_afb)$feature[which(marker_table(lef_out_afb)$pdaj_bh<0.05)]
pdf("Figures/Lefse_StoolStunting_metagAFB.pdf", height = 6, width = 6)
plot_ef_bar(lef_out_afb)
dev.off()

diffabun_afb_normo_sev <- subset_samples(diffabun_afb, Stunting_level %in% c("Normal_growth", "Severe_stunting"))
lef_out_afb_normo_sev <- run_lefse(diffabun_afb_normo_sev, wilcoxon_cutoff = 0.01, group = "Stunting_level", taxa_rank = "Species",
                        kw_cutoff = 0.05, norm = "CPM", multigrp_strat = TRUE, lda_cutoff = 2)
plot_ef_bar(lef_out_afb_normo_sev)
```

No clear higher abundance of a specific taxa when testing with methods for differential abundance analysis
However, the PCoA show a potential shift between severe and other samples.
Use envfit:
```{r}
ps_shift <- physeqAFB_sp
ps_shift <- subset_taxa(ps_shift, taxa_sums(ps_shift) > 0)
OTU_shift = as(otu_table(ps_shift), "matrix")
otu_shift <- as.data.frame(OTU_shift)
TAX_shift = as(tax_table(ps_shift), "matrix")
tax_shift <- as.data.frame(TAX_shift)
META_shift = as(sample_data(ps_shift), "matrix")
meta_shift <- as.data.frame(META_shift)

tax_otu <-cbind(tax_shift, otu_shift)
tax_otu$Name <- str_c(tax_otu$Kingdom, '_',tax_otu$Phylum, '_', tax_otu$Class, '_', tax_otu$Order, '_', tax_otu$Family, '_', tax_otu$Genus, '_', tax_otu$Species)
tax_otu[, 82] <- as.character(tax_otu[, 82])
rownames(tax_otu) <- tax_otu[,82]
tax_otu <- tax_otu[,-c(1:8,82)]
tax_otu <- t(tax_otu)
class(tax_otu) <- "numeric"
tax_otu <- as.data.frame(tax_otu)
dist <- vegdist(tax_otu[, 1:1647],  method = "bray")
PCOA <- ape::pcoa(dist)
vectorPCOA <- cbind(PCOA$vectors[, 1:2])
vectorPCOA <- as.data.frame(PCOA$vectors[, 1:2])

# Arrows and taxa name
env.fit <- envfit(vectorPCOA, tax_otu[,1:1647], perm = 999)
#Warning: the standard deviation is zero
env.fit.sig  <- as.data.frame(scores(env.fit, display = "vectors"))
env.fit.vect <- as.list(env.fit$vectors)
coordinates <- cbind(as.data.frame(env.fit.vect$arrows*sqrt(env.fit.vect$r)), as.data.frame(env.fit.vect$pvals))
coordinates_sign <- subset(coordinates, as.data.frame(env.fit.vect$pvals) <= 0.05)
coordinates_sign <- cbind(coordinates_sign, species = rownames(coordinates_sign))
arr_pvals_coord <- coordinates_sign %>%
  mutate(Coord = (Axis.1*Axis.1)+(Axis.2*Axis.2))
arr_pvals_coord[,5] <- as.numeric(arr_pvals_coord[,5])
df_envfit <- scores(env.fit,display=c("vectors"))
df_envfit <- df_envfit*vegan:::ordiArrowMul(df_envfit)
df_envfit <- as.data.frame(df_envfit)
df_envfit$species <- rownames(df_envfit)
df_envfit <- df_envfit %>% 
  filter(species %in% coordinates_sign$species)
arrows_names <- str_split_fixed(df_envfit$species, "_", 7)
arrows_names <- as.data.frame(arrows_names)
df_envfit <- df_envfit %>% 
  mutate(species_names = arrows_names$V7)

# Plotting
ft.scores <- as.data.frame(vectorPCOA)
ft.scores$fastqID <- rownames(tax_otu)
which(rownames(meta_shift) != rownames(ft.scores))
ft.scores$Bile_acids_Concentration <- meta_cheno$Bile_acids_Concentration
ft.scores$Stunting_level <- meta_shift$Stunting_level
ft.scores.d <- distinct(ft.scores)
per_explained <- 100*PCOA$values$Relative_eig / sum(PCOA$values$Relative_eig)
per_explained <- round(per_explained[1:2], digits = 1)
labs <- c(glue("Axis1 [{per_explained[1]}%]"),
          glue("Axis2 [{per_explained[2]}%]"))

ggplot() +
   geom_point(aes(x=ft.scores$Axis.1, y= ft.scores$Axis.2, color = ft.scores$Stunting_level, shape = ft.scores$Stunting_level),size = 3, show.legend = TRUE) +
  #scale_color_manual(values = pal_SIC, name = "SIC", guide = "none") +
  geom_segment(aes(x = 0, y = 0, xend = df_envfit$Axis.1*0.5,yend = df_envfit$Axis.2*0.5),
            arrow = arrow(length = unit(0.2, "cm")),color="#808080",alpha=0.5) +
  geom_text(aes(df_envfit$Axis.1*0.52, df_envfit$Axis.2*0.52, label = df_envfit$species_names, fontface=3), color="#303030", size=2) +
  ggtitle("") +
  xlab(labs[1]) +
  ylab(labs[2]) +
  theme_bw() + 
  theme(legend.position = "bottom", 
        axis.title=element_text(size=8,family = "sans"), 
        axis.text = element_text(size=8,family = "sans"),
        legend.text = element_text(size=8,family = "sans"),
        legend.title = element_text(size=8,family = "sans"))
```

The shift seems to come from the abundance of Segatella copri in some sample, the abundance of Segatella/Prevotella species and Bacteroides and other species in other samples. Around 320 species significative. No shift showing higher abundance of oral species in severely stunted children.


# MAGs abundance
MAGs were first dereplicated using dRep:
  -- strain level: Gordon lab parameters
  -- species level: sec. cluster at 95%
MAGs abundance in the sample was quantified using CoverM
Several methods are available and I tried 2:
--Trimmed mean: had to be normalized by sequencing depth. Estimate variation very large while haz is between -3 and 0, which made no biological sense.
--TPM: transcript per million - normalize per library size and for MAG length

Strain level - TPM
```{r}
coverm_dir_tpm <- "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/CoverM/tpm_strains"
coverm_tpm_tables <- list.files(coverm_dir_tpm, full.names = TRUE, pattern = "*.tsv")
coverm_tpm <- coverm_tpm_tables %>%
  map(read.delim) %>%
  purrr::reduce(full_join, by = "Genome")
colnames(coverm_tpm) <- str_remove_all(colnames(coverm_tpm), "_FASTP_BOWTIE2_R1_001.fastq.gz.TPM")

# matrix for ancom bc
coverm_ancombc_strains <- coverm_tpm

# Adding MAGs information
mags_df_strains  <- read.csv2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/inStrain/Only_MAGs/dRep_WinningGenomesMAGs_df.csv")
coverm_tpm <- merge(mags_df_strains, coverm_tpm, by.x = "genome", by.y = "Genome")
# Add metadata
coverm_tpm <- pivot_longer(coverm_tpm, cols = 8:136, names_to = "SampleID", values_to = "Abundance")
coverm_tpm <- merge(md_sat, coverm_tpm, by.x = "id", by.y = "SampleID")
coverm_tpm <- coverm_tpm[,c(9,1:8,10:16)]
# Select stool
coverm_fe_tpm <- coverm_tpm %>% filter(body_site == "Feces")
# Remove abundance 0 (272 genomes)
coverm_fe_tpm <- coverm_fe_tpm  %>%
  group_by(genome) %>%
  filter(sum(Abundance) > 0) %>%
  ungroup()
# Prevalence filtering 5/44 ~ 11%
coverm_fe_tpm <- coverm_fe_tpm %>%
  group_by(genome) %>%
  filter(sum(Abundance > 0) >= 22) %>%  # prevalence filter
  ungroup()


# Test for two bacteria 
coverm_fe_test <- coverm_fe_tpm %>%
  filter(genome == "AGN001FE.30")
summary(lm(haz_cont ~ log1p(Abundance), data = coverm_fe_test))
ggplot(coverm_fe_test, aes(x = log1p(Abundance), y = haz_cont)) +
  geom_point(alpha = 0.7) +
  geom_smooth(method = "lm", se = TRUE, color = "blue") +
  theme_minimal()

# Linear model for each MAGs
lm_results_tpm <- coverm_fe_tpm %>%
  mutate(log_abundance = log1p(Abundance)) %>%
  group_by(genome, classification) %>%
  summarise(
    model = list(lm(haz_cont ~ log_abundance + agem + sexe)),
    .groups = "drop"
  ) %>%
  mutate(
    summary = map(model, summary),
    estimate = map_dbl(summary, ~ coef(.x)["log_abundance", "Estimate"]),
    std.error = map_dbl(summary, ~ coef(.x)["log_abundance", "Std. Error"]),
    p.value   = map_dbl(summary, ~ coef(.x)["log_abundance", "Pr(>|t|)"])
  ) %>%
  mutate(
    p.adj = p.adjust(p.value, method = "fdr")
  ) %>%
  select(genome, classification, estimate, std.error, p.value, p.adj)

lm_results_tpm_sign <- filter(lm_results_tpm, p.adj <= 0.05)
# Not significant!

# Test with Ancombc
# Matrix of abundance
rownames(coverm_ancombc_strains) <- coverm_ancombc_strains$Genome
coverm_ancombc_strains <- coverm_ancombc_strains[,-1]
coverm_ancombc_strains <- otu_table(coverm_ancombc_strains, taxa_are_rows = TRUE)
# Matrix of taxnonomy
mags_ancombc_strains <- mags_df_strains
mags_ancombc_strains <- cbind(mags_ancombc_strains$genome, str_split_fixed(mags_ancombc_strains$classification, "\\;[p,c,o,f,g,s]__", 7))
mags_ancombc_strains <- as.data.frame(mags_ancombc_strains)
mags_ancombc_strains[,2] <- str_remove(mags_ancombc_strains[,1], "d__")
colnames(mags_ancombc_strains) <- c("MAGs","Kingdom", "Phylum", "Class","Order","Family","Genus","Species")
mags_ancombc_strains <- as.data.frame(mags_ancombc_strains)
mags_ancombc_strains$Species[which(mags_ancombc_strains$Species=="")] <- mags_ancombc_strains$Genus[which(mags_ancombc_strains$Species=="")]
rownames(mags_ancombc_strains) <- mags_ancombc_strains$MAGs
mags_ancombc_strains <- mags_ancombc_strains[,-1]
mags_ancombc_strains <- as.matrix(mags_ancombc_strains)
mags_ancombc_strains <- tax_table(mags_ancombc_strains)
# Adding metadata
md_ancombc_strains <- sample_data(md_sat)
# phyloseq 
ps_mags_strains <- phyloseq(coverm_ancombc_strains, mags_ancombc_strains, md_ancombc_strains)
ancombc_out_strains <- run_ancombc(ps_mags_strains, confounders = c("agem", "sexe"), group = "Stunting_status", taxa_rank = "Species",
                              p_adjust = "BH")
# No marker was identified
ps_mags_strains <- subset_samples(ps_mags_strains, body_site == "Feces")

# Trying contiuous variable: haz
ancombc_out_mags_cont <- ancombc(
  ps_mags_strains,
  formula = "haz_cont",
  p_adj_method = "BH",
  prv_cut = 0.1,
  tax_level = "Species"
)
View(ancombc_out_mags_cont$res$q_val)
# Nothing significant

```

Maaslin test
```{r}
coverm_maaslin <- coverm_fe_tpm %>%
  select(id, genome, Abundance) %>%
  pivot_wider(names_from = id, values_from = Abundance)
coverm_maaslin <- as.data.frame(coverm_maaslin)
rownames(coverm_maaslin) <- coverm_maaslin$genome
coverm_maaslin <- coverm_maaslin[,-1]

# metadata
md_maaslin <- coverm_fe_tpm %>%
  select(id, Project, Country, body_site, Stunting_status, haz_cont, haz, Stunting_level) %>%
  distinct()
md_maaslin <- as.data.frame(md_maaslin)
rownames(md_maaslin) <- md_maaslin$id
md_maaslin$Stunting_level <- factor(
  md_maaslin$Stunting_level,
  levels = c("Normal_growth", "Mild_stunting", "Moderate_stunting", "Severe_stunting"))

library(Maaslin2)
fit_data <- Maaslin2(
  input_data = coverm_maaslin,
  input_metadata = md_maaslin,
  output = "maaslin2_MAGs_HAZ_results",
  normalization = "TSS",       # Converts to relative abundance
  transform = "LOG",           # Log-transform data
  fixed_effects = c("haz_cont"),
  random_effects = NULL        # Or add batch info if needed
)
# "There are no associations to plot!"
```

Species level - TPM
```{r}
coverm_dir_sp <- "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/CoverM/tpm_species"
coverm_sp_tables <- list.files(coverm_dir_sp, full.names = TRUE, pattern = "*.tsv")
coverm_sp <- coverm_sp_tables %>%
  map(read.delim) %>%
  purrr::reduce(full_join, by = "Genome")
colnames(coverm_sp) <- str_remove_all(colnames(coverm_sp), "Satellite_drep_species.fa.")
colnames(coverm_sp) <- str_remove_all(colnames(coverm_sp), "_FASTP_BOWTIE2_R1_001.fastq.gz.TPM")
# 535 species MAGs

coverm_ancombc <- coverm_sp

# Add metadata
coverm_sp <- pivot_longer(coverm_sp, cols = 2:130, names_to = "SampleID", values_to = "Abundance")
# Satellite
coverm_sp <- merge(md_sat, coverm_sp, by.x = "id", by.y = "SampleID")
# Select stool
coverm_sp_fe <- coverm_sp %>% filter(body_site == "Feces")
# Remove abundance 0 (272 genomes)
coverm_sp_fe <- coverm_sp_fe  %>%
  group_by(Genome) %>%
  filter(sum(Abundance) > 0) %>%
  ungroup()
n_distinct(coverm_sp_fe$Genome)
# 499 MAGs now

# Linear model for each MAGs
lm_results_sp <- coverm_sp_fe %>%
  mutate(log_abundance = log1p(Abundance)) %>%
  group_by(Genome) %>%
  summarise(
    model = list(lm(haz_cont ~ log_abundance + agem + sexe)),
    .groups = "drop"
  ) %>%
  mutate(
    summary = map(model, summary),
    estimate = map_dbl(summary, ~ coef(.x)["log_abundance", "Estimate"]),
    std.error = map_dbl(summary, ~ coef(.x)["log_abundance", "Std. Error"]),
    p.value   = map_dbl(summary, ~ coef(.x)["log_abundance", "Pr(>|t|)"])
  ) %>%
  mutate(
    p.adj = p.adjust(p.value, method = "fdr")
  ) %>%
  select(Genome, estimate, std.error, p.value, p.adj)

## NOTHING SIGNIFICANT ##

# test with ancom-bc
# Matrix of abundance
rownames(coverm_ancombc) <- coverm_ancombc$Genome
coverm_ancombc <- coverm_ancombc[,-1]
coverm_ancombc <- otu_table(coverm_ancombc, taxa_are_rows = TRUE)
# Matrix of taxnonomy
mags_ancombc <- mags_df
mags_ancombc <- cbind(mags_ancombc$genome, str_split_fixed(mags_ancombc$classification, "\\;[p,c,o,f,g,s]__", 7))
mags_ancombc <- as.data.frame(mags_ancombc)
mags_ancombc[,2] <- str_remove(mags_ancombc[,1], "d__")
colnames(mags_ancombc) <- c("MAGs","Kingdom", "Phylum", "Class","Order","Family","Genus","Species")
mags_ancombc <- as.data.frame(mags_ancombc)
mags_ancombc$Species[which(mags_ancombc$Species=="")] <- mags_ancombc$Genus[which(mags_ancombc$Species=="")]
rownames(mags_ancombc) <- mags_ancombc$MAGs
mags_ancombc <- mags_ancombc[,-1]
mags_ancombc <- as.matrix(mags_ancombc)
mags_ancombc <- tax_table(mags_ancombc)
# Adding metadata
md_ancombc <- sample_data(md_sat)
# phyloseq 
ps_mags <- phyloseq(coverm_ancombc, mags_ancombc, md_ancombc)
ancombc_out_mags <- run_ancombc(ps_mags, confounders = c("agem", "sexe"), group = "Stunting_status", taxa_rank = "Species",
                              p_adjust = "BH")
# No marker was identified
lefse_out_mags <- run_lefse(ps_mags, group = "Stunting_status", taxa_rank = "Species", transform = "log10p", norm = "TSS")
# No marker was identified
# Trying contiuous variable: haz
ancombc_out_mags_cont <- ancombc(
  ps_mags_strains,
  formula = "haz_cont",
  p_adj_method = "BH",
  prv_cut = 0.1,
  tax_level = "Species"
)
View(ancombc_out_mags_cont$res$q_val)
# Nothing significant

```

Adding Afribiota samples!
Species level - TPM
```{r}
coverm_dir_sp_SA <- "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/CoverM/SATAFB_species"
coverm_sp_SA_tables <- list.files(coverm_dir_sp_SA, full.names = TRUE, pattern = "*.tsv")
coverm_sp_SA <- coverm_sp_SA_tables %>%
  map(read.delim) %>%
  purrr::reduce(full_join, by = "Genome")
colnames(coverm_sp_SA) <- str_remove_all(colnames(coverm_sp_SA), "SAT_AFB_drep_species.fa.")
colnames(coverm_sp_SA) <- str_remove_all(colnames(coverm_sp_SA), "_FASTP_BOWTIE2_R1_001.fastq.gz.TPM")
colnames(coverm_sp_SA) <- str_remove_all(colnames(coverm_sp_SA), "_bowtie2_R1_001.fastq.gz.TPM")
colnames(coverm_sp_SA) <- str_remove_all(colnames(coverm_sp_SA), "SE")
colnames(coverm_sp_SA) <- str_remove_all(colnames(coverm_sp_SA), "_")
# For Ancom bc
coverm_ancombc_SA <- coverm_sp_SA

# 791 species MAGs
# Add metadata
coverm_sp_SA <- pivot_longer(coverm_sp_SA, cols = 2:187, names_to = "SampleID", values_to = "Abundance")
coverm_sp_SA <- merge(md_AFB, coverm_sp_SA, by.x = "id", by.y = "SampleID")

# Select stool
coverm_sp_SA_fe <- coverm_sp_SA %>% filter(body_site == "Feces") %>%
  filter(Country == "RCA")
# Remove abundance 0 (272 genomes)
coverm_sp_SA_fe <- coverm_sp_SA_fe  %>%
  group_by(Genome) %>%
  filter(sum(Abundance) > 0) %>%
  ungroup()
n_distinct(coverm_sp_SA_fe$Genome)
# 749 MAGs now

# Linear model for each MAGs
lm_results_sp_SA <- coverm_sp_SA_fe %>%
  mutate(log_abundance = log1p(Abundance)) %>%
  group_by(Genome) %>%
  summarise(
    model = list(lm(haz_cont ~ log_abundance + Project + agem + sexe)),
    .groups = "drop"
  ) %>%
  mutate(
    summary = map(model, summary),
    estimate = map_dbl(summary, ~ coef(.x)["log_abundance", "Estimate"]),
    std.error = map_dbl(summary, ~ coef(.x)["log_abundance", "Std. Error"]),
    p.value   = map_dbl(summary, ~ coef(.x)["log_abundance", "Pr(>|t|)"])
  ) %>%
  mutate(
    p.adj = p.adjust(p.value, method = "fdr")
  ) %>%
  select(Genome, estimate, std.error, p.value, p.adj)

## NOTHING SIGNIFICANT ##
# test with ancom-bc
# Matrix of abundance
rownames(coverm_ancombc_SA) <- coverm_ancombc_SA$Genome
coverm_ancombc_SA <- coverm_ancombc_SA[,-1]
coverm_ancombc_SA <- otu_table(coverm_ancombc_SA, taxa_are_rows = TRUE)
# Matrix of taxnonomy
mags_afb <- read.delim2("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Afribiota/Dataset_PESeq/Afribiota_gtdbtk.bac120.summary.tsv")
mags_afb <- mags_afb[,1:2]
colnames(mags_afb)[1] <- "genome"
mags_df_afb <- mags_df[,c(1,7)]
mags_afb <- rbind(mags_afb, mags_df_afb)
mags_ancombc_SA <- mags_afb
mags_ancombc_SA <- cbind(mags_ancombc_SA$genome, str_split_fixed(mags_ancombc_SA$classification, "\\;[p,c,o,f,g,s]__", 7))
mags_ancombc_SA <- as.data.frame(mags_ancombc_SA)
mags_ancombc_SA[,2] <- str_remove(mags_ancombc_SA[,1], "d__")
colnames(mags_ancombc_SA) <- c("MAGs","Kingdom", "Phylum", "Class","Order","Family","Genus","Species")
mags_ancombc_SA <- as.data.frame(mags_ancombc_SA)
mags_ancombc_SA$Species[which(mags_ancombc_SA$Species=="")] <- mags_ancombc_SA$Genus[which(mags_ancombc_SA$Species=="")]
rownames(mags_ancombc_SA) <- mags_ancombc_SA$MAGs
mags_ancombc_SA <- mags_ancombc_SA[,-1]
mags_ancombc_SA <- as.matrix(mags_ancombc_SA)
mags_ancombc_SA <- tax_table(mags_ancombc_SA)
# Adding metadata
md_ancombc_SA <- sample_data(md_AFB)
# phyloseq 
ps_ancombc_SA <- phyloseq(coverm_ancombc_SA, mags_ancombc_SA, md_ancombc_SA)
ps_ancombc_SA <- subset_samples(ps_ancombc_SA, Country == "RCA" & body_site == "Feces")
ancombc_SA_out_mags <- run_ancombc(ps_ancombc_SA, confounders = c("agem", "sexe", "Project"), group = "Stunting_level", taxa_rank = "Species",
                              p_adjust = "fdr")
# No marker was identified
lefse_SA_out_mags <- run_lefse(ps_ancombc_SA, group = "Stunting_status", taxa_rank = "Species", transform = "log10p", norm = "TSS")
# No marker was identified
```

Strain level
```{r}
coverm_dir_st_SA <- "/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Assembly_based_output/CoverM/SATAFB_strains"
coverm_st_SA_tables <- list.files(coverm_dir_st_SA, full.names = TRUE, pattern = "*.tsv")
coverm_st_SA <- coverm_st_SA_tables %>%
  map(read.delim) %>%
  purrr::reduce(full_join, by = "Genome")
colnames(coverm_st_SA) <- str_remove_all(colnames(coverm_st_SA), "SAT_AFB_all_genomes.fa.")
colnames(coverm_st_SA) <- str_remove_all(colnames(coverm_st_SA), "_FASTP_BOWTIE2_R1_001.fastq.gz.TPM")
colnames(coverm_st_SA) <- str_remove_all(colnames(coverm_st_SA), "_bowtie2_R1_001.fastq.gz.TPM")
colnames(coverm_st_SA) <- str_remove_all(colnames(coverm_st_SA), "SE")
colnames(coverm_st_SA) <- str_remove_all(colnames(coverm_st_SA), "_")
# 3296 species MAGs
# Add metadata
coverm_st_SA <- pivot_longer(coverm_st_SA, cols = 2:187, names_to = "SampleID", values_to = "Abundance")
coverm_st_SA <- merge(md_AFB, coverm_st_SA, by.x = "id", by.y = "SampleID")

# Select stool
coverm_st_SA_fe <- coverm_st_SA %>% filter(body_site == "Feces") %>% filter(Country == "RCA")
# Remove abundance 0 (272 genomes)
coverm_st_SA_fe <- coverm_st_SA_fe  %>%
  group_by(Genome) %>%
  filter(sum(Abundance) > 0) %>%
  ungroup()
n_distinct(coverm_st_SA_fe$Genome)
# 3131 MAGs now

# Linear model for each MAGs
lm_results_st_SA <- coverm_st_SA_fe %>%
  mutate(log_abundance = log1p(Abundance)) %>%
  group_by(Genome) %>%
  summarise(
    model = list(lm(haz_cont ~ log_abundance + Project + agem + sexe)),
    .groups = "drop"
  ) %>%
  mutate(
    summary = map(model, summary),
    estimate = map_dbl(summary, ~ coef(.x)["log_abundance", "Estimate"]),
    std.error = map_dbl(summary, ~ coef(.x)["log_abundance", "Std. Error"]),
    p.value   = map_dbl(summary, ~ coef(.x)["log_abundance", "Pr(>|t|)"])
  ) %>%
  mutate(
    p.adj = p.adjust(p.value, method = "fdr")
  ) %>%
  select(Genome, estimate, std.error, p.value, p.adj)
```

```{r}
dist.bc_abc_SA <- phyloseq::distance(ps_ancombc_SA, method = "bray")

# Permanova
meta_pcoa_abc_SA = as.matrix(sample_data(ps_ancombc_SA))
meta_pcoa_abc_SA <- as.data.frame(meta_pcoa_abc_SA)
permanova_proj_SA <- vegan::adonis2(dist.bc_abc_SA ~ Project, data = meta_pcoa_abc_SA, permutations = 9999) # Not significant

permanova_SA <- vegan::adonis2(dist.bc_abc_SA ~ Stunting_level, data = meta_pcoa_abc_SA, permutations = 9999) # significant!!

test_PCoAbray_SA <- ordinate(ps_ancombc_SA, "PCoA", distance = dist.bc_abc_SA)
plot_ordination(ps_ancombc_SA, test_PCoAbray_SA, type = "samples", color = "Stunting_level", shape = "Stunting_level") +
  stat_ellipse(mapping= aes(fill = Stunting_level),geom = "polygon", alpha = 0.2, show.legend = TRUE) +
  theme_bw() +
  scale_shape_manual(values=c(15:18)) +
  scale_color_brewer(palette = "Set1")
```



id = general sample ID (1429AGN###)
sexe = sex of the individual
nagem = age in months 
nheight = height of the individual
nweight = weight of the individual
statut_nut = nutritional status at recruitment (different than HAZ)
voie_accou = delivery (vaginal or C-section)
atb_suivi = antibiotic (yes or no)
ph_estomac
ph_intestin
crp = C-reactive protein
SeqBatch = sequencing batch (either VH or SY)

atcd_malnut

# B:H ratio
Stool
```{r}
# Using absolute number of reads (not relative abundance)
# Filter for fecal samples, group by sample and select variable of interest, sum the abundance of all species (number of reads) and compute log (natural log) of sum abundance over number of host read removed
ps_abs_m <- psmelt(physeq_abs_filtered)
bh_ratio_feces <- ps_abs_m %>%
  filter(body_site == "Feces") %>%
  group_by(Sample, sexe, nagem, statut_nut, stunted, haz, haz_cont, stuntedandwasted, stuntedorwasted,
           voie_accou, atb_suivi,
           body_site, AAT...mg.g., CALPROTECTINE..g.g., Gut_inflammation_AAT, Gut_inflammation_Calpro,
           Systemic_inflammation, crp, Leaky_gut, Total, Retained, Removed) %>%
  summarise(sumAbundance = sum(Abundance)) %>%
  mutate(BH_ratio = sumAbundance/Removed) %>%
  mutate(lnBH_ratio = log(sumAbundance/Removed)) 
colnames(bh_ratio_feces)[13] <- "AAT"
colnames(bh_ratio_feces)[14] <- "Calprotectine"
bh_ratio_feces$crp <- as.numeric(bh_ratio_feces$crp)

# AGN001 has more host reads removed than there are bacterial reads
# removed as an outlier
bh_ratio_feces <- bh_ratio_feces %>% filter(Sample != "AGN001FE")
# We are left with 43 samples

# Effect of calprotectin inflammation on host reads removed
summary(lm(Removed ~ Gut_inflammation_Calpro, data = bh_ratio_feces))
```
Testing for effect 
```{r}
summary(lm(lnBH_ratio ~ haz_cont + sexe + nagem + voie_accou + Gut_inflammation_AAT + Gut_inflammation_Calpro, data = bh_ratio_feces))
# Gut_inflammation_Calpro significant with the full model: power reduction due to too many variables?
# Try a model stepwise
full_model <- lm(lnBH_ratio ~ haz_cont + sexe + nagem + voie_accou +
                   Gut_inflammation_AAT + Gut_inflammation_Calpro, data = bh_ratio_feces)
step_model <- step(full_model, direction = "both")
summary(step_model)
# It keep only haz_cont and Gut_inflammation_Calpro
# HAZ cont is sifinicative alongside gut inflammation with calprotectin 
summary(lm(lnBH_ratio ~ haz_cont + Gut_inflammation_Calpro, data = bh_ratio_feces))
#Visualization
ggplot(bh_ratio_feces, aes(x = haz_cont, y = lnBH_ratio, color = Gut_inflammation_Calpro)) +
  geom_point(size = 2) +
  geom_smooth(method = "lm", se = FALSE) +  # regression lines per group
  labs(
    title = "lnBH_ratio vs haz_cont by Gut_inflammation_Calpro",
    x = "haz_cont",
    y = "lnBH_ratio",
    color = "Gut Inflammation"
  ) +
    geom_text(aes(label = Sample), size = 3, color = "black") +
  theme_minimal()

```
```{r}
summary(lm(lnBH_ratio ~ haz_cont + Gut_inflammation_Calpro, data = bh_ratio_feces))
fit <- lm(lnBH_ratio ~ haz_cont + Gut_inflammation_Calpro, data = bh_ratio_feces)
# Extract summary info
summary_fit <- summary(fit)
# Extract p-values
p_haz_cont <- summary_fit$coefficients["haz_cont", "Pr(>|t|)"]
p_gut_inf <- summary_fit$coefficients["Gut_inflammation_CalproNo_inflammation", "Pr(>|t|)"]
# Extract R-squared and F-test p-value
r_squared <- summary_fit$r.squared
f_pvalue <- pf(summary_fit$fstatistic[1], summary_fit$fstatistic[2], summary_fit$fstatistic[3], lower.tail=FALSE)
# Format text annotation with some rounding
annotation_text <- paste0(
  "p(haz) = ", signif(p_haz_cont, 3), "\n",
  "p(calprotectin) = ", signif(p_gut_inf, 3), "\n",
  "R = ", signif(r_squared, 3), "\n",
  "F-test p = ", signif(f_pvalue, 3)
)

bh_ratio_feces$Gut_inflammation_Calpro <- str_replace_all(bh_ratio_feces$Gut_inflammation_Calpro, "_", " ")

# Plot
pdf("Figures/BHratio_haz.pdf", height = 6, width = 8)
ggplot(bh_ratio_feces, aes(x = haz_cont, y = lnBH_ratio, color = Gut_inflammation_Calpro)) +
  geom_point(size = 2) +
  geom_smooth(method = "lm", se = FALSE) +
  scale_color_manual(values = c( "#e06666", "#93c47d")) +
  labs(
    x = "heigth-for-age z-score",
    y = "ln(Bacterial-to-host reads ratio)",
    color = "Calprotectin"
  ) +
  annotate("text", 
           x = -0.7, y = 4, 
           label = annotation_text, 
           hjust = 0, vjust = 1,
           size = 4,
           color = "black") +
  theme_minimal()
dev.off()
```

Test gastric/duodenal samples
```{r}
ps_abs_gd <- ps_abs_m %>%
  group_by(Sample, sexe, nagem, statut_nut, stunted, haz, haz_cont, stuntedandwasted, stuntedorwasted,
           voie_accou, atb_suivi,
           body_site, AAT...mg.g., CALPROTECTINE..g.g., Gut_inflammation_AAT, Gut_inflammation_Calpro,
           Systemic_inflammation, crp, Leaky_gut, Total, Retained, Removed) %>%
  summarise(sumAbundance = sum(Abundance)) %>%
  filter(body_site == "Duodenum")
ps_abs_gd <- ps_abs_gd %>%
  mutate(BH_ratio = log(sumAbundance/Removed))
colnames(ps_abs_gd)[13] <- "AAT"
colnames(ps_abs_gd)[14] <- "Calprotectine"
ps_abs_gd$crp <- as.numeric(ps_abs_gd$crp)

summary(lm(BH_ratio ~ haz_cont + sexe + nagem + voie_accou + Leaky_gut + Gut_inflammation_AAT + Gut_inflammation_Calpro + Leaky_gut, data = ps_abs_gd))
# Gut_inflammation_Calpro significant with the full model: power reduction due to too many variables?
# Try a model stepwise
full_model_gd <- lm(BH_ratio ~ haz_cont + sexe + nagem + voie_accou +
                   Gut_inflammation_AAT + Gut_inflammation_Calpro, data = ps_abs_gd)
step_model_gd <- step(full_model_gd, direction = "both")
summary(step_model_gd)
```

##B:H ratio for individual taxa
Estimation of absolute abundance
Stool
```{r}
physeq_sp_ratio <- subset_samples(physeq_sp, body_site == "Feces")
physeq_sp_ratio <- prune_taxa(taxa_sums(physeq_sp_ratio) > 0, physeq_sp_ratio)
# Get prevalence
prevdf_sp_ratio = apply(X= otu_table(physeq_sp_ratio),
               MARGIN = ifelse(taxa_are_rows(physeq_sp_ratio),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_sp_ratio = data.frame(Prevalence = prevdf_sp_ratio,
                        Percentage_Prev_feces = 100*prevdf_sp_ratio/44,
                        tax_table(physeq_sp_ratio))
prevdf_sp_ratio_filt <- prevdf_sp_ratio %>% filter(Percentage_Prev_feces >= 10)
# 777 species are at least present in 10% of the samples
physeq_sp_ratio_filt <- subset_taxa(physeq_sp_ratio, Species %in% prevdf_sp_ratio_filt$Species)
physeq_sp_ratio_m <- psmelt(physeq_sp_ratio_filt)

# Adding ratio to melted ps
bh_ratio_df <- as.data.frame(bh_ratio_feces)
ps_sp_ratio_m <- merge(physeq_sp_ratio_m, bh_ratio_df, by = "Sample")
ps_sp_ratio_m <- ps_sp_ratio_m %>%
  mutate(Pseudo_relative_abundance = BH_ratio*Abundance)
ps_sp_ratio_m <- ps_sp_ratio_m %>%
  mutate(logPseudo_relative_abundance = log(Pseudo_relative_abundance+0.000001))
# Removing outlier
ps_sp_ratio_m <- ps_sp_ratio_m %>%
  filter(Sample != "AGN001FE")

# Testing linear model
taxa_lm_results <- ps_sp_ratio_m %>%
  group_by(Species) %>% 
  do(tidy(lm(logPseudo_relative_abundance ~ haz_cont.x + Gut_inflammation_Calpro.x, data = .))) %>%
  ungroup() %>%
  group_by(term) %>%
  mutate(p.adj = p.adjust(p.value, method = "BH")) %>%
  ungroup() %>%
  filter(term %in% c("haz_cont.x", "Gut_inflammation_Calpro.x")) %>%
  filter(p.adj < 0.05) %>%
  arrange(p.adj)

# Testing ps_sp_ratio_m
cor_results_abs <- ps_sp_ratio_m %>%
  group_by(Species) %>%  # replace 'OTU' with your taxa column name if different (e.g. Species)
  summarise(
    cor_spearman = cor(logPseudo_relative_abundance, haz_cont.x, method = "spearman", use = "complete.obs"),
    p_spearman = cor.test(logPseudo_relative_abundance, haz_cont.x, method = "spearman")$p.value,
  ) %>%
  ungroup() %>%
  mutate(p_adj_spearman = p.adjust(p_spearman, method = "BH"))

# Neither LM or spearman correlation return a significant species associated with HAZ

counts_mat_ratio <- as(otu_table(physeq_sp_ratio), "matrix")
# Remove AGN001
counts_mat_ratio <- counts_mat_ratio[,-1]
# Get B:H ratio vector
bh_ratios <- bh_ratio_df[, "BH_ratio"]
# Multiply each sample column by its B:H ratio
counts_abs <- sweep(counts_mat_ratio, 2, bh_ratios, FUN = "*")
# Recreate phyloseq with the new count table
physeq_sp_abs_ratio <- physeq_sp_ratio
otu_table(physeq_sp_abs_ratio) <- otu_table(counts_abs, taxa_are_rows = TRUE)
physeq_sp_abs_ratio <- tax_glom(physeq_sp_abs_ratio, "Species")
tax_table(physeq_sp_abs_ratio) <- tax_table(physeq_sp_abs_ratio)[, -8]
# Test ANCOM-BC
# Not significant when cofounder stunted added
ancombc_out_ratio <- run_ancombc(physeq_sp_abs_ratio, group = "Gut_inflammation_Calpro",
                                taxa_rank = "Species", confounders = "stunted",
                              p_adjust = "BH")
plot_ef_bar(ancombc_out_ratio)

# Not significant 
res_Ancombc <- ancombc(
  physeq_sp_abs_ratio,
  formula = "haz_cont",
  p_adj_method = "BH",
)
```


