---
title: "Satellite 16S rRNA data abalysis"
author: "Simon Yersin"
date: "2024-09-19"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, warning=FALSE,message=FALSE}
library(readstata13)
library(stringr)
library(magrittr)
library(devtools)
library(tidyr)
library(dplyr)
library(ggplot2)
library(tidyverse)
library(writexl)
library(readxl)
library(phyloseq)
library(RColorBrewer)
library(usedist)
library(vegan)
library(MicEco)
library(ComplexHeatmap)
library(circlize)
```


# Loading tables
Sample names correspondence table
```{r}
sample_corr <- read.csv("Tables/table_after_qc_20190313v2.csv", sep = ";")
# Keeping only Satellite samples
sample_corr <- sample_corr %>%
  filter(str_detect(ID_afri, "AGN"))

sample_corr$SampleType <- str_replace(sample_corr$SampleType, "duodenal", "DU")
sample_corr$SampleType <- str_replace(sample_corr$SampleType, "feces", "FE")
sample_corr$SampleType <- str_replace(sample_corr$SampleType, "gastric", "GA")
sample_corr$SampleType <- str_replace(sample_corr$SampleType, "saliva", "SA")

sample_corr <- sample_corr %>%
  mutate(id_modif = str_c(id, sep = "", SampleType))

sample_corr$id_modif <- str_remove(sample_corr$id_modif, "1429")
```
Taxonomy table
```{r}
tax = read.delim2("Tables/taxonomy_table_Satellite.txt", header = TRUE)

tax$Phylum[which(is.na(tax[,2]==TRUE))] <- "Unassigned"
tax$Class[which(is.na(tax[,3]==TRUE))] <- "Unassigned"
tax$Order[which(is.na(tax[,4]==TRUE))] <- "Unassigned"
tax$Family[which(is.na(tax[,5]==TRUE))] <- "Unassigned"
tax$Genus[which(is.na(tax[,6]==TRUE))] <- "Unassigned"
tax$Species[which(is.na(tax[,7]==TRUE))] <- "unassigned"
tax=as.matrix(tax)
taxonomy <- tax_table(tax)
``` 
Abundance table
```{r}
otu <- read.delim2("Tables/abundance_table_Satellite.txt", header = TRUE)

sample_corr <- cbind(sample_corr,colnames(otu))
setdiff(sample_corr$ID_metag, sample_corr$`colnames(otu)`)
# Col from the abundance table in the same order than the sample correspondence table
# Adapt colnames to the metadata sample names for clean output matching the shotgun metagenomic ids
colnames(otu) <- sample_corr$id_modif

otu<-as.matrix(otu)
class(otu) <- "numeric"

# Number of reads assigned in each samples
which(colSums(otu) < 5000)
# 5 samples with less than 5'000 reads assigned
# These samples will be removed from the analysis (see filtering later)
# AGN038DU AGN014GA AGN029GA AGN038GA AGN020SA 

abundance = otu_table(otu, taxa_are_rows = TRUE)
```
Metadata
```{r}
md_16S <- as.data.frame(read_xlsx("/Users/admin/Documents/VLab_projects/01_Projects_MAIN/Satellite/Analysis/Metadata/Metadata_satellite_full.xlsx"))
rownames(md_16S) <- md_16S$id_modif
md_16S_ps <- sample_data(md_16S)
```

# phyloseq
```{r}
physeq_16S <- phyloseq(taxonomy, abundance)
# Add metadata
physeq_16S <- merge_phyloseq(physeq_16S, md_16S_ps)
physeq_16S # 2'701 taxa in 146 samples with 548 variables
```

Filtering
```{r}
# remove sample with less than 5000 reads
physeq_16S_filt <- subset_samples(physeq_16S, id_modif != "AGN038DU" & id_modif != "AGN014GA" & id_modif != "AGN029GA" & id_modif != "AGN038GA" & id_modif != "AGN020SA")
# 141 samples remaining

# removing sample with gastric ph at 6 and above 
`%notin%` <- Negate(`%in%`)
physeq_16S_filt <- subset_samples(physeq_16S_filt, id_modif %notin% md_16S$id_modif[which(md_16S$ph_estomac >=6 & md_16S$body_site == "Gastric")])
# 13 samples removed

# removing sample with duodenal ph at 4 and below 
physeq_16S_filt <- subset_samples(physeq_16S_filt, id_modif %notin% md_16S$id_modif[which(md_16S$ph_intestin <=4 & md_16S$body_site == "Duodenum")])
# 2 sample removed


physeq_16S_filt
# 126 samples remaining
# Which ones are not matching the shotgun metagenomic data?

# Filter the dataset to keep only relevant data
get_taxa_unique(physeq_16S_filt, "Kingdom")  ##ex: Bacteria, Archaea

# filter out Chloroplast and Mitochondria
physeq_filtered <- physeq_16S_filt %>%
  subset_taxa(Family != "Mitochondria") %>%
  subset_taxa(Order != "Chloroplast")
physeq_filtered # 2'685 taxa, removed 6 taxa

# some taxa have only 0 --> filtered out as well
physeq_filtered <- prune_taxa(taxa_sums(physeq_filtered)> 0, physeq_filtered)
physeq_filtered # 2'472 taxa, removed 213 taxa with only 0 abundance

# relative abundance
physeq_ra <- transform_sample_counts(physeq_filtered, function(x) x / sum(x))
```

Sequencing depth
```{r}
body_site_list <- as.data.frame(unique(md_16S$body_site))
names(body_site_list)[1] <- "body_site"


sample_sums_df <- as.data.frame(sample_sums(physeq_filtered))
names(sample_sums_df)[1] <- "Reads_per_sample" 
sample_sums_df <- sample_sums_df %>%
  mutate("fastqID" = rownames(sample_sums_df))
sample_sums_df <- merge(sample_sums_df, md_16S, by.x="fastqID", by.y = "id_modif")

df_seq <- matrix(ncol =8, nrow = 4)
colnames(df_seq) <- c("#Samples", "mean", "max", "min", "sd", "sum", "<10'000", "n_taxa")
rownames(df_seq) <- body_site_list$body_site

for (j in 1:4){
      df_seq[j,1] <- length(which(sample_sums_df$body_site==body_site_list[j,]))
      df_seq[j,2] <- mean(sample_sums(subset_samples(physeq_filtered, body_site==body_site_list[j,])))
      df_seq[j,3] <-  max(sample_sums(subset_samples(physeq_filtered, body_site==body_site_list[j,])))
      df_seq[j,4] <- min(sample_sums(subset_samples(physeq_filtered, body_site==body_site_list[j,])))
      df_seq[j,5] <- sd(sample_sums(subset_samples(physeq_filtered, body_site==body_site_list[j,])))
      df_seq[j,6] <- sum(sample_sums(subset_samples(physeq_filtered, body_site==body_site_list[j,])))
      df_seq[j,7] <- length(which(sample_sums_df$Reads_per_sample < 10000 & sample_sums_df$body_site==body_site_list[j,]))
      #n_taxa_ps <- subset_samples(physeq_filtered, body_site==body_site_list[j,])
      #n_taxa_ps <-  prune_taxa(taxa_sums(n_taxa_ps)> 0, n_taxa_ps)
      #df_seq[j,8] <- as.numeric(ntaxa(n_taxa_ps))

}

# Save table with sequencing info after filtering
write.csv(df_seq, "Tables/16S_Satellite_sequencing_info_postfilt.csv")
```

Samples count per body sites
```{r}
md_16S_final <- as.matrix(sample_data(physeq_filtered))
md_16S_final <- as.data.frame(md_16S_final)

# Oral samples
n_distinct(md_16S_final$id[which(md_16S_final$body_site =="Saliva")]) # 41 samples
n_distinct(md_16S_final$id[which(md_16S_final$body_site =="Gastric")]) # 28 samples
n_distinct(md_16S_final$id[which(md_16S_final$body_site =="Duodenum")]) # 13 samples
n_distinct(md_16S_final$id[which(md_16S_final$body_site =="Feces")]) # 44 samples

# Pairs
n_distinct(intersect(md_16S_final$id[which(md_16S_final$body_site =="Gastric")],
          md_16S_final$id[which(md_16S_final$body_site =="Saliva")])) # 26 samples with both saliva and gastric
n_distinct(intersect(md_16S_final$id[which(md_16S_final$body_site =="Gastric")],
          md_16S_final$id[which(md_16S_final$body_site =="Duodenum")])) # 10 samples with both gastric and duodenal samples
n_distinct(intersect(md_16S_final$id[which(md_16S_final$body_site =="Gastric")],
          md_16S_final$id[which(md_16S_final$body_site =="Feces")])) # 28 samples with both gastric and fecal samples
n_distinct(intersect(md_16S_final$id[which(md_16S_final$body_site =="Saliva")],
          md_16S_final$id[which(md_16S_final$body_site =="Duodenum")])) # 13 samples with both saliva and duodenal samples
n_distinct(intersect(md_16S_final$id[which(md_16S_final$body_site =="Saliva")],
          md_16S_final$id[which(md_16S_final$body_site =="Feces")])) # 40 samples with both saliva and fecal samples
n_distinct(intersect(md_16S_final$id[which(md_16S_final$body_site =="Duodenum")],
          md_16S_final$id[which(md_16S_final$body_site =="Feces")])) # 13

n_distinct(intersect(intersect(md_16S_final$id[which(md_16S_final$body_site =="Gastric")],
          md_16S_final$id[which(md_16S_final$body_site =="Duodenum")]),
          md_16S_final$id[which(md_16S_final$body_site =="Saliva")])) # 10 samples with saliva, gastric and duodenal samples
n_distinct(intersect(intersect(md_16S_final$id[which(md_16S_final$body_site =="Gastric")],
          md_16S_final$id[which(md_16S_final$body_site =="Duodenum")]),
          md_16S_final$id[which(md_16S_final$body_site =="Feces")])) # 10 samples with fecal, gastric and duodenal samples
n_distinct(intersect(intersect(md_16S_final$id[which(md_16S_final$body_site =="Saliva")],
          md_16S_final$id[which(md_16S_final$body_site =="Duodenum")]),
          md_16S_final$id[which(md_16S_final$body_site =="Feces")])) # 13 samples with fecal, saliva and duodenal samples
n_distinct(intersect(intersect(md_16S_final$id[which(md_16S_final$body_site =="Gastric")],
          md_16S_final$id[which(md_16S_final$body_site =="Saliva")]),
          md_16S_final$id[which(md_16S_final$body_site =="Feces")])) # 26 samples with fecal, gastric and saliva samples


n_distinct(intersect(intersect(md_16S_final$id[which(md_16S_final$body_site =="Gastric")],
          md_16S_final$id[which(md_16S_final$body_site =="Saliva")]),
          intersect(md_16S_final$id[which(md_16S_final$body_site =="Duodenum")],
          md_16S_final$id[which(md_16S_final$body_site =="Feces")]))) # 10samples with all body sites
```

Rarefaction curve
```{r}
rare_curve_df <- otu_table(physeq_filtered)
class(rare_curve_df) <- "matrix"
rare_curve_df <- t(rare_curve_df)
pdf("Figures/RareCurve_16S.pdf", width = 10)
rarecurve(rare_curve_df, step=100, cex=0.5, lwd=1, ylab = "ASV")
dev.off()
```

# Analysis
Color palette
```{r}
palbrew_info <- brewer.pal.info[brewer.pal.info$category == "qual", ] 
palbrew_all <- unlist(mapply(brewer.pal,palbrew_info$maxcolors,rownames(palbrew_info)))
# Set color palette
pal_1 <- palbrew_all[c(63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14)]
pal_2  <- palbrew_all[c(69,66,67,68)]
pal_3 <- palbrew_all[16:26]
pal_4 <- palbrew_all[c(63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14,
                       63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14)]
pal_5 <- palbrew_all[c(63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14,
                       63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14,
                       63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14,
                       63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14,
                       63,46,64,47,48,16,72,66,67,50,69,51,52,53,61,70,74,1,2,3,10,11,12,13,14)]
```
# Diversity and Richness
## Alpha diversity
```{r}
ps_alpha <- physeq_filtered
results_adiv <- estimate_richness(ps_alpha)
# Warning for singletons, we don't have any... use like metaG data only the following metrics: Observed, Simpsons

write.csv(results_adiv, "Tables/16S_Satellite_AlphaDivResults.csv")
```

Observed index
counts the number of distinct species (or other taxonomic units) observed in the sample

Shannon index
Considers both species richness (the number of different species) and species evenness (the relative abundance of each species)
Higher Values: A higher Shannon index value indicates greater diversity within the community. This means there is a more even distribution of species abundances.
Lower Values: A lower Shannon index value suggests lower diversity, indicating that one or a few species dominate the community.

Simpson's index
accounts for both richness and evenness, providing a measure of diversity that considers the abundance distribution of species
probability that two individuals randomly selected from a sample will belong to the same species. It is a measure of dominance, with values ranging from 0 to 1
Lower Values: Indicate higher diversity (many species with similar abundances).
Higher Values: Indicate lower diversity (few species dominate)

Inverse Simpson's index
Provides an intuitive measure of diversity; higher values indicate higher diversity
Higher Values: Indicate higher diversity (many species with similar abundances)
Lower Values: Indicate lower diversity (few species dominate)

Index computation
In metagenomic sequencing we do not consider Chao1 or ACE index as they are taking into account rare species (singletons and doubletons) to estimate the "unobserved" species. We do not have singletons and doubletons using metaG. Here, even though we have 16S rRNA amplicon sequencing, we have a warning for the absence of singletons in the data, which is suspicious... 

```{r}
results_adiv <- results_adiv %>%
  mutate(SampleID = rownames(results_adiv))
results_adiv <- merge(results_adiv, md_16S, by.x = "SampleID", by.y="id_modif")

results_fe <- results_adiv %>%
  filter(body_site == "Feces")

results_ga <- results_adiv %>%
  filter(body_site == "Gastric")

results_du <- results_adiv %>%
  filter(body_site == "Duodenum")

results_sa <- results_adiv %>%
  filter(body_site == "Saliva")
```

Tests
Histograms
```{r}
ggplot(results_adiv, aes(x = Observed)) +
  geom_histogram(binwidth = 10, falpha = 0.7) +
  labs(title = "Histogram of Observed Index",
       x = "Observed Index",
       y = "Frequency")

ggplot(results_adiv, aes(x = Shannon)) +
  geom_histogram(binwidth = 0.2, falpha = 0.7) +
  labs(title = "Histogram of Shannon Index",
       x = "Shannon Index",
       y = "Frequency")

ggplot(results_adiv, aes(x = Simpson)) +
  geom_histogram(binwidth = 0.02, falpha = 0.7) +
  labs(title = "Histogram of Simpson Index",
       x = "Simpson Index",
       y = "Frequency")

ggplot(results_adiv, aes(x = InvSimpson)) +
  geom_histogram(binwidth = 5, falpha = 0.7) +
  labs(title = "Histogram of InvSimpson Index",
       x = "InvSimpson Index",
       y = "Frequency")
```


ANOVA (analysis of variance) 
for categorical metadata with normal distribution
If significant with multiple categories in the explanatory variables: Tukey test
```{r}
# Body site
summary(aov(Shannon ~ body_site, data = results_adiv))
# 0.00306
TukeyHSD(aov(Shannon ~ body_site, data = results_adiv))
# Significant for: Saliva - Feces : 0.0013711
summary(aov(Observed ~ body_site, data = results_adiv))
# 0.00086
TukeyHSD(aov(Observed ~ body_site, data = results_adiv))
# Significant for: Feces - Saliva 0.0004001
summary(aov(Simpson ~ body_site, data = results_adiv))
# 0.0485
TukeyHSD(aov(Simpson ~ body_site, data = results_adiv))
# Significant for saliva - feces 0.0348399
summary(aov(InvSimpson ~ body_site, data = results_adiv))
# 0.00687
TukeyHSD(aov(InvSimpson ~ body_site, data = results_adiv))
# Significant for saliva - feces 0.0060243

# Body site -> affect alpha diversity

# For all body sites:
# Stunting status (Stunted and Non-stunted)
summary(aov(Observed ~ Stunting_status, data = results_adiv))
summary(aov(Shannon ~ Stunting_status, data = results_adiv))
summary(aov(Simpson ~ Stunting_status, data = results_adiv))
summary(aov(InvSimpson ~ Stunting_status, data = results_adiv))
# Not significant, stunting status does not affect Observed diversity

# Stunting level
summary(aov(Observed ~ Stunting_level, data = results_adiv))
summary(aov(Shannon ~ Stunting_level, data = results_adiv))
summary(aov(Simpson ~ Stunting_level, data = results_adiv))
summary(aov(InvSimpson ~ Stunting_level, data = results_adiv))
# Nothing significant

#Presence of parasites
summary(aov(Shannon ~ giardiase + entamoeba + parasite_aut, data = results_adiv))
summary(aov(Shannon ~ entamoeba + parasite_aut, data = results_adiv))
summary(aov(Shannon ~ entamoeba, data = results_adiv))
summary(aov(Shannon ~ parasite_aut, data = results_adiv))
# Shannon diversity influenced by the presence of Entamoeba parasite
summary(aov(Simpson ~ giardiase + entamoeba + parasite_aut, data = results_adiv))
# Presence of other parasite does not affect Simpson index
summary(aov(InvSimpson ~ giardiase + entamoeba + parasite_aut, data = results_adiv))
# InvSimpson diversity is not influenced by the presence of Entamoeba parasite and other parasite
summary(aov(Observed ~ giardiase + entamoeba + parasite_aut, data = results_fe))
# Observed not influenced by parasite

# Antibiotics
summary(aov(Observed ~ atb_suivi, data = results_adiv))
summary(aov(Shannon ~ atb_suivi, data = results_adiv))
summary(aov(Simpson ~ atb_suivi, data = results_adiv))
summary(aov(InvSimpson ~ atb_suivi, data = results_adiv))
# Nothing significant


# Inflammation
summary(aov(Observed ~ Gut_inflammation_AAT + Gut_inflammation_Calpro + Systemic_inflammation, data = results_adiv))
summary(aov(Shannon ~ Gut_inflammation_AAT + Gut_inflammation_Calpro + Systemic_inflammation, data = results_adiv))
summary(aov(Simpson ~ Gut_inflammation_AAT + Gut_inflammation_Calpro + Systemic_inflammation, data = results_adiv))
summary(aov(InvSimpson ~ Gut_inflammation_AAT + Gut_inflammation_Calpro + Systemic_inflammation, data = results_adiv))
# Nothing significant

# Leaky gut
summary(aov(Observed ~ Leaky_gut, data = results_adiv))
summary(aov(Shannon ~ Leaky_gut, data = results_adiv))
summary(aov(Simpson ~ Leaky_gut, data = results_adiv))
summary(aov(InvSimpson ~ Leaky_gut, data = results_adiv))
# Nothing significant
```

Correlation test and plot (for continuous variables)
Indicates the strength and direction of the linear relationship (values range from -1 to 1)
```{r}
cor.test(results_adiv$Observed, results_adiv$haz_cont, method = "pearson")
ggplot(results_adiv, aes(x = Observed, y = haz_cont)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE, color = "black") +
  labs(title = paste("Correlation:", round(cor.test(results_adiv$Observed, results_adiv$haz_cont, method = "pearson")$estimate, 2)),
       x = "Observed",
       y = "haz")

cor.test(results_adiv$Shannon, results_adiv$haz_cont, method = "pearson")
ggplot(results_adiv, aes(x = Shannon, y = haz_cont)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE, color = "black") +
  labs(title = paste("Correlation:", round(cor.test(results_adiv$Shannon, results_adiv$haz_cont, method = "pearson")$estimate, 2)),
       x = "Shannon",
       y = "haz")
# No correlations between haz and alpha diversity index

# Test correlation between Shannon or Observed (both all data and only feces) with crp, calprotectin, ATT levels and Mannitol-lactilol ratio. 
# Do we need it?
```


Plots
```{r}
md_alpha <- as.matrix(sample_data(ps_alpha))
body_site_levels <- c("Saliva", "Gastric", "Duodenum", "Feces")
body_site_data <- data.frame(
  body_site = body_site_levels,
  bs_num = 1:length(body_site_levels))
md_alpha <- as.data.frame(md_alpha)
md_alpha <- dplyr::left_join(md_alpha, body_site_data, by= "body_site")
sample_data(ps_alpha) <- cbind(sample_data(ps_alpha), md_alpha$bs_num)
colnames(sample_data(ps_alpha))[549] <- "bs_num"
plot_rich <- plot_richness(ps_alpha, x= "bs_num",
                               measures=c("Observed", "Shannon", "Simpson", "InvSimpson"),
                               title = "Alpha Diversity indexes by body sites for 16S rRNA amplicon sequencing data",
                               color = "body_site") +
  scale_x_continuous(breaks=1:length(body_site_levels), labels=body_site_levels) +
  scale_color_manual(values = pal_2) +
  geom_boxplot(aes(x = bs_num, y = value), alpha = 0.1) +
  xlab("Body sites") +
  theme_bw() +
  theme(axis.text.x=element_text(angle = 45, hjust = .9))

plot_rich
ggsave(filename = "Figures/AlphaDivASVs_16SrRNA_satellite.svg", plot = plot_rich, width = 10, height = 8)

# Species level
ps_alpha_sp <- tax_glom(ps_alpha, "Species")
plot_rich_sp <- plot_richness(ps_alpha_sp, x= "bs_num",
                               measures=c("Observed", "Shannon", "Simpson", "InvSimpson"),
                               title = "Alpha Diversity indexes by body sites for 16S rRNA amplicon sequencing data",
                               color = "body_site") +
  scale_x_continuous(breaks=1:length(body_site_levels), labels=body_site_levels) +
  scale_color_manual(values = pal_2) +
  geom_boxplot(aes(x = bs_num, y = value), alpha = 0.1) +
  xlab("Body sites") +
  theme_bw() +
  theme(axis.text.x=element_text(angle = 45, hjust = .9))

plot_rich_sp
ggsave(filename = "Figures/AlphaDivSpecies_16SrRNA_satellite.svg", plot = plot_rich_sp, width = 10, height = 8)
```


## Beta-diversity
All dataset
we first measure "distance" between samples using Bray-Curtis
Bray-Curtis is a dissimilarity measure:
  0 indicates complete similarity (same composition)
  1 indicates complete dissimilarity (no common species)
We first compute the distance (obtaining a distance matrix), then do a PERMANOVA to test for significance differences in community composition between groups
We can then plot the distance (or "dissimilarity") 
And finally, we can use a PCoA to reduce the complexity of multivariate data by representing it in a few (two) dimensions for vizualisation
```{r}
# Bray-Curtis
ps_beta <- physeq_filtered
# Add log transformation? (tests don't change but PCoA axis values do change)
#ps_beta <- transform_sample_counts(ps_beta, function(x) log(1+x))
dist.bc <- phyloseq::distance(ps_beta, method = "bray")

# Permanova
meta_pcoa = as.matrix(sample_data(ps_beta))
meta_pcoa <- as.data.frame(meta_pcoa)
permanova_all <- vegan::adonis2(dist.bc ~ body_site, data = meta_pcoa, permutations = 9999) # Significant
# body site : 1e-04 ***
# structure of the anova table
permanova_all$`Pr(>F)` # 1e-04

pairwise_p <- numeric()
# Pairwise comparison
fe_sa_ado <- meta_pcoa %>% filter(body_site == "Feces" | body_site == "Saliva")
fe_sa_ado_dist <- dist_subset(dist.bc, fe_sa_ado$id_modif)
fe_sa_tado <- adonis2(fe_sa_ado_dist ~ body_site, data = fe_sa_ado,permutations = 1000)
pairwise_p["FE_SA"] <- fe_sa_tado$`Pr(>F)`[1]

fe_ga_ado <- meta_pcoa %>% filter(body_site == "Feces" | body_site == "Gastric")
fe_ga_ado_dist <- dist_subset(dist.bc, fe_ga_ado$id_modif)
fe_ga_tado <- adonis2(fe_ga_ado_dist ~ body_site, data = fe_ga_ado,permutations = 1000)
pairwise_p["FE_GA"] <- fe_ga_tado$`Pr(>F)`[1]

fe_du_ado <- meta_pcoa %>% filter(body_site == "Feces" | body_site == "Duodenum")
fe_du_ado_dist <- dist_subset(dist.bc, fe_du_ado$id_modif)
fe_du_tado <- adonis2(fe_du_ado_dist ~ body_site, data = fe_du_ado,permutations = 1000)
pairwise_p["FE_DU"] <- fe_du_tado$`Pr(>F)`[1]

ga_sa_ado <- meta_pcoa %>% filter(body_site == "Gastric" | body_site == "Saliva")
ga_sa_ado_dist <- dist_subset(dist.bc, ga_sa_ado$id_modif)
ga_sa_tado <- adonis2(ga_sa_ado_dist ~ body_site, data = ga_sa_ado,permutations = 1000)
pairwise_p["GA_SA"] <- ga_sa_tado$`Pr(>F)`[1]

du_ga_ado <- meta_pcoa %>% filter(body_site == "Duodenum" | body_site == "Saliva")
du_ga_ado_dist <- dist_subset(dist.bc, du_ga_ado$id_modif)
du_ga_tado <- adonis2(du_ga_ado_dist ~ body_site, data = du_ga_ado,permutations = 1000)
pairwise_p["DU_SA"] <- du_ga_tado$`Pr(>F)`[1]

ga_du_ado <- meta_pcoa %>% filter(body_site == "Gastric" | body_site == "Duodenum")
ga_du_ado_dist <- dist_subset(dist.bc, ga_du_ado$id_modif)
ga_du_tado <- adonis2(ga_du_ado_dist ~ body_site, data = ga_du_ado,permutations = 1000)
pairwise_p["GA_DU"] <- ga_du_tado$`Pr(>F)`[1]

pairwise_padj <- p.adjust(pairwise_p, method = "BH")
pairwise_padj
# Feces significant from everything
# Saliva significant from everything
# Gastric and duodenum not different from each other
write.csv(pairwise_padj, "Tables/BetaDiv_Pairwise_adonis2.csv")

# Plot distance intra and inter groups
dist.bc_df <- as.data.frame(as.matrix(dist.bc))
dist.bc_df <- dist.bc_df %>% 
  mutate(Row = rownames(dist.bc_df))
dist.bc_long <- dist.bc_df %>%
  pivot_longer(cols = -Row, names_to = "Column", values_to = "distances")
dist.bc_unique <- dist.bc_long %>%
  filter(Row != Column) %>%
  filter(Row < Column)
dist.bc_unique <- dist.bc_unique %>%
  mutate(Body_site_comp = "")
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "FE") & str_detect(dist.bc_unique$Column, "FE"))] <- "FE_FE"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "FE") & str_detect(dist.bc_unique$Column, "SA"))] <- "FE_SA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "FE") & str_detect(dist.bc_unique$Column, "GA"))] <- "FE_GA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "FE") & str_detect(dist.bc_unique$Column, "DU"))] <- "FE_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "SA") & str_detect(dist.bc_unique$Column, "SA"))] <- "SA_SA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "SA") & str_detect(dist.bc_unique$Column, "GA"))] <- "SA_GA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "SA") & str_detect(dist.bc_unique$Column, "DU"))] <- "SA_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "GA") & str_detect(dist.bc_unique$Column, "GA"))] <- "GA_GA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "GA") & str_detect(dist.bc_unique$Column, "DU"))] <- "GA_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Row, "DU") & str_detect(dist.bc_unique$Column, "DU"))] <- "DU_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "FE") & str_detect(dist.bc_unique$Row, "FE"))] <- "FE_FE"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "FE") & str_detect(dist.bc_unique$Row, "SA"))] <- "FE_SA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "FE") & str_detect(dist.bc_unique$Row, "GA"))] <- "FE_GA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "FE") & str_detect(dist.bc_unique$Row, "DU"))] <- "FE_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "SA") & str_detect(dist.bc_unique$Row, "SA"))] <- "SA_SA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "SA") & str_detect(dist.bc_unique$Row, "GA"))] <- "SA_GA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "SA") & str_detect(dist.bc_unique$Row, "DU"))] <- "SA_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "GA") & str_detect(dist.bc_unique$Row, "GA"))] <- "GA_GA"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "GA") & str_detect(dist.bc_unique$Row, "DU"))] <- "GA_DU"
dist.bc_unique$Body_site_comp[which(str_detect(dist.bc_unique$Column, "DU") & str_detect(dist.bc_unique$Row, "DU"))] <- "DU_DU"

pdf("Figures/BetaDiv_distanceIntraInter.pdf")
dist.bc_unique %>%
  ggplot(aes(x=Body_site_comp, y=distances)) +
  geom_point() +
  geom_boxplot() +
  theme_bw()
dev.off()
#PCoA
test_PCoAbray <- ordinate(ps_beta, "PCoA", distance = dist.bc)
plot_ordi <- plot_ordination(ps_beta, test_PCoAbray, type = "samples", color = "body_site", shape = "Stunting_level") +
  geom_point(size=3) +
  theme_bw() +
  scale_shape_manual(values=c(15:18)) +
  scale_color_brewer(palette = "Set1") +
  ggtitle("PCoA (Bray-Curtis distance) on 16S rRNA sequencing data")
# Clear separation between feces and other body sites
# Also separation from saliva and gastric and duodenum
plot_ordi

ggsave(filename = "Figures/PCoA_16SrRNA_satellite.svg", plot = plot_ordi, width = 10, height = 8)



#Covariates analysis using EnvFit
#sexe, voie_accou, lait_enf, age_allait, eau_traitee, nb_repas, alim_malnut, haz, Stunting_level, Stunting_status, parasite, Gut_inflammation_AAT, Gut_inflammation_Calpro, Systemic_inflammation, ferritine_seuil, body_site

vectorPCOA <- as.data.frame(test_PCoAbray$vectors[,1:2])
# Get metadata
META_envfit = as(sample_data(physeq_ra), "matrix")
meta_envfit <- as.data.frame(META_envfit)
# Covariates:
envfit_sd <- meta_envfit[, c("sexe", "voie_accou", "lait_enf", "age_allaite",
                "eau_traitee", "nb_repas", "alim_malnut", "haz", "Stunting_level",
                "Stunting_status", "parasite", "Gut_inflammation_AAT", "Gut_inflammation_Calpro",
                "Systemic_inflammation", "ferritine_seuil", "body_site")]

env.fit_results <- envfit(vectorPCOA, envfit_sd, perm = 999, na.rm=TRUE)
env.fit_results$factors$pvals
env.fit_results$factors$r
# Variance mostly explained by body site (make sense)
```

Feces only
```{r}
# Bray-Curtis
ps_beta_fe <- subset_samples(ps_beta, body_site == "Feces")
dist.bc_fe <- phyloseq::distance(ps_beta_fe, method = "bray")
test_PCoAbray_fe <- ordinate(ps_beta_fe, "PCoA", distance = dist.bc_fe)

plot_ordination(ps_beta_fe, test_PCoAbray_fe, type = "samples", color = "Stunting_level", shape = "Stunting_status") +
  theme_bw() +
  scale_shape_manual(values=c(0:9)) +
  scale_color_brewer(palette = "Set1") +
  ggtitle("PCoA (Bray-Curtis distance) on fecal samples")

# No clear separation between stunted and non-stunted children in the feces (expected)
# Tests
meta_pcoa_fe = as.matrix(sample_data(ps_beta_fe))
meta_pcoa_fe <- as.data.frame(meta_pcoa_fe)
vegan::adonis2(dist.bc_fe ~ Stunting_level, data = meta_pcoa_fe, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_fe ~ Stunting_status, data = meta_pcoa_fe, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_fe ~ Systemic_inflammation, data = meta_pcoa_fe, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_fe ~ sexe, data = meta_pcoa_fe, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_fe ~ Gut_inflammation_AAT, data = meta_pcoa_fe, permutations = 9999) # 0.0445 *
vegan::adonis2(dist.bc_fe ~ Gut_inflammation_Calpro, data = meta_pcoa_fe, permutations = 9999) # Not significant
#Parasites
meta_pcoa_fe_para <- meta_pcoa_fe
meta_pcoa_fe_para <- meta_pcoa_fe_para[-which(is.na(meta_pcoa_fe_para$Parasites)),]
dist.bc_fe_para <- dist_subset(dist.bc_fe, meta_pcoa_fe_para$id_modif)
vegan::adonis2(dist.bc_fe_para ~ Parasites, data = meta_pcoa_fe_para, permutations = 9999) # Not significant
#Leaky gut
meta_pcoa_fe_leaky <- meta_pcoa_fe
meta_pcoa_fe_leaky <- meta_pcoa_fe_leaky[-which(is.na(meta_pcoa_fe_leaky$Leaky_gut)),]
dist.bc_fe_leaky <- dist_subset(dist.bc_fe, meta_pcoa_fe_leaky$id_modif)
vegan::adonis2(dist.bc_fe_leaky ~ Leaky_gut, data = meta_pcoa_fe_leaky, permutations = 9999) # Not significant

# Using envfit
vectorPCOA_fe <- as.data.frame(test_PCoAbray_fe$vectors[,1:2])
# Get metadata
META_envfit_fe = as(sample_data(ps_beta_fe), "matrix")
meta_envfit_fe <- as.data.frame(META_envfit_fe)
# Covariates:
envfit_sd_fe <- meta_envfit_fe[, c("sexe", "voie_accou", "lait_enf", "age_allaite",
                "eau_traitee", "nb_repas", "alim_malnut", "haz", "Stunting_level",
                "Stunting_status", "parasite", "Gut_inflammation_AAT", "Gut_inflammation_Calpro",
                "Systemic_inflammation", "ferritine_seuil")]

env.fit_results_fe <- envfit(vectorPCOA_fe, envfit_sd_fe, perm = 999, na.rm=TRUE)
env.fit_results_fe$factors$pvals
env.fit_results_fe$factors$r
```

Saliva, gastric and duodenal samples
```{r}
# Bray-Curtis
ps_beta_upper <- subset_samples(ps_beta, body_site != "Feces")
dist.bc_upper <- phyloseq::distance(ps_beta_upper, method = "bray")
test_PCoAbray_upper <- ordinate(ps_beta_upper, "PCoA", distance = dist.bc_upper)

plot_ordination(ps_beta_upper, test_PCoAbray_upper, type = "samples", color = "body_site", shape = "body_site") +
  theme_bw() +
  scale_shape_manual(values=c(0:9)) +
  scale_color_brewer(palette = "Set1") +
  ggtitle("PCoA (Bray-Curtis distance) on saliva, gastric and duodenal samples")

# Clear separation between saliva and other samples
meta_pcoa_upper = as.matrix(sample_data(ps_beta_upper))
meta_pcoa_upper <- as.data.frame(meta_pcoa_upper)
#Significant
vegan::adonis2(dist.bc_upper ~ body_site, data = meta_pcoa_upper, permutations = 9999) # 1e-04 ***

vectorPCOA_upper <- as.data.frame(test_PCoAbray_upper$vectors[,1:2])
# Get metadata
META_envfit_upper = as(sample_data(ps_beta_upper), "matrix")
meta_envfit_upper <- as.data.frame(META_envfit_upper)
# Covariates:
envfit_sd_upper <- meta_envfit_upper[, c("sexe", "voie_accou", "lait_enf", "age_allaite",
                "eau_traitee", "nb_repas", "alim_malnut", "haz", "Stunting_level",
                "Stunting_status", "parasite", "Gut_inflammation_AAT", "Gut_inflammation_Calpro",
                "Systemic_inflammation", "ferritine_seuil", "body_site")]

env.fit_results_upper <- envfit(vectorPCOA_upper, envfit_sd_upper, perm = 999, na.rm=TRUE)
env.fit_results_upper$factors$pvals
env.fit_results_upper$factors$r
# Variance mostly explained by body site (30.6%)
# Also stunting status --> does it hold when separating the body sites?
```
Saliva
```{r}
# Bray-Curtis
ps_beta_sa <- subset_samples(ps_beta, body_site == "Saliva")
dist.bc_sa <- phyloseq::distance(ps_beta_sa, method = "bray")
test_PCoAbray_sa <- ordinate(ps_beta_sa, "PCoA", distance = dist.bc_sa)

plot_ordination(ps_beta_sa, test_PCoAbray_sa, type = "samples", color = "Stunting_level", shape = "Stunting_level") +
  theme_bw() +
  scale_shape_manual(values=c(0:9)) +
  scale_color_brewer(palette = "Set1") +
  ggtitle("PCoA (Bray-Curtis distance) on saliva samples")
# No clear separation between stunted and non-stunted children in the saliva samples

meta_pcoa_sa = as.matrix(sample_data(ps_beta_sa))
meta_pcoa_sa <- as.data.frame(meta_pcoa_sa)
vegan::adonis2(dist.bc_sa ~ Systemic_inflammation, data = meta_pcoa_sa, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_sa ~ Stunting_level, data = meta_pcoa_sa, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_sa ~ Stunting_status, data = meta_pcoa_sa, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_sa ~ sexe, data = meta_pcoa_sa, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_sa ~ Gut_inflammation_AAT, data = meta_pcoa_sa, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_sa ~ Gut_inflammation_Calpro, data = meta_pcoa_sa, permutations = 9999) # Not significant
#Parasites
meta_pcoa_sa_para <- meta_pcoa_sa
meta_pcoa_sa_para <- meta_pcoa_sa_para[-which(is.na(meta_pcoa_sa_para$Parasites)),]
dist.bc_sa_para <- dist_subset(dist.bc_sa, meta_pcoa_sa_para$id_modif)
vegan::adonis2(dist.bc_sa_para ~ Parasites, data = meta_pcoa_sa_para, permutations = 9999) # Not significant
# Leaky gut
meta_pcoa_sa_leaky <- meta_pcoa_sa
meta_pcoa_sa_leaky <- meta_pcoa_sa_leaky[-which(is.na(meta_pcoa_sa_leaky$Leaky_gut)),]
dist.bc_sa_leaky <- dist_subset(dist.bc_sa, meta_pcoa_sa_leaky$id_modif)
vegan::adonis2(dist.bc_sa_leaky ~ Leaky_gut, data = meta_pcoa_sa_leaky, permutations = 9999) # Not significant

# Using EnvFit
vectorPCOA_sa <- as.data.frame(test_PCoAbray_sa$vectors[,1:2])
# Get metadata
META_envfit_sa = as(sample_data(ps_beta_sa), "matrix")
meta_envfit_sa <- as.data.frame(META_envfit_sa)
# Covariates:
envfit_sd_sa <- meta_envfit_sa[, c("sexe", "voie_accou", "lait_enf", "age_allaite",
                "eau_traitee", "nb_repas", "alim_malnut", "haz", "Stunting_level",
                "Stunting_status", "parasite", "Gut_inflammation_AAT", "Gut_inflammation_Calpro",
                "Systemic_inflammation", "ferritine_seuil")]

env.fit_results_sa <- envfit(vectorPCOA_sa, envfit_sd_sa, perm = 999, na.rm=TRUE)
env.fit_results_sa$factors$pvals
env.fit_results_sa$factors$r
# Nothing
```
Gastric and duodenal samples
```{r}
# Bray-Curtis
ps_beta_gd <- subset_samples(ps_beta, body_site == "Duodenum" | body_site == "Gastric")
dist.bc_gd <- phyloseq::distance(ps_beta_gd, method = "bray")
test_PCoAbray_gd <- ordinate(ps_beta_gd, "PCoA", distance = dist.bc_gd)

plot_ordination(ps_beta_gd, test_PCoAbray_gd, type = "samples", color = "Stunting_level", shape = "Stunting_level") +
  theme_bw() +
  scale_shape_manual(values=c(0:9)) +
  scale_color_brewer(palette = "Set1") +
  ggtitle("PCoA (Bray-Curtis distance) on saliva, gastric and duodenal samples")

meta_pcoa_gd = as.matrix(sample_data(ps_beta_gd))
meta_pcoa_gd <- as.data.frame(meta_pcoa_gd)

#Parasites
meta_pcoa_gd_para <- meta_pcoa_gd
dist.bc_gd_para <- dist_subset(dist.bc_gd, meta_pcoa_gd_para$id_modif)
vegan::adonis2(dist.bc_gd_para ~ Parasites, data = meta_pcoa_gd_para, permutations = 9999) # Not significant
# Not significant
vegan::adonis2(dist.bc_gd ~ body_site, data = meta_pcoa_gd, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_gd ~ Stunting_level, data = meta_pcoa_gd, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_gd ~ Stunting_status, data = meta_pcoa_gd, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_gd ~ sexe, data = meta_pcoa_gd, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_gd ~ Systemic_inflammation, data = meta_pcoa_gd, permutations = 9999) # 0.026 *
vegan::adonis2(dist.bc_gd ~ Gut_inflammation_AAT, data = meta_pcoa_gd, permutations = 9999) # Not significant
vegan::adonis2(dist.bc_gd ~ Gut_inflammation_Calpro, data = meta_pcoa_gd, permutations = 9999) # 0.0419 *
#Leaky gut
meta_pcoa_gd_leaky <- meta_pcoa_gd
meta_pcoa_gd_leaky <- meta_pcoa_gd_leaky[-which(is.na(meta_pcoa_gd_leaky$Leaky_gut)),]
dist.bc_gd_leaky <- dist_subset(dist.bc_gd, meta_pcoa_gd_leaky$id_modif)
vegan::adonis2(dist.bc_gd_leaky ~ Leaky_gut, data = meta_pcoa_gd_leaky, permutations = 9999) # Not significant
# Do correction BH if included in manuscript


# USing EnvFit
vectorPCOA_gd <- as.data.frame(test_PCoAbray_gd$vectors[,1:2])
# Get metadata
META_envfit_gd = as(sample_data(ps_beta_gd), "matrix")
meta_envfit_gd <- as.data.frame(META_envfit_gd)
# Covariates:
envfit_sd_gd <- meta_envfit_gd[, c("sexe", "voie_accou", "lait_enf", "age_allaite",
                "eau_traitee", "nb_repas", "alim_malnut", "haz", "Stunting_level",
                "Stunting_status", "parasite", "Gut_inflammation_AAT", "Gut_inflammation_Calpro",
                "Systemic_inflammation", "ferritine_seuil")]

env.fit_results_gd <- envfit(vectorPCOA_gd, envfit_sd_gd, perm = 999, na.rm=TRUE)
env.fit_results_gd$factors$pvals
env.fit_results_gd$factors$r
# age_allait?
```

# Composition
```{r}
ps_composition <- physeq_ra

ps_ra_feces <- subset_samples(ps_composition, body_site == "Feces")
ps_ra_saliva <- subset_samples(ps_composition, body_site == "Saliva")
ps_ra_gadu <- subset_samples(ps_composition, body_site == "Duodenum" | body_site == "Gastric")
```

Add differential abundance analysis with Deseq2 (try Lefse as well)

## Oral microbiota
```{r}
ps_ra_saliva <- prune_taxa(taxa_sums(ps_ra_saliva)> 0, ps_ra_saliva)
ps_ra_saliva_gen <- tax_glom(ps_ra_saliva, "Genus") # 169 genera

TopGen_sal <- names(sort(taxa_sums(ps_ra_saliva_gen), TRUE)[1:20])
SalGen20 <- prune_taxa(TopGen_sal, ps_ra_saliva_gen)
topotuG_sal <- otu_table(SalGen20)
toptaxaG_sal <- tax_table(SalGen20)
SalGen20 = phyloseq(topotuG_sal, toptaxaG_sal)
SalGen20 = merge_phyloseq(sample_data(ps_ra_saliva_gen), SalGen20)
# Grouping other Genilies
OtherG_sal <- names(sort(taxa_sums(ps_ra_saliva_gen), TRUE)[c(21:169)])
SalG_other <- prune_taxa(OtherG_sal, ps_ra_saliva_gen)
others_otu_sal <- t(as.matrix(apply(otu_table(SalG_other), 2, sum)))
rownames(others_otu_sal) <- "Others Genera"
others_otu_sal=otu_table(others_otu_sal, taxa_are_rows = TRUE)
other_taxG_sal <- c("Others", "Others","Others","Others","Others","Others","Others")
other_taxG_sal <- as.matrix(other_taxG_sal)
rownames(other_taxG_sal) <- colnames(tax_table(SalG_other))
other_taxG_sal <- t(other_taxG_sal)
rownames(other_taxG_sal) <- "Others Genera"
other_taxG_sal =  tax_table(other_taxG_sal)
others_salG= phyloseq(others_otu_sal,other_taxG_sal)
others_salG = merge_phyloseq(sample_data(ps_ra_saliva_gen), others_salG)
others_salG
# Regrouping phyloseq
salG <- merge_phyloseq(SalGen20, others_salG)
salG_melt <- psmelt(salG)


boxplot_oral <- ggplot(salG_melt, aes(x = Genus, y = Abundance)) +
  geom_boxplot(aes(fill = Genus), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Genus), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9, face = 3)) +
  ggtitle("Oral microbiota composition - 16S rRNA data - Top 20 Genera")

ggsave(filename = "Figures/Boxplot_oral_16SrRNA_satellite.svg", plot = boxplot_oral, width = 10, height = 8)

barplot_oral <- salG_melt %>%
  ggplot(aes_string(x = "id_modif", y = "Abundance", fill = "Genus")) +
  geom_bar(stat = "identity", position = "stack") + 
  #facet_grid(~Stunting_level, scales = "free_x") +
  scale_fill_manual(values = pal_1) +
  theme_bw() + 
  theme(legend.position = "right", 
        axis.title.x=element_blank(),
        axis.text.x=element_text(angle = 60, hjust = 1),
        axis.ticks.x=element_blank()) +
  guides(fill=guide_legend(ncol=1))+
  ylab("Relative abundance")+
  ggtitle("Oral microbiota composition - 16S rRNA data - Top 20 Genera") +
  theme(legend.text = element_text(face = 3))

ggsave(filename = "Figures/Barplot_oral_16SrRNA_satellite.svg", plot = barplot_oral, width = 10, height = 8)
```

## Gastric-duodenal microbiota
```{r}
ps_ra_gadu <- prune_taxa(taxa_sums(ps_ra_gadu)> 0, ps_ra_gadu)
ps_ra_gadu_gen <- tax_glom(ps_ra_gadu, "Genus") # 275 genera

TopGen_gadu <- names(sort(taxa_sums(ps_ra_gadu_gen), TRUE)[1:20])
gaduGen20 <- prune_taxa(TopGen_gadu, ps_ra_gadu_gen)
topotuG_gadu <- otu_table(gaduGen20)
toptaxaG_gadu <- tax_table(gaduGen20)
gaduGen20 = phyloseq(topotuG_gadu, toptaxaG_gadu)
gaduGen20 = merge_phyloseq(sample_data(ps_ra_gadu_gen), gaduGen20)
# Grouping other Genilies
OtherG_gadu <- names(sort(taxa_sums(ps_ra_gadu_gen), TRUE)[c(21:275)])
gaduG_other <- prune_taxa(OtherG_gadu, ps_ra_gadu_gen)
others_otu_gadu <- t(as.matrix(apply(otu_table(gaduG_other), 2, sum)))
rownames(others_otu_gadu) <- "Others Genera"
others_otu_gadu=otu_table(others_otu_gadu, taxa_are_rows = TRUE)
other_taxG_gadu <- c("Others", "Others","Others","Others","Others","Others","Others")
other_taxG_gadu <- as.matrix(other_taxG_gadu)
rownames(other_taxG_gadu) <- colnames(tax_table(gaduG_other))
other_taxG_gadu <- t(other_taxG_gadu)
rownames(other_taxG_gadu) <- "Others Genera"
other_taxG_gadu =  tax_table(other_taxG_gadu)
others_gaduG= phyloseq(others_otu_gadu,other_taxG_gadu)
others_gaduG = merge_phyloseq(sample_data(ps_ra_gadu_gen), others_gaduG)
others_gaduG
# Regrouping phyloseq
gaduG <- merge_phyloseq(gaduGen20, others_gaduG)
gaduG_melt <- psmelt(gaduG)


boxplot_gadu <- ggplot(gaduG_melt, aes(x = Genus, y = Abundance)) +
  geom_boxplot(aes(fill = Genus), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Genus), size =1) +
  facet_grid(~body_site, scales = "free_x") +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9, face = 3))+
  ggtitle("Gastric and duodenal microbiota composition - 16S rRNA data - Top 20 Genus")

ggsave(filename = "Figures/Boxplot_gadu_16SrRNA_satellite.svg", plot = boxplot_gadu, width = 10, height = 8)

barplot_gadu <- gaduG_melt %>%
  ggplot(aes_string(x = "id_modif", y = "Abundance", fill = "Genus")) +
  geom_bar(stat = "identity", position = "stack") + 
  #facet_grid(~body_site, scales = "free_x") +
  scale_fill_manual(values = pal_1) +
  theme_bw() + 
  theme(legend.position = "right", 
        axis.title.x=element_blank(),
        axis.text.x=element_text(angle = 60, hjust = 1),
        axis.ticks.x=element_blank()) +
  guides(fill=guide_legend(ncol=1))+
  ylab("Relative abundance")+
  ggtitle("Gastric and duodenal microbiota composition - 16S rRNA data- Top 20 Genera") +
  theme(legend.text = element_text(face = 3))

ggsave(filename = "Figures/Barplot_gadu_16SrRNA_satellite.svg", plot = barplot_gadu, width = 10, height = 8)
```

## Feces
```{r}
ps_ra_feces <- prune_taxa(taxa_sums(ps_ra_feces)> 0, ps_ra_feces)
ps_ra_feces_gen <- tax_glom(ps_ra_feces, "Genus") # 217 genera

TopGen_fec <- names(sort(taxa_sums(ps_ra_feces_gen), TRUE)[1:20])
fecGen20 <- prune_taxa(TopGen_fec, ps_ra_feces_gen)
topotuG_fec <- otu_table(fecGen20)
toptaxaG_fec <- tax_table(fecGen20)
fecGen20 = phyloseq(topotuG_fec, toptaxaG_fec)
fecGen20 = merge_phyloseq(sample_data(ps_ra_feces_gen), fecGen20)
# Grouping other Genilies
OtherG_fec <- names(sort(taxa_sums(ps_ra_feces_gen), TRUE)[c(21:217)])
fecG_other <- prune_taxa(OtherG_fec, ps_ra_feces_gen)
others_otu_fec <- t(as.matrix(apply(otu_table(fecG_other), 2, sum)))
rownames(others_otu_fec) <- "Others Genera"
others_otu_fec=otu_table(others_otu_fec, taxa_are_rows = TRUE)
other_taxG_fec <- c("Others", "Others","Others","Others","Others","Others","Others")
other_taxG_fec <- as.matrix(other_taxG_fec)
rownames(other_taxG_fec) <- colnames(tax_table(fecG_other))
other_taxG_fec <- t(other_taxG_fec)
rownames(other_taxG_fec) <- "Others Genera"
other_taxG_fec =  tax_table(other_taxG_fec)
others_fecG= phyloseq(others_otu_fec,other_taxG_fec)
others_fecG = merge_phyloseq(sample_data(ps_ra_feces_gen), others_fecG)
others_fecG
# Regrouping phyloseq
fecG <- merge_phyloseq(fecGen20, others_fecG)
fecG_melt <- psmelt(fecG)


boxplot_feces <- ggplot(fecG_melt, aes(x = Genus, y = Abundance)) +
  geom_boxplot(aes(fill = Genus), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Genus), size =1) +
  scale_color_manual(values = pal_1) +
  scale_fill_manual(values = pal_1) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9, face = 3))+
  ggtitle("Fecal microbiota composition - 16S rRNA data")

ggsave(filename = "Figures/Boxplot_feces_16SrRNA_satellite.svg", plot = boxplot_feces, width = 10, height = 8)

barplot_feces <- fecG_melt %>%
  ggplot(aes_string(x = "id_modif", y = "Abundance", fill = "Genus")) +
  geom_bar(stat = "identity", position = "stack") + 
  #facet_grid(~Stunting_level, scales = "free_x") +
  scale_fill_manual(values = pal_1) +
  theme_bw() + 
  theme(legend.position = "right", 
        axis.title.x=element_blank(),
        axis.text.x=element_text(angle = 60, hjust = 1),
        axis.ticks.x=element_blank()) +
  guides(fill=guide_legend(ncol=1))+
  ylab("Relative abundance")+
  ggtitle("Fecal microbiota composition - 16S rRNA data- Top 20 Genera") +
  theme(legend.text = element_text(face = 3))

ggsave(filename = "Figures/Barplot_feces_16SrRNA_satellite.svg", plot = barplot_feces, width = 10, height = 8)
```

# Intersect
## venn diagramm
Overview of taxa shared between compartments
Venn diagram: taxa present at least one in the compartment, does not take into account the prevalence in the compartment
```{r}
ps_inter <- physeq_ra

physeq_fam <- tax_glom(ps_inter, taxrank = "Family")
# 161 families in the dataset
physeq_fam_m <- psmelt(physeq_fam)

ps_venn(physeq_fam,
        group = "body_site",
        plot = TRUE)

venn_list <- ps_venn(physeq_fam,
        group = "body_site",
        plot = FALSE)
# Only shared Families between all body sites
physeq_fam_shared <- subset_taxa(physeq_fam, rownames(tax_table(physeq_fam)) %in% venn_list$Duodenum__Feces__Gastric__Saliva)

physeq_fam_shared_melt <- psmelt(physeq_fam_shared)
physeq_fam_shared_melt$body_site = factor(physeq_fam_shared_melt$body_site, levels = c("Saliva", "Gastric", "Duodenum", "Feces"))

ggplot(physeq_fam_shared_melt, aes(x = Family, y = Abundance)) +
  geom_boxplot(aes(fill = Family), alpha = .3) +
  geom_point(colour = "black", size = 1.5) +
  geom_point(aes(color = Family), size =1) +
  scale_color_manual(values = pal_4) +
  scale_fill_manual(values = pal_4) +
  theme_bw() +
  theme(legend.position = "none", axis.text.x =  element_text(angle = 45, hjust = .9)) +
  facet_grid(body_site~., scales = "free")
```
Genus
```{r}
physeq_gen <- tax_glom(ps_inter, taxrank = "Genus")
# 385 Genera (with abundance >0%) in the dataset

# All samples, all body sites
physeq_gen %>%
  ps_venn(group = "body_site",
        plot = TRUE)

#### Samples with all 4 body sites in 16S rRNA amplicon sequencing
physeq_gen_4bs <- subset_samples(physeq_gen, id %in% intersect(intersect(md_16S_final$id[which(md_16S_final$body_site =="Gastric")],
          md_16S_final$id[which(md_16S_final$body_site =="Saliva")]),
          intersect(md_16S_final$id[which(md_16S_final$body_site =="Duodenum")],
          md_16S_final$id[which(md_16S_final$body_site =="Feces")])))# 10 samples with all body sites)
physeq_gen_4bs <- prune_taxa(taxa_sums(physeq_gen_4bs)> 0, physeq_gen_4bs)
# 212 taxa 
ps_venn(physeq_gen_4bs, group = "body_site",
        plot = TRUE)

#### Oral and gastric
# All samples
intersect(md_16S$id[which(md_16S$body_site =="Gastric")], md_16S$id[which(md_16S$body_site =="Saliva")])

physeq_gen %>%
  subset_samples(body_site == "Saliva" | body_site == "Gastric") %>%
  ps_venn(
        group = "body_site",
        plot = TRUE)



# Gastric and Duodenal
physeq_gen %>%
  subset_samples(body_site == "Duodenum" | body_site == "Gastric") %>%
  ps_venn(
        group = "body_site",
        plot = TRUE)

# Gastric and fecal
physeq_gen %>%
  subset_samples(body_site == "Feces" | body_site == "Gastric") %>%
  ps_venn(
        group = "body_site",
        plot = TRUE)

# Gastric, duodenum and fecal
physeq_gen %>%
  subset_samples(body_site == "Feces" | body_site == "Gastric" | body_site == "Duodenum") %>%
  ps_venn(
        group = "body_site",
        plot = TRUE)
```


Classifying species in 6 categories
1) Uniquely oral (never in feces, 0% on feces)
2) predominantly in oral samples (= oral origin) (more than 10% of the oral samples and less than 10% (>0%) of the fecal samples)
3) Low prevalence taxa (<10% for both fecal and oral)
4) Mostly shared (prevalence between 10 and 60% for both sample type)
5) Highly shared (prevalence between 60 and 100 for both sample type)
4) completely shared (100% of samples both fecal and oral)
5) predominantly fecal (=fecal origin) (more than 10% of the fecal samples and less than 10% of the oral samples)
6) Uniquely fecal (never in oral, 0% in oral)

For children with paired saliva and fecal samples
First remove children for which we do not have paired saliva and fecal sample


Should add a coverage filter (Schmidt et al.)
average vertical coverage (depth) >= 0.25
Average vertical coverage (sequencing depth) and horizontal coverage (breadth) per microbial genome in each sample were quantified using the qaCompute utility in metaSNV

OR a filter to select only the species for which we have a MAG
(might have to change the metaphlan table to GTDB taxonomy and merge them)
```{r}
physeq_gen_sf <- subset_samples(physeq_gen, body_site == "Saliva" | body_site == "Feces")
# Remove samples with unpaired saliva and feces
physeq_gen_sf <- subset_samples(physeq_gen_sf, id != "1429AGN007" & id != "1429AGN012" & id != "1429AGN020")

# If consider only feces and saliva do: 
physeq_gen_sf <- subset_taxa(physeq_gen_sf, taxa_sums(physeq_gen_sf)/82 > 10^-6)

prevdf_gen_sf = apply(X= otu_table(physeq_gen_sf),
               MARGIN = ifelse(taxa_are_rows(physeq_gen_sf),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_gen_sf = data.frame(Prevalence = prevdf_gen_sf,
                        Percentage_Prev = 100*prevdf_gen_sf/82,
                        tax_table(physeq_gen_sf))

n_distinct(rownames(prevdf_gen_sf)[which(prevdf_gen_sf$Percentage_Prev >= 10)]) # 142 ASV have above >= 10 prevalence
physeq_gen_sf = subset_taxa(physeq_gen_sf,  rownames(tax_table(physeq_gen_sf)) %in% rownames(prevdf_gen_sf)[which(prevdf_gen_sf$Percentage_Prev >= 10)])
# 142 genera have prevalence >= 10% in fecal and saliva sample AND average relative abundance > 10^-6
# For now, only prevalence and abundance filter and only saliva and feces

# Prevalence and abundance in feces for all species
physeq_gen_fe <- subset_samples(physeq_gen_sf, body_site == "Feces")
prevdf_gen_fe = apply(X= otu_table(physeq_gen_fe),
               MARGIN = ifelse(taxa_are_rows(physeq_gen_fe),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_gen_fe = data.frame(Prevalence_feces = prevdf_gen_fe,
                       Percentage_Prev_feces = 100*prevdf_gen_fe/41,
                       Mean_Abundance_feces = 100*taxa_sums(physeq_gen_fe)/41,
                       Total_Abundance_feces = taxa_sums(physeq_gen_fe),
                        tax_table(physeq_gen_fe))
prevdf_gen_fe <- prevdf_gen_fe[, -11]

# Prevalence and abundance in saliva for all species
physeq_gen_sa <- subset_samples(physeq_gen_sf, body_site == "Saliva")
prevdf_gen_sa = apply(X= otu_table(physeq_gen_sa),
               MARGIN = ifelse(taxa_are_rows(physeq_gen_sa),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_gen_sa = data.frame(Prevalence_saliva = prevdf_gen_sa,
                       Percentage_Prev_saliva = 100*prevdf_gen_sa/41,
                       Mean_Abundance_saliva = 100*taxa_sums(physeq_gen_sa)/41,
                       Total_Abundance_saliva = taxa_sums(physeq_gen_sa),
                        tax_table(physeq_gen_sa))
prevdf_gen_sa <- prevdf_gen_sa[, -11]


# Merging dataset
prevdf_gen <- cbind(prevdf_gen_sa[,c(1:4)], prevdf_gen_fe)
# Adding categories:
prevdf_gen <- prevdf_gen %>%
  mutate(Intersect_category = "")
prevdf_gen$Intersect_category[which(prevdf_gen$Percentage_Prev_saliva > 0 
                                    & prevdf_gen$Percentage_Prev_feces == 0)] <- "Uniquely oral"
prevdf_gen$Intersect_category[which(prevdf_gen$Percentage_Prev_saliva == 0 
                                    & prevdf_gen$Percentage_Prev_feces > 0)] <- "Uniquely fecal"

prevdf_gen$Intersect_category[which(prevdf_gen$Percentage_Prev_saliva < 10 & prevdf_gen$Percentage_Prev_feces < 10)] <- "Low prevalence"
prevdf_gen$Intersect_category[which(prevdf_gen$Percentage_Prev_saliva == 100 & prevdf_gen$Percentage_Prev_feces == 100)] <- "Completely shared"


prevdf_gen$Intersect_category[which(prevdf_gen$Percentage_Prev_saliva > 10 
                                    & prevdf_gen$Percentage_Prev_feces < 10 & prevdf_gen$Percentage_Prev_feces > 0)] <- "Oral origin"

prevdf_gen$Intersect_category[which(prevdf_gen$Percentage_Prev_saliva < 10 & prevdf_gen$Percentage_Prev_saliva > 0 
                                    & prevdf_gen$Percentage_Prev_feces > 10)] <- "Fecal origin"

prevdf_gen$Intersect_category[which(prevdf_gen$Percentage_Prev_saliva >= 10 & prevdf_gen$Percentage_Prev_saliva < 100
                                    & prevdf_gen$Percentage_Prev_feces >= 10 )] <- "Mostly shared"
prevdf_gen$Intersect_category[which(prevdf_gen$Percentage_Prev_saliva >= 10
                                    & prevdf_gen$Percentage_Prev_feces >= 10 & prevdf_gen$Percentage_Prev_feces < 100)] <- "Mostly shared"
prevdf_gen$Intersect_category[which(prevdf_gen$Percentage_Prev_saliva >= 60 & prevdf_gen$Percentage_Prev_saliva < 100
                                    & prevdf_gen$Percentage_Prev_feces >= 60)] <- "Highly shared"
prevdf_gen$Intersect_category[which(prevdf_gen$Percentage_Prev_saliva >= 60 
                                    & prevdf_gen$Percentage_Prev_feces >= 60 & prevdf_gen$Percentage_Prev_feces < 100)] <- "Highly shared"

prevdf_gen$Intersect_category <- as.factor(prevdf_gen$Intersect_category)
prevdf_gen$Intersect_category <- factor(prevdf_gen$Intersect_category, levels=c("Uniquely oral", "Oral origin", "Low prevalence",
                                                                                "Mostly shared", "Highly shared", "Completely shared",
                                                                                "Fecal origin", "Uniquely fecal"), ordered = TRUE)

prevdf_gen_ordered <- prevdf_gen[order(prevdf_gen$Intersect_category, -prevdf_gen$Percentage_Prev_saliva, prevdf_gen$Percentage_Prev_feces),]
# Fix unassigned names to higher taxonomic ranks
prevdf_gen_ordered <- prevdf_gen_ordered %>%
  mutate(Name = case_when(
    str_detect(Genus, "Unassigned", negate = TRUE) ~ Genus,
    str_detect(Family, "Unassigned", negate = TRUE) ~ paste("Unassigned ", Family),
    str_detect(Order, "Unassigned", negate = TRUE) ~ paste("Unassigned ", Order),
    str_detect(Class, "Unassigned", negate = TRUE) ~ paste("Unassigned ", Class),
    str_detect(Phylum, "Unassigned", negate = TRUE) ~ paste("Unassigned ", Phylum),
    TRUE ~ "Unassigned"
  ))

```

```{r}
prevdf_gen_hm <- prevdf_gen_ordered
rownames(prevdf_gen_hm) <- prevdf_gen_hm$Name
hm_tab <- t(prevdf_gen_hm[,c(3, 7)])

rownames(hm_tab)[1] <- "Saliva"
rownames(hm_tab)[2] <- "Feces"

col_fun = colorRamp2(c(0,0,0.1,1,6,30), c("white","white", "#B8DE29FF", "#29AF7FFF","#33638DFF", "#440154FF"))

row_annot <- rowAnnotation(ls = prevdf_gen_hm$Intersect_category,
                                  col = list(ls=c("Uniquely oral" = "khaki2",
                                                  "Oral origin" = "orange2",
                                                  "Mostly shared" = "pink2",
                                                  "Highly shared" = "indianred3",
                                                  "Completely shared" = "#440154FF",
                                                  "Fecal origin" = "darkcyan",
                                                  "Uniquely fecal" = "lightblue2"
                                                  )),
                                  annotation_label = "Sharing", annotation_name_side = "top", annotation_name_rot = 45)

# How many per categories
n_distinct(prevdf_gen_hm$Name[which(prevdf_gen_hm$Intersect_category == "Uniquely fecal")])

split_hm = rep("Uniquely oral", 142)
split_hm[25:35] = "Oral origin"
split_hm[36:56] = "Mostly shared"
split_hm[57:67] = "Highly shared"
split_hm[68] = "Completely shared"
split_hm[69:104] = "Fecal origin"
split_hm[105:142] = "Uniquely fecal"
split_hm <- as.factor(split_hm)
split_hm <- factor(split_hm, levels=c("Uniquely oral", "Oral origin",
                                     "Mostly shared", "Highly shared", "Completely shared",
                                     "Fecal origin", "Uniquely fecal"), ordered = TRUE)



heatmap_sharing_t <- Heatmap(t(hm_tab), name = "Mean relative abundance",row_split = split_hm,
                             row_order =  rownames(prevdf_gen_hm),
        col = col_fun, right_annotation = row_annot, show_column_dend = FALSE, row_names_side = "left", column_names_side = "top",
        column_names_rot = 45, border = TRUE, row_title = NULL, row_names_gp = gpar(fontsize = 8, fontface = 3))    

pdf("Figures/Heatmap_sharing_t_16S.pdf", height = 16, width = 6)
heatmap_sharing_t
dev.off()

```

```{r}
ps_hm_venn <- subset_taxa(physeq_gen_sf, Genus %in% rownames(prevdf_gen_hm))
# Venn diagram 
pdf("Figures/Venndiagramm_FecesOral.pdf", height = 6, width = 6)
ps_venn(ps_hm_venn,
        group = "body_site",
        plot = TRUE, 
        fill = c("lightblue2", "khaki2", "white"),
        label=list(cex=2),
        quantities = list(cex=4)
        )
dev.off()
```


Same but oral, gastric and fecal samples
```{r}
physeq_gen_sgf <- subset_samples(physeq_gen, body_site == "Saliva" | body_site == "Feces" | body_site == "Gastric")
# Remove samples with unpaired saliva and feces
physeq_gen_sgf <- subset_samples(physeq_gen_sgf, id %in% intersect(intersect(md_16S_final$id[which(md_16S_final$body_site =="Gastric")],
          md_16S_final$id[which(md_16S_final$body_site =="Saliva")]),
          md_16S_final$id[which(md_16S_final$body_site =="Feces")]))

physeq_gen_sgf <- subset_taxa(physeq_gen_sgf, taxa_sums(physeq_gen_sgf)/78 > 10^-6)

prevdf_gen_sgf = apply(X= otu_table(physeq_gen_sgf),
               MARGIN = ifelse(taxa_are_rows(physeq_gen_sgf),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_gen_sgf = data.frame(Prevalence = prevdf_gen_sgf,
                        Percentage_Prev = 100*prevdf_gen_sgf/78,
                        tax_table(physeq_gen_sgf))

physeq_gen_sgf = subset_taxa(physeq_gen_sgf,  rownames(tax_table(physeq_gen_sgf)) %in% rownames(prevdf_gen_sgf)[which(prevdf_gen_sgf$Percentage_Prev >= 10)])
# 148 genera have prevalence > 10% in gastric, fecal and saliva sample AND average relative abundance > 10^-6

# Prevalence and abundance in feces for all species
physeq_gen_sgf_fe <- subset_samples(physeq_gen_sgf, body_site == "Feces")
prevdf_gen_sgf_fe = apply(X= otu_table(physeq_gen_sgf_fe),
               MARGIN = ifelse(taxa_are_rows(physeq_gen_sgf_fe),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_gen_sgf_fe = data.frame(Prevalence_feces = prevdf_gen_sgf_fe,
                       Percentage_Prev_feces = 100*prevdf_gen_sgf_fe/26,
                       Mean_Abundance_feces = 100*taxa_sums(physeq_gen_sgf_fe)/26,
                       Total_Abundance_feces = taxa_sums(physeq_gen_sgf_fe),
                        tax_table(physeq_gen_sgf_fe))
prevdf_gen_sgf_fe <- prevdf_gen_sgf_fe[, -11]

# Prevalence and abundance in saliva for all species
physeq_gen_sgf_sa <- subset_samples(physeq_gen_sgf, body_site == "Saliva")
prevdf_gen_sgf_sa = apply(X= otu_table(physeq_gen_sgf_sa),
               MARGIN = ifelse(taxa_are_rows(physeq_gen_sgf_sa),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_gen_sgf_sa = data.frame(Prevalence_saliva = prevdf_gen_sgf_sa,
                       Percentage_Prev_saliva = 100*prevdf_gen_sgf_sa/26,
                       Mean_Abundance_saliva = 100*taxa_sums(physeq_gen_sgf_sa)/26,
                       Total_Abundance_saliva = taxa_sums(physeq_gen_sgf_sa),
                        tax_table(physeq_gen_sgf_sa))
prevdf_gen_sgf_sa <- prevdf_gen_sgf_sa[, -11]

# Prevalence and abundance in gastric for all species
physeq_gen_sgf_ga <- subset_samples(physeq_gen_sgf, body_site == "Gastric")
prevdf_gen_sgf_ga = apply(X= otu_table(physeq_gen_sgf_ga),
               MARGIN = ifelse(taxa_are_rows(physeq_gen_sgf_ga),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_gen_sgf_ga = data.frame(Prevalence_ga = prevdf_gen_sgf_ga,
                       Percentage_Prev_ga = 100*prevdf_gen_sgf_ga/26,
                       Mean_Abundance_ga = 100*taxa_sums(physeq_gen_sgf_ga)/26,
                       Total_Abundance_ga = taxa_sums(physeq_gen_sgf_ga),
                        tax_table(physeq_gen_sgf_ga))
prevdf_gen_sgf_ga <- prevdf_gen_sgf_ga[, -11]


# Merging dataset
prevdf_gen_sgf <- cbind(prevdf_gen_sgf_sa[,c(1:4)], prevdf_gen_sgf_fe[,c(1:4)], prevdf_gen_sgf_ga)

# Adding categories:
prevdf_gen_sgf <- prevdf_gen_sgf %>%
  mutate(Intersect_category = "")
prevdf_gen_sgf$Intersect_category[which(prevdf_gen_sgf$Percentage_Prev_saliva > 10 
                                    & prevdf_gen_sgf$Percentage_Prev_feces < 10
                                    & prevdf_gen_sgf$Percentage_Prev_ga < 10)] <- "Oral origin"

prevdf_gen_sgf$Intersect_category[which(prevdf_gen_sgf$Percentage_Prev_saliva < 10
                                    & prevdf_gen_sgf$Percentage_Prev_feces > 10 
                                    & prevdf_gen_sgf$Percentage_Prev_ga < 10)] <- "Fecal origin"

prevdf_gen_sgf$Intersect_category[which(prevdf_gen_sgf$Percentage_Prev_saliva < 10
                                    & prevdf_gen_sgf$Percentage_Prev_ga > 10 
                                    & prevdf_gen_sgf$Percentage_Prev_feces < 10 )] <- "Gastric origin"
#Unique to a body site
prevdf_gen_sgf$Intersect_category[which(prevdf_gen_sgf$Percentage_Prev_saliva > 0 
                                    & prevdf_gen_sgf$Percentage_Prev_feces == 0
                                     & prevdf_gen_sgf$Percentage_Prev_ga == 0)] <- "Uniquely oral"
prevdf_gen_sgf$Intersect_category[which(prevdf_gen_sgf$Percentage_Prev_saliva == 0 
                                    & prevdf_gen_sgf$Percentage_Prev_feces > 0
                                     & prevdf_gen_sgf$Percentage_Prev_ga == 0)] <- "Uniquely fecal"
prevdf_gen_sgf$Intersect_category[which(prevdf_gen_sgf$Percentage_Prev_ga > 0 
                                    & prevdf_gen_sgf$Percentage_Prev_feces == 0
                                     & prevdf_gen_sgf$Percentage_Prev_saliva == 0)] <- "Uniquely gastric"


prevdf_gen_sgf$Intersect_category[which(prevdf_gen_sgf$Percentage_Prev_saliva < 10 
                                   & prevdf_gen_sgf$Percentage_Prev_feces < 10 &
                                     prevdf_gen_sgf$Percentage_Prev_ga < 10)] <- "Low prevalence"

# Shared by pairs
prevdf_gen_sgf$Intersect_category[which(prevdf_gen_sgf$Percentage_Prev_saliva > 10 
                                    & prevdf_gen_sgf$Percentage_Prev_feces < 10
                                    & prevdf_gen_sgf$Percentage_Prev_ga > 10)] <- "Shared in upper tract"
prevdf_gen_sgf$Intersect_category[which(prevdf_gen_sgf$Percentage_Prev_saliva < 10
                                    & prevdf_gen_sgf$Percentage_Prev_feces > 10 
                                    & prevdf_gen_sgf$Percentage_Prev_ga > 10)] <- "Shared in lower tract"
prevdf_gen_sgf$Intersect_category[which(prevdf_gen_sgf$Percentage_Prev_saliva > 10
                                    & prevdf_gen_sgf$Percentage_Prev_feces > 10 
                                    & prevdf_gen_sgf$Percentage_Prev_ga < 10)] <- "Saliva-feces"

prevdf_gen_sgf$Intersect_category[which(prevdf_gen_sgf$Percentage_Prev_saliva >= 10
                                    & prevdf_gen_sgf$Percentage_Prev_feces >= 10 
                                    & prevdf_gen_sgf$Percentage_Prev_ga >= 10 )] <- "Mostly shared"


prevdf_gen_sgf$Intersect_category[which(prevdf_gen_sgf$Percentage_Prev_saliva >= 60
                                    & prevdf_gen_sgf$Percentage_Prev_feces >= 60
                                    & prevdf_gen_sgf$Percentage_Prev_ga >= 60)] <- "Highly shared"

prevdf_gen_sgf$Intersect_category[which(prevdf_gen_sgf$Percentage_Prev_saliva == 100 & 
                                         prevdf_gen_sgf$Percentage_Prev_feces == 100 &
                                         prevdf_gen_sgf$Percentage_Prev_ga == 100)] <- "Completely shared"

prevdf_gen_sgf$Intersect_category <- as.factor(prevdf_gen_sgf$Intersect_category)
prevdf_gen_sgf$Intersect_category <- factor(prevdf_gen_sgf$Intersect_category, levels=c("Uniquely oral", "Oral origin", "Shared in upper tract", 
                                                                              "Gastric origin", "Uniquely gastric",
                                                                                "Mostly shared", "Highly shared", "Completely shared",
                                                                              "Shared in lower tract",
                                                                                "Fecal origin", "Uniquely fecal", "Saliva-feces"), ordered = TRUE)

prevdf_gen_sgf_ordered <- prevdf_gen_sgf[order(prevdf_gen_sgf$Intersect_category, -prevdf_gen_sgf$Percentage_Prev_saliva,
                                             prevdf_gen_sgf$Percentage_Prev_ga, prevdf_gen_sgf$Percentage_Prev_feces),]

# Fix unassigned names to higher taxonomic ranks
prevdf_gen_sgf_ordered <- prevdf_gen_sgf_ordered %>%
  mutate(Name = case_when(
    str_detect(Genus, "Unassigned", negate = TRUE) ~ Genus,
    str_detect(Family, "Unassigned", negate = TRUE) ~ paste("Unassigned ", Family),
    str_detect(Order, "Unassigned", negate = TRUE) ~ paste("Unassigned ", Order),
    str_detect(Class, "Unassigned", negate = TRUE) ~ paste("Unassigned ", Class),
    str_detect(Phylum, "Unassigned", negate = TRUE) ~ paste("Unassigned ", Phylum),
    TRUE ~ "Unassigned"
  ))
```

```{r}
prevdf_gen_sgf_hm <- prevdf_gen_sgf_ordered
rownames(prevdf_gen_sgf_hm) <- prevdf_gen_sgf_hm$Name
hm_tab <- t(prevdf_gen_sgf_hm[,c(3,11,7)])

max(prevdf_gen_sgf_hm$Mean_Abundance_feces) # 17.50909
max(prevdf_gen_sgf_hm$Mean_Abundance_saliva) # 28.30646

rownames(hm_tab)[1] <- "Saliva"
rownames(hm_tab)[3] <- "Feces"
rownames(hm_tab)[2] <- "Gastric"

col_fun = colorRamp2(c(0,0,0.1,1,10,30), c("white","white", "#B8DE29FF", "#29AF7FFF","#33638DFF", "#440154FF"))

row_annot <- rowAnnotation(ls = prevdf_gen_sgf_hm$Intersect_category,
                                  col = list(ls=c("Uniquely oral" = "khaki2",
                                                  "Oral origin" = "orange2",
                                                  "Shared in upper tract" = "darkgoldenrod",
                                                  "Gastric origin" = "olivedrab4" ,
                                                  "Uniquely gastric" = "yellow4" ,
                                                  "Mostly shared" = "pink2",
                                                  "Highly shared" = "indianred3",
                                                  "Completely shared" = "#440154FF",
                                                  "Shared in lower tract" = "mediumorchid",
                                                  "Fecal origin" = "darkcyan",
                                                  "Uniquely fecal" = "lightblue2",
                                                  "Saliva-feces" = "honeydew4"
                                                  )),
                                  annotation_label = "Sharing", annotation_name_side = "top", annotation_name_rot = 45)

n_distinct(prevdf_gen_sgf_hm$Name[which(prevdf_gen_sgf_hm$Intersect_category == "Saliva-feces")])

n_distinct(prevdf_gen_sgf_hm$Name) #148
split_hm_sgf = rep("Uniquely oral", 148)
split_hm_sgf[1:2] = "Oral origin"
split_hm_sgf[3:40] = "Shared in upper tract"
split_hm_sgf[41:46] = "Gastric origin"
split_hm_sgf[47] = "Uniquely gastric"
split_hm_sgf[48:66] = "Mostly shared"
split_hm_sgf[67:77] = "Highly shared"
split_hm_sgf[78] = "Completely shared"
split_hm_sgf[79:93] = "Shared in lower tract"
split_hm_sgf[94:128] = "Fecal origin"
split_hm_sgf[129:147] = "Uniquely fecal"
split_hm_sgf[148] = "Saliva-feces"
split_hm_sgf <- as.factor(split_hm_sgf)
split_hm_sgf <- factor(split_hm_sgf, levels=c("Uniquely oral", "Oral origin", "Shared in upper tract", 
                                                                              "Gastric origin", "Uniquely gastric",
                                                                                "Mostly shared", "Highly shared", "Completely shared",
                                                                              "Shared in lower tract",
                                                                                "Fecal origin", "Uniquely fecal", "Saliva-feces"), ordered = TRUE)

heatmap_sharing_t_sgf <- Heatmap(t(hm_tab), name = "Mean relative abundance", row_split = split_hm_sgf,
                             row_order =  rownames(prevdf_gen_sgf_hm),
        col = col_fun, right_annotation = row_annot, show_column_dend = FALSE, row_names_side = "left", column_names_side = "top",
        column_names_rot = 45, border = TRUE, row_title = NULL, row_names_gp = gpar(fontsize = 6, fontface = 3))    

pdf("Figures/Heatmap_sharing_t_SGF_16S.pdf", height = 15, width = 6)
heatmap_sharing_t_sgf
dev.off()
```

Gastric - Duodenum analysis 
```{r}
physeq_gen_gd <- subset_samples(physeq_gen, body_site == "Duodenum" | body_site == "Gastric")
# Remove samples with unpaired saliva and feces
physeq_gen_gd <- subset_samples(physeq_gen_gd, id %in% intersect(md_16S_final$id[which(md_16S_final$body_site =="Gastric")],
          md_16S_final$id[which(md_16S_final$body_site =="Duodenum")]))

physeq_gen_gd <- subset_taxa(physeq_gen_gd, taxa_sums(physeq_gen_gd)/20 > 10^-6)

prevdf_gen_gd = apply(X= otu_table(physeq_gen_gd),
               MARGIN = ifelse(taxa_are_rows(physeq_gen_gd),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_gen_gd = data.frame(Prevalence = prevdf_gen_gd,
                        Percentage_Prev = 100*prevdf_gen_gd/20,
                        tax_table(physeq_gen_gd))

# Filter based on ASV (rownames) instead of genus (due to "Unassigned" genera)
physeq_gen_gd = subset_taxa(physeq_gen_gd,  rownames(tax_table(physeq_gen_gd)) %in% rownames(prevdf_gen_gd)[which(prevdf_gen_gd$Percentage_Prev >= 10)])
# 120 genera have prevalence >= 10% in gastric and duodenal samples AND average relative abundance > 10^-6

# Should add a coverage filter (Schmidt et al.)
# average vertical coverage (depth) >= 0.25
# Average vertical coverage (sequencing depth) and horizontal coverage (breadth) per microbial genome in each sample were quantified using the qaCompute utility in metaSNV

# OR a filter to select only the species for which we have a MAG
# (might have to change the metaphlan table to GTDB taxonomy and merge them)

# For now, only prevalence and abundance filter and only saliva and feces


# Prevalence and abundance in gastric for all species
physeq_gen_gd_ga <- subset_samples(physeq_gen_gd, body_site == "Gastric")
prevdf_gen_gd_ga = apply(X= otu_table(physeq_gen_gd_ga),
               MARGIN = ifelse(taxa_are_rows(physeq_gen_gd_ga),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_gen_gd_ga = data.frame(Prevalence_ga = prevdf_gen_gd_ga,
                       Percentage_Prev_ga = 100*prevdf_gen_gd_ga/10,
                       Mean_Abundance_ga = 100*taxa_sums(physeq_gen_gd_ga)/10,
                       Total_Abundance_ga = taxa_sums(physeq_gen_gd_ga),
                        tax_table(physeq_gen_gd_ga))
prevdf_gen_gd_ga <- prevdf_gen_gd_ga[, -11]


# Prevalence and abundance in duodenal for all species
physeq_gen_gd_da <- subset_samples(physeq_gen_gd, body_site == "Duodenum")
prevdf_gen_gd_da = apply(X= otu_table(physeq_gen_gd_da),
               MARGIN = ifelse(taxa_are_rows(physeq_gen_gd_da),
              yes =1, no =2), FUN = function(x){sum(x>0)})
prevdf_gen_gd_da = data.frame(Prevalence_da = prevdf_gen_gd_da,
                       Percentage_Prev_da = 100*prevdf_gen_gd_da/10,
                       Mean_Abundance_da = 100*taxa_sums(physeq_gen_gd_da)/10,
                       Total_Abundance_da = taxa_sums(physeq_gen_gd_da),
                        tax_table(physeq_gen_gd_da))
prevdf_gen_gd_da <- prevdf_gen_gd_da[, -11]


# Merging dataset
prevdf_gen_gd <- cbind(prevdf_gen_gd_ga[,c(1:4)], prevdf_gen_gd_da)

# Adding categories:
prevdf_gen_gd <- prevdf_gen_gd %>%
  mutate(Intersect_category = "")
prevdf_gen_gd$Intersect_category[which(prevdf_gen_gd$Percentage_Prev_ga > 10 
                                    & prevdf_gen_gd$Percentage_Prev_da < 10)] <- "Gastric origin"

prevdf_gen_gd$Intersect_category[which(prevdf_gen_gd$Percentage_Prev_ga < 10
                                    & prevdf_gen_gd$Percentage_Prev_da > 10)] <- "Duodenal origin"
#Unique to a body site
prevdf_gen_gd$Intersect_category[which(prevdf_gen_gd$Percentage_Prev_ga > 0 
                                    & prevdf_gen_gd$Percentage_Prev_da == 0)] <- "Uniquely gastric"

prevdf_gen_gd$Intersect_category[which(prevdf_gen_gd$Percentage_Prev_ga == 0 
                                    & prevdf_gen_gd$Percentage_Prev_da > 0)] <- "Uniquely duodenal"

prevdf_gen_gd$Intersect_category[which(prevdf_gen_gd$Percentage_Prev_ga < 10 
                                   & prevdf_gen_gd$Percentage_Prev_da < 10)] <- "Low prevalence"

# Shared
prevdf_gen_gd$Intersect_category[which(prevdf_gen_gd$Percentage_Prev_ga >= 10
                                    & prevdf_gen_gd$Percentage_Prev_da >= 10)] <- "Mostly shared"

prevdf_gen_gd$Intersect_category[which(prevdf_gen_gd$Percentage_Prev_ga >= 60
                                    & prevdf_gen_gd$Percentage_Prev_da >= 60)] <- "Highly shared"

prevdf_gen_gd$Intersect_category[which(prevdf_gen_gd$Percentage_Prev_ga == 100 & 
                                         prevdf_gen_gd$Percentage_Prev_da == 100)] <- "Completely shared"

prevdf_gen_gd$Intersect_category <- as.factor(prevdf_gen_gd$Intersect_category)
prevdf_gen_gd$Intersect_category <- factor(prevdf_gen_gd$Intersect_category, levels=c("Uniquely gastric", "Gastric origin",
                                                                                "Mostly shared", "Highly shared", "Completely shared",
                                                                                "Duodenal origin", "Uniquely duodenal"), ordered = TRUE)


prevdf_gen_gd_ordered <- prevdf_gen_gd[order(prevdf_gen_gd$Intersect_category, -prevdf_gen_gd$Percentage_Prev_ga,
                                             prevdf_gen_gd$Percentage_Prev_ga),]

# Fix unassigned names to higher taxonomic ranks
prevdf_gen_gd_ordered <- prevdf_gen_gd_ordered %>%
  mutate(Name = case_when(
    str_detect(Genus, "Unassigned", negate = TRUE) ~ Genus,
    str_detect(Family, "Unassigned", negate = TRUE) ~ paste("Unassigned ", Family),
    str_detect(Order, "Unassigned", negate = TRUE) ~ paste("Unassigned ", Order),
    str_detect(Class, "Unassigned", negate = TRUE) ~ paste("Unassigned ", Class),
    str_detect(Phylum, "Unassigned", negate = TRUE) ~ paste("Unassigned ", Phylum),
    TRUE ~ "Unassigned"
  ))
```

```{r}
prevdf_gen_gd_hm <- prevdf_gen_gd_ordered
rownames(prevdf_gen_gd_hm) <- prevdf_gen_gd_hm$Name
hm_tab <- t(prevdf_gen_gd_hm[,c(3,7)])

max(prevdf_gen_gd_hm$Mean_Abundance_ga) # 16
max(prevdf_gen_gd_hm$Mean_Abundance_da) # 248.30646

rownames(hm_tab)[1] <- "Gastric"
rownames(hm_tab)[2] <- "Duodenum"

col_fun = colorRamp2(c(0,0,0.1,1,10,25), c("white","white", "#B8DE29FF", "#29AF7FFF","#33638DFF", "#440154FF"))

row_annot <- rowAnnotation(ls = prevdf_gen_gd_hm$Intersect_category,
                                  col = list(ls=c("Uniquely gastric" = "yellow4",
                                                  "Gastric origin" = "olivedrab4",
                                                  "Mostly shared" = "pink2",
                                                  "Highly shared" = "indianred3",
                                                  "Completely shared" = "#440154FF",
                                                  "Duodenal origin" = "#6575aa",
                                                  "Uniquely duodenal" = "#5099aa"
                                                  )),
                                  annotation_label = "Sharing", annotation_name_side = "top", annotation_name_rot = 45)

n_distinct(prevdf_gen_gd_hm$Name[which(prevdf_gen_gd_hm$Intersect_category == "Uniquely duodenal")])

n_distinct(prevdf_gen_gd_hm$Name) #120
split_hm_gd = rep("Uniquely gastric", 120)
#split_hm_gd[15:46] = "Gastric origin"
split_hm_gd[15:84] = "Mostly shared"
split_hm_gd[85:105] = "Highly shared"
split_hm_gd[106:119] = "Completely shared"
#split_hm_gd[94:128] = "Duodenal origin"
split_hm_gd[120] = "Uniquely duodenal"
split_hm_gd <- as.factor(split_hm_gd)
split_hm_gd <- factor(split_hm_gd, levels=c("Gastric origin", "Uniquely gastric",
                                            "Mostly shared", "Highly shared", "Completely shared",
                                            "Duodenal origin", "Uniquely duodenal"), ordered = TRUE)

heatmap_sharing_t_gd <- Heatmap(t(hm_tab), name = "Mean relative abundance", row_split = split_hm_gd,
                             row_order =  rownames(prevdf_gen_gd_hm),
        col = col_fun, right_annotation = row_annot, show_column_dend = FALSE, row_names_side = "left", column_names_side = "top",
        column_names_rot = 45, border = TRUE, row_title = NULL, row_names_gp = gpar(fontsize = 6, fontface = 3))    

pdf("Figures/Heatmap_sharing_t_gd_16S.pdf", height = 15, width = 6)
heatmap_sharing_t_gd
dev.off()
```





